---
title: "Motor neuron nucleocytoplasmic mislocalisation"
author: "Oliver Ziff"
date: "`r format(Sys.time(), '%d %h %Y %H:%M')`"
output:
  html_document:
    fig_width: 7
    fig_height: 6
    toc_depth: 3
    theme: simplex
    dev: jpeg
editor_options:
  chunk_output_type: console
---

# Setup

```{r setup, include=FALSE}
# set here path https://cran.r-project.org/web/packages/here/vignettes/here.html
library(here)
# Mac
# nemo_path = here("/Volumes/lab-patanir/home/users/ziffo")
# proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
# script_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts")
# shared_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working")
# collab_path = here("/Volumes/lab-patanir/home/users/ziffo/patani-collab")
## OnDemand
nemo_path = here("/nemo/lab/patanir/home/users/ziffo")
# .libPaths("/nemo/lab/patanir/home/users/ziffo/R/x86_64-pc-linux-gnu-library/4.0")
proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
script_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts/als_rna_protein_distribution_ml240")
shared_path = here("/nemo/project/proj-luscombn-patani/working")
collab_path = here("/nemo/lab/patanir/home/users/ziffo/patani-collab")
setwd(proj_path)
set_here(proj_path)
here(proj_path)
# here::i_am("als_rna_protein_distribution_ml240.Rmd")
here()
dr_here()

source(here(nemo_path,"scripts/functions/R_workspace.R"))

# nemo_path = here("/Volumes/lab-patanir/home/users/ziffo")
# proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
# load(here(proj_path,"scripts/mn_mislocalisation.RData")) # overwrites with old functions
load(here(proj_path,"scripts/metadata_deseq_objects.RData"))
# load(here(proj_path,"scripts/untreated_deseq_objects.RData"))

# untreated objects
load(here(proj_path,"scripts/untreated_sample_deseq_objects.RData"))
load(here(proj_path, "scripts/dep_mass_spec_objects.RData"))

# ML240 objects
load(here(proj_path,"scripts/ml240_sample_deseq_objects.RData")) # load(here(proj_path,"scripts/ml240_deseq_objects.RData"))
load(here(proj_path, "scripts/dep_mass_spec_ml240_objects.RData"))

# whole cell objects & splicing
load(here(proj_path, "scripts/whole_untreated_majiq_objects.RData"))
load(here(proj_path,"scripts/whole_untreated_deseq_objects.RData"))

load(here(proj_path, "scripts/whole_ml240_majiq_objects.RData"))
load(here(proj_path,"scripts/whole_ml240_deseq_objects.RData"))



# Keep total environment space < 1GB or gets too slow
rm(list=ls(pattern="^RNAct"))
rm(list=ls(pattern="^rnact"))
rm(list=ls(pattern="transite"))


```

Run the DESeq2 object script to create mn_mislocalisation.RData from OnDemand. Takes ~ 1 hour 20 mins

```{r}
# GEO submission
salmon.merged.gene_tpm.tsv = read_tsv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/geo-submission-sep22/salmon.merged.gene_tpm.tsv")) %>% select(-contains("whole"))
write_tsv(salmon.merged.gene_tpm.tsv, here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/geo-submission-sep22/salmon.merged.gene_tpm.tsv"))

geo_md5sum.txt = read_delim(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/geo-submission-sep22/geo_md5sum.txt"), col_names = c("file_checksum", "file_name"), delim = "  ")
write_csv(geo_md5sum.txt, here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/geo-submission-sep22/geo_md5sum.csv"))

read_csv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/sample-details/samplesheet.csv")) %>% glimpse
geo_metadata = read_csv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/sample-details/samplesheet.csv")) %>% distinct(sample, .keep_all = TRUE) %>% 
  mutate(title = paste0(fraction,"_",treatment,"_",cellline), organism = "Homo sapiens", cell_type = "ipsn", genotype = case_when(grepl("ctrl", cellline) ~ "wildtype",  grepl("tdpn", cellline) ~ "tardbp mutant", grepl("cb1|gli", cellline) ~ "vcp mutant", grepl("ncrm", cellline) ~ "vcp knock-in"), molecule = case_when(grepl("nuc", fraction) ~ "nuclear polyA RNA", grepl("cyt", fraction) ~ "cytoplasmic polyA RNA"), single_or_paired_end = "paired-end", instrument_model = "Illumina NovoSeq 6000",
         raw_file_1 = paste0(sample,"_1.merged.fastq.gz"), raw_file_2 = paste0(sample,"_2.merged.fastq.gz")) %>%
  filter(fraction != "whole") %>%
  select(library_name = sample, title, organism, cell_type, cellline, genotype, treatment, molecule, single_or_paired_end, instrument_model, raw_file_1, raw_file_2, strandedness, fraction, library_type)
geo_metadata
write_csv(geo_metadata, here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/geo-submission-sep22/geo_metadata.csv"))

```

## Update git / github

https://happygitwithr.com/common-remote-setups.html

```{r}
library(here)
library(usethis) # usethis & gert https://jeroen.github.io/gert2019 https://www.garrickadenbuie.com/blog/pull-request-flow-usethis/?interactive=1&steps=
use_git_config(user.name = "ojziff", user.email = "oliver.ziff@crick.ac.uk")
# make github repro on github.com ojziff/als_rna_protein_distribution_ml240
# on HPC clone to scripts folder: git clone https://github.com/ojziff/als_rna_protein_distribution_ml240.git # instead of password type PAT code https://github.com/settings/tokens 
git config --global user.name "ojziff"
git config --global user.email "o.ziff@ucl.ac.uk"
# usethis::create_github_token()
usethis::edit_r_environ()
credentials::set_github_pat("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")  
library(gert)
proj_path = here("/Volumes/lab-patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
setwd(here(proj_path,"scripts/als_rna_protein_distribution_ml240")) # setwd then dont need to specify repo argument
# git_init(path = here(proj_path,"scripts/als_rna_protein_distribution_ml240")) # initialise repository
git_commit_all("update") # stage all files & commit in one command - implies git_add(".")
pr_init("figures_and_tables") # initialise a new branch
pr_push() # this will open github in chrome > pull request > merge

git_sitrep() # print git and github info
git_config()
git_info()
git_ls() # list git tree
git_log() # check recent commits
git_status() # check which files are modified
# pr_resume("figures_and_tables") # resume a branch
# git_add(".") #stage the files you want to commmit - "." indicates all files.
# git_commit("update") # commit with comment
pr_pull()

```

# Table S2 RNAseq QC

nuclear & cytoplasm fractions only

```{r}
multiqc_stat_paths = metadata %>% distinct(database_dir) %>% mutate(multiqc_stat_paths = paste0(database_dir,"/nfcore/multiqc/star_salmon/multiqc_data/multiqc_general_stats.txt"), multiqc_stat_paths = gsub("/nemo/lab/patanir/home/users/ziffo|/nemo/project","/Volumes/lab-patanir/home/users/ziffo",multiqc_stat_paths)) %>% pull(multiqc_stat_paths)
multiqc_stats.tsv = read_tsv(multiqc_stat_paths) 
multiqc_stats = multiqc_stats.tsv %>% clean_names() %>% left_join(metadata) %>%
  filter(!grepl("whole", sample)) %>% #mutate(sample = gsub("_d35|_R","",sample)) %>%
  drop_na(samtools_mqc_generalstats_samtools_flagstat_total) %>%
  select(sample, fraction, treatment, cellline, mutation, condition, mapped_passed = samtools_mqc_generalstats_samtools_mapped_passed, reads_mapped = samtools_mqc_generalstats_samtools_reads_mapped, reads_mapped_percent = samtools_mqc_generalstats_samtools_reads_mapped_percent, reads_aligned = quali_map_mqc_generalstats_qualimap_reads_aligned,  uniquely_mapped_percent = star_mqc_generalstats_star_uniquely_mapped_percent, uniquely_mapped = star_mqc_generalstats_star_uniquely_mapped, r_rna_percent = biotype_counts_mqc_generalstats_biotype_counts_percent_r_rna, bias_5_3 = quali_map_mqc_generalstats_qualimap_5_3_bias, proper_pairs_percent = r_se_qc_mqc_generalstats_rseqc_proper_pairs_percent, percent_duplication = picard_mqc_generalstats_picard_percent_duplication, error_rate = samtools_mqc_generalstats_samtools_error_rate, non_primary_alignments = samtools_mqc_generalstats_samtools_non_primary_alignments, reads_properly_paired_percent = samtools_mqc_generalstats_samtools_reads_properly_paired_percent) %>%
  arrange(treatment, fraction)
multiqc_stats %>% glimpse
write_csv(multiqc_stats, here(proj_path,"alignment/multiqc_general_stats.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(multiqc_stats, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S2 RNAseqQC")

multiqc_stats %>% my_summarise(var = mapped_passed)
# mean_mapped_passed median_mapped_passed sd_mapped_passed Q1_mapped_passed Q3_mapped_passed min_mapped_passed max_mapped_passed sum_mapped_passed n_mapped_passed
#                <dbl>                <dbl>            <dbl>            <dbl>            <dbl>             <dbl>             <dbl>             <dbl>           <int>
# 1         188792042.            164235355        51988254.        157442803        203395615         143286655         367650434        8873225969              47

metadata %>% filter(fraction != "whole", treatment == "untreated") %>% count(mutation)

```


# Fig. 1 Gene Expression

Fractionation schematic made in Biorender.

```{r}
# import with magick 
fractionation_schematic = magick::image_read(here(proj_path,"biorender/mn_differentiation_fractionation.png"))
fractionation_schematic.gg = ggdraw() + draw_image(fractionation_schematic)
# fractionation_schematic.gg

# motor neuron markers ICC
ipsn_mn_markers = magick::image_read(here(proj_path,"ml240-assays/ipsn_mn_markers.png"))
ipsn_mn_markers.gg = ggdraw() + draw_image(ipsn_mn_markers)
ipsn_mn_markers.gg

```


## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",   "SLC4A4", "DIO2")#"SOX9", ,"MLC1""FGFR3", 
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1","ETV4","FOXP1","ALDH1A2", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
# oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
# interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)
dataset.samples.df = untreated.metadata %>% filter(fraction != "whole") %>% select(sample, fraction)

untreated.nuc_vs_cyt$vsd.counts = as_tibble(assay(untreated.nuc_vs_cyt$vsd), rownames = "gene_id") %>% left_join(gene2ens)
min_vst = min(untreated.nuc_vs_cyt$vsd.counts[,2])
celltype.markers.marker <- untreated.nuc_vs_cyt$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) 
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
# celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
# rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$sample]
# dataset.samples.df$sample[!dataset.samples.df$sample %in% rownames(celltype.markers.marker.centered_mat)]
celltype.markers.rnaseq.cluster.heatmap <- as.ggplot(
  Heatmap(celltype.markers.marker.mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          row_order = celltype.markers.df.filt$gene_name,
          row_names_gp = gpar(fontsize = 5), 
          row_split = celltype.markers.df.filt$group,
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelia", "Myeloid", "Oligo-\ndendrocyte","Motor\nneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_title = NULL, 
          show_column_names = FALSE, #row_names_gp = gpar(fontsize = 6),
          # row_order = dataset.samples.df$dataset_sample,
          column_split = factor(dataset.samples.df$fraction, levels = c("nucleus", "cytoplasm")),
          bottom_annotation = columnAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",2)),  labels = c("Nuclear samples", "Cytoplasm samples"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE))
celltype.markers.rnaseq.cluster.heatmap

```



## QualiMap read origin

Export data from multiQC QualiMap origin of reads & RSeQC Read Distribution

```{r}
rseqc_read_distribution = read_tsv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt")) %>%
  clean_names() %>% select(sample, CDS = cds_exons_tag_pct, `5UTR` = x5_utr_exons_tag_pct, `3UTR` = x3_utr_exons_tag_pct, Intron = introns_tag_pct) %>% mutate(Intergenic = 100 - (CDS + `5UTR` + `3UTR` + Intron)) %>%
  pivot_longer(!sample, names_to = "region", values_to = "percent") %>%
  mutate(cellline = str_split_fixed(sample, "_", 3)[,1], 
         sample_label = case_when(cellline == "glia" ~ "VCP1^R191Q", cellline == "glib" ~ "VCP2^R191Q", cellline == "cb1d" ~ "VCP1^R155C", cellline == "cb1e" ~ "VCP2^R155C", cellline == "ncrmc2" ~ "VCPki1^R191Q", cellline == "ncrme6" ~ "VCPki2^R191Q",  cellline == "tdpn1" ~ "TARDBP1^G298S", cellline == "tdpn2" ~ "TARDBP2^G298S", TRUE ~ cellline), sample_label = gsub("ctrl","CTRL",sample_label),
         # sample_label = case_when(cellline == "glia" ~ "VCP1", cellline == "glib" ~ "VCP2", cellline == "cb1d" ~ "VCP3", cellline == "cb1e" ~ "VCP4", cellline == "ncrmc2" ~ "VCPki1", cellline == "ncrme6" ~ "VCPki2",  cellline == "tdpn1" ~ "TARDBP1", cellline == "tdpn2" ~ "TARDBP2", TRUE ~ cellline), 
         # cellline = case_when(grepl("CTRL_R1", cellline) ~ "CTRL1", grepl("CTRL_R2", cellline) ~ "CTRL2", grepl("CTRL_R3", cellline) ~ "CTRL3", grepl("CTRL_R4", cellline) ~ "CTRL4",grepl("VCP_R1", cellline) ~ "VCP1", grepl("VCP_R2", cellline) ~ "VCP2", grepl("VCP_R3", cellline) ~ "VCP3", grepl("VCP_R4", cellline) ~ "VCP4"),
         sample_label = factor(sample_label, levels = rev(c("CTRL1", "CTRL3", "CTRL4", "CTRL6", "TARDBP1^G298S", "TARDBP2^G298S", "VCP1^R155C", "VCP2^R155C", "VCP1^R191Q", "VCP2^R191Q", "VCPki1^R191Q", "VCPki2^R191Q"))),
         fraction = str_split_fixed(sample, "_", 3)[,3], fraction = gsub("nucleus","Nuclear Samples",fraction), fraction = gsub("cytoplasm","Cytoplasm Samples",fraction),
         treatment = str_split_fixed(sample, "_", 3)[,2],
         region = factor(region, levels = c("5UTR", "CDS", "3UTR", "Intron", "Intergenic"))) %>% drop_na(cellline) %>%
  filter(treatment == "untreated", fraction != "whole", sample != "ctrl3_untreated_nucleus")

rseqc_read_distribution.summarised.barplot <- ggbarplot(rseqc_read_distribution, x="sample_label" , y="percent", fill="region", label = FALSE, position = position_fill(), xlab = "")  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 1)) +
  facet_grid(fraction ~ .,scales="free_y") +
  coord_flip() +
  ylab(expression(Read~type~proportions)) +
  scale_x_discrete(labels = scales::parse_format()) 
rseqc_read_distribution.summarised.barplot

rseqc_read_distribution %>% filter(fraction == "cytoplasm", region %in% c("5UTR","3UTR","CDS")) %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum)) #88.0
rseqc_read_distribution %>% filter(fraction == "nucleus", region %in% c("5UTR","3UTR","CDS")) %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum)) #67.5
rseqc_read_distribution %>% filter(fraction == "cytoplasm", region == "Intron") %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum))#9.4
rseqc_read_distribution %>% filter(fraction == "nucleus", region == "Intron") %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum))#28.1
  
```


## Cyto & Nuc Heatmap

Keep nuc R155C.1 in heatmap but exclude in all downstream analyses

```{r}
# Import gene list of Nuc & Cyto markers
price_2020_nuclear_cyto_genes <- read_excel(here(nemo_path, "genomes/nuc-cyt-localisation/price_2020_nuclear_cytoplasmic_transcriptomes_supp_tables.xlsx"), sheet = "Table S2 DEGs by Fraction") %>% filter(Symbol %in% gene2ens$gene_name) # 4284
price_2020_nuclear_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Nuclear: Both","Nuclear: Prenatal Only", "Nuclear: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_cyto_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Cytoplasmic: Both","Cytoplasmic: Prenatal Only", "Cytoplasmic: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_nuc_cyto_markers <- c(price_2020_cyto_genes, price_2020_nuclear_genes)
nuc_cyt_markers.df = data.frame("gene_name" = price_2020_nuc_cyto_markers) %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>% mutate(group = case_when(gene_name %in% price_2020_nuclear_genes ~ "nuclear", gene_name %in% price_2020_cyto_genes ~ "cytoplasm")) %>% distinct()
nuc_cyt_markers.annot <- tibble(gene_name = nuc_cyt_markers.df$gene_name, group = factor(nuc_cyt_markers.df$group)) %>% arrange(group) # Annotation 

untreated.nuc_vs_cyt$vsd.counts = as_tibble(assay(untreated.nuc_vs_cyt$vsd), rownames = "gene_id") %>% left_join(gene2ens)
mn_price.nuc.cyt.marker <- untreated.nuc_vs_cyt$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>% #, rowSums(across(where(is.numeric))) > 162) %>% 
  select(-gene_id, -contains("whole")) 
mn_price.nuc.cyt.marker.topVar <- mn_price.nuc.cyt.marker %>%
  dplyr::mutate_if(is.numeric, funs(ifelse(. <0, 0, .))) %>%   # dplyr::mutate_if(is.numeric, funs(log2(.))) %>%
  pivot_longer(!gene_name, names_to = "sample", values_to = "count") %>% 
  mutate(fraction = case_when(grepl("cyt", sample) ~ "cytoplasm", grepl("nuc", sample) ~ "nuclear", TRUE ~ "whole")) %>%
  group_by(gene_name, fraction) %>% summarise(mean = mean(count, na.rm = TRUE)) %>% ungroup %>% pivot_wider(names_from = fraction, values_from = mean) %>% mutate(delta.mean = nuclear - cytoplasm) %>% 
  slice_max(abs(delta.mean), n = 3000) %>% # retain the most variable nuc vs cyt genes between samples
  mutate(fraction_database = case_when(gene_name %in% price_2020_cyto_genes ~ "cytoplasm", gene_name %in% price_2020_nuclear_genes ~ "nuclear")) %>% arrange(fraction_database)

colnames(mn_price.nuc.cyt.marker) = gsub("untreated_", "", colnames(mn_price.nuc.cyt.marker))
mn_price.nuc.cyt.marker.filt <- mn_price.nuc.cyt.marker %>% filter(gene_name %in% mn_price.nuc.cyt.marker.topVar$gene_name) %>% filter_all(all_vars(!is.infinite(.))) %>% drop_na()
mn_price.nuc.cyt.marker.mat <- make_matrix(select(mn_price.nuc.cyt.marker.filt,-gene_name), pull(mn_price.nuc.cyt.marker.filt,gene_name))
# colnames(mn_price.nuc.cyt.marker.mat) <- mn.metadata$name.long[ mn.metadata$sample %in% colnames(mn_price.nuc.cyt.marker.mat) ]
# colnames(mn_price.nuc.cyt.marker.mat) <- gsub("mn_|d35_|_R", "", colnames(mn_price.nuc.cyt.marker.mat) )

# price.nuc.cyt.marker.topVar[price.nuc.cyt.marker.topVar > 14] <- 14 # crop outliers to improve colour distribution
mn_price.nuc.cyt.marker.scaled_mat = t(scale(t(mn_price.nuc.cyt.marker.mat))) # row scale
mn_price.nuc.cyt.marker.scaled_mat <- mn_price.nuc.cyt.marker.scaled_mat[!is.infinite(rowSums(mn_price.nuc.cyt.marker.scaled_mat)),]

cytoplasm_markers = mn_price.nuc.cyt.marker.filt$gene_name[mn_price.nuc.cyt.marker.filt$gene_name %in% price_2020_cyto_genes]
nuclear_markers = mn_price.nuc.cyt.marker.filt$gene_name[mn_price.nuc.cyt.marker.filt$gene_name %in% price_2020_nuclear_genes]
mn_price.nuc.cyt.marker.column_order = untreated.metadata %>% filter(!grepl("whole",sample)) %>% arrange(fraction)

mn_price.nuc.cyt.marker.heatmap <- as.ggplot(
  Heatmap(mn_price.nuc.cyt.marker.scaled_mat,
          row_km = 2,   # row_order = mn_price.nuc.cyt.marker.topVar$gene_name,
          show_row_names = FALSE,
          # row_split =  factor(mn_price.nuc.cyt.marker.topVar$fraction_database, levels = c("nuclear", "cytoplasm")), #c(rep("cyt", length(cytoplasm_markers)), rep("nuc", length(nuclear_markers))),
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Cytoplasm\nmarkers", "Nuclear\nmarkers"), labels_gp = gpar(fontsize = 8))), row_title = NULL,
          column_order = gsub("untreated_", "", mn_price.nuc.cyt.marker.column_order$sample), 
          column_km = 2,
          # column_split = factor(mn_price.nuc.cyt.marker.column_order$fraction, levels = c("cytoplasm", "nuclear", "whole")),
          bottom_annotation = HeatmapAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Nuclear samples", "Cytoplasm samples"), labels_gp = gpar(fontsize = 8))), column_title = NULL,
          show_column_names = FALSE, #column_names_gp = gpar(fontsize = 6),
          cluster_row_slices = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE, cluster_rows = FALSE,
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical")))
# mn_price.nuc.cyt.marker.heatmap

```


## Cyto & Nuclear Density

Known markers from Bahar Halpern https://www.sciencedirect.com/science/article/pii/S2211124715013510

Human cortex markers from Price et al https://genome.cshlp.org/content/30/1/1.full#ref-2

```{r}
price_2020_nuclear_cyto_genes <- read_excel(here(nemo_path, "genomes/nuc-cyt-localisation/price_2020_nuclear_cytoplasmic_transcriptomes_supp_tables.xlsx"), sheet = "Table S2 DEGs by Fraction") %>% filter(Symbol %in% gene2ens$gene_name) # 4284
price_2020_nuclear_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Nuclear: Both","Nuclear: Prenatal Only", "Nuclear: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_cyto_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Cytoplasmic: Both","Cytoplasmic: Prenatal Only", "Cytoplasmic: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_nuc_cyto_markers <- c(price_2020_cyto_genes, price_2020_nuclear_genes, "LMNA", "NEAT1", "GCGR", "HC-4")
nuc_vs_cyt.volcano.labels <-  c("MALAT1", "MLXIPL", "XIST", "SOX9", "RBFOX3", "LMNA", "HC-4",  # nuclear
                                      "GLUL",  "STAU1") # cytoplasmic

# Density
untreated.nuc_vs_cyt.res.price_markers <- untreated.nuc_vs_cyt$res %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>%
  mutate(Markers = case_when(gene_name %in% price_2020_cyto_genes ~ "Cytoplasmic", TRUE ~ "Nuclear"))
# redraw figure 1e (volcano plot of nuclear and cytoplasmic markers) to instead be a superimposed density plot of the two distributions of marker intensities - the volcano plot is quite misleading as it squashes the key points near log-foldchange=0 to a very small vertical region preventing us from seeing whether the two distributions are properly balanced around the origin (which represents the point where the _normalised_ cytoplasmic = _normalised_ nuclear).
untreated.nuc_vs_cyt.price_markers.density = density_plot(untreated.nuc_vs_cyt.res.price_markers, color_by = Markers, cols = c("dodgerblue2", "firebrick2"), ymax = 0.8, xlims = c(-2.5,4), numerator = "Nucleus", denomenator = "Cytoplasm", dpi = 300)
untreated.nuc_vs_cyt.price_markers.density


# # # Volcano
# untreated.nuc_vs_cyt.price_markers.volcano <- volcano_plot(untreated.nuc_vs_cyt.res.price_markers, gene_labels = nuc_vs_cyt.volcano.labels, numerator = "Nucleus", denomenator = "Cytoplasm", ymax = 150, xmax = 5, dpi = 300) 
# untreated.nuc_vs_cyt.price_markers.volcano


### identifier of cell fraction: GSEA of cytoplasm markers

### Which fraction is a better identifier of cell identity: GSEA of nuclear markers
# cytoplasm
mn_nuc_vs_cyt.ranks <- untreated.nuc_vs_cyt$res %>% select(gene_name, stat) %>% na.omit() %>% distinct() %>% group_by(gene_name) %>% summarize(stat = mean(stat)) %>% deframe()
price_2020_cyto.list <- list("cytoplasm_markers" = unique(price_2020_cyto_genes[price_2020_cyto_genes %in% names(mn_nuc_vs_cyt.ranks)])) 
price_2020_cyto.list$cytoplasm_markers = price_2020_cyto.list$cytoplasm_markers[price_2020_cyto.list$cytoplasm_markers %in% mn_price.nuc.cyt.marker.topVar$gene_name]
mn.nuc_vs_cyt.res.price_cytoplasm.fgseaRes <- fgsea(pathways = price_2020_cyto.list, stats = mn_nuc_vs_cyt.ranks, nPerm = 10000)
mn.nuc_vs_cyt.res.price_cytoplasm.fgseaRes
#             pathway  pval  padj log2err         ES       NES size                                  leadingEdge
# 1: cytoplasm_markers 1e-50 1e-50      NA -0.8250897 -3.401073 1714 CBLL1,MAPK1IP1L,EIF4B,RNF2,FAM204A,WBP11,...

# nuclear
price_2020_nuclear.list <- list("nuclear_markers" = unique(price_2020_nuclear_genes[price_2020_nuclear_genes %in% untreated.nuc_vs_cyt$res$gene_name])) 
mn.nuc_vs_cyt.res.price_nuclear.fgseaRes <- fgsea(pathways = price_2020_nuclear.list, stats = mn_nuc_vs_cyt.ranks, nPerm = 1000)
mn.nuc_vs_cyt.res.price_nuclear.fgseaRes
#  pathway  pval  padj log2err        ES      NES size                                        leadingEdge
# 1: nuclear_markers 1e-50 1e-50      NA 0.7645324 2.695193 1726 NPEPL1,AGER,IL11RA,STX16-NPEPL1,SLC25A34,IFRD2,...

```



## Coverage plots

Run on CAMP

```{r}
# library(here)
# proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
# script_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts/als_rna_protein_distribution_ml240")
# shared_path = here("/nemo/project/proj-luscombn-patani/working")
# collab_path = here("/nemo/lab/patanir/home/users/ziffo/patani-collab")
# source(here(nemo_path,"scripts/functions/R_workspace.R"))
# # load(here(proj_path,"scripts/untreated_deseq_objects.RData"))
# 
# ctrl_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_nuclear.bam") # CTRL3 removed
# ctrl_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_cytoplasm.bam") # CTRL3 removed to balance reads
# vcp_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_nuclear.bam")
# vcp_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_cytoplasm.bam")
# tardbp_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_nuclear.bam")
# tardbp_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_cytoplasm.bam")
# 
# 
# library(TxDb.Hsapiens.UCSC.hg38.knownGene)
# library(org.Hs.eg.db)
# # txdb <- makeTxDbFromGFF(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf"), format="gtf") # make txdb from GTF - turn on only when needed
# # seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations & scaffolds which are also in the BAM
# # saveRDS(txdb, here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.rds"))
# txdb = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.rds"))
# 
# track_viewer_nuc_cyt(gene.name = "MALAT1", ctrl.nuc.bam = ctrl_nuclear.bam, ctrl.cyt.bam = ctrl_cytoplasm.bam, print2screen = FALSE, save = TRUE) # plot only control track
# track_viewer_nuc_cyt(gene.name = "GAPDH", ctrl.nuc.bam = ctrl_nuclear.bam, ctrl.cyt.bam = ctrl_cytoplasm.bam, print2screen = FALSE, save = TRUE) # plot only control track


MALAT1.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/MALAT1.ctrl_only.nuc_cyt_coverage_plot.png"))
MALAT1.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(MALAT1.nuc_cyt_coverage_plot.png)
# MALAT1.nuc_cyt_coverage_plot.gg

GAPDH.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/GAPDH.ctrl_only.nuc_cyt_coverage_plot.png"))
GAPDH.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(GAPDH.nuc_cyt_coverage_plot.png)
# GAPDH.nuc_cyt_coverage_plot.gg

```


## PCA 

```{r}
untreated.nuc_vs_cyt.pcaData <- plotPCA(untreated.nuc_vs_cyt$vsd, intgroup=c("condition", "sample", "cellline", "fraction", "mutation"), returnData=TRUE) %>% as_tibble() %>%
  mutate(sample_label = case_when(cellline == "glia" ~ "VCP1^R191Q", cellline == "glib" ~ "VCP2^R191Q", cellline == "cb1d" ~ "VCP1^R155C", cellline == "cb1e" ~ "VCP2^R155C", cellline == "ncrmc2" ~ "VCPki1^R191Q", cellline == "ncrme6" ~ "VCPki2^R191Q",  
                                  cellline == "tdpn1" ~ "TARDBP1^G298S", cellline == "tdpn2" ~ "TARDBP2^G298S", TRUE ~ as.character(cellline)), 
         gender = case_when(cellline %in% c("ctrl3","ctrl4","cb1e","cb1d") ~ "Female", TRUE~"Male"),
         sample_label = gsub("ctrl","CTRL",sample_label), fraction = gsub("nucleus","Nuclear Samples",fraction), fraction = gsub("cytoplasm","Cytoplasm Samples",fraction))

untreated.nuc_vs_cyt.pca <- ggplot(untreated.nuc_vs_cyt.pcaData, aes(PC1, PC2, color=fraction)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=as.character(sample_label)), parse = TRUE, size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
    xlab(paste0("PC1: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData, "percentVar"))[2],"% variance")) +
  ggforce::geom_mark_ellipse(show.legend = FALSE)
untreated.nuc_vs_cyt.pca



# colour by gender for reviewer
untreated.nuc_vs_cyt.pca.gender <- ggplot(untreated.nuc_vs_cyt.pcaData, aes(PC1, PC2, color=gender)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=as.character(sample_label)), parse = TRUE, size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
    xlab(paste0("PC1: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData, "percentVar"))[2],"% variance"))# +
  # ggforce::geom_mark_ellipse(show.legend = FALSE)
untreated.nuc_vs_cyt.pca.gender

untreated.nuc_vs_cyt.pcaData.3_4.gender = plotPCA.san(untreated.nuc_vs_cyt$vsd, intgroup=c("condition", "sample", "cellline", "fraction", "mutation"), PC_x = 3, PC_y = 4, returnData=TRUE) %>% as_tibble() %>% mutate(sample_label = case_when(cellline == "glia" ~ "VCP1^R191Q", cellline == "glib" ~ "VCP2^R191Q", cellline == "cb1d" ~ "VCP1^R155C", cellline == "cb1e" ~ "VCP2^R155C", cellline == "ncrmc2" ~ "VCPki1^R191Q", cellline == "ncrme6" ~ "VCPki2^R191Q", cellline == "tdpn1" ~ "TARDBP1^G298S", cellline == "tdpn2" ~ "TARDBP2^G298S", TRUE ~ as.character(cellline)), 
         gender = case_when(cellline %in% c("ctrl3","ctrl4","cb1e","cb1d") ~ "Female", TRUE~"Male"), sample_label = gsub("ctrl","CTRL",sample_label), fraction = gsub("nucleus","Nuclear Samples",fraction), fraction = gsub("cytoplasm","Cytoplasm Samples",fraction))
untreated.nuc_vs_cyt.pca.3_4.gender <- ggplot(untreated.nuc_vs_cyt.pcaData.3_4.gender, aes(PCX, PCY, color=gender)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=as.character(sample_label)), parse = TRUE, size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
    xlab(paste0("PC3: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.3_4.gender, "percentVar"))[1],"% variance")) +  ylab(paste0("PC4: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.3_4.gender, "percentVar"))[2],"% variance"))
untreated.nuc_vs_cyt.pca.3_4.gender

untreated.nuc_vs_cyt.pcaData.5_6.gender = plotPCA.san(untreated.nuc_vs_cyt$vsd, intgroup=c("condition", "sample", "cellline", "fraction", "mutation"), PC_x = 5, PC_y = 6, returnData=TRUE) %>% as_tibble() %>% mutate(sample_label = case_when(cellline == "glia" ~ "VCP1^R191Q", cellline == "glib" ~ "VCP2^R191Q", cellline == "cb1d" ~ "VCP1^R155C", cellline == "cb1e" ~ "VCP2^R155C", cellline == "ncrmc2" ~ "VCPki1^R191Q", cellline == "ncrme6" ~ "VCPki2^R191Q", cellline == "tdpn1" ~ "TARDBP1^G298S", cellline == "tdpn2" ~ "TARDBP2^G298S", TRUE ~ as.character(cellline)), 
         gender = case_when(cellline %in% c("ctrl3","ctrl4","cb1e","cb1d") ~ "Female", TRUE~"Male"), sample_label = gsub("ctrl","CTRL",sample_label), fraction = gsub("nucleus","Nuclear Samples",fraction), fraction = gsub("cytoplasm","Cytoplasm Samples",fraction))
untreated.nuc_vs_cyt.pca.5_6.gender <- ggplot(untreated.nuc_vs_cyt.pcaData.5_6.gender, aes(PCX, PCY, color=gender)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=as.character(sample_label)), parse = TRUE, size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
    xlab(paste0("PC5: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.5_6.gender, "percentVar"))[1],"% variance")) +  ylab(paste0("PC6: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.5_6.gender, "percentVar"))[2],"% variance"))
untreated.nuc_vs_cyt.pca.5_6.gender

untreated.nuc_vs_cyt.pcaData.7_8.gender = plotPCA.san(untreated.nuc_vs_cyt$vsd, intgroup=c("condition", "sample", "cellline", "fraction", "mutation"), PC_x = 7, PC_y = 8, returnData=TRUE) %>% as_tibble() %>% mutate(sample_label = case_when(cellline == "glia" ~ "VCP1^R191Q", cellline == "glib" ~ "VCP2^R191Q", cellline == "cb1d" ~ "VCP1^R155C", cellline == "cb1e" ~ "VCP2^R155C", cellline == "ncrmc2" ~ "VCPki1^R191Q", cellline == "ncrme6" ~ "VCPki2^R191Q", cellline == "tdpn1" ~ "TARDBP1^G298S", cellline == "tdpn2" ~ "TARDBP2^G298S", TRUE ~ as.character(cellline)), 
         gender = case_when(cellline %in% c("ctrl3","ctrl4","cb1e","cb1d") ~ "Female", TRUE~"Male"), sample_label = gsub("ctrl","CTRL",sample_label), fraction = gsub("nucleus","Nuclear Samples",fraction), fraction = gsub("cytoplasm","Cytoplasm Samples",fraction))
untreated.nuc_vs_cyt.pca.7_8.gender <- ggplot(untreated.nuc_vs_cyt.pcaData.7_8.gender, aes(PCX, PCY, color=gender)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=as.character(sample_label)), parse = TRUE, size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_blank(), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) +
    xlab(paste0("PC7: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.7_8.gender, "percentVar"))[1],"% variance")) +  ylab(paste0("PC8: ",round(100 * attr(untreated.nuc_vs_cyt.pcaData.7_8.gender, "percentVar"))[2],"% variance"))
untreated.nuc_vs_cyt.pca.7_8.gender

untreated.nuc_vs_cyt.pca.gender.PC1_PC8 = plot_grid(untreated.nuc_vs_cyt.pca.gender, untreated.nuc_vs_cyt.pca.3_4.gender, untreated.nuc_vs_cyt.pca.5_6.gender, untreated.nuc_vs_cyt.pca.7_8.gender,
          nrow = 2, ncol = 2)
untreated.nuc_vs_cyt.pca.gender.PC1_PC8
ggsave(untreated.nuc_vs_cyt.pca.gender.PC1_PC8, filename = here(proj_path, "expression/deseq2/untreated.nuc_vs_cyt.pca.gender.PC1_PC8.png"), height = 5, width = 9)


```




## Merge

```{r}
pca_characterisation.merged <- plot_grid(
  plot_grid(NULL, fractionation_schematic.gg, labels = c("a",""), rel_widths = c(0.02,1)),
  plot_grid(NULL, ipsn_mn_markers.gg, labels = c("b",""), rel_widths = c(0.02,1)),
  plot_grid(NULL, celltype.markers.rnaseq.cluster.heatmap, rseqc_read_distribution.summarised.barplot, ncol = 3, rel_widths = c(0.05,1,1), labels = c("c", "", "d")),
  plot_grid(NULL, mn_price.nuc.cyt.marker.heatmap, untreated.nuc_vs_cyt.price_markers.density, ncol = 3, rel_widths = c(0.05,1,1), labels = c("e", "", "f")),
    plot_grid(NULL, plot_grid(MALAT1.nuc_cyt_coverage_plot.gg, GAPDH.nuc_cyt_coverage_plot.gg, nrow = 2), untreated.nuc_vs_cyt.pca, ncol = 3, rel_widths = c(0.04,0.7, 1.3), labels = c("g","","h")),
  nrow = 5, rel_heights = c(0.6,1,1.7,1.1,1))
# pca_characterisation.merged
ggsave(pca_characterisation.merged, filename = here(proj_path, "expression/deseq2/Figure1.pca_characterisation.merged.png"), height = 11, width = 9)
ggsave(pca_characterisation.merged, filename = here(proj_path, "expression/deseq2/Figure1.pca_characterisation.merged.pdf"), height = 11, width = 9)

```


# Fig S1 Coverage tracks nuclear & cytoplasm transcripts

Coverage tracks of known nuclear & cytoplasm markers - run on CAMP

```{r}
library(here)
proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
script_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts/ipsc_mn_als_meta")
shared_path = here("/nemo/project/proj-luscombn-patani/working")
collab_path = here("/nemo/lab/patanir/home/users/ziffo/patani-collab")
source(here(nemo_path,"scripts/functions/R_workspace.R"))
# load(here(proj_path,"scripts/untreated_deseq_objects.RData"))

ctrl_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_nuclear.bam") # CTRL3 removed
ctrl_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_cytoplasm.bam") # CTRL3 removed to balance reads
tardbp_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_nuclear.bam")
tardbp_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_cytoplasm.bam")
vcp_r155c_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_nuclear.bam")
vcp_r155c_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_cytoplasm.bam")
vcp_r191q_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_nuclear.bam")
vcp_r191q_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_cytoplasm.bam")
vcp_iso_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_nuclear.bam")
vcp_iso_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_cytoplasm.bam")

library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
txdb <- makeTxDbFromGFF(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf"), format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations & scaffolds which are also in the BAM

track_viewer_nuc_cyt(level = "gene", gene_name = "MALAT1", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "GAPDH", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "LINC00599", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "NPEPL1", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "MAN2C1", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "GFAP", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "TUBB", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "KRT19", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)
track_viewer_nuc_cyt(level = "gene", gene_name = "HIF1A", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)

# Merge into a single figure:

# nuclear markers
MALAT1.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/MALAT1.5_conditions.nuc_cyt_coverage_plot.png"))
MALAT1.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(MALAT1.nuc_cyt_coverage_plot.png)
# MALAT1.nuc_cyt_coverage_plot.gg

LINC00599.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/LINC00599.5_conditions.nuc_cyt_coverage_plot.png"))
LINC00599.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(LINC00599.nuc_cyt_coverage_plot.png)
# LINC00599.nuc_cyt_coverage_plot.gg

NPEPL1.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/NPEPL1.5_conditions.nuc_cyt_coverage_plot.png"))
NPEPL1.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(NPEPL1.nuc_cyt_coverage_plot.png)
# NPEPL1.nuc_cyt_coverage_plot.gg

MAN2C1.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/MAN2C1.5_conditions.nuc_cyt_coverage_plot.png"))
MAN2C1.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(MAN2C1.nuc_cyt_coverage_plot.png)
# MAN2C1.nuc_cyt_coverage_plot.gg

# cytoplasm markers 
GAPDH.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/GAPDH.5_conditions.nuc_cyt_coverage_plot.png"))
GAPDH.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(GAPDH.nuc_cyt_coverage_plot.png)
# GAPDH.nuc_cyt_coverage_plot.gg

GFAP.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/GFAP.5_conditions.nuc_cyt_coverage_plot.png"))
GFAP.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(GFAP.nuc_cyt_coverage_plot.png)
# GFAP.nuc_cyt_coverage_plot.gg

TUBB.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/TUBB.5_conditions.nuc_cyt_coverage_plot.png"))
TUBB.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(TUBB.nuc_cyt_coverage_plot.png)
# TUBB.nuc_cyt_coverage_plot.gg

HIF1A.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/HIF1A.5_conditions.nuc_cyt_coverage_plot.png"))
HIF1A.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(HIF1A.nuc_cyt_coverage_plot.png)
# HIF1A.nuc_cyt_coverage_plot.gg

KRT19.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/KRT19.5_conditions.nuc_cyt_coverage_plot.png"))
KRT19.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(KRT19.nuc_cyt_coverage_plot.png)
# KRT19.nuc_cyt_coverage_plot.gg

# CENPA.nuc_cyt_coverage_plot.png = magick::image_read(here(proj_path,"expression/coverage/CENPA.nuc_cyt_coverage_plot.png"))
# CENPA.nuc_cyt_coverage_plot.gg = ggdraw() + draw_image(CENPA.nuc_cyt_coverage_plot.png)
# CENPA.nuc_cyt_coverage_plot.gg


nuc_cyt_coverage_plot.merged = plot_grid(
  MALAT1.nuc_cyt_coverage_plot.gg, LINC00599.nuc_cyt_coverage_plot.gg, NPEPL1.nuc_cyt_coverage_plot.gg, MAN2C1.nuc_cyt_coverage_plot.gg, GAPDH.nuc_cyt_coverage_plot.gg, TUBB.nuc_cyt_coverage_plot.gg, HIF1A.nuc_cyt_coverage_plot.gg, KRT19.nuc_cyt_coverage_plot.gg, ncol = 2, nrow = 4, labels = "auto")
# nuc_cyt_coverage_plot.merged
ggsave(nuc_cyt_coverage_plot.merged, filename = here(proj_path, "expression/coverage/nuc_cyt_coverage_plot.merged.png"), height = 11.75, width = 8.25)


# histone_transcripts = readRDS(file = here(nemo_path, "genomes/ensembl/histone.interpro.genes.rds"))
# histone_transcripts
# 
# untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% histone_transcripts$ensembl_transcript_id) %>% arrange(baseMean) %>% print(n = Inf)
# untreated.nuc_vs_cyt$res %>% filter(gene_id %in% histone_transcripts$ensembl_gene_id) %>% arrange(baseMean) %>% print(n = Inf)

```


# Fig S2 PC1 & PC2 gene loadings

https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
https://github.com/mikelove/DESeq2/blob/master/R/plots.R

```{r}
untreated.vsd.rv <- rowVars(assay(untreated.nuc_vs_cyt$vsd)) # calculate the variance for each gene
untreated.vsd.ntop_genes <- order(untreated.vsd.rv, decreasing=TRUE)[seq_len(min(500, length(untreated.vsd.rv)))] # select the ntop genes by variance
untreated.vsd.pca <- prcomp(t(assay(untreated.nuc_vs_cyt$vsd)[untreated.vsd.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
untreated.vsd.loadings <- as_tibble(untreated.vsd.pca$rotation, rownames = "gene_id") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
untreated.vsd.loadings %>% arrange(PC1)
untreated.vsd.loadings %>% arrange(-PC1)

# scatter all PC1 vs PC2
untreated.vsd.loadings.scatter = ggplot(untreated.vsd.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    theme_oz() +
  geom_text_repel(data = filter(untreated.vsd.loadings, gene_name %in% price_2020_nuclear_genes, PC1 < -0), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", max.overlaps = 15, size = 2.3) +
    geom_text_repel(data = filter(untreated.vsd.loadings, gene_name %in% price_2020_cyto_genes & PC1  > 0), aes(x = PC1, y = PC2, label = gene_name), colour = "dodgerblue", max.overlaps = 15, size = 2.3) +
  # geom_text_repel(data = filter(untreated.vsd.loadings,  gene_name %in% c(nuc_cyt_markers.annot$gene_name)), aes(x = PC1, y = PC2, label = gene_name), colour = "black", size = 2.3) +
  annotate("text", x = 0.06, y = 0.12, label = "PC1+ PC2+",size = 2.8, color = "black") + annotate("text", x = -0.08, y = -0.12, label = "PC1- PC2-", size = 2.8, color = "black") + 
  annotate("text", x = 0.06, y = -0.12, label = "PC1+ PC2-", size = 2.8, color = "black") + annotate("text", x = -0.08, y = 0.12, label = "PC1- PC2+", size = 2.8, color = "black")
untreated.vsd.loadings.scatter
ggsave(untreated.vsd.loadings.scatter, filename = here(proj_path,"expression/deseq2/untreated.vsd.loadings.scatter.png"), height = 9, width = 9)
# drive_upload(media = here(proj_path,"expression/deseq2/untreated.vsd.loadings.scatter.png"), path = "mRNA redistribution ALS/figS1.vsd.loadings.png", overwrite = TRUE)

```


# Fig S3 Protein nucleocytoplasmic QC

## Western blot

```{r}
western_blot.png =  magick::image_read(here(proj_path,"expression/jacob_western_fractionation_untreated.png"))
western_blot.gg = ggdraw() + draw_image(western_blot.png)
# western_blot.gg

```

## Mass spec intensities

cytoplasmic (ALDOA, STMN1, GAPDH, TUBB3) & nuclear (LMNB1, HIST2H3A, HIST1H4A, HIST2H3A, SUGP2)

```{r}
sample_submission = read_excel(here(proj_path, "mass-spec/Nuc_Cyto_Sample_submission_5.4.22.xlsx"), skip = 1) %>% clean_names() %>% mutate(sample_name = gsub(" ","",tolower(sample_decoded)), cellline = str_split_fixed(sample_name,"_",3)[,1], treatment = str_split_fixed(sample_name,"_",3)[,2], fraction = str_split_fixed(sample_name,"_",3)[,3], mutation = case_when(mutation == "Wildtype" ~ "ctrl", TRUE ~ tolower(mutation))) %>%
  filter(!sample_name %in% c("ctrl6_ml240_nuclear", "glib_ml240_nuclear", "tdpn1_ml240_nuclear", "tdpn1_ml240_cytoplasm", "ctrl3_untreated_nuclear")) %>% # remove samples that failed qc
  select(sample_number, sample_name, cellline, treatment, fraction, mutation)
sample_replicates = bind_rows(mutate(sample_submission, technical_replicate = 1), mutate(sample_submission, technical_replicate = 2), mutate(sample_submission, technical_replicate = 3)) %>%
  mutate(lfq_id = paste0("lfq_intensity_", sample_number, "_", technical_replicate), lfq_name = gsub(" ", "", paste0("lfq_intensity_", sample_name, "_", technical_replicate)))

# MaxQuant LFQ intentisities
lfq_intensities = read_excel(here(proj_path, "mass-spec/IB2652_02092022.xlsx")) %>% clean_names() %>%
  select(!starts_with("lfq_intensity_2_")) %>% # remove untreated ctrl3 nuclear
  filter(reverse != "+", potential_contaminant != "+") %>%
  rename_at(vars(starts_with("lfq_intensity")), ~ str_replace_all(., setNames(sample_replicates$lfq_name, sample_replicates$lfq_id))) %>% 
  select(-lfq_intensity_glia_untreated_nuclear_1, -lfq_intensity_glia_untreated_nuclear_2, -lfq_intensity_glia_untreated_nuclear_3)
lfq_intensities.unique <- make_unique(lfq_intensities, "gene_names", "protein_i_ds", delim = ";")
sample_replicates.untreated.nuc_vs_cyt = sample_replicates %>% filter(treatment == "untreated") %>% group_by(cellline, technical_replicate) %>% mutate(replicate = cur_group_id()) %>% ungroup %>%
  select(label = lfq_name, condition = fraction, replicate, technical_replicate, cellline, treatment, fraction, mutation) %>%
  filter(!label %in% c("lfq_intensity_glia_untreated_nuclear_1", "lfq_intensity_glia_untreated_nuclear_2", "lfq_intensity_glia_untreated_nuclear_3"))
lfq_intensities.unique.untreated.nuc_vs_cyt = lfq_intensities.unique %>% select(!matches("ml240"))
DEP.se = make_se(lfq_intensities.unique.untreated.nuc_vs_cyt, grep("lfq_", colnames(lfq_intensities.unique.untreated.nuc_vs_cyt)), sample_replicates.untreated.nuc_vs_cyt)
DEP.filt = filter_missval(DEP.se, thr = 8)
DEP.norm = DEP.filt %>% normalize_vsn() 
DEP.res = DEP.norm %>% impute(., fun = "MinProb", q = 0.01) %>% test_diff(., type = "control", control = "cytoplasm", design_formula = ~0 + condition + technical_replicate + cellline) 
mass_spec_intensities_cytoplasmic = plot_single_oz(DEP.res, proteins = c("ALDOA", "STMN1", "GAPDH", "TUBB3"), type = "centered", facet_nrow = 1, facet_ncol = 4)
mass_spec_intensities_nuclear = plot_single_oz(DEP.res, proteins = c("LMNB1", "HIST2H3A", "HIST1H4A", "SUGP2"), type = "centered", facet_nrow = 1, facet_ncol = 4)
mass_spec_intensities.nuc_cyt_markers = plot_grid(mass_spec_intensities_cytoplasmic, mass_spec_intensities_nuclear, nrow = 2)
mass_spec_intensities.nuc_cyt_markers

```

## Proteomics PCA

https://bioconductor.org/packages/devel/bioc/vignettes/DEP/inst/doc/DEP.html

Process on CAMP

```{r}
load(here(proj_path, "scripts/dep_mass_spec_objects.RData"))

untreated.nuc_vs_cyt.dep.pca.plot = DEP::plot_pca(untreated.nuc_vs_cyt.dep$dep, point_size = 1, indicate = "condition") + scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold2")) +
    theme_oz() + theme(panel.grid = element_blank(), axis.text=element_text(size=8), axis.title=element_text(size=10), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
    guides(color=guide_legend(title="", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) + ggtitle("") +
    ggforce::geom_mark_ellipse(aes(color = condition), show.legend = FALSE) + 
  geom_text_repel(aes(label=toupper(colData(untreated.nuc_vs_cyt.dep$dep)$cellline)), size = 2.4, max.overlaps = 15) + coord_equal(ratio = 0.35)
untreated.nuc_vs_cyt.dep.pca.plot
# ggsave(untreated.nuc_vs_cyt.dep.pca.plot, filename = here(proj_path, "mass-spec/figures/untreated.nuc_vs_cyt.dep.pca.png"), height = 4, width = 10)

## PC1 & PC2 gene loadings
# https://stackoverflow.com/questions/12760108/principal-components-analysis-how-to-get-the-contribution-of-each-paramete
# https://github.com/mikelove/DESeq2/blob/master/R/plots.R
untreated.nuc_vs_cyt.dep.rv <- rowVars(assay(untreated.nuc_vs_cyt.dep$dep)) # calculate the variance for each gene
untreated.nuc_vs_cyt.dep.ntop_genes <- order(untreated.nuc_vs_cyt.dep.rv, decreasing=TRUE)[seq_len(min(500, length(untreated.nuc_vs_cyt.dep.rv)))] # select the ntop genes by variance
untreated.nuc_vs_cyt.dep.pca <- prcomp(t(assay(untreated.nuc_vs_cyt.dep$dep)[untreated.nuc_vs_cyt.dep.ntop_genes,])) # perform a PCA on the data in assay(x) for the selected genes
untreated.nuc_vs_cyt.dep.loadings <- as_tibble(untreated.nuc_vs_cyt.dep.pca$rotation, rownames = "gene_name") %>% left_join(gene2ens) %>% arrange(PC1) %>% select(gene_name, gene_id, everything())
untreated.nuc_vs_cyt.dep.loadings %>% arrange(PC1)
untreated.nuc_vs_cyt.dep.loadings %>% arrange(-PC1)

# scatter all PC1 vs PC2
untreated.nuc_vs_cyt.dep.loadings.scatter = ggplot(untreated.nuc_vs_cyt.dep.loadings, aes(x = PC1, y = PC2 )) + 
    labs(x = "PC1 gene loading", y = "PC2 gene loading") +
    geom_hline(yintercept = 0, linetype = 3) + geom_vline(xintercept = 0, linetype = 3) +
    geom_point(size = 1, alpha = 0.1) +
    theme_oz() +
        geom_text_repel(data = filter(untreated.nuc_vs_cyt.dep.loadings, grepl("^HIST",gene_name) | gene_name %in% c(bahar_halpern.nuclear_cytoplasmic.markers, price_2020_nuclear_genes), PC1 < -0.02), aes(x = PC1, y = PC2, label = gene_name), colour = "firebrick2", max.overlaps = 15, size = 2.3) +
        geom_text_repel(data = filter(untreated.nuc_vs_cyt.dep.loadings, grepl("^TUBB|GAPDH",gene_name) | (gene_name %in% c(bahar_halpern.nuclear_cytoplasmic.markers, price_2020_cyto_genes) & PC1 > 0.05)), aes(x = PC1, y = PC2, label = gene_name), colour = "dodgerblue", max.overlaps = 15, size = 2.3) +
  annotate("text", x = 0.06, y = 0.12, label = "PC1+ PC2+",size = 2.8, color = "black") + annotate("text", x = -0.08, y = -0.14, label = "PC1- PC2-", size = 2.8, color = "black") + 
  annotate("text", x = 0.06, y = -0.14, label = "PC1+ PC2-", size = 2.8, color = "black") + annotate("text", x = -0.08, y = 0.12, label = "PC1- PC2+", size = 2.8, color = "black")
# untreated.nuc_vs_cyt.dep.loadings.scatter
# ggsave(untreated.nuc_vs_cyt.dep.loadings.scatter, filename = here(proj_path,"mass-spec/figures/untreated.nuc_vs_cyt.dep.loadings.scatter.png"), height = 9, width = 9)



```

## Merge

```{r}
untreated.nuc_vs_cyt.dep.pca.merged = plot_grid(
  western_blot.gg, 
  mass_spec_intensities.nuc_cyt_markers,
  untreated.nuc_vs_cyt.dep.pca.plot, 
  untreated.nuc_vs_cyt.dep.loadings.scatter, 
  nrow = 4, labels = "auto", rel_heights = c(0.7,1.5,1,1))
# untreated.nuc_vs_cyt.dep.pca.merged
ggsave(untreated.nuc_vs_cyt.dep.pca.merged, filename = here(proj_path,"mass-spec/figures/untreated.nuc_vs_cyt.dep.pca.merged.png"), height = 11.5, width = 9)

```




# Fig 2 ALS vs CTRL & Nuc vs Cyt

## TARDBP Volcano  & GO

multi-factor design

```{r}
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) # 381
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 153
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 228
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 8
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) # 0
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 46
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) # MATR3, ELP3, HNRNPA2B1
# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 10 
# nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))

# transcript level volcano
untreated.nuc_vs_cyt.tardbp_vs_ctrl.rbp.labels <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.als.labels <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.volcano <- volcano_plot(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, gene_labels = c(untreated.nuc_vs_cyt.tardbp_vs_ctrl.als.labels, untreated.nuc_vs_cyt.tardbp_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "TARDBP mutant") + 
  xlab(expression(TARDBP~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 15*0.88, yend= 15 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 15*0.88, yend= 15*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 15*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 15*0.93, label = "Cytoplasm", size = 3)
# untreated.nuc_vs_cyt.tardbp_vs_ctrl.volcano

### GO mislocalised
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt <- untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        84        32 
untreated.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_nuclear <- untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
untreated.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_cytoplasm <- untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
# "nucleoplasm",
"microtubule organizing center",
"intracellular anatomical structure",
# "regulation of cellular component organization",
# "centrosome",
"axon cytoplasm",
# "transcription coactivator activity",
"protein modification process",
# "peptidyl-amino acid modification",
"transcription coregulator activity",
"neuron projection cytoplasm",
"cellular localization",
"protein metabolic process",
"protein binding"
# "cytoplasm"
)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_nuclear <- untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
# "transcription factor binding",
# "regulation of cell death",
# "ion binding",
# "phosphorylation",
# "protein phosphorylation",
# "binding",
# "RNA biosynthetic process",
# "nucleus",
"protein metabolic process",
"histone deacetylase complex",
"cellular response to stress",
# "Chromatin modifying enzymes",
"Chromatin organization",
"nervous system development",
# "regulation of gene expression",
"nuclear speck",
"regulation of RNA metabolic process",
"protein modification process",
# "organelle lumen",
"intracellular anatomical structure",
# "nuclear lumen",
# "cytoplasm",
# "nucleoplasm",
"protein binding")
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm <- untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
untreated.nuc_vs_cyt.tardbp_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
untreated.nuc_vs_cyt.tardbp_vs_ctrl.go_curated

# untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% mutate(log10pvalue = -log10(pvalue), log10baseMean = log10(baseMean)) %>% ggscatter(x = "log10baseMean", y = "log10pvalue", size = 0.5, cor.coef = TRUE) 

```


## VCP R155c volcano & GO

multi-factor design

```{r}
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res %>% filter(padj < 0.05) # 11
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) # 358
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 181
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 177
# # untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj > 0.05) # 1017
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 35
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) #7
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) #0
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) #FUS
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 5 


# transcript level volcano
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.rbp.labels <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.als.labels <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano <- volcano_plot(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, gene_labels = c(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.als.labels, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "VCP R155C mutant") + 
  xlab(expression(VCP~R155C~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 15*0.88, yend= 15*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 15*0.88, yend= 15*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 15*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 15*0.93, label = "Cytoplasm", size = 3)
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano
# # density plot
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.density = density_plot(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, xlab_prefix = "", ymax = NULL, xmax = 5, xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.density

### GO mislocalised
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        11         7 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_nuclear <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_cytoplasm <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cell junction organization",
"synapse",
"cell junction",
"postsynapse",
"cytosol",
"cytoplasm",
"protein binding"
)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_nuclear <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"intracellular anatomical structure",
# "cytoplasm",
"ion binding",
"positive regulation of double-strand break repair",
"peptidyl-amino acid modification",
"cellular macromolecule metabolic process",
"RNA polymerase",
"organelle organization",
"cell cortex",
"catalytic activity, acting on a nucleic acid")
# "cytosol")
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("cellular ","",term_name), term_name = gsub("acting on a ","",term_name), term_name = gsub("peptidyl-","",term_name))
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated

```


## VCP R191Q volcano & GO

multi-factor design

```{r}
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res %>% filter(padj < 0.05) # 11
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) # 602
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 181
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 421
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 59
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) #9
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) #C9orf72
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) #WSR1
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 10 

# transcript level volcano
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.rbp.labels <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.als.labels <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano <- volcano_plot(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, gene_labels = c(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.als.labels, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q mutant") + 
  xlab(expression(VCP~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 15*0.88, yend= 15*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 15*0.88, yend= 15*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 15*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 15*0.93, label = "Cytoplasm", size = 3)
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano

# # density plot
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.density = density_plot(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, xlab_prefix = "", ymax = NULL, xmax = 5, xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.density

### GO mislocalised
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        37        14 
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_nuclear <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_cytoplasm <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
# "ESYT1-ESYT2 complex",
"extrinsic component of presynaptic membrane",
# "organelle organization",
"Focal adhesion",
# "EGFR tyrosine kinase inhibitor resistance",
"intracellular anatomical structure",
# "extrinsic component of presynaptic endocytic zone membrane",
"extrinsic component of synaptic membrane",
"cellular localization",
"protein binding",
"intracellular transport",
"Cilium Assembly"
# "cytosol",
# "cytoplasm"
)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear %<>% mutate(term_name = gsub("extrinsic component of ", "", term_name), term_name = gsub("cellular ","",term_name), term_name = gsub("EGFR ","",term_name))
paste('"',unique(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"actin cytoskeleton organization",
"Ubiquitin E3 ligase (FBXW7, CUL1, SKP1A, RBX1)",
# "regulation of organelle organization",
"translation factor activity, RNA binding",
# "intracellular membrane-bounded organelle",
# "nBAF complex",
# "positive regulation of cellular component organization",
"neuron projection",
# "regulation of molecular function",
"nervous system development",
# "regulation of cellular process",
# "catalytic activity, acting on a nucleic acid",
# "organic substance biosynthetic process",
# "nuclear lumen",
# "regulation of supramolecular fiber organization",
# "biosynthetic process",
# "regulation of cellular component biogenesis",
# "enzyme regulator activity",
# "cellular process",
# "cellular biosynthetic process",
# "cell cortex",
# "enzyme binding",
# "organonitrogen compound metabolic process",
# "GTPase regulator activity",
# "nucleoside-triphosphatase regulator activity",
"intracellular anatomical structure",
"protein metabolic process",
"regulation of cellular component organization",
"cellular protein metabolic process",
# "organelle organization",
# "nucleoplasm",
"catalytic activity, acting on DNA",
"protein binding")
# "cellular macromolecule metabolic process",
# "cytosol",
# "cytoplasm")
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("acting on a ","",term_name), term_name = gsub(" \\(FBXW7, CUL1, SKP1A, RBX1\\)","",term_name), term_name=gsub("translation factor activity, ","",term_name))
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated

```


## VCP R191Q knock-in volcano & GO

multi-factor design

```{r}
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) # 569
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 190
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 379
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 76
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) # 0
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) # 0
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 7
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 12 

# transcript level volcano
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.rbp.labels <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.als.labels <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano <- volcano_plot(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, gene_labels = c(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.als.labels, untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q knock-in") + 
  xlab(expression(VCPki~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 15*0.88, yend= 15 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 15*0.88, yend= 15*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 15*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 15*0.93, label = "Cytoplasm", size = 3)
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano

# # violin plot
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% mutate(condition = "Nuc/Cyt vcp_iso vs CTRL")
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.violin = violin_plot(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", ylims = c(-5,5), null_value = 0, one_sample = TRUE, stats_test = "wilcox_test", stat_ypos = 4.5, title = NULL, subtitle = NULL, numerator = NULL, denomenator = NULL, arrow_labels = FALSE, dpi = 300) +
#   ylab(expression(vcp_iso~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) + theme(axis.text.x = element_blank(), axis.title.x = element_blank()) +#guides(colour = "none") +
#       annotation_custom(grid::textGrob(label = "Nuclear", x = unit(0.8, "npc"), y = unit(0.8, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 90)) + 
#       annotation_custom(grid::textGrob(label = "Cytoplasm", x = unit(0.8, "npc"), y = unit(0.2, "npc"), gp = grid::gpar(fontsize = 8, fontface=1), rot = 90)) + 
#       annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), y = c(0.95,0.65), x = unit(0.97, "npc"))) +
#       annotation_custom(grid::linesGrob(arrow=arrow(type="open", ends="first", length=unit(3,"mm")), gp=gpar(col="black", lwd=1), y = c(0.05,0.35), x = unit(0.97, "npc"))) 
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.violin
# [1] "Using one sample wilcox_test"
# # A tibble: 1 × 8
#   group               .y.   group1 group2          n   statistic     p p.signif
#   <chr>               <chr> <chr>  <chr>       <int>       <dbl> <dbl> <chr>   
# 1 Nuc/Cyt vcp_iso vs CTRL lfc   1      null model 159286 7537567416.     0 ****    
# # A tibble: 1 × 7
#   .y.   group1 group2     effsize group                    n magnitude 
# * <chr> <chr>  <chr>        <dbl> <chr>                <int> <ord>     
# 1 lfc   1      null model   0.104 Nuc/Cyt vcp_iso vs CTRL 227818 

# # density plot
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.density = density_plot(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, xlab_prefix = "", ymax = NULL, xmax = 5, xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.density


### GO mislocalised
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#       181        85 
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_nuclear <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_cytoplasm <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
# "TSC1-TSC2 complex",
# "purine-containing compound metabolic process",
# "enoyl-CoA hydratase activity",
"identical protein binding",
# "chaperone complex",
# "transferase activity, transferring phosphorus-containing groups",
# "neuron to neuron synapse",
"postsynaptic specialization",
# "catabolic process",
"protein C-terminus binding",
"glutamatergic synapse",
# "system development",
# "cell differentiation",
# "cellular catabolic process",
"intracellular signal transduction",
# "cellular developmental process",
"organelle organization",
# "organonitrogen compound metabolic process",
# "cell population proliferation",
# "nucleoplasm",
# "developmental process",
"cell junction",
"enzyme binding",
"Hsp90 protein binding",
# "postsynapse",
# "phosphorus metabolic process",
"synapse")
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_nuclear <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
# "cytoplasmic translational initiation",
"protein metabolic process",
"cytoplasmic ribonucleoprotein granule",
# "cell projection",
# "regulation of cellular component organization",
# "endomembrane system",
# "regulation of synapse structure or activity",
"chromosome organization",
"nucleosome binding",
"nervous system development",
"chromatin organization",
"cellular protein metabolic process",
"synapse",
# "neuron projection",
"transcription coregulator activity",
"axon",
# "organelle organization",
"cytoplasmic stress granule",
# "intracellular organelle",
"protein binding")
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("cellular ", "",term_name), term_name=gsub(" activity","",term_name))
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated

```


## Combine ALS mutants

```{r}
untreated.nuc_vs_cyt.als_vs_ctrl = readRDS(file = here(proj_path, "scripts/untreated.nuc_vs_cyt.als_vs_ctrl.rds"))

# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05) # 330
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 134
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 196
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 7
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) # CFAP410
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 36
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) # 0
# untreated.nuc_vs_cyt.als_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) #8
# nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))

# transcript level volcano
untreated.nuc_vs_cyt.als_vs_ctrl.rbp.labels <- untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.als_vs_ctrl.als.labels <- untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.nuc_vs_cyt.als_vs_ctrl.volcano <- volcano_plot(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, gene_labels = c(untreated.nuc_vs_cyt.als_vs_ctrl.als.labels, untreated.nuc_vs_cyt.als_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 25, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "Mutants combined") + 
  xlab(expression(ALS~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 25*0.25, xend = 25*0.95, y= 15*0.88, yend= 15 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -25*0.25, xend = -25*0.95, y= 15*0.88, yend= 15*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 35*0.6, y = 15*0.93, label = "Nuclear", size = 3) + annotate("text", x = -25*0.6, y = 15*0.93, label = "Cytoplasm", size = 3)
# untreated.nuc_vs_cyt.als_vs_ctrl.volcano

# # # # density plot
# untreated.nuc_vs_cyt.als_vs_ctrl.density = density_plot(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, ymax = 1, xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# untreated.nuc_vs_cyt.als_vs_ctrl.density

### GO mislocalised
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost <- untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt <- untreated.nuc_vs_cyt.als_vs_ctrl.res.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        19        13 
untreated.nuc_vs_cyt.als_vs_ctrl.gprofiles_nuclear <- untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
untreated.nuc_vs_cyt.als_vs_ctrl.gprofiles_cytoplasm <- untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.nuc_vs_cyt.als_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"regulation of response to stimulus",
"intracellular signal transduction",
"cytoplasm",
"intracellular anatomical structure",
"protein binding"
)
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_nuclear <- untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.nuc_vs_cyt.als_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"intracellular organelle lumen",
"ion binding",
"regulation of cellular response to stress",
"nucleoplasm",
"cytoplasm")
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_cytoplasm <- untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
untreated.nuc_vs_cyt.als_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.nuc_vs_cyt.als_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.nuc_vs_cyt.als_vs_ctrl.go_curated

```


## Correlation heatmap

```{r}
# heatmap of correlation coefficients
untreated.nuc_vs_cyt.als_vs_ctrl_stat = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, `TARDBP^G298S` = stat) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, `VCP^R155C` = stat)) %>%
  left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, `VCP^R191Q` = stat)) %>%
  left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, `VCPki^R191Q` = stat)) %>%
  # left_join(select(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, transcript_id, `Combined` = stat)) %>%
  drop_na() %>%
  column_to_rownames("transcript_id")
cormat <- round(cor(untreated.nuc_vs_cyt.als_vs_ctrl_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# reorder_cormat <- function(cormat){ # Use correlation between variables as distance
#   dd <- as.dist((1-cormat)/2)
#   hc <- hclust(dd)
#   cormat <-cormat[hc$order, hc$order]
# }
# cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
  # mutate(Var1 = fct_relevel(Var1, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"), Var2 = fct_relevel(Var2, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))
untreated.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
untreated.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap

```


## Upset plot

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCP R191Q knock-in"))

untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset = upset_plot(filter(untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx, stat > 0), overlap_by = "mutation", overlap_level = "transcript_id", split_direction = FALSE, ymax = 200, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Nuclear redistributed", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset

untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset = upset_plot(filter(untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx, stat < 0), overlap_by = "mutation", overlap_level = "transcript_id", split_direction = FALSE, ymax = 340, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Cytoplasm redistributed", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset

# Fisher exact overlap test
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)

newGeneOverlap(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up, genome.size=nrow(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=181
# Intersection size=20
# Overlapping p-value=1.7e-38
# Jaccard Index=0.1
newGeneOverlap(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down, genome.size=nrow(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=228
# listB size=177
# Intersection size=34
# Overlapping p-value=2.1e-67
# Jaccard Index=0.1

newGeneOverlap(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=181
# Intersection size=16
# Overlapping p-value=2.2e-29
# Jaccard Index=0.1
newGeneOverlap(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=228
# listB size=421
# Intersection size=53
# Overlapping p-value=1e-94
# Jaccard Index=0.1

newGeneOverlap(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=181
# listB size=181
# Intersection size=25
# Overlapping p-value=1.6e-48
# Jaccard Index=0.1
newGeneOverlap(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=177
# listB size=421
# Intersection size=50
# Overlapping p-value=3.3e-94
# Jaccard Index=0.1

newGeneOverlap(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.up, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=190
# listB size=181
# Intersection size=22
# Overlapping p-value=5.1e-41
# Jaccard Index=0.1
newGeneOverlap(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.down, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=379
# listB size=421
# Intersection size=65
# Overlapping p-value=1.1e-106
# Jaccard Index=0.1


# # # ggvenn
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes <- list("TARDBP" = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, "VCP" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes <- list("TARDBP" = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, "VCP" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn <- ggvenn(untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5 ,text_size = 4) + labs(caption = "Nuclear")  + theme(plot.caption = element_text(vjust = 4, hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn
# 
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn <- ggvenn(untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5,text_size = 4) + labs(caption = "Cytoplasm")  + theme(plot.caption = element_text(vjust = 4,hjust = 0.5, colour = "dodgerblue2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn

# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn = plot_grid(NULL, untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn, NULL, untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn, rel_heights =c(0.02,1,0.05,1), nrow = 4)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn

```


## NEFL FISH

```{r}
# representative image
# import with majick
nefl_fish_image.png = magick::image_read(here(proj_path,"polya-fish/NEFL/nefl_fish_image.png"))
nefl_fish_image.gg = ggdraw() + draw_image(nefl_fish_image.png)
# nefl_fish_image.gg


# Experiment 1 - low throughput
fish_nuclear_whole_mRNA_counts = read_excel(here(proj_path,"polya-fish/NEFL/ViewRNA D6 MN exp1 NEFL_TMEM259 results.xlsx"), sheet = "optimised") %>% clean_names() %>% 
  mutate(cytoplasm_nefl = whole_nefl_m_rna_count - nuclei_nefl_m_rna_count, cytoplasm_tmem259 = whole_tmem259_m_rna_count - nuclei_tmem259_m_rna_count, 
         nuc_cyt_nefl_intensity = nuclei_nefl_m_rna_count / cytoplasm_nefl, nuc_cyt_tmem259_intensity = nuclei_tmem259_m_rna_count / cytoplasm_tmem259,  
         nuc_cyt_nefl_lfc = log2(nuc_cyt_nefl_intensity), nuc_cyt_tmem259_lfc = log2(nuc_cyt_tmem259_intensity), cellline = case_when(grepl("CTRL1", file_name_biii) ~ "CTRL1", grepl("CTRL5", file_name_biii) ~ "CTRL5",  grepl("CTRL2", file_name_biii) ~ "CTRL2", grepl("CTRL3", file_name_biii) ~ "CTRL3", grepl("CTRL6", file_name_biii) ~ "CTRL6", grepl("CB1D", file_name_biii) ~ "CB1D", grepl("CB1D", file_name_biii) ~ "CB1D", grepl("CB1E", file_name_biii) ~ "CB1E", grepl("GliA", file_name_biii) ~ "GliA", grepl("NCRM C2", file_name_biii) ~ "NCRMC2", grepl("NCRM E6", file_name_biii) ~ "NCRME6", grepl("TDP1", file_name_biii) ~ "TDPN1", grepl("TDP2", file_name_biii) ~ "TDPN2"), 
                                                                           mutation = case_when(grepl("CTRL", cellline) ~ "CTRL", grepl("CB1", cellline) ~ "VCP\nR155C",  grepl("Gli", cellline) ~ "VCP\nR191Q",  grepl("NCRM", cellline) ~ "VCP\nR191Q ki",  grepl("TDP", cellline) ~ "TARDBP\nG298S"), condition = case_when(mutation == "CTRL" ~ "CTRL", grepl("VCP",mutation)~ "VCP", grepl("TARDBP",mutation)~ "TARDBP"))

fish_nuclear_whole_mRNA_counts
fish_nuclear_whole_mRNA_counts %>% glimpse
fish_nuclear_whole_mRNA_counts %>% count(mutation, cellline)
fish_nuclear_whole_mRNA_counts %>% my_summarise(nuclei_nefl_m_rna_count, mutation)
fish_nuclear_whole_mRNA_counts %>% my_summarise(cyt_mean_intensity, mutation)
fish_nuclear_whole_mRNA_counts %>% my_summarise(nuc_cyt_mean_intensity, mutation)
fish_nuclear_whole_mRNA_counts %>% my_summarise(nuc_cyt_mean_intensity, mutation, replicate)

fish_nuclear_whole_mRNA_counts %>% my_summarise(nuc_cyt_mean_intensity, condition)

# Nuclear / Cytoplasm intensity ratio
nuc_cyt_nefl_lfc.violin = violin_plot(fish_nuclear_whole_mRNA_counts, color_by = "mutation", continuous = "nuc_cyt_nefl_lfc", ylims = c(-3,4.2), null_value = 0, stat_ypos = c(3.7,0,4.2,rep(0,7)), ylabel = "NEFL FISH\nNuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif")
nuc_cyt_nefl_lfc.violin
# .y.   group1          group2             n1    n2 statistic         p    p.adj p.adj.signif y.position
#    <chr> <chr>           <chr>           <int> <int>     <dbl>     <dbl>    <dbl> <chr>             <dbl>
#  1 lfc   "CTRL"          "TARDBP\nG298S"   867   424   156854. 0.0000172 0.000172 ***                 3.7
#  2 lfc   "CTRL"          "VCP\nR155C"      867   419   168964. 0.042     0.293    ns                  0  
#  3 lfc   "CTRL"          "VCP\nR191Q"      867   189    70766. 0.003     0.029    *                   4.2
#  4 lfc   "CTRL"          "VCP\nR191Q ki"   867   152    63948  0.56      1        ns                  0  
# # A tibble: 10 × 7
#    .y.   group1          group2          effsize    n1    n2 magnitude 
#  * <chr> <chr>           <chr>             <dbl> <int> <int> <ord>     
#  1 lfc   "CTRL"          "TARDBP\nG298S" -0.275   1240   626 small     
#  2 lfc   "CTRL"          "VCP\nR155C"    -0.127   1240   529 negligible
#  3 lfc   "CTRL"          "VCP\nR191Q"    -0.244   1240   320 small     
#  4 lfc   "CTRL"          "VCP\nR191Q ki" -0.0551  1240   247 negligible

violin_plot(fish_nuclear_whole_mRNA_counts, color_by = "mutation", continuous = "nuclei_nefl_m_rna_count", ylims = c(0,20), stat_ypos = 20, ylabel = "NEFL FISH\nnuclear mRNA count", dpi = 300, stats_test = "wilcox_test", stat_label = "p.adj.signif", arrow_labels = FALSE)

violin_plot(fish_nuclear_whole_mRNA_counts, color_by = "mutation", continuous = "cytoplasm_nefl", ylims = c(0,20), stat_ypos = 20, ylabel = "NEFL FISH\ncytoplasm mRNA count", dpi = 300, stats_test = "wilcox_test", stat_label = "p.adj.signif", arrow_labels = FALSE)

# nuc_cyt_tmem259_lfc.violin = violin_plot(fish_nuclear_whole_mRNA_counts, color_by = "mutation", continuous = "nuc_cyt_tmem259_lfc", ylims = c(-3,3), null_value = 0, stat_ypos = c(3.7,0,4,rep(0,7)), ylabel = "TMEM259 Nuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif")
# nuc_cyt_tmem259_lfc.violin
# 
# nuc_cyt_tmem259_raw.violin = violin_plot(fish_nuclear_whole_mRNA_counts, color_by = "mutation", continuous = "nuc_cyt_tmem259_intensity", ylims = c(0,10), null_value = 0, stat_ypos = c(3.7,0,4,rep(0,7)), ylabel = "TMEM259 Nuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif")
# nuc_cyt_tmem259_raw.violin

# # Experiment 2 - high throughput. different induction.
# fish_nuclear_cytoplasm_mRNA_counts.exp2 = read_excel(here(proj_path,"polya-fish/NEFL/jacob_nefl_tmem259_fish_experiment2.xlsx"), sheet = "Objects_Population - 488 Pos Ce", skip = 10) %>% clean_names() %>% 
#   mutate(nefl_nuc_cyt_spots.lfc = log((nefl_nuc_spots)) - log((nefl_cyt_spots)),
#     #nefl_nuc_cyt_spots.lfc = log((nefl_nuc_spots + 0.01)) - log((nefl_cyt_spots + 0.01)), # add psuedocount to avoid NAs, log fold change
#          tmem259_nuc_cyt_spots.lfc = log((tmem259_nuc_spots + 0.01)) / log((tmem259_cyt_spots + 0.01)),
#          cellline = case_when(grepl("C2", cellline) ~ "CTRL2", grepl("C3", cellline) ~ "CTRL3",  grepl("C4", cellline) ~ "CTRL4", grepl("C5", cellline) ~ "CTRL5", grepl("C6", cellline) ~ "CTRL6", grepl("CB1D", cellline) ~ "CB1D", grepl("CB1E", cellline) ~ "CB1E", grepl("GliA", cellline) ~ "GliA", grepl("GliB", cellline) ~ "GliB", grepl("NCRM C2", cellline) ~ "NCRMC2", grepl("NCRM E6", cellline) ~ "NCRME6", grepl("VCP-F10", cellline) ~ "VCP-F10", grepl("TDP n1", cellline) ~ "TDPN1", grepl("TDP n2", cellline) ~ "TDPN2", TRUE ~ celline),
#          mutation = case_when(grepl("CTRL", cellline) ~ "CTRL", grepl("CB1", cellline) ~ "VCP\nR155C",  grepl("Gli", cellline) ~ "VCP\nR191Q",  grepl("NCRM", cellline) ~ "VCP\nR191Q ki",  grepl("TDP", cellline) ~ "TARDBP\nG298S", grepl("F10", cellline) ~ "VCP\nR155C corrected"), condition = case_when(mutation == "CTRL" ~ "CTRL", grepl("VCP",mutation)~ "VCP", grepl("TARDBP",mutation)~ "TARDBP")) %>%
#   filter(cellline != "VCP-F10")
# fish_nuclear_cytoplasm_mRNA_counts.exp2
# 
# fish_nuclear_cytoplasm_mRNA_counts.exp2 %>% count(cellline)
#  
# fish_nuclear_cytoplasm_mRNA_counts.nefl = fish_nuclear_cytoplasm_mRNA_counts.exp2 %>% filter(nefl_nuc_spots + nefl_cyt_spots > 0) %>% select(mutation, cellline, nefl_nuc_spots, nefl_cyt_spots, nefl_nuc_cyt_spots.lfc)
# 
# fish_nuclear_cytoplasm_mRNA_counts.nefl %>% glimpse
# fish_nuclear_cytoplasm_mRNA_counts.nefl %>% my_summarise(nefl_nuc_cyt_spots.lfc, mutation)
# 
# 
# # Nuclear / Cytoplasm intensity ratio
# nuc_cyt_nefl_lfc.violin = violin_plot(fish_nuclear_cytoplasm_mRNA_counts.nefl, color_by = "cellline", continuous = "nefl_nuc_cyt_spots.lfc", ylims = c(-2,3), null_value = 0, stat_ypos = c(3), ylabel = "NEFL FISH\nNuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif", cols = viridis_pal()(11))
# nuc_cyt_nefl_lfc.violin
# 
# 
# fish_nuclear_cytoplasm_mRNA_counts.exp2.stat = fish_nuclear_cytoplasm_mRNA_counts.exp2 %>% filter(nefl_nuc_spots + nefl_cyt_spots > 0) %>% 
#   select(mutation, nefl_nuc_spots, nefl_cyt_spots, nefl_nuc_cyt_spots.lfc, nuc_cyt_nefl_intensity, nuc_cyt_nefl_lfc)
# drop_na(fish_nuclear_cytoplasm_mRNA_counts.exp2, nuc_cyt_nefl_lfc) %>% wilcox_test(nuc_cyt_nefl_lfc ~ mutation) %>% add_significance() %>% add_xy_position(x = "mutation") %>% 
#   mutate(p.signif = case_when(group1=="CTRL"&p<0.001~"****", group1=="CTRL"&p<0.001~"***", group1=="CTRL"&p<0.01~"**", group1=="CTRL"&p<0.05~"*", TRUE ~ "ns")) %>% select(-contains("adj")) %>%
#   filter(group1 == "CTRL")
# fish_nuclear_cytoplasm_mRNA_counts.exp2.stat
#       
# nuc_cyt_nefl_lfc.violin = violin_plot(drop_na(fish_nuclear_cytoplasm_mRNA_counts.exp2, nuc_cyt_nefl_lfc), color_by = "mutation", continuous = "nuc_cyt_nefl_lfc", ylims = c(-3,4.2), null_value = 0, ylabel = "NEFL FISH\nNuclear/Cytoplasm LFC", dpi = 300, plot_stats = FALSE, numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95) +
#   stat_pvalue_manual(fish_nuclear_cytoplasm_mRNA_counts.exp2.stat, label = "p.signif", tip.length = .03, size = 4, hide.ns = TRUE, y.position = c(3.6,3.7,3.8))
# nuc_cyt_nefl_lfc.violin
# 
# 
# # # A tibble: 10 × 10
# #    .y.   group1          group2             n1    n2 statistic         p    p.adj p.adj.signif y.position
# #    <chr> <chr>           <chr>           <int> <int>     <dbl>     <dbl>    <dbl> <chr>             <dbl>
# #  1 lfc   "CTRL"          "TARDBP\nG298S"   867   424   156854. 0.0000172 0.000172 ***                 3.7
# #  2 lfc   "CTRL"          "VCP\nR155C"      867   419   168964. 0.042     0.293    ns                  0  
# #  3 lfc   "CTRL"          "VCP\nR191Q"      867   189    70766. 0.003     0.029    *                   4  
# #  4 lfc   "CTRL"          "VCP\nR191Q ki"   867   152    63948  0.56      1        ns                  0  
# #  5 lfc   "TARDBP\nG298S" "VCP\nR155C"      424   419    95724  0.051     0.304    ns                  0  
# #  6 lfc   "TARDBP\nG298S" "VCP\nR191Q"      424   189    40285  0.915     1        ns                  0  
# #  7 lfc   "TARDBP\nG298S" "VCP\nR191Q ki"   424   152    36386  0.018     0.14     ns                  0  
# #  8 lfc   "VCP\nR155C"    "VCP\nR191Q"      419   189    36945  0.185     0.74     ns                  0  
# #  9 lfc   "VCP\nR155C"    "VCP\nR191Q ki"   419   152    33310  0.399     1        ns                  0  
# # 10 lfc   "VCP\nR191Q"    "VCP\nR191Q ki"   189   152    16077  0.056     0.304    ns                  0  
# 
# 
# nuc_cyt_tmem259_lfc.violin = violin_plot(fish_nuclear_cytoplasm_mRNA_counts.exp2, color_by = "mutation", continuous = "nuc_cyt_tmem259_lfc", ylims = c(-2,4), null_value = 0, stat_ypos = c(3.7,0,4,rep(0,7)), ylabel = "TMEM259 Nuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif")
# nuc_cyt_tmem259_lfc.violin
# 
# nuc_cyt_tmem259_raw.violin = violin_plot(fish_nuclear_cytoplasm_mRNA_counts.exp2, color_by = "mutation", continuous = "nuc_cyt_tmem259_intensity", ylims = c(0,10), null_value = 0, stat_ypos = c(3.7,0,4,rep(0,7)), ylabel = "TMEM259 Nuclear/Cytoplasm LFC", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.95, stat_label = "p.adj.signif")
# nuc_cyt_tmem259_raw.violin
# 
# 
# fish_nuclear_whole_mRNA_counts %>% summarise(nuc_cyt_tmem259_lfc)
# fish_nuclear_whole_mRNA_counts %>% my_summarise(whole_tmem259_m_rna_count)
# whole_tmem259_m_rna_count
# 
# glm(data=nuc_cyt_mean_intensities, nuc_cyt_mean_intensity ~ mutation + replicate, family = gaussian)%>% summary()
# 
# # Coefficients:
# #                     Estimate Std. Error t value Pr(>|t|)    
# # (Intercept)         1.354035   0.014902  90.864   <2e-16 ***
# # mutationVCP\nR155C  0.018789   0.009161   2.051   0.0404 *  
# # mutationVCP\nR191Q  0.009781   0.012049   0.812   0.4170    
# 
# glm(data=nuc_cyt_mean_intensities, nuc_cyt_mean_intensity ~ condition + replicate, family = gaussian)%>% summary()
# # Coefficients:
# #               Estimate Std. Error t value Pr(>|t|)    
# # (Intercept)   1.357778   0.013898  97.699   <2e-16 ***
# # conditionVCP  0.015935   0.008192   1.945   0.0519 .  
#   
# polya_fish.merged = plot_grid(PolyAFISHfigure.gg, 
#           plot_grid(nuc_cyt_mean_intensities.violin, ncol = 2),
#           nrow = 2, labels =  "auto", rel_heights = c(1,1.2))
# # polya_fish.merged
# ggsave(polya_fish.merged, filename = here(proj_path, "polya-fish/polya_fish.merged.png"), height = 7, width = 9)
# # googledrive::drive_upload(media = here(proj_path, "polya-fish/polya_fish.merged.png"), path = "mRNA redistribution ALS/figS2.polya_fish.merged.png", overwrite = TRUE)




### NEFL counts

# NEFL validation
whole.untreated.tardbp_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
whole.untreated.vcp_r155c_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
whole.untreated.vcp_r191q_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
whole.untreated.vcp_iso_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))

ctrl.untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
tardbp.untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
vcp_r155c.untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
vcp_r191q.untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
vcp_iso.untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))

nuc.untreated.tardbp_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
cyt.untreated.tardbp_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
nuc.untreated.vcp_r155c_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
cyt.untreated.vcp_r155c_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
nuc.untreated.vcp_r191q_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
cyt.untreated.vcp_r191q_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
nuc.untreated.vcp_iso_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))
cyt.untreated.vcp_iso_vs_ctrl$res.tx %>% filter(transcript_id %in% c("ENST00000619417"))


untreated.nuc_vs_cyt.tardbp_vs_ctrl$res %>%  filter(gene_name %in% c("NFX1","NIT1"))
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res %>%  filter(gene_name %in% c("NFX1","NIT1"))
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res %>%  filter(gene_name %in% c("NFX1","NIT1"))
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res %>%  filter(gene_name %in% c("NFX1","NIT1"))
whole.untreated.tardbp_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
whole.untreated.vcp_r155c_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
whole.untreated.vcp_r191q_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
whole.untreated.vcp_iso_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
ctrl.untreated.nuc_vs_cyt$res %>% filter(gene_name %in% c("NFX1","NIT1"))
tardbp.untreated.nuc_vs_cyt$res %>% filter(gene_name %in% c("NFX1","NIT1"))
vcp_r155c.untreated.nuc_vs_cyt$res %>% filter(gene_name %in% c("NFX1","NIT1"))
vcp_r191q.untreated.nuc_vs_cyt$res %>% filter(gene_name %in% c("NFX1","NIT1"))
vcp_iso.untreated.nuc_vs_cyt$res %>% filter(gene_name %in% c("NFX1","NIT1"))
nuc.untreated.tardbp_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
cyt.untreated.tardbp_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
nuc.untreated.vcp_r155c_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
cyt.untreated.vcp_r155c_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
nuc.untreated.vcp_r191q_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
cyt.untreated.vcp_r191q_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
nuc.untreated.vcp_iso_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))
cyt.untreated.vcp_iso_vs_ctrl$res %>% filter(gene_name %in% c("NFX1","NIT1"))


# untreated.nuc_vs_cyt.tx = readRDS(here(proj_path,"expression/deseq2/untreated.nuc_vs_cyt.tx.rds"))
# untreated.nuc_vs_cyt.tx$vsd.tx.counts = as_tibble(assay(untreated.nuc_vs_cyt.tx$vsd.tx), rownames = "transcript_id") %>% left_join(tx2gene) %>% left_join(gene2ens)
# 
# # NEFL
# untreated.nuc_vs_cyt.tx.vsd.nefl.long = filter(untreated.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id == "ENST00000619417") %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>%
#   mutate(fraction = case_when(grepl("nucleus",sample)~"nucleus", grepl("cytoplasm",sample)~"cytoplasm"), mutation = case_when(grepl("ctrl",sample)~"CTRL",grepl("tdpn",sample)~"TARDBP",grepl("gli",sample)~"VCP^R191Q",grepl("cb1",sample)~"VCP^R155C",grepl("ncrm",sample)~"VCPki^R191Q"))
# untreated.nuc_vs_cyt.tx.vsd.nefl.long %>% my_summarise(vsd)
# untreated.nuc_vs_cyt.tx.vsd.nefl.violin = violin_plot(untreated.nuc_vs_cyt.tx.vsd.nefl.long, color_by = "fraction", continuous = "vsd", ylims = c(0,13), facet_by = "mutation", facet_ncol = 5, facet_labeller = "label_parsed", plot_violin = FALSE, plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 13, #rep(c(4.1,4.4,4.2),4))
#             ylabel = expression(NEFL~normalised~expression), xlabel = "", cols = c("dodgerblue2","firebrick2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.tx.vsd.nefl.violin
# 
# untreated.nuc_vs_cyt.tx.vsd.nefl.long = filter(untreated.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id == "ENST00000619417") %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>%
#   mutate(fraction = case_when(grepl("nucleus",sample)~"nucleus", grepl("cytoplasm",sample)~"cytoplasm"), mutation = case_when(grepl("ctrl",sample)~"CTRL",grepl("tdpn",sample)~"TARDBP",grepl("gli",sample)~"VCP^R191Q",grepl("cb1",sample)~"VCP^R155C",grepl("ncrm",sample)~"VCPki^R191Q"))
# untreated.nuc_vs_cyt.tx.vsd.nefl.long %>% my_summarise(vsd)
# untreated.nuc_vs_cyt.tx.vsd.nefl.violin = violin_plot(untreated.nuc_vs_cyt.tx.vsd.nefl.long, color_by = "fraction", continuous = "vsd", ylims = c(0,13), facet_by = "mutation", facet_ncol = 5, facet_labeller = "label_parsed", plot_violin = FALSE, plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 13, #rep(c(4.1,4.4,4.2),4))
#             ylabel = expression(NEFL~normalised~expression), xlabel = "", cols = c("dodgerblue2","firebrick2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.tx.vsd.nefl.violin
# 
# # NEFL protein
# untreated.ctrl_nuc_vs_cyt.dep$res %>% filter(gene_name == "NEFL")
# untreated.tardbp_nuc_vs_cyt.dep$res %>% filter(gene_name == "NEFL")
# untreated.vcp_r155c_nuc_vs_cyt.dep$res %>% filter(gene_name == "NEFL")
# untreated.vcp_r191q_nuc_vs_cyt.dep$res %>% filter(gene_name == "NEFL")
# untreated.vcp_iso_nuc_vs_cyt.dep$res %>% filter(gene_name == "NEFL")
# 
# untreated.nuc_vs_cyt.tx = readRDS(here(proj_path,"expression/deseq2/untreated.nuc_vs_cyt.tx.rds"))
# untreated.nuc_vs_cyt.tx$vsd.tx.counts = as_tibble(assay(untreated.nuc_vs_cyt.tx$vsd.tx), rownames = "transcript_id") %>% left_join(tx2gene) %>% left_join(gene2ens)
# 
# ml240.nuc_vs_cyt.tx = readRDS(here(proj_path,"expression/deseq2/ml240.nuc_vs_cyt.tx.rds"))
# ml240.nuc_vs_cyt.tx$vsd.tx.counts = as_tibble(assay(ml240.nuc_vs_cyt.tx$vsd.tx), rownames = "transcript_id") %>% left_join(tx2gene) %>% left_join(gene2ens)
# 
# # NEFL
# untreated.nuc_vs_cyt.tx.vsd.nefl.long = filter(untreated.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id == "ENST00000589055") %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>% mutate(treatment = "Untreated")
# ml240.nuc_vs_cyt.tx.vsd.nefl.long = filter(ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id == "ENST00000589055") %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>% mutate(treatment = "ML240")
# 
# untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.long = bind_rows(untreated.nuc_vs_cyt.tx.vsd.nefl.long, ml240.nuc_vs_cyt.tx.vsd.nefl.long) %>% 
#   mutate(treatment = factor(treatment, levels = c("Untreated","ML240")), fraction = case_when(grepl("nucleus",sample)~"nucleus", grepl("cytoplasm",sample)~"cytoplasm"), mutation = case_when(grepl("ctrl",sample)~"CTRL",grepl("tdpn",sample)~"TARDBP",grepl("gli",sample)~"VCP^R191Q",grepl("cb1",sample)~"VCP^R155C",grepl("ncrm",sample)~"VCPki^R191Q"))
# untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.long %>% my_summarise(vsd, treatment)
# untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.violin = violin_plot(untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.long, color_by = "fraction", continuous = "vsd", ylims = c(0,13), facet_by = "mutation", facet_by2 = "treatment", facet_ncol = 5, facet_nrow = 2, facet_labeller = "label_parsed", plot_violin = FALSE, plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 13, #rep(c(4.1,4.4,4.2),4))
#                                                       ylabel = expression(NEFL~ENST00000619417~normalised~expression), xlabel = "", cols = c("dodgerblue2","firebrick2"), dpi = 300, arrow_labels = FALSE) + theme_bw()
# untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.violin
# ggsave(untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.violin, filename = here(proj_path, "expression/deseq2/NEFL.ENST00000619417.untreated_ml240.nuc_vs_cyt.tx.vsd.nefl.violin.pdf"), height = 7, width = 8.25)
# 
# # VSD counts for untreated and ML240 for a transcript
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot <- function(untreated_vsd.tx.counts, ml240_vsd.tx.counts, transcript_id, gene_name, stats = TRUE){
#   untreated.nuc_vs_cyt.tx.vsd.long = filter(untreated_vsd.tx.counts, transcript_id == {{transcript_id}}) %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>% mutate(treatment = "Untreated")
#   ml240.nuc_vs_cyt.tx.vsd.long = filter(ml240_vsd.tx.counts, transcript_id == {{transcript_id}}) %>% pivot_longer(!c(transcript_id, gene_id, gene_name), names_to = "sample", values_to = "vsd") %>% mutate(treatment = "ML240")
#   
#   untreated_ml240.nuc_vs_cyt.tx.vsd.long = bind_rows(untreated.nuc_vs_cyt.tx.vsd.long, ml240.nuc_vs_cyt.tx.vsd.long) %>% 
#     mutate(treatment = factor(treatment, levels = c("Untreated","ML240")), fraction = case_when(grepl("nucleus",sample)~"nucleus", grepl("cytoplasm",sample)~"cytoplasm"), mutation = case_when(grepl("ctrl",sample)~"CTRL",grepl("tdpn",sample)~"TARDBP",grepl("gli",sample)~"VCP^R191Q",grepl("cb1",sample)~"VCP^R155C",grepl("ncrm",sample)~"VCPki^R191Q"))
#   untreated_ml240.nuc_vs_cyt.tx.vsd.long %>% my_summarise(vsd, treatment)
#   untreated_ml240.nuc_vs_cyt.tx.vsd.violin = violin_plot(untreated_ml240.nuc_vs_cyt.tx.vsd.long, color_by = "fraction", continuous = "vsd", ylims = c(0,13), facet_by = "mutation", facet_by2 = "treatment", facet_ncol = 5, facet_nrow = 2, facet_labeller = "label_parsed", 
#                                                          plot_violin = FALSE, plot_stats = stats, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 13, #rep(c(4.1,4.4,4.2),4))
#                                                          ylabel = paste(gene_name, transcript_id, "normalised expression"), xlabel = "", cols = c("dodgerblue2","firebrick2"), dpi = 300, arrow_labels = FALSE) + theme_bw()
#   ggsave(untreated_ml240.nuc_vs_cyt.tx.vsd.violin, filename = here(proj_path, "expression/deseq2",paste0(gene_name,".",transcript_id,".untreated_ml240.nuc_vs_cyt.tx.vsd.violin.pdf")), height = 5, width = 8.25)
#   return(untreated_ml240.nuc_vs_cyt.tx.vsd.violin)
# }
# 
# untreated.nuc_vs_cyt.tx = readRDS(here(proj_path,"expression/deseq2/untreated.nuc_vs_cyt.tx.rds"))
# untreated.nuc_vs_cyt.tx$vsd.tx.counts = as_tibble(assay(untreated.nuc_vs_cyt.tx$vsd.tx), rownames = "transcript_id") %>% left_join(tx2gene) %>% left_join(gene2ens)
# 
# ml240.nuc_vs_cyt.tx = readRDS(here(proj_path,"expression/deseq2/ml240.nuc_vs_cyt.tx.rds"))
# ml240.nuc_vs_cyt.tx$vsd.tx.counts = as_tibble(assay(ml240.nuc_vs_cyt.tx$vsd.tx), rownames = "transcript_id") %>% left_join(tx2gene) %>% left_join(gene2ens)
# 
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000619417", gene_name = "NEFL")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000589055", gene_name = "TMEM259", stats = FALSE)
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000397573", gene_name = "BMF")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000589825", gene_name = "EFTUD2")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000611351", gene_name = "ZMIZ1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000572707", gene_name = "GPS2")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000645840", gene_name = "CSNK2A1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000552766", gene_name = "PA2G4")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000465301", gene_name = "RLT9")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000537274", gene_name = "YAP1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000541774", gene_name = "ERBB2", stats = FALSE)
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000370244", gene_name = "BCAR3")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000373980", gene_name = "PRKG1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000642556", gene_name = "CERT1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000482967", gene_name = "CDK5KRAP1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000662330", gene_name = "RMST")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000642953", gene_name = "SMCHD1")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000479281", gene_name = "MAGED4")
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000394706", gene_name = "TBCK", stats = FALSE)
# untreated_ml240.nuc_vs_cyt.tx.vsd.violin_plot(untreated.nuc_vs_cyt.tx$vsd.tx.counts, ml240.nuc_vs_cyt.tx$vsd.tx.counts, transcript_id = "ENST00000316606", gene_name = "ZSCAN26")
# 
# 
# # TPM plots
# 
# #import TPMs
# salmon.merged.transcript_counts.tsv = read_tsv(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/salmon.merged.transcript_counts.tsv"))
# salmon.merged.transcript_counts.tsv %>% glimpse
# salmon.merged.transcript_counts.tsv %>% filter(tx == "ENST00000619417") %>% glimpse
# 
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot <- function(salmon.merged.transcript_counts.tsv, transcript_id, stats = FALSE){
#   gene_name = gtf %>% filter(transcript == {{transcript_id}}) %>% distinct(gene_name) %>% pull(gene_name)
#   nuc_vs_cyt.tx.tpm.long = filter(salmon.merged.transcript_counts.tsv, tx == {{transcript_id}}) %>% pivot_longer(!c(tx, gene_id), names_to = "sample", values_to = "tpm") %>% 
#     mutate(treatment = str_split_fixed(sample,"_",3)[,2], treatment = factor(treatment, levels = c("untreated","ml240")), fraction = str_split_fixed(sample,"_",3)[,3], 
#            mutation = case_when(grepl("ctrl",sample)~"CTRL",grepl("tdpn",sample)~"TARDBP",grepl("gli",sample)~"VCP^R191Q",grepl("cb1",sample)~"VCP^R155C",grepl("ncrm",sample)~"VCPki^R191Q"))
#   nuc_vs_cyt.tx.tpm.long %>% my_summarise(tpm, mutation, treatment)
#   nuc_vs_cyt.tx.tpm.violin = violin_plot(nuc_vs_cyt.tx.tpm.long, color_by = "fraction", continuous = "tpm", facet_by = "mutation", facet_by2 = "treatment", facet_ncol = 5, facet_nrow = 2, facet_labeller = "label_parsed", 
#                                                          plot_violin = FALSE, plot_stats = stats, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = stat_ypos, #rep(c(4.1,4.4,4.2),4))
#                                                          ylabel = paste(gene_name, transcript_id, "TPM"), xlabel = "", dpi = 300, arrow_labels = FALSE) + theme_bw() + theme(axis.text.x = element_text(angle = 30, hjust = 1))
#   ggsave(nuc_vs_cyt.tx.tpm.violin, filename = paste0("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/expression/deseq2/",gene_name,".",transcript_id,".untreated_ml240.nuc_vs_cyt.tx.tpm.violin.pdf"), height = 5, width = 8.25)
#   return(nuc_vs_cyt.tx.tpm.violin)
# }
# 
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000619417") # NEFL
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000589055") #gene_name = "TMEM259", 
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000397573")#, gene_name = "BMF")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000589825")#, gene_name = "EFTUD2")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000611351")#, gene_name = "ZMIZ1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000572707")#, gene_name = "GPS2")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000645840")#, gene_name = "CSNK2A1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000552766")#, gene_name = "PA2G4")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000465301")#, gene_name = "RLT9")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000537274")#, gene_name = "YAP1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000541774")#, gene_name = "ERBB2", stats = FALSE)
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000370244")#, gene_name = "BCAR3")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000373980")#, gene_name = "PRKG1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000642556")#, gene_name = "CERT1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000482967")#, gene_name = "CDK5KRAP1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000662330")#, gene_name = "RMST")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000642953")#, gene_name = "SMCHD1")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000479281")#, gene_name = "MAGED4")
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000394706")#, gene_name = "TBCK", stats = FALSE)
# untreated_ml240.nuc_vs_cyt.tx.tpm.violin_plot(salmon.merged.transcript_counts.tsv, transcript_id = "ENST00000316606")#, gene_name = "SCAN26")

```



## Merge

```{r}
vcp_tardbp_tx_redistribution.merged <- plot_grid(
  # analysis_schematic.gg
  plot_grid(untreated.nuc_vs_cyt.tardbp_vs_ctrl.volcano, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano, untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano, untreated.nuc_vs_cyt.als_vs_ctrl.volcano, ncol = 5, labels = "auto"),
  plot_grid(untreated.nuc_vs_cyt.tardbp_vs_ctrl.go_curated, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated, untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated, untreated.nuc_vs_cyt.als_vs_ctrl.go_curated, ncol = 5, labels = letters[6:10]), 
  plot_grid(untreated.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap, untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset, untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset, ncol = 3, rel_widths = c(0.8,1,1), labels = letters[11:12]), 
  plot_grid(NULL, nefl_fish_image.gg, nuc_cyt_nefl_lfc.violin, ncol = 3, rel_widths = c(0.02,2,1), labels = c("m","","n")),
  # untreated.nuc_vs_cyt.tx.vsd.nefl.violin,
  nrow = 4, rel_heights = c(1,1.2,0.6,1))
# vcp_tardbp_tx_redistribution.merged
ggsave(vcp_tardbp_tx_redistribution.merged, filename = here(proj_path, "expression/deseq2/Figure2.vcp_tardbp_tx_redistribution.merged.png"), height = 14, width = 13)
ggsave(vcp_tardbp_tx_redistribution.merged, filename = here(proj_path, "expression/deseq2/Figure2.vcp_tardbp_tx_redistribution.merged.pdf"), height = 14, width = 13)


# plot_grid(tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl_stat.corr, vcp_iso_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl_stat.corr, ncol = 2, rel_widths = c(1,1), labels = letters[8:9]), 

```



## Post-mortem comparison

No figure. For text only and Table S3.

```{r}
nygc_postmortem_spinal_cord.tx = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/expression/deseq2/nygc_postmortem_spinal_cord.tx.rds"))

# Fisher exact overlap test
nygc_postmortem_spinal_cord.tx.res.dge  <- nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)
untreated.nuc_vs_cyt.als_vs_ctrl.res.dge  <- untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge  <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge  <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge  <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge  <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05 ) %>% pull(transcript_id)

# Combined mutants
newGeneOverlap(untreated.nuc_vs_cyt.als_vs_ctrl.res.dge , nygc_postmortem_spinal_cord.tx.res.dge , genome.size=nrow(nygc_postmortem_spinal_cord.tx$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=330
# listB size=33736
# Intersection size=43
# Overlapping p-value=0.84
# Jaccard Index=0.0
48/330#14.5
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.als_vs_ctrl.res.dge) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.als_vs_ctrl.res.dge, gene_name %in% c(nuclear_export.markers)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.als_vs_ctrl.res.dge, gene_name %in% c(ALS_genes)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.als_vs_ctrl.res.dge, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_split(stat > 0)

# TARDBP
newGeneOverlap(untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge , nygc_postmortem_spinal_cord.tx.res.dge , genome.size=nrow(nygc_postmortem_spinal_cord.tx$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=381
# listB size=33736
# Intersection size=68
# Overlapping p-value=0.058
# Jaccard Index=0.0
68/381#17.8
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge, gene_name %in% c(nuclear_export.markers)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge, gene_name %in% c(ALS_genes)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_split(stat > 0)

# R155C
newGeneOverlap(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge, nygc_postmortem_spinal_cord.tx.res.dge, genome.size=nrow(nygc_postmortem_spinal_cord.tx$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=358
# listB size=33736
# Intersection size=59
# Overlapping p-value=0.21
# Jaccard Index=0.0
59/358#16.5
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge, gene_name %in% c(nuclear_export.markers)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge, gene_name %in% c(ALS_genes)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_split(stat > 0)

#R191Q
newGeneOverlap(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge, nygc_postmortem_spinal_cord.tx.res.dge, genome.size=nrow(nygc_postmortem_spinal_cord.tx$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=602
# listB size=33736
# Intersection size=117
# Overlapping p-value=1.2e-03
# Jaccard Index=0.0
117/602#19.4
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge, gene_name %in% c(nuclear_export.markers)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge, gene_name %in% c(ALS_genes)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_split(stat > 0)

#knock-in
newGeneOverlap(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge, nygc_postmortem_spinal_cord.tx.res.dge, genome.size=nrow(nygc_postmortem_spinal_cord.tx$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=569
# listB size=33736
# Intersection size=124
# Overlapping p-value=5.3e-06
# Jaccard Index=0.0
124/569#21.8
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge, gene_name %in% c(nuclear_export.markers)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge, gene_name %in% c(ALS_genes)) %>% group_split(stat > 0)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% group_split(stat > 0)



# # heatmap of correlation coefficients
# nygc.untreated.nuc_vs_cyt.als_vs_ctrl_stat = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, `TARDBP^G298S` = stat) %>% 
#   left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, `VCP^R155C` = stat)) %>%
#   left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, `VCP^R191Q` = stat)) %>%
#   left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, `VCPki^R191Q` = stat)) %>%
#   left_join(select(nygc_postmortem_spinal_cord.tx$res.tx, transcript_id, `Postmortem` = stat)) %>%
#   drop_na() %>%
#   column_to_rownames("transcript_id")
# cormat <- round(cor(nygc.untreated.nuc_vs_cyt.als_vs_ctrl_stat),2)
# head(cormat)
# melted_cormat <- melt(cormat)
# head(melted_cormat)
# get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
#   cormat[upper.tri(cormat)] <- NA
#   return(cormat)
# }
# get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
#   cormat[lower.tri(cormat)]<- NA
#   return(cormat)
#   }
# upper_tri <- get_upper_tri(cormat)
# melted_cormat <- melt(upper_tri, na.rm = TRUE)
# # reorder_cormat <- function(cormat){ # Use correlation between variables as distance
# #   dd <- as.dist((1-cormat)/2)
# #   hc <- hclust(dd)
# #   cormat <-cormat[hc$order, hc$order]
# # }
# # cormat <- reorder_cormat(cormat)
# upper_tri <- get_upper_tri(cormat)
# melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
#   # mutate(Var1 = fct_relevel(Var1, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"), Var2 = fct_relevel(Var2, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))
# nygc.untreated.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
#   geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
#   geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
#   theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
#                           legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
#   guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
#   scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
# nygc.untreated.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap
# 
# 
# ### Upset plot
# nygc.untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCP R191Q knock-in"),   mutate(nygc_postmortem_spinal_cord.tx$res.tx, mutation = "Postmortem"))
# 
# nygc.untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.upset = upset_plot(nygc.untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx, overlap_by = "mutation", overlap_level = "transcript_id", split_direction = FALSE, ymax = 400, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in", "Postmortem"), title = "Nuclear redistributed", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
# nygc.untreated.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.upset

```


# Table S3 Transcript redistribution

```{r}
untreated.nuc_vs_cyt.als_vs_ctrl = readRDS(file = here(proj_path, "scripts/untreated.nuc_vs_cyt.als_vs_ctrl.rds"))
nygc_postmortem_spinal_cord.tx = readRDS(here(nemo_path, "projects/ipsc-mn-als-meta/expression/deseq2/nygc_postmortem_spinal_cord.tx.rds"))

nygc_postmortem_spinal_cord.tx.res.dge.up <- nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, stat > 0) %>% pull(transcript_id)
nygc_postmortem_spinal_cord.tx.res.dge.down <- nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05, stat < 0) %>% pull(transcript_id)

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join = mutate(select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean),
                                                             tardbp.ml240 = case_when(transcript_id %in% pull(filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05, (stat > 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05, stat > 0),transcript_id)) | (stat < 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05, stat < 0),transcript_id))),transcript_id) ~ "rescued", TRUE ~ "")) %>%
  left_join(mutate(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean),
            vcp_r155c.ml240 = case_when(transcript_id %in% pull(filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05, (stat > 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05, stat > 0),transcript_id)) | (stat < 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05, stat < 0),transcript_id))),transcript_id) ~ "rescued", TRUE ~ ""))) %>%
  left_join(mutate(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean),
            vcp_r191q.ml240 = case_when(transcript_id %in% pull(filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05, (stat > 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05, stat > 0),transcript_id)) | (stat < 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05, stat < 0),transcript_id))),transcript_id) ~ "rescued", TRUE ~ ""))) %>%
  left_join(mutate(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, vcp_ki_r191q.stat = stat, vcp_ki_r191q.padj = padj, vcp_ki_r191q.lfc = log2FoldChange, vcp_ki_r191q.baseMean = baseMean),
            vcp_ki.ml240 = case_when(transcript_id %in% pull(filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05, (stat > 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05, stat > 0),transcript_id)) | (stat < 0 & !transcript_id %in% pull(filter(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05, stat < 0),transcript_id))),transcript_id) ~ "rescued", TRUE ~ ""))) %>%
  left_join(select(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, transcript_id, gene_name, combined.stat = stat, combined.padj = padj, combined.lfc = log2FoldChange, combined.baseMean = baseMean)) %>% 
  filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_ki_r191q.stat) == FALSE) %>%
  filter(tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_ki_r191q.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_ki_r191q.stat))) %>%
  mutate(overlapping = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki_r191q.padj < 0.05 & tardbp.stat > 0 & vcp_r155c.stat > 0 & vcp_r191q.stat > 0 & vcp_ki_r191q.stat > 0) ~ "Nuclear", 
                             (tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_ki_r191q.padj < 0.05 & vcp_ki_r191q.padj < 0.05 & tardbp.stat < 0 & vcp_r155c.stat < 0 & vcp_r191q.stat < 0 & vcp_ki_r191q.stat < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% c(go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION) ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS) ~ "Nuclear export", 
                              gene_name %in% RNA_BINDING_PROTEINS ~ "RBP", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ ""),
         postmortem = case_when(transcript_id %in% nygc_postmortem_spinal_cord.tx.res.dge.up ~ "increased", transcript_id %in% nygc_postmortem_spinal_cord.tx.res.dge.down ~ "decreased", TRUE ~ "")) # %>%
    # rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_iso")), ~ str_replace_all(., "vcp_iso", "vcp_ki"))
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join
write_csv(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, here(proj_path,"expression/untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S3 ALS redist")

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(gene_name %in% "XIST")
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(gene_name %in% c("XIST",'RPS4Y1', 'EIFAY', 'DDX3Y', 'KDM5D'))

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(overlapping != "", category != "")

```


# Fig S4 Scatter correlation

```{r}
untreated.nuc_vs_cyt.als_vs_ctrl_list = list("TARDBP G298S" = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, "VCP R155C" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, "VCP R191Q" = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, "VCP R191Q knock-in" = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx)

tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))
tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))
tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))
vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))
vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))
vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat) %>% left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, vcp.stat = stat)) %>% filter((vcp.stat > 5 & tardbp.stat  > 5) | (vcp.stat < -5 & tardbp.stat  < -5))

untreated.nuc_vs_cyt.als_vs_ctrl_list_stat_plot <- 
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R155C", join_by = c("transcript_id", "gene_name"), transcript_labels = pull(tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q", join_by = c("transcript_id", "gene_name"), transcript_labels = pull(tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q knock-in", join_by = c("transcript_id", "gene_name"),  transcript_labels = pull(tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q", join_by = c("transcript_id", "gene_name"), transcript_labels = pull(vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q knock-in", join_by = c("transcript_id", "gene_name"), transcript_labels = pull(vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
plot_de_correlation(model_list1 = untreated.nuc_vs_cyt.als_vs_ctrl_list, model_list2 = untreated.nuc_vs_cyt.als_vs_ctrl_list, mutation1 = "VCP R191Q", mutation2 = "VCP R191Q knock-in", join_by = c("transcript_id", "gene_name"), transcript_labels = pull(vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,transcript_id))  +
  plot_layout(ncol = 2, nrow = 3) &  
  xlim(-8,8) & ylim(-8,8)
# untreated.nuc_vs_cyt.als_vs_ctrl_list_stat_plot
ggsave(untreated.nuc_vs_cyt.als_vs_ctrl_list_stat_plot, filename = here(proj_path, "expression/deseq2/untreated.nuc_vs_cyt.als_vs_ctrl_list_stat_plot.png"), height = 10, width = 8.25)

```


# Fig S5 Whole cell Heatmap nuclear export & NPC

```{r}
whole.untreated.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) #832
whole.untreated.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05)
whole.untreated.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05)
whole.untreated.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05)


whole.untreated.tardbp_vs_ctrl$res.tx %>% filter(transcript_id %in% "ENST00000619417") #832

untreated.nuc_vs_cyt$res.tx %>% filter(transcript_id %in% "ENST00000619417") #832
whole.untreated.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% nucleocytoplasmic_genes) 

# define gene markers
go.pathways.msigdb$GO_NUCLEAR_EXPORT
go.pathways.msigdb$GO_NUCLEAR_EXPORT_SIGNAL_RECEPTOR_ACTIVITY
go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY
go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX
go.pathways.msigdb$GO_MRNA_EXPORT_FROM_NUCLEUS
go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS
go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION
go.pathways.msigdb$GO_NUCLEAR_PORE
go.pathways.msigdb$GO_NUCLEAR_PORE_NUCLEAR_BASKET

### nuclear export markers
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))
nuclear_export.markers.df = tibble("gene_name" = nuclear_export.markers) %>% filter(gene_name %in% nuclear_export.markers) %>% 
  mutate(group = case_when(gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE_NUCLEAR_BASKET ~ "nuclear.basket", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION ~ "npc.organisation", 
                           gene_name %in% go.pathways.msigdb$GO_NUCLEAR_EXPORT_SIGNAL_RECEPTOR_ACTIVITY ~ "export.signal", gene_name %in% go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY ~ "carrier.activity", gene_name %in% go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX ~ "TREX", gene_name %in% go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS ~ "rna.export", 
    gene_name %in% go.pathways.msigdb$GO_NUCLEAR_EXPORT ~ "nuclear.export", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "nuclear.pore"),
         group = factor(group, levels = c("nuclear.basket", "npc.organisation", "nuclear.pore", "export.signal","carrier.activity","TREX","rna.export", "nuclear.export"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)


# L2FC
nuclear_export.l2fc.marker <- select(whole.untreated.tardbp_vs_ctrl$res, gene_name, TARDBP = log2FoldChange) %>% 
  full_join(select(whole.untreated.vcp_r155c_vs_ctrl$res, gene_name, VCP.R155C = log2FoldChange)) %>% 
  full_join(select(whole.untreated.vcp_r191q_vs_ctrl$res, gene_name, VCP.R191Q = log2FoldChange)) %>% 
  full_join(select(whole.untreated.vcp_iso_vs_ctrl$res, gene_name, VCP.KI = log2FoldChange)) %>%
  filter(gene_name %in% nuclear_export.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  arrange(factor(gene_name, levels = nuclear_export.markers.df$gene_name))
nuclear_export.markers.df.filt = nuclear_export.markers.df %>% filter(gene_name %in% nuclear_export.l2fc.marker$gene_name)
nuclear_export.l2fc.marker.mat <- make_matrix(select(nuclear_export.l2fc.marker,-gene_name), pull(nuclear_export.l2fc.marker,gene_name))

### Cluster samples ###
nuclear_export.l2fc.marker.centered_mat = t(scale(t(nuclear_export.l2fc.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
nuclear_export.l2fc.marker.scaled_mat = t(scale(t(nuclear_export.l2fc.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered

nuclear_export.markers.rnaseq.cluster.heatmap <- Heatmap(nuclear_export.l2fc.marker.mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_names_rot = 90, column_names_gp = gpar(fontsize = 10), 
          row_order = nuclear_export.markers.df.filt$gene_name, 
          row_split = nuclear_export.markers.df.filt$group,
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels = c("NPC\nBasket", "NPC\nOrganisation", "Nuclear\nPore","Export\nSignal","Carrier\nActivity","TREX", "RNA\nExport","Nuclear\nExport"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_title = NULL,
          row_names_gp = gpar(fontsize = 5), #show_row_names = FALSE, 
          column_title = NULL, cluster_columns = FALSE, cluster_row_slices = FALSE, cluster_rows = FALSE)
nuclear_export.markers.rnaseq.cluster.heatmap
ggsave(as.ggplot(nuclear_export.markers.rnaseq.cluster.heatmap), filename = here(proj_path,"expression/deseq2/nuclear_export.markers.rnaseq.cluster.heatmap.png"), width = 9, height = 14)

nuclear_export.l2fc.marker %>% left_join(nuclear_export.markers.df) %>% filter(TARDBP < 0, VCP.R155C < 0, VCP.R191Q < 0, VCP.KI < 0) %>% mutate(sum = TARDBP + VCP.R155C + VCP.R191Q + VCP.KI) %>% arrange(sum)

nuclear_export.marker.bind <- mutate(whole.untreated.tardbp_vs_ctrl$res, mutation = "tardbp") %>% 
  bind_rows(mutate(whole.untreated.vcp_r155c_vs_ctrl$res, mutation = "vcp.r155c")) %>% 
  bind_rows(mutate(whole.untreated.vcp_r191q_vs_ctrl$res, mutation = "vcp.r191q")) %>% 
  bind_rows(mutate(whole.untreated.vcp_iso_vs_ctrl$res, mutation = "cxp.ki")) %>%
  filter(gene_name %in% nuclear_export.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE)
nuclear_export.marker.bind %>% arrange(padj)

## nuclear export GSEA
# # # https://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html
# library(msigdbr)
# # msigdbr_hs = msigdbr(species = "Homo sapiens")
# msigdbr_go = msigdbr(species = "Homo sapiens", category = "C5") # C5 = GO pathways
# msigdbr_go.t2g = msigdbr_go %>% select(gs_name, ensembl_gene)
# 
# whole.untreated.tardbp_vs_ctrl.geneList <- whole.untreated.tardbp_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
# names(whole.untreated.tardbp_vs_ctrl.geneList) <- whole.untreated.tardbp_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_id) %>% as.character()
# whole.untreated.tardbp_vs_ctrl.geneList = sort(whole.untreated.tardbp_vs_ctrl.geneList, decreasing = TRUE)
# whole.untreated.tardbp_vs_ctrl.gsea <- GSEA(whole.untreated.tardbp_vs_ctrl.geneList, TERM2GENE = msigdbr_go.t2g)
# as_tibble(whole.untreated.tardbp_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID)) %>% arrange(-abs(NES)) %>% print(n=100)
# as_tibble(whole.untreated.tardbp_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("RNA",ID)) %>% arrange(-abs(NES))  %>% print(n=Inf)
# as_tibble(whole.untreated.tardbp_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID)) %>% print(n=Inf)
# as_tibble(whole.untreated.tardbp_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID))
# whole.untreated.tardbp_vs_ctrl.gsea.filt = whole.untreated.tardbp_vs_ctrl.gsea %>% filter(!grepl("^HP_", ID)) %>% mutate(Description = gsub("GOBP_|GOCC_|GOMF_","",Description), Description = gsub("NUCLEAR_TRANSCRIBED_"," ",Description), Description = gsub("_"," ",Description), Description = tolower(Description), Description = gsub("rna ","RNA ",Description), Description = gsub("dna ","DNA ",Description), Description = gsub("poly a","polyA",Description), Description = gsub("regulation of ","",Description))
# library(enrichplot)
# whole.untreated.tardbp_vs_ctrl.gseaplot = gseaplot2(whole.untreated.tardbp_vs_ctrl.gsea.filt, geneSetID = c("GOBP_NUCLEAR_TRANSPORT", "GOBP_NUCLEAR_PORE_ORGANIZATION", "GOBP_NUCLEAR_PORE_COMPLEX_ASSEMBLY", "GOCC_NUCLEAR_PORE", "GOCC_NUCLEAR_PORE_NUCLEAR_BASKET"), ES_geom = "line", subplots = 1) + geom_hline(yintercept = 0, linetype = "dashed") + theme_oz() + theme(legend.position = "top", legend.title = element_blank(), legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=7), legend.key.size = unit(0.4, "cm")) +
#   guides(fill = guide_legend(nrow = 2)) + scale_color_npg()
# whole.untreated.tardbp_vs_ctrl.gseaplot
# 
# whole.untreated.vcp_r155c_vs_ctrl.geneList <- whole.untreated.vcp_r155c_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
# names(whole.untreated.vcp_r155c_vs_ctrl.geneList) <- whole.untreated.vcp_r155c_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_id) %>% as.character()
# whole.untreated.vcp_r155c_vs_ctrl.geneList = sort(whole.untreated.vcp_r155c_vs_ctrl.geneList, decreasing = TRUE)
# whole.untreated.vcp_r155c_vs_ctrl.gsea <- GSEA(whole.untreated.vcp_r155c_vs_ctrl.geneList, TERM2GENE = msigdbr_go.t2g)
# as_tibble(whole.untreated.vcp_r155c_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID)) %>% arrange(-abs(NES)) %>% print(n=100)
# as_tibble(whole.untreated.vcp_r155c_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("RNA",ID)) %>% arrange(-abs(NES))  %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_r155c_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID)) %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_r155c_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID))
# whole.untreated.vcp_r155c_vs_ctrl.gsea.filt = whole.untreated.vcp_r155c_vs_ctrl.gsea %>% filter(!grepl("^HP_", ID)) %>% mutate(Description = gsub("GOBP_|GOCC_|GOMF_","",Description), Description = gsub("NUCLEAR_TRANSCRIBED_"," ",Description), Description = gsub("_"," ",Description), Description = tolower(Description), Description = gsub("rna ","RNA ",Description), Description = gsub("dna ","DNA ",Description), Description = gsub("poly a","polyA",Description), Description = gsub("regulation of ","",Description))
# whole.untreated.vcp_r155c_vs_ctrl.gseaplot = gseaplot2(whole.untreated.vcp_r155c_vs_ctrl.gsea.filt, geneSetID = c("GOBP_NUCLEAR_TRANSPORT", "GOBP_NUCLEAR_PORE_ORGANIZATION", "GOBP_NUCLEAR_PORE_COMPLEX_ASSEMBLY", "GOCC_NUCLEAR_PORE", "GOCC_NUCLEAR_PORE_NUCLEAR_BASKET"), ES_geom = "line", subplots = 1) + geom_hline(yintercept = 0, linetype = "dashed") + theme_oz() + theme(legend.position = "top", legend.title = element_blank(), legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=7), legend.key.size = unit(0.4, "cm")) +
#   guides(fill = guide_legend(nrow = 2)) + scale_color_npg()
# whole.untreated.vcp_r155c_vs_ctrl.gseaplot
# 
# whole.untreated.vcp_r191q_vs_ctrl.geneList <- whole.untreated.vcp_r191q_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
# names(whole.untreated.vcp_r191q_vs_ctrl.geneList) <- whole.untreated.vcp_r191q_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_id) %>% as.character()
# whole.untreated.vcp_r191q_vs_ctrl.geneList = sort(whole.untreated.vcp_r191q_vs_ctrl.geneList, decreasing = TRUE)
# whole.untreated.vcp_r191q_vs_ctrl.gsea <- GSEA(whole.untreated.vcp_r191q_vs_ctrl.geneList, TERM2GENE = msigdbr_go.t2g)
# as_tibble(whole.untreated.vcp_r191q_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID)) %>% arrange(-abs(NES)) %>% print(n=100)
# as_tibble(whole.untreated.vcp_r191q_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("RNA",ID)) %>% arrange(-abs(NES))  %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_r191q_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID)) %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_r191q_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID))
# whole.untreated.vcp_r191q_vs_ctrl.gsea.filt = whole.untreated.vcp_r191q_vs_ctrl.gsea %>% filter(!grepl("^HP_", ID)) %>% mutate(Description = gsub("GOBP_|GOCC_|GOMF_","",Description), Description = gsub("NUCLEAR_TRANSCRIBED_"," ",Description), Description = gsub("_"," ",Description), Description = tolower(Description), Description = gsub("rna ","RNA ",Description), Description = gsub("dna ","DNA ",Description), Description = gsub("poly a","polyA",Description), Description = gsub("regulation of ","",Description))
# whole.untreated.vcp_r191q_vs_ctrl.gseaplot = gseaplot2(whole.untreated.vcp_r191q_vs_ctrl.gsea.filt, geneSetID = c("GOBP_NUCLEAR_TRANSPORT", "GOBP_NUCLEAR_PORE_ORGANIZATION", "GOBP_NUCLEAR_PORE_COMPLEX_ASSEMBLY", "GOCC_NUCLEAR_PORE", "GOCC_NUCLEAR_PORE_NUCLEAR_BASKET"), ES_geom = "line", subplots = 1) + geom_hline(yintercept = 0, linetype = "dashed") + theme_oz() + theme(legend.position = "top", legend.title = element_blank(), legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=7), legend.key.size = unit(0.4, "cm")) +
#   guides(fill = guide_legend(nrow = 2)) + scale_color_npg()
# whole.untreated.vcp_r191q_vs_ctrl.gseaplot
# 
# whole.untreated.vcp_iso_vs_ctrl.geneList <- whole.untreated.vcp_iso_vs_ctrl$res %>% drop_na(stat) %>% pull(stat)
# names(whole.untreated.vcp_iso_vs_ctrl.geneList) <- whole.untreated.vcp_iso_vs_ctrl$res %>% drop_na(stat) %>% pull(gene_id) %>% as.character()
# whole.untreated.vcp_iso_vs_ctrl.geneList = sort(whole.untreated.vcp_iso_vs_ctrl.geneList, decreasing = TRUE)
# whole.untreated.vcp_iso_vs_ctrl.gsea <- GSEA(whole.untreated.vcp_iso_vs_ctrl.geneList, TERM2GENE = msigdbr_go.t2g)
# as_tibble(whole.untreated.vcp_iso_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID)) %>% arrange(-abs(NES)) %>% print(n=100)
# as_tibble(whole.untreated.vcp_iso_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("RNA",ID)) %>% arrange(-abs(NES))  %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_iso_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID)) %>% print(n=Inf)
# as_tibble(whole.untreated.vcp_iso_vs_ctrl.gsea) %>% filter(!grepl("^HP_", ID), p.adjust < 0.05, grepl("NUCLEAR",ID))
# whole.untreated.vcp_iso_vs_ctrl.gsea.filt = whole.untreated.vcp_iso_vs_ctrl.gsea %>% filter(!grepl("^HP_", ID)) %>% mutate(Description = gsub("GOBP_|GOCC_|GOMF_","",Description), Description = gsub("NUCLEAR_TRANSCRIBED_"," ",Description), Description = gsub("_"," ",Description), Description = tolower(Description), Description = gsub("rna ","RNA ",Description), Description = gsub("dna ","DNA ",Description), Description = gsub("poly a","polyA",Description), Description = gsub("regulation of ","",Description))
# whole.untreated.vcp_iso_vs_ctrl.gseaplot = gseaplot2(whole.untreated.vcp_iso_vs_ctrl.gsea.filt, geneSetID = c("GOBP_NUCLEAR_TRANSPORT", "GOBP_NUCLEAR_PORE_ORGANIZATION", "GOBP_NUCLEAR_PORE_COMPLEX_ASSEMBLY", "GOCC_NUCLEAR_PORE", "GOCC_NUCLEAR_PORE_NUCLEAR_BASKET"), ES_geom = "line", subplots = 1) + geom_hline(yintercept = 0, linetype = "dashed") + theme_oz() + theme(legend.position = "top", legend.title = element_blank(), legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=7), legend.key.size = unit(0.4, "cm")) +
#   guides(fill = guide_legend(nrow = 2)) + scale_color_npg()
# whole.untreated.vcp_iso_vs_ctrl.gseaplot



```


# Fig S6 polyA FISH

```{r}
# import image
PolyAFISHfigure.png = magick::image_read(here(proj_path,"polya-fish/PolyAFISHfigure_wide.png"))
# PolyAFISHfigure.png = magick::image_read(here(proj_path,"polya-fish/PolyAFISHfigure.png"))
PolyAFISHfigure.gg = ggdraw() + draw_image(PolyAFISHfigure.png)
PolyAFISHfigure.gg

nuc_cyt_mean_intensities.xlsx = read_excel(here(proj_path,"polya-fish/nuc_cyt_mean_intensities.xlsx"), sheet = "raw") %>% clean_names()
nuc_cyt_mean_intensities.xlsx
nuc_cyt_mean_intensities = nuc_cyt_mean_intensities.xlsx %>% mutate(cellline = case_when(grepl("CB1D", file_name_biii) ~ "CB1D", grepl("CB1D", file_name_biii) ~ "CB1D", grepl("CB1E", file_name_biii) ~ "CB1E", grepl("CTRl4", file_name_biii) ~ "CTRL4", grepl("CTRL4", file_name_biii) ~ "CTRL4",  grepl("CTRL2", file_name_biii) ~ "CTRL2", grepl("GliA", file_name_biii) ~ "GliA",grepl("CTRL3", file_name_biii) ~ "CTRL3",grepl("CTRl3", file_name_biii) ~ "CTRL3"), mutation = case_when(grepl("CTRL", cellline) ~ "CTRL", grepl("CB1", cellline) ~ "VCP\nR155C",  grepl("Gli", cellline) ~ "VCP\nR191Q"), condition = case_when(mutation == "CTRL" ~ "CTRL", TRUE ~ "VCP"))

nuc_cyt_mean_intensities %>% count(replicate, mutation, cellline)
nuc_cyt_mean_intensities %>% my_summarise(nuc_mean_intensity, mutation)
nuc_cyt_mean_intensities %>% my_summarise(cyt_mean_intensity, mutation)
nuc_cyt_mean_intensities %>% my_summarise(nuc_cyt_mean_intensity, mutation)
nuc_cyt_mean_intensities %>% my_summarise(nuc_cyt_mean_intensity, mutation, replicate)

nuc_cyt_mean_intensities %>% my_summarise(nuc_cyt_mean_intensity, condition)


# mutation mean_nuc_cyt_mean_intensity median_nuc_cyt_mean_intensity sd_nuc_cyt_mean_intensity Q1_nuc_cyt_mean_intensity Q3_nuc_cyt_mean_intensity min_nuc_cyt_mean_intensity max_nuc_cyt_mean_intensity sum_nuc_cyt_mean_intensity n_nuc_cyt_mean_intensity
#   <chr>                       <dbl>                         <dbl>                     <dbl>                     <dbl>                     <dbl>                      <dbl>                      <dbl>                      <dbl>                    <int>
# 1 CTRL                         1.12                          1.12                     0.213                     1.01                       1.25                      0.252                       1.94                      1244.                     1106
# 2 VCP                          1.12                          1.11                     0.204                     0.996                      1.23                      0.331                       2.04                      1322.                     1177

# Nuclear Intensity
nuc_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "nuc_mean_intensity", stat_ypos = 0.010, ylabel = "Nuclear intensity a.u.", dpi = 300, stats_test = "wilcox_test", arrow_labels = FALSE, facet_by = "replicate", stat_label = "p.adj.signif")
nuc_mean_intensity.violin
# [1] "Faceting by replicate"
# [1] "Using two sample wilcox_test"
# # A tibble: 2 × 13
#   facet .y.   mutation1 mutation2    n1    n2 statistic            p p.signif y.position mutations        xmin  xmax
#   <dbl> <chr> <chr>  <chr>  <int> <int>     <dbl>        <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1     1 lfc   CTRL   VCP      530   432    121355 0.109        ns             0.01 <chr [2]>        1     2
# 2     2 lfc   CTRL   VCP      576   745    251518 0.0000000765 ****           0.01 <chr [2]>        1     2
# # A tibble: 2 × 8
#   .y.   mutation1 mutation2 effsize facet    n1    n2 magnitude 
# * <chr> <chr>  <chr>    <dbl> <dbl> <int> <int> <ord>     
# 1 lfc   CTRL   VCP    0.00122     1   530   432 negligible
# 2 lfc   CTRL   VCP    0.313       2   576   745 small     

# Cytoplasm Instensity
cyt_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "cyt_mean_intensity", stat_ypos = 0.010, ylabel = "Cytoplasm intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE, facet_by = "replicate", stat_label = "p.adj.signif")
cyt_mean_intensity.violin
# facet .y.   mutation1 mutation2    n1    n2 statistic    df        p p.signif y.position mutations        xmin  xmax
#   <dbl> <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>    <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1     1 lfc   CTRL   VCP      530   432     -1.37  918. 1.7 e- 1 ns             0.01 <chr [2]>        1     2
# 2     2 lfc   CTRL   VCP      576   745      9.78 1313. 7.15e-22 ****           0.01 <chr [2]>        1     2
# # A tibble: 2 × 8
#   .y.   mutation1 mutation2 effsize facet    n1    n2 magnitude 
# * <chr> <chr>  <chr>    <dbl> <dbl> <int> <int> <ord>     
# 1 lfc   CTRL   VCP    -0.0891     1   530   432 negligible
# 2 lfc   CTRL   VCP     0.530      2   576   745 moderate  

# Nuclear / Cytoplasm intensity ratio
nuc_cyt_mean_intensities.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "condition", continuous = "nuc_cyt_mean_intensity", ylims = c(0.2,2.1), null_value = 1, stat_ypos = 2.0, ylabel = "Nuclear/Cytoplasm ratio", dpi = 300, stats_test = "wilcox_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.85)#, stat_label = "p.adj.signif")
nuc_cyt_mean_intensities.violin
# [1] "Using two sample t_test"
# # A tibble: 1 × 10
#   .y.   mutation1 mutation2    n1    n2 statistic    df     p p.signif y.position
#   <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl> <dbl> <chr>         <dbl>
# 1 lfc   CTRL   VCP     1106  1177     0.144 2258. 0.886 ns                2
# # A tibble: 1 × 7
#   .y.   mutation1 mutation2 effsize    n1    n2 magnitude 
# * <chr> <chr>  <chr>    <dbl> <int> <int> <ord>     
# 1 lfc   CTRL   VCP    0.00602  1106  1177 negligible

glm(data=nuc_cyt_mean_intensities, nuc_cyt_mean_intensity ~ mutation + replicate, family = gaussian)%>% summary()

# Coefficients:
#                     Estimate Std. Error t value Pr(>|t|)    
# (Intercept)         1.354035   0.014902  90.864   <2e-16 ***
# mutationVCP\nR155C  0.018789   0.009161   2.051   0.0404 *  
# mutationVCP\nR191Q  0.009781   0.012049   0.812   0.4170    

glm(data=nuc_cyt_mean_intensities, nuc_cyt_mean_intensity ~ condition + replicate, family = gaussian)%>% summary()
# Coefficients:
#               Estimate Std. Error t value Pr(>|t|)    
# (Intercept)   1.357778   0.013898  97.699   <2e-16 ***
# conditionVCP  0.015935   0.008192   1.945   0.0519 .  
  
polya_fish.merged = plot_grid(PolyAFISHfigure.gg, 
          plot_grid(nuc_cyt_mean_intensities.violin, ncol = 2),
          nrow = 2, labels =  "auto", rel_heights = c(1,1.2))
# polya_fish.merged
ggsave(polya_fish.merged, filename = here(proj_path, "polya-fish/polya_fish.merged.png"), height = 7, width = 9)
# googledrive::drive_upload(media = here(proj_path, "polya-fish/polya_fish.merged.png"), path = "mRNA redistribution ALS/figS2.polya_fish.merged.png", overwrite = TRUE)


# # replicate 2 only
# nuc_cyt_mean_intensities.rep2 = nuc_cyt_mean_intensities %>% filter(replicate ==2)
# # Nuclear Intensity
# nuc_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities.rep2, color_by = "mutation", continuous = "nuc_mean_intensity", ylims = c(0,0.011), stat_ypos = 0.010, ylabel = "Nuclear intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# nuc_mean_intensity.violin
# 
# # Cytoplasm Instensity
# cyt_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities.rep2, color_by = "mutation", continuous = "cyt_mean_intensity", ylims = c(0,0.011), stat_ypos = 0.010, ylabel = "Cytoplasm intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# cyt_mean_intensity.violin
# 
# # Nuclear / Cytoplasm intensity ratio
# nuc_cyt_mean_intensities.violin = violin_plot(nuc_cyt_mean_intensities.rep2, color_by = "mutation", continuous = "nuc_cyt_mean_intensity", ylims = c(0.2,2.1), null_value = 1, stat_ypos = 2.0, ylabel = "Nuclear/Cytoplasm ratio", dpi = 300, stats_test = "t_test", numerator = "Nucleus", denomenator = "Cytoplasm", arrow_label_x = 0.85)
# nuc_cyt_mean_intensities.violin

# plot_grid(nuc_mean_intensity.violin, cyt_mean_intensity.violin,nuc_cyt_mean_intensities.violin, ncol = 3)


# # Nuclear Intensity
# nuc_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "nuc_mean_intensity", ylims = c(0,0.02), stat_ypos = 0.019, ylabel = "Nuclear intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# nuc_mean_intensity.violin
# 
# # Cytoplasm Instensity
# cyt_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "cyt_mean_intensity", ylims = c(0,0.02), stat_ypos = 0.019, ylabel = "Cytoplasm intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# cyt_mean_intensity.violin
# 
# # Nuclear / Cytoplasm intensity ratio
# nuc_cyt_mean_intensities.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "nuc_cyt_mean_intensity", ylims = c(0,2.3), null_value = 1, stat_ypos = 2.2, ylabel = "Nuclear/Cytoplasm ratio a.u.", dpi = 300, stats_test = "t_test", numerator = "Nucleus", denomenator = "Cytoplasm")
# nuc_cyt_mean_intensities.violin
# 
# plot_grid(nuc_mean_intensity.violin, cyt_mean_intensity.violin,nuc_cyt_mean_intensities.violin, ncol = 3)
# 
# # nuc_cyt_sum_intensities.violin = violin_plot(nuc_cyt_mean_intensities, color_by = "mutation", continuous = "nuc_cyt_sum_intensity", ylims = c(0,22), stat_ypos = 22, ylabel = "Nuclear/Cytoplasm sum ratio a.u.", dpi = 30, stats_test = "t_test")
# # nuc_cyt_sum_intensities.violin
# 
# # replicate 1 only
# nuc_cyt_mean_intensities.rep1 = nuc_cyt_mean_intensities %>% filter(replicate ==1)
# # Nuclear Intensity
# nuc_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities.rep1, color_by = "mutation", continuous = "nuc_mean_intensity", ylims = c(0,0.02), stat_ypos = 0.019, ylabel = "Nuclear intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# nuc_mean_intensity.violin
# 
# # Cytoplasm Instensity
# cyt_mean_intensity.violin = violin_plot(nuc_cyt_mean_intensities.rep1, color_by = "mutation", continuous = "cyt_mean_intensity", ylims = c(0,0.02), stat_ypos = 0.019, ylabel = "Cytoplasm intensity a.u.", dpi = 300, stats_test = "t_test", arrow_labels = FALSE)
# cyt_mean_intensity.violin
# 
# # Nuclear / Cytoplasm intensity ratio
# nuc_cyt_mean_intensities.violin = violin_plot(nuc_cyt_mean_intensities.rep1, color_by = "mutation", continuous = "nuc_cyt_mean_intensity", ylims = c(0,2.3), null_value = 1, stat_ypos = 2.2, ylabel = "Nuclear/Cytoplasm ratio a.u.", dpi = 300, stats_test = "t_test", numerator = "Nucleus", denomenator = "Cytoplasm")
# nuc_cyt_mean_intensities.violin
# 
# plot_grid(nuc_mean_intensity.violin, cyt_mean_intensity.violin,nuc_cyt_mean_intensities.violin, ncol = 3)

```


# Fig 3 Sequence features: Biotypes, lengths and IR


## Redistributed transcript features

```{r}
# Biorender schematic mechanisms of transcript nuclear accumulation
# # import with majick 
# nuclear_accumulation_mrna_mechanisms.png = magick::image_read(here(proj_path,"biorender/nuclear_accumulation_mrna_mechanisms.png"))
# nuclear_accumulation_mRNA_mechanisms.gg = ggdraw() + draw_image(nuclear_accumulation_mrna_mechanisms.png)
# nuclear_accumulation_mRNA_mechanisms.gg

homo_sapiens.gtf = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.rds"))
homo_sapiens.gtf.gene <- homo_sapiens.gtf %>% filter(type == "gene")
homo_sapiens.gtf.transcript <- homo_sapiens.gtf %>% filter(type == "transcript") # NB this is unspliced width
homo_sapiens.gtf.spliced.transcript <- homo_sapiens.gtf %>% group_by(transcript_id) %>% select(transcript_id, transcript_biotype, width, type) %>% pivot_wider(c(transcript_id, transcript_biotype), names_from = type, values_from = width, values_fn = sum) %>% mutate(width = sum(exon, CDS, five_prime_utr, three_prime_utr, na.rm = TRUE)) %>% select(transcript_id, transcript_biotype, width)
homo_sapiens.gtf.5UTR.transcript <- homo_sapiens.gtf %>% filter(type == "five_prime_utr") %>% group_by(transcript_id) %>% summarise(five_prime_utr = mean(width, na.rm = TRUE)) # mean 5'UTR length isoform
homo_sapiens.gtf.3UTR.transcript <- homo_sapiens.gtf %>% filter(type == "three_prime_utr") %>% group_by(transcript_id) %>% summarise(three_prime_utr = mean(width, na.rm = TRUE)) # mean 3UTR length isoform
homo_sapiens.gtf.CDS.transcript <- homo_sapiens.gtf %>% filter(type == "CDS") %>% group_by(transcript_id) %>% summarise(CDS = mean(width, na.rm = TRUE))
homo_sapiens.gtf.exon.transcript <- homo_sapiens.gtf %>% filter(type == "exon") %>% group_by(transcript_id) %>% summarise(exon_number = n()) # total exon number - capped at 9???
homo_sapiens.gtf.transcript %>% group_by(transcript_biotype) %>% summarise(lengths = mean(width, na.rm = TRUE)) %>% print(n=Inf) # mean lenths of biotypes

untreated.nuc_vs_cyt.als_vs_ctrl.tx.features <- bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(select(homo_sapiens.gtf.spliced.transcript, transcript_id, transcript_biotype, width)) %>%
  left_join(select(homo_sapiens.gtf.5UTR.transcript, transcript_id, five_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.3UTR.transcript, transcript_id, three_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.CDS.transcript, transcript_id, CDS)) %>%
  left_join(select(homo_sapiens.gtf.exon.transcript, transcript_id, exon_number)) %>%
  mutate(mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), 
         log10.width = log10(width), log10.five_prime_utr = log10(five_prime_utr), log10.three_prime_utr = log10(three_prime_utr), log10.CDS = log10(CDS), log10.exon_number = log10(exon_number),
         intronless = case_when(exon_number == 1 ~ "Intronless", exon_number >1 ~ "Intron Containing", TRUE ~ "NA"),
         mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))

### Transcript lengths

# mean lenths of groups in bases (not kb)
untreated.nuc_vs_cyt.als_vs_ctrl.tx.features %>% group_by(mutation, mislocalisation) %>% summarise(lengths = mean(width, na.rm = TRUE))
# mutation     mislocalisation      lengths
#    <fct>        <fct>                  <dbl>
#  1 TARDBP^G298S "Non\nRedistributed"   2536.
#  2 TARDBP^G298S "Nuc"                  5891.
#  3 TARDBP^G298S "Cyto"                 5588.
#  4 VCP^R155C    "Non\nRedistributed"   2537.
#  5 VCP^R155C    "Nuc"                  5657.
#  6 VCP^R155C    "Cyto"                 5161.
#  7 VCP^R191Q    "Non\nRedistributed"   2533.
#  8 VCP^R191Q    "Nuc"                  5593.
#  9 VCP^R191Q    "Cyto"                 5524.
# 10 VCPki^R191Q  "Non\nRedistributed"   2534.
# 11 VCPki^R191Q  "Nuc"                  6008.
# 12 VCPki^R191Q  "Cyto"                 5476.

untreated.nuc_vs_cyt.als_vs_ctrl.width.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "mislocalisation", continuous = "log10.width", ylims = c(2.5,4.5), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(4.1,4.4,4.2),4), 
                                                               ylabel = expression(log[10]~Transcript~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.als_vs_ctrl.width.tx.violin
# facet        .y.   group1               group2    n1    n2 statistic            p        p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>  <int> <int>     <dbl>        <dbl>        <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    97820    86  3111240  0.0000292    0.0000584    ****             420000 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   97820   135  5033928  0.00000177   0.00000531   ****             480000 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto      86   135     5849  0.925        0.925        ns               380000 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    97850    96  3603555  0.0000787    0.000236     ***              420000 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   97850    95  3758820  0.001        0.002        **               480000 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto      96    95     4662  0.79         0.79         ns               380000 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    97699   110  4326742  0.000405     0.00081      ***              420000 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   97699   232  8944320  0.0000000279 0.0000000837 ****             480000 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto     110   232    12383  0.659        0.659        ns               380000 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    97700   107  3735474. 0.000000323  0.000000646  ****             420000 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   97700   234  9077633  0.0000000509 0.000000153  ****             480000 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto     107   234    13619  0.193        0.193        ns               380000 <chr [2]>        2     3

# check for length-bias
untreated.nuc_vs_cyt.als_vs_ctrl.tx.features %>% drop_na(padj) %>% group_by(mutation, mislocalisation) %>% summarise(lengths = mean(width, na.rm = TRUE))
untreated.nuc_vs_cyt.als_vs_ctrl.tx.features %>% filter(baseMean > 100) %>% drop_na(padj) %>% group_by(mutation, mislocalisation) %>% summarise(lengths = mean(width, na.rm = TRUE))
ggscatter(mutate(untreated.nuc_vs_cyt.als_vs_ctrl.tx.features, baseMean.log10 = log10(baseMean)), x = "stat", y = "width", #color = "Classification", palette = rev(c("firebrick2", "grey", "dodgerblue2")),
          cor.coef = TRUE, cor.coeff.args = list(method = "pearson", label.y = 7.5, label.x = -1), cor.coef.size = 3, xlab = expression(Stat~ALS~vs~CTRL), ylab = expression(log[10]~baseMean), size = 0.5) + 
  theme_classic() + theme(legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) +
  geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") +
  facet_wrap(~mutation)


### exon number
untreated.nuc_vs_cyt.als_vs_ctrl.tx.features %>% group_by(mutation, mislocalisation) %>% summarise(exons = mean(exon_number, na.rm = TRUE))
# mutation           mislocalisation      exons
#    <fct>              <fct>                <dbl>
#  1 TARDBP G298S       "Non\nRedistributed"  6.02
#  2 TARDBP G298S       "Nucleus"            11.2 
#  3 TARDBP G298S       "Cytoplasm"          11.8 
#  4 VCP R155C          "Non\nRedistributed"  6.03
#  5 VCP R155C          "Nucleus"            10.5 
#  6 VCP R155C          "Cytoplasm"          11.8 
#  7 VCP R191Q          "Non\nRedistributed"  6.02
#  8 VCP R191Q          "Nucleus"            11.3 
#  9 VCP R191Q          "Cytoplasm"          11.1 
# 10 VCP R191Q knock-in "Non\nRedistributed"  6.02
# 11 VCP R191Q knock-in "Nucleus"            11.4 
# 12 VCP R191Q knock-in "Cytoplasm"          10.2 

untreated.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "mislocalisation", continuous = "log10.exon_number", ylims = c(0,1.7), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(1.4,1.6,1.3),4), 
                                                               ylabel = expression(log[10]~exon~number), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE, violin_adjust = 2.5)
# untreated.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin
# facet              .y.   group1               group2        n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>              <chr> <chr>                <chr>      <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP G298S       lfc   "Non\nRedistributed" Nucleus   227039   360 23557686  1.64e-44 3.28e-44 ****               1.52 <chr [2]>        1     2
#  2 TARDBP G298S       lfc   "Non\nRedistributed" Cytoplasm 227039   419 23710398. 1.78e-71 5.34e-71 ****               1.6  <chr [2]>        1     3
#  3 TARDBP G298S       lfc   "Nucleus"            Cytoplasm    360   419    69372  5.3 e- 2 5.3 e- 2 ns                 1.5  <chr [2]>        2     3
#  4 VCP R155C          lfc   "Non\nRedistributed" Nucleus   227068   407 29043742. 6.31e-39 1.26e-38 ****               1.52 <chr [2]>        1     2
#  5 VCP R155C          lfc   "Non\nRedistributed" Cytoplasm 227068   343 19799234  1.26e-56 3.78e-56 ****               1.6  <chr [2]>        1     3
#  6 VCP R155C          lfc   "Nucleus"            Cytoplasm    407   343    62280. 1.1 e- 2 1.1 e- 2 *                  1.5  <chr [2]>        2     3
#  7 VCP R191Q          lfc   "Non\nRedistributed" Nucleus   226867   373 23571447  3.43e-50 6.86e-50 ****               1.52 <chr [2]>        1     2
#  8 VCP R191Q          lfc   "Non\nRedistributed" Cytoplasm 226867   578 34630211  8.61e-87 2.58e-86 ****               1.6  <chr [2]>        1     3
#  9 VCP R191Q          lfc   "Nucleus"            Cytoplasm    373   578   106182. 6.96e- 1 6.96e- 1 ns                 1.5  <chr [2]>        2     3
# 10 VCP R191Q knock-in lfc   "Non\nRedistributed" Nucleus   226567   476 31691587  2.74e-55 5.48e-55 ****               1.52 <chr [2]>        1     2
# 11 VCP R191Q knock-in lfc   "Non\nRedistributed" Cytoplasm 226567   775 50293470. 4.13e-95 1.24e-94 ****               1.6  <chr [2]>        1     3
# 12 VCP R191Q knock-in lfc   "Nucleus"            Cytoplasm    476   775   186834. 7   e- 1 7   e- 1 ns                 1.5  <chr [2]>        2     3
# # A tibble: 12 × 8
#    .y.   group1               group2    effsize facet                  n1    n2 magnitude 
#  * <chr> <chr>                <chr>       <dbl> <fct>               <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nucleus   -0.786  TARDBP G298S       227039   360 moderate  
#  2 lfc   "Non\nRedistributed" Cytoplasm -0.915  TARDBP G298S       227039   419 large     
#  3 lfc   "Nucleus"            Cytoplasm -0.134  TARDBP G298S          360   419 negligible
#  4 lfc   "Non\nRedistributed" Nucleus   -0.700  VCP R155C          227068   407 moderate  
#  5 lfc   "Non\nRedistributed" Cytoplasm -0.897  VCP R155C          227068   343 large     
#  6 lfc   "Nucleus"            Cytoplasm -0.202  VCP R155C             407   343 small     
#  7 lfc   "Non\nRedistributed" Nucleus   -0.827  VCP R191Q          226867   373 large     
#  8 lfc   "Non\nRedistributed" Cytoplasm -0.849  VCP R191Q          226867   578 large     
#  9 lfc   "Nucleus"            Cytoplasm -0.0238 VCP R191Q             373   578 negligible
# 10 lfc   "Non\nRedistributed" Nucleus   -0.780  VCP R191Q knock-in 226567   476 moderate  
# 11 lfc   "Non\nRedistributed" Cytoplasm -0.743  VCP R191Q knock-in 226567   775 moderate  
# 12 lfc   "Nucleus"            Cytoplasm  0.0393 VCP R191Q knock-in    476   775 negligible


### 3'UTR length
untreated.nuc_vs_cyt.als_vs_ctrl.tx.features %>% group_by(mutation, mislocalisation) %>% summarise(exons = mean(three_prime_utr, na.rm = TRUE))

untreated.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "mislocalisation", continuous = "log10.three_prime_utr", ylims = c(1.5,4), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(3.6,3.9,3.5),4), 
                                                               ylabel = expression(log[10]~3~UTR~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin
# facet              .y.   group1               group2       n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>              <chr> <chr>                <chr>     <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP G298S       lfc   "Non\nRedistributed" Nucleus   76505   249  7369985  6.69e-10 1.34e- 9 ****                3.8 <chr [2]>        1     2
#  2 TARDBP G298S       lfc   "Non\nRedistributed" Cytoplasm 76505   306  8828225  1.07e-13 3.21e-13 ****                3.9 <chr [2]>        1     3
#  3 TARDBP G298S       lfc   "Nucleus"            Cytoplasm   249   306    37222. 6.42e- 1 6.42e- 1 ns                  3.7 <chr [2]>        2     3
#  4 VCP R155C          lfc   "Non\nRedistributed" Nucleus   76532   278  8764513  3.84e- 7 7.68e- 7 ****                3.8 <chr [2]>        1     2
#  5 VCP R155C          lfc   "Non\nRedistributed" Cytoplasm 76532   250  7741051  1.82e- 7 5.46e- 7 ****                3.9 <chr [2]>        1     3
#  6 VCP R155C          lfc   "Nucleus"            Cytoplasm   278   250    33882  6.2 e- 1 6.2 e- 1 ns                  3.7 <chr [2]>        2     3
#  7 VCP R191Q          lfc   "Non\nRedistributed" Nucleus   76379   246  6899328  5.85e-13 1.76e-12 ****                3.8 <chr [2]>        1     2
#  8 VCP R191Q          lfc   "Non\nRedistributed" Cytoplasm 76379   435 13589609  5.58e-11 1.12e-10 ****                3.9 <chr [2]>        1     3
#  9 VCP R191Q          lfc   "Nucleus"            Cytoplasm   246   435    58364. 4.9 e- 2 4.9 e- 2 *                   3.7 <chr [2]>        2     3
# 10 VCP R191Q knock-in lfc   "Non\nRedistributed" Nucleus   76187   316  9367754. 9.45e-12 1.89e-11 ****                3.8 <chr [2]>        1     2
# 11 VCP R191Q knock-in lfc   "Non\nRedistributed" Cytoplasm 76187   557 17357951  1.27e-13 3.81e-13 ****                3.9 <chr [2]>        1     3
# 12 VCP R191Q knock-in lfc   "Nucleus"            Cytoplasm   316   557    92597  2   e- 1 2   e- 1 ns                  3.7 <chr [2]>        2     3
# # A tibble: 12 × 8
#    .y.   group1               group2    effsize facet                  n1    n2 magnitude 
#  * <chr> <chr>                <chr>       <dbl> <fct>               <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nucleus   -0.374  TARDBP G298S       227039   360 small     
#  2 lfc   "Non\nRedistributed" Cytoplasm -0.408  TARDBP G298S       227039   419 small     
#  3 lfc   "Nucleus"            Cytoplasm -0.0351 TARDBP G298S          360   419 negligible
#  4 lfc   "Non\nRedistributed" Nucleus   -0.277  VCP R155C          227068   407 small     
#  5 lfc   "Non\nRedistributed" Cytoplasm -0.301  VCP R155C          227068   343 small     
#  6 lfc   "Nucleus"            Cytoplasm -0.0256 VCP R155C             407   343 negligible
#  7 lfc   "Non\nRedistributed" Nucleus   -0.442  VCP R191Q          226867   373 small     
#  8 lfc   "Non\nRedistributed" Cytoplasm -0.293  VCP R191Q          226867   578 small     
#  9 lfc   "Nucleus"            Cytoplasm  0.163  VCP R191Q             373   578 negligible
# 10 lfc   "Non\nRedistributed" Nucleus   -0.352  VCP R191Q knock-in 226567   476 small     
# 11 lfc   "Non\nRedistributed" Cytoplasm -0.288  VCP R191Q knock-in 226567   775 small     
# 12 lfc   "Nucleus"            Cytoplasm  0.0705 VCP R191Q knock-in    476   775 negligible

```


## RNAct RNA view

https://arrow.apache.org/overview/
Summarise scores per gene and scores per protein - Run on CAMP

```{r}
# import with magick RNA-RBP interaction Schematic from biorender
# rna_rbp_schematic = magick::image_read(here(proj_path,"biorender/RNA_RBP_interactions.png"))
# rna_rbp_schematic.gg = ggdraw() + draw_image(rna_rbp_schematic)
# rna_rbp_schematic.gg

# # Run on CAMP
# nemo_path = here("/nemo/lab/patanir/home/users/ziffo")
# RNAct_database.protein.summarised <- arrow::read_tsv_arrow(file = here::here(nemo_path, "genomes/rbp-databases/rnact/catrapid_human_basic_normalised.txt")) %>%
#   group_by(uniprot_accession) %>% summarise(mean_score = mean(catrapid_score_max_normalised, na.rm = FALSE), number = n(), sum_score = sum(catrapid_score_max_normalised)) %>%
#   collect()
# arrow::write_parquet(RNAct_database.protein.summarised, here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.protein.summarised.parquet"))
# saveRDS(RNAct_database.protein.summarised, here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.protein.summarised.rds"))
# 
# nemo_path = here("/nemo/lab/patanir/home/users/ziffo")
# «arrow::write_parquet(RNAct_database.transcript.summarised, here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.transcript.summarised.parquet"))
# saveRDS(RNAct_database.transcript.summarised, here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.transcript.summarised.rds"))

```


```{r}
RNAct_database.transcript.summarised = readRDS(here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.transcript.summarised.rds"))
RNAct_database.transcript.summarised
RNAct_database.transcript.summarised$transcript_id[!RNAct_database.transcript.summarised$transcript_id %in% untreated.nuc_vs_cyt.vcp_vs_ctrl$res.tx$transcript_id] # 570 missing based on transcript_id.

rnact_rna_view.untreated.nuc_vs_cyt.als_vs_ctrl = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(RNAct_database.transcript.summarised) %>%
  mutate(log10.mean_score = log10(mean_score), log10.sum_score = log10(sum_score), log10.number = log10(number), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(mean_score)#, 
         # across(contains(c("score","number")), ~replace(., is.na(.)|is.infinite(.), 0))) # replace NA with 0 - this is crucial before taking the mean otherwise doesnt account for number of RBP-RNA interactions
rnact_rna_view.untreated.nuc_vs_cyt.als_vs_ctrl

rnact_rna_view.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(sum_score, mislocalisation, mutation)
# mislocalisation      mutation     mean_sum_score median_sum_score sd_sum_score Q1_sum_score Q3_sum_score min_sum_score max_sum_score sum_sum_score n_sum_score
#    <fct>                <fct>                 <dbl>            <dbl>        <dbl>        <dbl>        <dbl>         <dbl>         <dbl>         <dbl>       <int>
#  1 "Non\nRedistributed" TARDBP^G298S        223444.          216898.      117038.      130072.      305182.       -97515.       690384.  21857330779.       97820
#  2 "Non\nRedistributed" VCP^R155C           223463.          216990.      117012.      130089.      305189.       -97515.       690384.  21865858559.       97850
#  3 "Non\nRedistributed" VCP^R191Q           223412.          216894.      117043.      130020.      305132.       -97515.       690384.  21827142667.       97699
#  4 "Non\nRedistributed" VCPki^R191Q         223397.          216875.      117045.      130019.      305134.       -97515.       690384.  21825904737.       97700
#  5 "Nuc"                TARDBP^G298S        269116.          265497.       83139.      211598.      323522.       100921.       490745.     23143974.          86
#  6 "Nuc"                VCP^R155C           271298.          260861.      106895.      197653.      321463.        74536.       664032.     26044590.          96
#  7 "Nuc"                VCP^R191Q           257529.          248413.       91117.      185293.      320803.        56338.       453837.     28328169.         110
#  8 "Nuc"                VCPki^R191Q         274322.          287253.       87270.      215188.      324895.        92781.       485999.     29352401.         107
#  9 "Cyto"               TARDBP^G298S        267357.          261227.       97585.      199937.      326436.        28145.       583505.     36093142.         135
# 10 "Cyto"               VCP^R155C           259629.          264491.      105087.      184509.      333997.        70510.       547116.     24664746.          95
# 11 "Cyto"               VCP^R191Q           263349.          262718.      102441.      185706.      329108.        67581.       543643.     61097059.         232
# 12 "Cyto"               VCPki^R191Q         262012.          261516.      100438.      187686.      324380.        34921.       547116.     61310756.         234

# Mislocalisation Nuclear vs Cytoplasm vs None
rnact.rna_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(rnact_rna_view.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "sum_score", ylims = c(50000,500000), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(420000, 480000, 380000),4), 
                                                               ylabel = expression(RNA:protein~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.rna_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
#  facet         .y.   group1               group2       n1    n2 statistic           p       p.adj p.adj.signif y.position groups        xmin  xmax
#   <fct>         <chr> <chr>                <chr>     <int> <int>     <dbl>       <dbl>       <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 TARDBP mutant lfc   "Non\nRedistributed" Nuclear   97592   201  8219721  0.0000711   0.000142    ***              550000 <chr [2]>        1     2
# 2 TARDBP mutant lfc   "Non\nRedistributed" Cytoplasm 97592   248 10013610. 0.0000026   0.0000078   ****             600000 <chr [2]>        1     3
# 3 TARDBP mutant lfc   "Nuclear"            Cytoplasm   201   248    24582  0.803       0.803       ns               570000 <chr [2]>        2     3
# 4 VCP mutant    lfc   "Non\nRedistributed" Nuclear   97482   326 13589260. 0.00000619  0.0000124   ****             550000 <chr [2]>        1     2
# 5 VCP mutant    lfc   "Non\nRedistributed" Cytoplasm 97482   233  9037034. 0.000000069 0.000000207 ****             600000 <chr [2]>        1     3
# 6 VCP mutant    lfc   "Nuclear"            Cytoplasm   326   233    35292. 0.154       0.154       ns               570000 <chr [2]>        2     3
# 7 VCP knock-in  lfc   "Non\nRedistributed" Nuclear   97501   324 13198328  0.00000031  0.00000093  ****             550000 <chr [2]>        1     2
# 8 VCP knock-in  lfc   "Non\nRedistributed" Cytoplasm 97501   216  8609644. 0.00000353  0.00000706  ****             600000 <chr [2]>        1     3
# 9 VCP knock-in  lfc   "Nuclear"            Cytoplasm   324   216    34030  0.588       0.588       ns               570000 <chr [2]>        2     3
# # A tibble: 9 × 8
#   .y.   group1               group2    effsize facet            n1    n2 magnitude 
# * <chr> <chr>                <chr>       <dbl> <fct>         <int> <int> <ord>     
# 1 lfc   "Non\nRedistributed" Nuclear   -0.261  TARDBP mutant 97592   201 small     
# 2 lfc   "Non\nRedistributed" Cytoplasm -0.275  TARDBP mutant 97592   248 small     
# 3 lfc   "Nuclear"            Cytoplasm -0.0156 TARDBP mutant   201   248 negligible
# 4 lfc   "Non\nRedistributed" Nuclear   -0.234  VCP mutant    97482   326 small     
# 5 lfc   "Non\nRedistributed" Cytoplasm -0.331  VCP mutant    97482   233 small     
# 6 lfc   "Nuclear"            Cytoplasm -0.109  VCP mutant      326   233 negligible
# 7 lfc   "Non\nRedistributed" Nuclear   -0.264  VCP knock-in  97501   324 small     
# 8 lfc   "Non\nRedistributed" Cytoplasm -0.296  VCP knock-in  97501   216 small     
# 9 lfc   "Nuclear"            Cytoplasm -0.0361 VCP knock-in    324   216 negligible


# # correlate stat with sum_score
# ggscatter(rnact_rna_view.untreated.nuc_vs_cyt.als_vs_ctrl, x = "stat", y = "sum_score", #color = "Classification", palette = rev(c("firebrick2", "grey", "dodgerblue2")),
#           cor.coef = TRUE, cor.coeff.args = list(method = "pearson", label.y = 7.5, label.x = -4), cor.coef.size = 3, xlab = expression(Stat~ALS~vs~CTRL), ylab = expression(RNAct~score), size = 0.5) + 
#   theme_classic() + theme(legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) +
#   geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") +
#   facet_wrap(~mutation)


```


## Transite mRNA motifs

Can be run through server at https://transite.mit.edu/ or locally on R https://github.com/kkrismer/transite/blob/ea66ffac0fdf7598c9e915ad5bfdddde1c4c463e/R/main.R 
https://github.com/kkrismer/transite/blob/master/vignettes/spma.Rmd and get sequences with biomart https://rdrr.io/bioc/biomaRt/man/getSequence.html

Transite motif database contains only 174 motifs https://transite.mit.edu/#tab-2131-4 from RBPDB and CISBP-RNA

Use Transite to scan the more extensive ATtRACT database.

We use TMSA rather than SPMA
TMSA: Transcript set motif Analysis focuses on significantly upregulated and downregulated sets of transcripts and identifies RBPs whose binding sites are overrepresented among those transcripts
SPMA: Spectrum Motif Analysis to assess how RBP targets are distrubted across a spectrum of transcripts ordered by LFC

```{r}
# # run this script in Rstudio server OnDemand Terminal Tab (not with conda env r4.0.3) with:
# # cd /nemo/home/ziffo/home/projects/motor-neuron-mislocalisation/scripts
# # sbatch -N 1 -c 8 --mem=72G -t 24:00:00 --wrap="Rscript /nemo/home/ziffo/home/projects/motor-neuron-mislocalisation/scripts/transcrite_motif_enrichment.mrna.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=transite_mrna
# # sbatch --cpus-per-task=10 --mem=300G -N 1 --partition=hmem -t 24:00:00 --wrap="Rscript /nemo/home/ziffo/home/projects/motor-neuron-mislocalisation/scripts/transcrite_motif_enrichment.mrna.R" --mail-type=ALL,ARRAY_TASKS --mail-user=oliver.ziff@crick.ac.uk --job-name=transite_mrna
# # .libPaths("/nemo/lab/patanir/home/users/ziffo/R/x86_64-pc-linux-gnu-library/4.0")
# library(tidyverse)
# library(biomaRt)
# library(transite)
# library(here)
# nemo_path = here("/nemo/lab/patanir/home/users/ziffo")
# proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
# load(here(proj_path,"scripts/untreated_deseq_objects.RData"))
# 
# # Sys.setenv("http_proxy" = "http://my.proxy.org:9999")
# # options(RCurlOptions = list(proxy="uscache.kcc.com:80",proxyuserpwd="------:-------"))
# # httr::set_config(httr::config(ssl_verifypeer = FALSE))
# # httr::set_config(httr::config(ssl_cipher_list = "DEFAULT@SECLEVEL=1"))
# ensembl = useEnsembl(biomart='ensembl', dataset='hsapiens_gene_ensembl', mirror = "uswest")# Select biomaRt database (genes) and dataset (hsapiens)
# 
# # define foreground & background sets ------------------------------------
# tardbp_nuc_redistributed <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0)
# tardbp_cyt_redistributed <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) 
# tardbp_background <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj > 0.05) %>% slice_sample(n=10000)
# 
# vcp_r155c_nuc_redistributed <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0)
# vcp_r155c_cyt_redistributed <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) 
# vcp_r155c_background <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj > 0.05) %>% slice_sample(n=10000)
# 
# vcp_r191q_nuc_redistributed <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0)
# vcp_r191q_cyt_redistributed <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) 
# vcp_r191q_background <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj > 0.05) %>% slice_sample(n=10000)
# 
# vcp_iso_nuc_redistributed <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0)
# vcp_iso_cyt_redistributed <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) 
# vcp_iso_background <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj > 0.05) %>% slice_sample(n=10000)
# 
# # get gene_exon sequences of sets
# tardbp_nuc_redistributed.gene_exon.getseq = biomaRt::getSequence(id = tardbp_nuc_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# tardbp_cyt_redistributed.gene_exon.getseq = biomaRt::getSequence(id = tardbp_cyt_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# tardbp_background.gene_exon.getseq = biomaRt::getSequence(id = tardbp_background$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# 
# vcp_r155c_nuc_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_r155c_nuc_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_r155c_cyt_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_r155c_cyt_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_r155c_background.gene_exon.getseq = biomaRt::getSequence(id = vcp_r155c_background$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# 
# vcp_r191q_nuc_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_r191q_nuc_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_r191q_cyt_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_r191q_cyt_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_r191q_background.gene_exon.getseq = biomaRt::getSequence(id = vcp_r191q_background$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable")
# 
# vcp_iso_nuc_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_iso_nuc_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_iso_cyt_redistributed.gene_exon.getseq = biomaRt::getSequence(id = vcp_iso_cyt_redistributed$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# vcp_iso_background.gene_exon.getseq = biomaRt::getSequence(id = vcp_iso_background$transcript_id, type = "ensembl_transcript_id", seqType = "gene_exon", mart = ensembl) %>% filter(`gene_exon` != "Sequence unavailable", !grepl("N",gene_exon))
# 
# # name sequences as: ENST ID | gene_exon - required to cache results
# tardbp_nuc_redistributed.gene_exon_seqs = gsub("T", "U", tardbp_nuc_redistributed.gene_exon.getseq$`gene_exon`)
# names(tardbp_nuc_redistributed.gene_exon_seqs) <- paste0(tardbp_nuc_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# tardbp_cyt_redistributed.gene_exon_seqs = gsub("T", "U", tardbp_cyt_redistributed.gene_exon.getseq$`gene_exon`)
# names(tardbp_cyt_redistributed.gene_exon_seqs) <- paste0(tardbp_cyt_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# tardbp_nuc_cyt_redistributed.gene_exon.foreground_sets <- list(tardbp_nuc_redistributed.gene_exon_seqs, tardbp_cyt_redistributed.gene_exon_seqs)
# tardbp_background.gene_exon_seqs = gsub("T", "U", tardbp_background.gene_exon.getseq$`gene_exon`)
# names(tardbp_background.gene_exon_seqs) <- paste0(tardbp_background.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# 
# vcp_r155c_nuc_redistributed.gene_exon_seqs = gsub("T", "U", vcp_r155c_nuc_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_r155c_nuc_redistributed.gene_exon_seqs) <- paste0(vcp_r155c_nuc_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_r155c_cyt_redistributed.gene_exon_seqs = gsub("T", "U", vcp_r155c_cyt_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_r155c_cyt_redistributed.gene_exon_seqs) <- paste0(vcp_r155c_cyt_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_r155c_nuc_cyt_redistributed.gene_exon.foreground_sets <- list(vcp_r155c_nuc_redistributed.gene_exon_seqs, vcp_r155c_cyt_redistributed.gene_exon_seqs)
# vcp_r155c_background.gene_exon_seqs = gsub("T", "U", vcp_r155c_background.gene_exon.getseq$`gene_exon`)
# names(vcp_r155c_background.gene_exon_seqs) <- paste0(vcp_r155c_background.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# 
# vcp_r191q_nuc_redistributed.gene_exon_seqs = gsub("T", "U", vcp_r191q_nuc_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_r191q_nuc_redistributed.gene_exon_seqs) <- paste0(vcp_r191q_nuc_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_r191q_cyt_redistributed.gene_exon_seqs = gsub("T", "U", vcp_r191q_cyt_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_r191q_cyt_redistributed.gene_exon_seqs) <- paste0(vcp_r191q_cyt_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_r191q_nuc_cyt_redistributed.gene_exon.foreground_sets <- list(vcp_r191q_nuc_redistributed.gene_exon_seqs, vcp_r191q_cyt_redistributed.gene_exon_seqs)
# vcp_r191q_background.gene_exon_seqs = gsub("T", "U", vcp_r191q_background.gene_exon.getseq$`gene_exon`)
# names(vcp_r191q_background.gene_exon_seqs) <- paste0(vcp_r191q_background.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# 
# vcp_iso_nuc_redistributed.gene_exon_seqs = gsub("T", "U", vcp_iso_nuc_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_iso_nuc_redistributed.gene_exon_seqs) <- paste0(vcp_iso_nuc_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_iso_cyt_redistributed.gene_exon_seqs = gsub("T", "U", vcp_iso_cyt_redistributed.gene_exon.getseq$`gene_exon`)
# names(vcp_iso_cyt_redistributed.gene_exon_seqs) <- paste0(vcp_iso_cyt_redistributed.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# vcp_iso_nuc_cyt_redistributed.gene_exon.foreground_sets <- list(vcp_iso_nuc_redistributed.gene_exon_seqs, vcp_iso_cyt_redistributed.gene_exon_seqs)
# vcp_iso_background.gene_exon_seqs = gsub("T", "U", vcp_iso_background.gene_exon.getseq$`gene_exon`)
# names(vcp_iso_background.gene_exon_seqs) <- paste0(vcp_iso_background.gene_exon.getseq$ensembl_transcript_id, "|gene_exon")
# 
# # run cached version of TSMA with all Transite motifs (recommended):
# tardbp_gene_exon.transite <- run_matrix_tsma(tardbp_nuc_cyt_redistributed.gene_exon.foreground_sets, tardbp_background.gene_exon_seqs)
# vcp_r155c_gene_exon.transite <- run_matrix_tsma(vcp_r155c_nuc_cyt_redistributed.gene_exon.foreground_sets, vcp_r155c_background.gene_exon_seqs)
# vcp_r191q_gene_exon.transite <- run_matrix_tsma(vcp_r191q_nuc_cyt_redistributed.gene_exon.foreground_sets, vcp_r191q_background.gene_exon_seqs)
# vcp_iso_gene_exon.transite <- run_matrix_tsma(vcp_iso_nuc_cyt_redistributed.gene_exon.foreground_sets, vcp_iso_background.gene_exon_seqs)
# print("3'UTR completed")
# 
# save(tardbp_gene_exon.transite, vcp_r155c_gene_exon.transite, vcp_r191q_gene_exon.transite, vcp_iso_gene_exon.transite,
#      file = here(proj_path, "expression/rna-rbp-interaction/transite/transite.gene_exon.redistributed_transcript.Rdata"))
# ### Spectrum Motif Analysis SPMA Examines the distribution of RBP binding sites across the entire spectrum of transcripts, sorted according to their fold change
# # untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res %>% left_join(ens2refseq) %>% filter(refseq != "") %>% select(refseq, log2FoldChange) %>% write.csv(here(proj_path,"expression/rna-rbp-interaction/transite/SPMA.untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.csv")
# # SPMA takes too long
```

upload csv to https://transite.mit.edu/ and run on their server - takes a few hours
set foreground csv
set background custom as csv
sequence region - run 3 times each for mature mRNA 3'UTR & 5'UTR separately. This defines the sequence region that will be analyzed for putative RBP binding sites. Usually, RNA-binding proteins that change the stability of its targets bind to the 3' UTR
interpret results https://transite.mit.edu/runs/f05444b6cba4bb29857883844525f7/index.html


Plot Top motifs
https://omarwagih.github.io/ggseqlogo/
https://bioconductor.org/packages/release/bioc/vignettes/motifStack/inst/doc/motifStack_HTML.html

```{r}
# load(here(proj_path, "expression/rna-rbp-interaction/transite/transite.gene_exon.redistributed_transcript.Rdata"))
# 
# tardbp_gene_exon.transite.nuclear = tardbp_gene_exon.transite$enrichment_dfs[[1]]
# tardbp_gene_exon.transite.cytoplasm = tardbp_gene_exon.transite$enrichment_dfs[[2]]
# vcp_r155c_gene_exon.transite.nuclear = vcp_r155c_gene_exon.transite$enrichment_dfs[[1]]
# vcp_r155c_gene_exon.transite.cytoplasm = vcp_r155c_gene_exon.transite$enrichment_dfs[[2]]
# vcp_r191q_gene_exon.transite.nuclear = vcp_r191q_gene_exon.transite$enrichment_dfs[[1]]
# vcp_r191q_gene_exon.transite.cytoplasm = vcp_r191q_gene_exon.transite$enrichment_dfs[[2]]
# vcp_iso_gene_exon.transite.nuclear = vcp_iso_gene_exon.transite$enrichment_dfs[[1]]
# vcp_iso_gene_exon.transite.cytoplasm = vcp_iso_gene_exon.transite$enrichment_dfs[[2]]
# 
# tardbp_gene_exon.transite.nuclear %>% filter(adj_p_value < 0.05) # 9
# tardbp_gene_exon.transite.cytoplasm %>% filter(adj_p_value < 0.05) # 7
# vcp_r155c_gene_exon.transite.nuclear %>% filter(adj_p_value < 0.05) #0
# vcp_r155c_gene_exon.transite.cytoplasm %>% filter(adj_p_value < 0.05) #0
# vcp_r191q_gene_exon.transite.nuclear %>% filter(adj_p_value < 0.05) # 64
# vcp_r191q_gene_exon.transite.cytoplasm %>% filter(adj_p_value < 0.05) # 60
# vcp_iso_gene_exon.transite.nuclear %>% filter(adj_p_value < 0.05) # 102
# vcp_iso_gene_exon.transite.cytoplasm %>% filter(adj_p_value < 0.05) #33
# 
# # How many unique motifs are tested?
# vcp_iso_gene_exon.transite.cytoplasm %>% distinct(motif_id) # 174
# # How many unique RBPs are in the motifs?
# vcp_iso_gene_exon.transite.cytoplasm %>% pull(motif_rbps) %>% str_split(", ") %>% unlist(., recursive = FALSE) %>% unique() #142 unique RBPs
# 
# # overlap
# transite.tmsa.mrna.join = rename(select(tardbp_gene_exon.transite.nuclear,-p_value_n), tardbp.nuc.enrichment = enrichment, tardbp.nuc.pval = p_value, tardbp.nuc.padj = adj_p_value) %>% 
#   left_join(rename(select(tardbp_gene_exon.transite.cytoplasm,-p_value_n), tardbp.cyt.enrichment = enrichment, tardbp.cyt.pval = p_value, tardbp.cyt.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_r155c_gene_exon.transite.nuclear,-p_value_n), vcp_r155c.nuc.enrichment = enrichment, vcp_r155c.nuc.pval = p_value, vcp_r155c.nuc.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_r155c_gene_exon.transite.cytoplasm,-p_value_n), vcp_r155c.cyt.enrichment = enrichment, vcp_r155c.cyt.pval = p_value, vcp_r155c.cyt.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_r191q_gene_exon.transite.nuclear,-p_value_n), vcp_r191q.nuc.enrichment = enrichment, vcp_r191q.nuc.pval = p_value, vcp_r191q.nuc.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_r191q_gene_exon.transite.cytoplasm,-p_value_n), vcp_r191q.cyt.enrichment = enrichment, vcp_r191q.cyt.pval = p_value, vcp_r191q.cyt.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_iso_gene_exon.transite.nuclear,-p_value_n), vcp_ki.nuc.enrichment = enrichment, vcp_ki.nuc.pval = p_value, vcp_ki.nuc.padj = adj_p_value)) %>% 
#   left_join(rename(select(vcp_iso_gene_exon.transite.cytoplasm,-p_value_n), vcp_ki.cyt.enrichment = enrichment, vcp_ki.cyt.pval = p_value, vcp_ki.cyt.padj = adj_p_value)) %>% 
#   filter(tardbp.nuc.padj < 0.05 | tardbp.cyt.padj < 0.05 | vcp_r155c.nuc.padj < 0.05 | vcp_r155c.cyt.padj < 0.05 | vcp_r191q.nuc.padj < 0.05 | vcp_r191q.cyt.padj < 0.05 | vcp_ki.nuc.padj < 0.05 | vcp_ki.cyt.padj < 0.05 ) %>%
#   mutate(sum_enrichment = (tardbp.nuc.enrichment + tardbp.cyt.enrichment + vcp_r155c.nuc.enrichment + vcp_r155c.cyt.enrichment + vcp_r191q.nuc.enrichment + vcp_r191q.cyt.enrichment + vcp_ki.nuc.enrichment + vcp_ki.cyt.enrichment),
#          sum_nuc_enrichment = (tardbp.nuc.enrichment + vcp_r155c.nuc.enrichment + vcp_r191q.nuc.enrichment + vcp_ki.nuc.enrichment),
#          sum_cyt_enrichment = (tardbp.cyt.enrichment + vcp_r155c.cyt.enrichment + vcp_r191q.cyt.enrichment + vcp_ki.cyt.enrichment),
#          motif_id = case_when(motif_id == "244_7543047" ~ "M274_0.6", motif_id == "790_10094314" ~ "M331_0.6", motif_id == "950_7908267" ~ "M349_0.6", motif_id == "682_10811881" ~ "M325_0.6", 
#    motif_id == "784_7972035" ~ "M330_0.6", motif_id == "951_12324455" ~ "M350_0.6", motif_id == "953_7543047" ~ "M352_0.6", TRUE ~ motif_id)) %>% as_tibble() %>% arrange(-sum_enrichment)
# transite.tmsa.mrna.join
# 
# transite.tmsa.mrna.join
# transite.tmsa.mrna.join %>% arrange(-sum_cyt_enrichment)
# # motif_id    motif_r…¹ tardb…² tardb…³ tardb…⁴ tardb…⁵ tardb…⁶ tardb…⁷ vcp_r…⁸ vcp_r…⁹ vcp_r…˟ vcp_r…˟
# #    <chr>       <chr>       <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
# #  1 M085_0.6    ZCRB1       1.04    0.605   0.893    1.14 1.30e-2 0.113     1.05   0.616    0.979    1.34
# #  2 M160_0.6    KHDRBS1     0.932   0.153   0.457    1.11 9.99e-3 0.109     1.03   0.587    0.979    1.14
# #  3 M082_0.6    YBX2, CS…   0.962   0.525   0.839    1.13 1.20e-2 0.113     0.944  0.502    0.979    1.09
# #  4 M001_0.6    A1CF        0.951   0.514   0.837    1.01 8.50e-1 0.923     1.09   0.373    0.979    1.29
# #  5 M073_0.6    STAR-PAP    0.896   0.227   0.533    1.06 4.61e-1 0.720     1.32   0.0130   0.377    1.03
# #  6 M136_0.6    SNRPB2      0.974   0.718   0.926    1.21 9.09e-4 0.0263    1.14   0.155    0.897    1.10
# #  7 M163_0.6    IGF2BP3     1.05    0.292   0.583    1.11 2.00e-2 0.139     0.947  0.423    0.979    1.14
# #  8 M162_0.6    PABPC5      1.07    0.129   0.440    1.18 2.95e-5 0.00256   1.03   0.637    0.979    1.15
# #  9 M032_0.6    IGF2BP2     1.05    0.359   0.664    1.09 6.69e-2 0.271     1.01   0.881    0.979    1.22
# # 10 661_1717938 SNRPA       1.03    0.655   0.919    1.22 5.92e-4 0.0206    1.10   0.324    0.979    1.04
# transite.tmsa.mrna.join %>% arrange(-sum_nuc_enrichment)
# # motif_id    motif_r…¹ tardb…² tardb…³ tardb…⁴ tardb…⁵ tardb…⁶ tardb…⁷ vcp_r…⁸ vcp_r…⁹ vcp_r…˟ vcp_r…˟
# #    <chr>       <chr>       <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
# #  1 M274_0.6    SFRS2       1.14  0.0410   0.228    1.08    0.210   0.485   0.939 0.560     0.979   0.845
# #  2 M023_0.6    HNRNPA1L…   1.02  0.826    0.948    0.938   0.428   0.717   1.13  0.270     0.979   1.05 
# #  3 M022_0.6    HNRNPA1,…   1.00  0.998    1        0.965   0.631   0.802   1.16  0.180     0.920   1.18 
# #  4 M153_0.6    LIN28A, …   1.04  0.439    0.753    1.01    0.859   0.923   1.03  0.714     0.979   0.850
# #  5 662_1717938 SNRPA       0.985 0.852    0.948    1.09    0.129   0.393   1.16  0.0979    0.897   1.02 
# #  6 M145_0.6    RBM5        1.14  0.00256  0.0496   1.00    0.935   0.946   1.04  0.605     0.979   0.841
# #  7 M040_0.6    MSI1, MS…   0.960 0.681    0.919    0.946   0.551   0.762   0.960 0.769     0.979   1.01 
# #  8 M167_0.6    MSI1, MS…   0.960 0.689    0.919    0.946   0.542   0.761   0.960 0.764     0.979   1.01 
# #  9 M142_0.6    RBM42       1.11  0.212    0.526    1.13    0.116   0.367   1.32  0.00899   0.377   0.931
# # 10 M024_0.6    HNRNPA2B…   1.04  0.635    0.899    0.968   0.677   0.818   1.05  0.673     0.979   1.11
# 
# transite.tmsa.mrna.join %>% filter(tardbp.nuc.padj < 0.05 | tardbp.cyt.padj < 0.05 | vcp_r155c.nuc.padj < 0.05 | vcp_r155c.cyt.padj < 0.05 | vcp_r191q.nuc.padj < 0.05 | vcp_r191q.cyt.padj < 0.05 | vcp_ki.nuc.padj < 0.05 | vcp_ki.cyt.padj < 0.05 )
# transite.tmsa.mrna.join %>% filter(tardbp.nuc.padj < 0.05, vcp_r155c.nuc.padj < 0.05, vcp_r191q.nuc.padj < 0.05, vcp_ki.nuc.padj < 0.05)
# transite.tmsa.mrna.join %>% filter(tardbp.cyt.padj < 0.05, vcp_r155c.cyt.padj < 0.05, vcp_r191q.cyt.padj < 0.05, vcp_ki.cyt.padj < 0.05)
# transite.tmsa.mrna.join %>% filter(tardbp.cyt.padj < 0.05, vcp_r155c.cyt.padj < 0.05) # 
# # motif_id      rbp             tardbp.nuc.enrichment tardbp.nuc.padj tardbp.cyt.enrichment tardbp.cyt.padj vcp_r155c.nuc.enrichment vcp_r155c.nuc.adj_…¹ vcp_r…² vcp_r…³ vcp_r…⁴ vcp_r…⁵ vcp_r…⁶ vcp_r…⁷ vcp_k…⁸ vcp_k…⁹ vcp_k…˟ vcp_k…˟
# #   <chr>         <chr>                           <dbl>                  <dbl>                 <dbl>                  <dbl>                    <dbl>                <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>   <dbl>
# # 1 M144_0.6      PABPC1L, PABPC3                  1.26                 0.0499                  1.26                0.00361                     1.06                0.772    1.14   0.472    1.35 0.00132    1.13   0.205   0.992  0.947     1.00   0.984
# # 2 1216_19457263 KHDRBS3                          1.14                 0.440        
# 
# 
# 
# 
# 
# # regions
# load(here(proj_path, "expression/rna-rbp-interaction/transite/transite.3utr.redistributed_transcript.Rdata"))
# load(here(proj_path, "expression/rna-rbp-interaction/transite/transite.5utr.redistributed_transcript.Rdata"))
# load(here(proj_path, "expression/rna-rbp-interaction/transite/transite.transcript_exon_intron.redistributed_transcript.Rdata"))
# tardbp_3utr.transite %>% glimpse
# tardbp_3utr.transite.nuclear = tardbp_3utr.transite$enrichment_dfs[[1]]
# tardbp_3utr.transite.cytoplasm = tardbp_3utr.transite$enrichment_dfs[[2]]
# vcp_r155c_3utr.transite.nuclear = vcp_r155c_3utr.transite$enrichment_dfs[[1]]
# vcp_r155c_3utr.transite.cytoplasm = vcp_r155c_3utr.transite$enrichment_dfs[[2]]
# vcp_r191q_3utr.transite.nuclear = vcp_r191q_3utr.transite$enrichment_dfs[[1]]
# vcp_r191q_3utr.transite.cytoplasm = vcp_r191q_3utr.transite$enrichment_dfs[[2]]
# vcp_iso_3utr.transite.nuclear = vcp_iso_3utr.transite$enrichment_dfs[[1]]
# vcp_iso_3utr.transite.cytoplasm = vcp_iso_3utr.transite$enrichment_dfs[[2]]
# 
# tardbp_5utr.transite.nuclear = tardbp_5utr.transite$enrichment_dfs[[1]]
# tardbp_5utr.transite.cytoplasm = tardbp_5utr.transite$enrichment_dfs[[2]]
# vcp_r155c_5utr.transite.nuclear = vcp_r155c_5utr.transite$enrichment_dfs[[1]]
# vcp_r155c_5utr.transite.cytoplasm = vcp_r155c_5utr.transite$enrichment_dfs[[2]]
# vcp_r191q_5utr.transite.nuclear = vcp_r191q_5utr.transite$enrichment_dfs[[1]]
# vcp_r191q_5utr.transite.cytoplasm = vcp_r191q_5utr.transite$enrichment_dfs[[2]]
# vcp_iso_5utr.transite.nuclear = vcp_iso_5utr.transite$enrichment_dfs[[1]]
# vcp_iso_5utr.transite.cytoplasm = vcp_iso_5utr.transite$enrichment_dfs[[2]]
# 
# tardbp_transcript_exon_intron.transite.nuclear = tardbp_transcript_exon_intron.transite$enrichment_dfs[[1]]
# tardbp_transcript_exon_intron.transite.cytoplasm = tardbp_transcript_exon_intron.transite$enrichment_dfs[[2]]
# vcp_r155c_transcript_exon_intron.transite.nuclear = vcp_r155c_transcript_exon_intron.transite$enrichment_dfs[[1]]
# vcp_r155c_transcript_exon_intron.transite.cytoplasm = vcp_r155c_transcript_exon_intron.transite$enrichment_dfs[[2]]
# vcp_r191q_transcript_exon_intron.transite.nuclear = vcp_r191q_transcript_exon_intron.transite$enrichment_dfs[[1]]
# vcp_r191q_transcript_exon_intron.transite.cytoplasm = vcp_r191q_transcript_exon_intron.transite$enrichment_dfs[[2]]
# vcp_iso_transcript_exon_intron.transite.nuclear = vcp_iso_transcript_exon_intron.transite$enrichment_dfs[[1]]
# vcp_iso_transcript_exon_intron.transite.cytoplasm = vcp_iso_transcript_exon_intron.transite$enrichment_dfs[[2]]
# 
# tardbp_3utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# tardbp_3utr.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r155c_3utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r155c_3utr.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r191q_3utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r191q_3utr.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_iso_3utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_iso_3utr.transite.cytoplasm %>% filter(adj_p_value < 0.05) # 22
# 
# tardbp_5utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# tardbp_5utr.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r155c_5utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r155c_5utr.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r191q_5utr.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r191q_5utr.transite.cytoplasm %>% filter(adj_p_value < 0.05) # SRSF1
# vcp_iso_5utr.transite.nuclear %>% filter(adj_p_value < 0.05) # SRSF7
# vcp_iso_5utr.transite.cytoplasm %>% filter(adj_p_value < 0.05) #0
# 
# tardbp_transcript_exon_intron.transite.nuclear %>% filter(adj_p_value < 0.05)
# tardbp_transcript_exon_intron.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r155c_transcript_exon_intron.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r155c_transcript_exon_intron.transite.cytoplasm %>% filter(adj_p_value < 0.05)
# vcp_r191q_transcript_exon_intron.transite.nuclear %>% filter(adj_p_value < 0.05)
# vcp_r191q_transcript_exon_intron.transite.cytoplasm %>% filter(adj_p_value < 0.05) # SRSF1
# vcp_iso_transcript_exon_intron.transite.nuclear %>% filter(adj_p_value < 0.05) # SRSF7
# vcp_iso_transcript_exon_intron.transite.cytoplasm %>% filter(adj_p_value < 0.05) #0


# download CISBP-RNA motif database from http://cisbp-rna.ccbr.utoronto.ca/
# download RBPDB motif database from http://rbpdb.ccbr.utoronto.ca/download.php
CISBP.M085_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M085_0.6.txt")) %>% select(-Pos) %>% t # matrix of position frequency matrix (rows = letters, column = position)
CISBP.M160_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M160_0.6.txt")) %>% select(-Pos) %>% t
CISBP.M082_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M082_0.6.txt")) %>% select(-Pos) %>% t # matrix of position frequency matrix (rows = letters, column = position)
CISBP.M274_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M274_0.6.txt")) %>% select(-Pos) %>% t # matrix of position frequency matrix (rows = letters, column = position)
CISBP.M023_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M023_0.6.txt")) %>% select(-Pos) %>% t
CISBP.M022_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M022_0.6.txt")) %>% select(-Pos) %>% t # matrix of position frequency matrix (rows = letters, column = position)
# For motifs without the CISBP IDs then find CISBP motif ID name from searching http://cisbp-rna.ccbr.utoronto.ca/TFreport.php?searchTF=T38010_0.6&orderby=M
# CISBP.M349_0.6 = read_tsv(here(nemo_path, "genomes/rbp-databases/CISBP-RNA/pwms_all_motifs/M349_0.6.txt")) %>% select(-Pos) %>% t # 950_7908267 == M349_0.6

# Merge into a single plot
require(ggseqlogo)
library(ggtext)
motifs.merged = list(CISBP.M274_0.6, CISBP.M023_0.6, CISBP.M022_0.6, CISBP.M085_0.6, CISBP.M160_0.6, CISBP.M082_0.6)
names(motifs.merged) = c("Nuclear CISBP 274\nSRSF2", "Nuclear CISBP 23\nHNRNPA1L2 HNRNPA3", "Nuclear CISBP 22\nHNRNPA1 HNRNPA3", "Cytoplasm CISBP 85\nZCRB1", "Cytoplasm CISBP 160\nKHDRBS1", "Cytoplasm CISBP 82\nYBX2 YBX3")
motifs.merged.plot = ggseqlogo(motifs.merged, nrow = 2) + facet_wrap(~seq_group, scales = "free_x", nrow = 2, ncol = 3) +#, labeller = "label_parsed") + 
  theme_void() + theme(strip.text.x = element_text(size = 8))
motifs.merged.plot

```



## CLIP regions

```{r}
clip_5utr.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_5utr.HepG2_K562_Hela.rds"))
clip_3utr.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_3utr.HepG2_K562_Hela.rds"))
clip_cds.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_cds.HepG2_K562_Hela.rds"))
# clip_exon.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_exon.HepG2_K562_Hela.rds"))
clip_intron.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_intron.HepG2_K562_Hela.rds"))
clip_transcript.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_transcript.HepG2_K562_Hela.rds"))

clip_5utr.HepG2_K562_Hela.tx = clip_5utr.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_5utr.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)

clip_3utr.HepG2_K562_Hela.tx = clip_3utr.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_3utr.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)

clip_cds.HepG2_K562_Hela.tx = clip_cds.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_cds.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)

clip_intron.HepG2_K562_Hela.tx = clip_intron.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_intron.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)


# Compare transcript regions Cohens D effect sizes:
clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind = bind_rows(mutate(clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl, region = "5UTR"), mutate(clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl, region = "3UTR"), mutate(clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl, region = "CDS"), mutate(clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl, region = "Intron"))

clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size = clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind %>% group_by(region, mutation) %>% cohens_d(log10.mean_iclip_score ~ mislocalisation, var.equal = TRUE) 
clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size
clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.cohens_effect_size.filt = filter(clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size, group1 != "Nuc") %>% mutate(effsize = effsize * -1, group2 = gsub("Nuc","Nuclear",group2), group2 = gsub("Cyto","Cytoplasm",group2), group2 = factor(group2, levels = c("Nuclear","Cytoplasm")))

clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size.violin = violin_plot(clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.cohens_effect_size.filt, color_by = "region", continuous = "effsize", ylims = c(-0.9,1), facet_by = "group2", facet_ncol = 4,null_value = 0, plot_violin = FALSE, plot_stats = FALSE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(0,0,0.95,0.8,0.85,0.78),2), ylabel = expression(Effect~size~CLIP~xlinks), xlabel = "", cols = pal_npg("nrc")(4), dpi = 300, arrow_labels = FALSE)
clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size.violin



clip_intron.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(across(is.numeric, sum))

```

CLIP coverage track

```{r}
library(here)
proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
script_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts/ipsc_mn_als_meta")
shared_path = here("/nemo/project/proj-luscombn-patani/working")
collab_path = here("/nemo/lab/patanir/home/users/ziffo/patani-collab")
source(here(nemo_path,"scripts/functions/R_workspace.R"))
# load(here(proj_path,"scripts/untreated_deseq_objects.RData"))

ctrl_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_nuclear.bam") # CTRL3 removed
ctrl_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_cytoplasm.bam") # CTRL3 removed to balance reads
tardbp_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_nuclear.bam")
tardbp_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_cytoplasm.bam")
vcp_r155c_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_nuclear.bam")
vcp_r155c_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_cytoplasm.bam")
vcp_r191q_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_nuclear.bam")
vcp_r191q_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_cytoplasm.bam")
vcp_iso_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_nuclear.bam")
vcp_iso_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_cytoplasm.bam")

library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
txdb <- makeTxDbFromGFF(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf"), format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations & scaffolds which are also in the BAM

track_viewer_nuc_cyt(level = "gene", gene_name = "NEFL", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6, clip_track = "FUS")

```


## Merge

```{r}
sequence_features.rna_rbp_interactions.merged <- plot_grid(
  plot_grid(untreated.nuc_vs_cyt.als_vs_ctrl.width.tx.violin, untreated.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin, 
  untreated.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin, rnact.rna_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, ncol = 2, nrow = 2, labels = "auto"), 
  plot_grid(motifs.merged.plot, clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.bind.cohens_effect_size.violin, ncol = 2, labels = c("e","f"), rel_widths = c(3,2)),
  nrow = 2, rel_heights = c(2,1.5))
# sequence_features.rna_rbp_interactions.merged
ggsave(sequence_features.rna_rbp_interactions.merged, filename = here(proj_path, "expression/deseq2/Figure3.sequence_features.rna_rbp_interactions.merged.png"), height = 8.5, width = 10)
ggsave(sequence_features.rna_rbp_interactions.merged, filename = here(proj_path, "expression/deseq2/Figure3.sequence_features.rna_rbp_interactions.merged.pdf"), height = 8.5, width = 10)

```



# Fig S7: Lineplot sequence features & GC content & PhastCons 

## Lineplots

```{r}
homo_sapiens.gtf = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.rds"))
Homo_sapiens.GRCh38.99.gc.phast = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.gc.phase.rds"))
homo_sapiens.gtf.spliced.transcript <- homo_sapiens.gtf %>% group_by(transcript_id) %>% select(transcript_id, transcript_biotype, width, type) %>% pivot_wider(c(transcript_id, transcript_biotype), names_from = type, values_from = width, values_fn = sum) %>% mutate(width = sum(exon, CDS, five_prime_utr, three_prime_utr, na.rm = TRUE)) %>% select(transcript_id, transcript_biotype, width)
homo_sapiens.gtf.5UTR.transcript <- homo_sapiens.gtf %>% filter(type == "five_prime_utr") %>% group_by(transcript_id) %>% summarise(five_prime_utr = mean(width, na.rm = TRUE)) # mean 5'UTR length isoform
homo_sapiens.gtf.3UTR.transcript <- homo_sapiens.gtf %>% filter(type == "three_prime_utr") %>% group_by(transcript_id) %>% summarise(three_prime_utr = mean(width, na.rm = TRUE)) # mean 3UTR length isoform
homo_sapiens.gtf.CDS.transcript <- homo_sapiens.gtf %>% filter(type == "CDS") %>% group_by(transcript_id) %>% summarise(CDS = mean(width, na.rm = TRUE))
homo_sapiens.gtf.exon.transcript <- homo_sapiens.gtf %>% filter(type == "exon") %>% group_by(transcript_id) %>% summarise(exon_number = n()) # total exon number - capped at 9???

# geom_smooth loess in nuc_vs_cyt
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCP R191Q knock-in")) %>%
  left_join(select(homo_sapiens.gtf.spliced.transcript, transcript_id, width)) %>% 
  left_join(select(homo_sapiens.gtf.5UTR.transcript, transcript_id, five_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.3UTR.transcript, transcript_id, three_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.CDS.transcript, transcript_id, CDS)) %>%
  left_join(select(homo_sapiens.gtf.exon.transcript, transcript_id, exon_number)) %>%
  left_join(select(Homo_sapiens.GRCh38.99.gc.phast, transcript_id, gc_content, phast)) %>%
  mutate(mislocalisation = case_when(padj < 0.05 & stat > 0 ~ "Nuc", padj < 0.05 & stat < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), 
         log10.width = log10(width), log10.five_prime_utr = log10(five_prime_utr), log10.three_prime_utr = log10(three_prime_utr), log10.CDS = log10(CDS), log10.exon_number = log10(exon_number),
         intronless = case_when(exon_number == 1 ~ "Intronless", exon_number >1 ~ "Intron Containing", TRUE ~ "NA"), mutation = factor(mutation, levels = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in")))


untreated.nuc_vs_cyt.als_vs_ctrl.res.tx %>% drop_na(log10.width, log2FoldChange) %>% group_by(mutation) %>% summarise(cor(log10.width, log2FoldChange))
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx %>% drop_na(log10.five_prime_utr, log2FoldChange) %>% group_by(mutation) %>% summarise(cor(log10.five_prime_utr, log2FoldChange)) 
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx %>% drop_na(log10.three_prime_utr, log2FoldChange) %>% group_by(mutation) %>% summarise(cor(log10.three_prime_utr, log2FoldChange)) 
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx %>% drop_na(log10.CDS, log2FoldChange) %>% group_by(mutation) %>% summarise(cor(log10.CDS, log2FoldChange)) 
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx %>% drop_na(log10.exon_number, log2FoldChange) %>% group_by(mutation) %>% summarise(cor(log10.exon_number, log2FoldChange))

###  Transcript Lengths
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.width.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = width, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "top") + #guides(colour = guide_legend(nrow = 2)) + #legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"),
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(Transcript~length))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.width.scatter

### 5'UTR length
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.5UTR.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = five_prime_utr, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(5~UTR~length))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.5UTR.scatter

### 3'UTR length
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.3UTR.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = three_prime_utr, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(3~UTR~length))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.3UTR.scatter

### Exon count
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.exon_number.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = exon_number, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(Exon~number))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.exon_number.scatter

### GC content
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.gc.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = gc_content, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(GC~content))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.gc.scatter

### PhastCons
untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.phast.scatter = ggplot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, aes(x = phast, y = log2FoldChange, colour = mutation, fill = mutation)) +
  geom_smooth() +  scale_colour_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) + scale_fill_manual(values = c("dodgerblue2", "firebrick2", "forestgreen", "gold")) +
  theme_oz() + theme(legend.position = "none") + 
  geom_hline(yintercept = 0, linetype = 2)  + ylab(label = expression(ALS~vs~CTRL~log[2]~FC~Nuc/Cyto) ) + xlab(label = expression(Conservation~score))
# untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.phast.scatter


# Violins

### 5'UTR length
untreated.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, color_by = "mislocalisation", continuous = "log10.five_prime_utr", ylims = c(1,3), facet_by = "mutation", facet_ncol = 4, 
                                                               plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(2.7,2.9,2.8),4),
                                                               ylabel = expression(log[10]~5~UTR~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin
# facet              .y.   group1               group2       n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>              <chr> <chr>                <chr>     <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP G298S       lfc   "Non\nRedistributed" Nucleus   83523   227  9649856. 0.64  1     ns                  2.7 <chr [2]>        1     2
#  2 TARDBP G298S       lfc   "Non\nRedistributed" Cytoplasm 83523   296 12201953  0.701 1     ns                  2.9 <chr [2]>        1     3
#  3 TARDBP G298S       lfc   "Nucleus"            Cytoplasm   227   296    32464. 0.509 1     ns                  2.8 <chr [2]>        2     3
#  4 VCP R155C          lfc   "Non\nRedistributed" Nucleus   83534   268 11267550  0.852 0.852 ns                  2.7 <chr [2]>        1     2
#  5 VCP R155C          lfc   "Non\nRedistributed" Cytoplasm 83534   244 10790020  0.112 0.336 ns                  2.9 <chr [2]>        1     3
#  6 VCP R155C          lfc   "Nucleus"            Cytoplasm   268   244    34400. 0.308 0.616 ns                  2.8 <chr [2]>        2     3
#  7 VCP R191Q          lfc   "Non\nRedistributed" Nucleus   83411   234  9831124. 0.845 1     ns                  2.7 <chr [2]>        1     2
#  8 VCP R191Q          lfc   "Non\nRedistributed" Cytoplasm 83411   401 17044918  0.507 1     ns                  2.9 <chr [2]>        1     3
#  9 VCP R191Q          lfc   "Nucleus"            Cytoplasm   234   401    47413  0.824 1     ns                  2.8 <chr [2]>        2     3
# 10 VCP R191Q knock-in lfc   "Non\nRedistributed" Nucleus   83208   297 12052834. 0.464 0.786 ns                  2.7 <chr [2]>        1     2
# 11 VCP R191Q knock-in lfc   "Non\nRedistributed" Cytoplasm 83208   541 22987006. 0.393 0.786 ns                  2.9 <chr [2]>        1     3
# 12 VCP R191Q knock-in lfc   "Nucleus"            Cytoplasm   297   541    84282. 0.239 0.717 ns                  2.8 <chr [2]>        2     3
# # A tibble: 12 × 8
#    .y.   group1               group2     effsize facet                  n1    n2 magnitude 
#  * <chr> <chr>                <chr>        <dbl> <fct>               <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nucleus    0.0142  TARDBP G298S       227039   360 negligible
#  2 lfc   "Non\nRedistributed" Cytoplasm -0.0222  TARDBP G298S       227039   419 negligible
#  3 lfc   "Nucleus"            Cytoplasm -0.0413  TARDBP G298S          360   419 negligible
#  4 lfc   "Non\nRedistributed" Nucleus    0.00575 VCP R155C          227068   407 negligible
#  5 lfc   "Non\nRedistributed" Cytoplasm  0.114   VCP R155C          227068   343 negligible
#  6 lfc   "Nucleus"            Cytoplasm  0.110   VCP R155C             407   343 negligible
#  7 lfc   "Non\nRedistributed" Nucleus    0.00856 VCP R191Q          226867   373 negligible
#  8 lfc   "Non\nRedistributed" Cytoplasm  0.0608  VCP R191Q          226867   578 negligible
#  9 lfc   "Nucleus"            Cytoplasm  0.0522  VCP R191Q             373   578 negligible
# 10 lfc   "Non\nRedistributed" Nucleus   -0.0532  VCP R191Q knock-in 226567   476 negligible
# 11 lfc   "Non\nRedistributed" Cytoplasm  0.0201  VCP R191Q knock-in 226567   775 negligible
# 12 lfc   "Nucleus"            Cytoplasm  0.0812  VCP R191Q knock-in    476   775 negligible


# GC content
untreated.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, color_by = "mislocalisation", continuous = "gc_content", ylims = c(0.3,0.7), facet_by = "mutation", facet_ncol = 4, 
                                                            stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(0.66,0.68,0.64),4), 
                                                            ylabel = "GC content", xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin
# facet         .y.   group1               group2        n1    n2 statistic     p p.adj p.adj.signif y.position groups        xmin  xmax
#   <fct>         <chr> <chr>                <chr>      <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 TARDBP mutant lfc   "Non\nRedistributed" Nucleus   222016   358 38110200  0.179 0.179 ns                 0.66 <chr [2]>        1     2
# 2 TARDBP mutant lfc   "Non\nRedistributed" Cytoplasm 222016   397 47402454. 0.009 0.02  *                  0.68 <chr [2]>        1     3
# 3 TARDBP mutant lfc   "Nucleus"            Cytoplasm    358   397    79186. 0.007 0.02  *                  0.64 <chr [2]>        2     3
# 4 VCP mutant    lfc   "Non\nRedistributed" Nucleus   221835   524 61007488. 0.049 0.148 ns                 0.66 <chr [2]>        1     2
# 5 VCP mutant    lfc   "Non\nRedistributed" Cytoplasm 221835   412 44730721  0.457 0.457 ns                 0.68 <chr [2]>        1     3
# 6 VCP mutant    lfc   "Nucleus"            Cytoplasm    524   412   100369  0.065 0.148 ns                 0.64 <chr [2]>        2     3
# 7 VCP knock-in  lfc   "Non\nRedistributed" Nucleus   221591   439 45444439  0.017 0.052 ns                 0.66 <chr [2]>        1     2
# 8 VCP knock-in  lfc   "Non\nRedistributed" Cytoplasm 221591   741 82089680  0.996 0.996 ns                 0.68 <chr [2]>        1     3
# 9 VCP knock-in  lfc   "Nucleus"            Cytoplasm    439   741   173312. 0.06  0.119 ns                 0.64 <chr [2]>        2     3
# # A tibble: 9 × 8
#   .y.   group1               group2      effsize facet             n1    n2 magnitude 
# * <chr> <chr>                <chr>         <dbl> <fct>          <int> <int> <ord>     
# 1 lfc   "Non\nRedistributed" Nucleus   -0.101    TARDBP mutant 227029   373 negligible
# 2 lfc   "Non\nRedistributed" Cytoplasm  0.131    TARDBP mutant 227029   416 negligible
# 3 lfc   "Nucleus"            Cytoplasm  0.233    TARDBP mutant    373   416 small     
# 4 lfc   "Non\nRedistributed" Nucleus    0.0984   VCP mutant    226818   560 negligible
# 5 lfc   "Non\nRedistributed" Cytoplasm -0.0445   VCP mutant    226818   440 negligible
# 6 lfc   "Nucleus"            Cytoplasm -0.149    VCP mutant       560   440 negligible
# 7 lfc   "Non\nRedistributed" Nucleus   -0.128    VCP knock-in  226582   466 negligible
# 8 lfc   "Non\nRedistributed" Cytoplasm  0.000991 VCP knock-in  226582   770 negligible
# 9 lfc   "Nucleus"            Cytoplasm  0.130    VCP knock-in     466   770 negligible


# PhastCons
untreated.nuc_vs_cyt.als_vs_ctrl.phast.tx.violin = violin_plot(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx, color_by = "mislocalisation", continuous = "phast", ylims = c(0,0.35), facet_by = "mutation", facet_ncol = 4, 
                                                           stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(0.3,0.34,0.32),4), 
                                                               ylabel = "Conservation score", xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# untreated.nuc_vs_cyt.als_vs_ctrl.phast.tx.violin
# facet         .y.   group1               group2        n1    n2 statistic             p        p.adj p.adj.signif y.position groups        xmin  xmax
#   <fct>         <chr> <chr>                <chr>      <int> <int>     <dbl>         <dbl>        <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
# 1 TARDBP mutant lfc   "Non\nRedistributed" Nucleus   187262   283 22832986. 0.0000566     0.00017      ***                0.3  <chr [2]>        1     2
# 2 TARDBP mutant lfc   "Non\nRedistributed" Cytoplasm 187262   290 25878552. 0.167         0.167        ns                 0.34 <chr [2]>        1     3
# 3 TARDBP mutant lfc   "Nucleus"            Cytoplasm    283   290    45463  0.025         0.051        ns                 0.32 <chr [2]>        2     3
# 4 VCP mutant    lfc   "Non\nRedistributed" Nucleus   187099   406 34202156. 0.000523      0.001        ***                0.3  <chr [2]>        1     2
# 5 VCP mutant    lfc   "Non\nRedistributed" Cytoplasm 187099   330 27259436  0.000235      0.000705     ***                0.34 <chr [2]>        1     3
# 6 VCP mutant    lfc   "Nucleus"            Cytoplasm    406   330    65186  0.53          0.53         ns                 0.32 <chr [2]>        2     3
# 7 VCP knock-in  lfc   "Non\nRedistributed" Nucleus   186900   360 27677302  0.00000000585 0.0000000176 ****               0.3  <chr [2]>        1     2
# 8 VCP knock-in  lfc   "Non\nRedistributed" Cytoplasm 186900   575 49117794. 0.000368      0.000736     ***                0.34 <chr [2]>        1     3
# 9 VCP knock-in  lfc   "Nucleus"            Cytoplasm    360   575   114654. 0.006         0.006        **                 0.32 <chr [2]>        2     3
# # A tibble: 9 × 8
#   .y.   group1               group2    effsize facet             n1    n2 magnitude 
# * <chr> <chr>                <chr>       <dbl> <fct>          <int> <int> <ord>     
# 1 lfc   "Non\nRedistributed" Nucleus    0.0418 TARDBP mutant 227029   373 negligible
# 2 lfc   "Non\nRedistributed" Cytoplasm  0.127  TARDBP mutant 227029   416 negligible
# 3 lfc   "Nucleus"            Cytoplasm  0.153  TARDBP mutant    373   416 negligible
# 4 lfc   "Non\nRedistributed" Nucleus    0.0841 VCP mutant    226818   560 negligible
# 5 lfc   "Non\nRedistributed" Cytoplasm  0.0350 VCP mutant    226818   440 negligible
# 6 lfc   "Nucleus"            Cytoplasm -0.0780 VCP mutant       560   440 negligible
# 7 lfc   "Non\nRedistributed" Nucleus   -0.0190 VCP knock-in  226582   466 negligible
# 8 lfc   "Non\nRedistributed" Cytoplasm  0.0786 VCP knock-in  226582   770 negligible
# 9 lfc   "Nucleus"            Cytoplasm  0.148  VCP knock-in     466   770 negligible
```


## CLIP countOverlaps

```{r}
clip_exon.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_exon.HepG2_K562_Hela.rds"))

# exons i.e. mRNA
clip_exon.HepG2_K562_Hela # 1,376,499 exons
clip_exon.HepG2_K562_Hela %>% count(transcript_id) # 227,818 transcripts. Transcripts have > 1 exon, as composed of multiple exons. Sum them.
clip_exon.HepG2_K562_Hela.tx = clip_exon.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_exon.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl

clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl %>% filter(padj < 0.05) %>% arrange(-sum_iclip_score) %>% print(n=50)

clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)

clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,3), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(2.1, 2.7, 2.0),4), 
                                                               ylabel = expression(log[10]~mRNA~CLIP~xlinks), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# facet        .y.   group1               group2     n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>   <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    208188   151  7319834. 5.98e-30 1.2 e-29 ****                2.6 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   208188   225 12561172  2.16e-33 6.48e-33 ****                3.1 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto      151   225    19151  3.6 e- 2 3.6 e- 2 *                   2.5 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    208209   180 10420272  6.22e-25 1.87e-24 ****                2.6 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   208209   175 10271138. 1.66e-23 3.32e-23 ****                3.1 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto      180   175    15625  8.98e- 1 8.98e- 1 ns                  2.5 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    207965   181 10088794  3.18e-27 6.36e-27 ****                2.6 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   207965   418 24089914. 4.97e-56 1.49e-55 ****                3.1 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto      181   418    39157  4.95e- 1 4.95e- 1 ns                  2.5 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    207997   189  9578390  2.98e-34 5.96e-34 ****                2.6 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   207997   378 19049791  2.27e-67 6.81e-67 ****                3.1 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto      189   378    36334. 7.39e- 1 7.39e- 1 ns                  2.5 <chr [2]>        2     3

```

## Merge

```{r}
## Merge with common legend
untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.legend = cowplot::get_legend(untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.width.scatter)  
untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.lineplots <- plot_grid(
  untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.width.scatter + theme(legend.position = "none") , untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.exon_number.scatter, untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.5UTR.scatter,
  untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.3UTR.scatter, untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.gc.scatter, untreated.nuc_vs_cyt.als_vs_ctrl.res.tx.phast.scatter,  
  ncol = 3, nrow = 2, labels = "auto") 
untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.violins = plot_grid(untreated.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin, untreated.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin, untreated.nuc_vs_cyt.als_vs_ctrl.phast.tx.violin, clip_exon.untreated.nuc_vs_cyt.als_vs_ctrl.violin, nrow = 4, labels = letters[7:10]) 
untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.merged.legend <- plot_grid(untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.legend, untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.lineplots, untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.violins, nrow = 3, rel_heights = c(0.05,1,1.7)) 
# untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.merged.legend
ggsave(untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.merged.legend, filename = here(proj_path, "expression/deseq2/untreated.nuc_vs_cyt.als_vs_ctrl.sequence_features.merged.png"), height = 12.5, width = 9)

```


# Fig S8 RNAct RBP view violins

```{r}
# TDP43
RNAct.TDP43.als.redistributed = bind_rows(mutate(RNAct.TDP43.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.TDP43.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.TDP43.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.TDP43.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
RNAct.TDP43.als.redistributed %>% my_summarise(catrapid_score_max_normalised, mislocalisation, mutation)

# Mislocalisation Nuclear vs Cytoplasm vs None
rnact.TDP43_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.TDP43.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(TDP43:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.TDP43_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# acet        .y.   group1               group2    n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>  <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    97606   185  7706410  5.68e- 4 1   e- 3 ***                  27 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   97606   250 10193113  6.78e- 6 2.03e- 5 ****                 29 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto     185   250    22554. 6.6 e- 1 6.6 e- 1 ns                   28 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    97623   213  8565073  8.63e- 6 2.59e- 5 ****                 27 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   97623   205  8601048. 5.03e- 4 1   e- 3 ***                  29 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto     213   205    22631  5.18e- 1 5.18e- 1 ns                   28 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    97512   204  9076405  3.1 e- 2 3.1 e- 2 *                    27 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   97512   325 12779515  1.62e- 9 4.86e- 9 ****                 29 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto     204   325    28994. 1.5 e- 2 3   e- 2 *                    28 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    97315   250 10429075  9.55e- 5 1.91e- 4 ***                  27 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   97315   476 17953244. 2.33e-17 6.99e-17 ****                 29 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto     250   476    53397  2.3 e- 2 2.3 e- 2 *                    28 <chr [2]>        2     3
# # A tibble: 12 × 8
#    .y.   group1               group2 effsize facet           n1    n2 magnitude 
#  * <chr> <chr>                <chr>    <dbl> <fct>        <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nuc    -0.231  TARDBP^G298S 97606   185 small     
#  2 lfc   "Non\nRedistributed" Cyto   -0.258  TARDBP^G298S 97606   250 small     
#  3 lfc   "Nuc"                Cyto   -0.0300 TARDBP^G298S   185   250 negligible
#  4 lfc   "Non\nRedistributed" Nuc    -0.304  VCP^R155C    97623   213 small     
#  5 lfc   "Non\nRedistributed" Cyto   -0.215  VCP^R155C    97623   205 small     
#  6 lfc   "Nuc"                Cyto    0.0957 VCP^R155C      213   205 negligible
#  7 lfc   "Non\nRedistributed" Nuc    -0.120  VCP^R191Q    97512   204 negligible
#  8 lfc   "Non\nRedistributed" Cyto   -0.300  VCP^R191Q    97512   325 small     
#  9 lfc   "Nuc"                Cyto   -0.207  VCP^R191Q      204   325 small     
# 10 lfc   "Non\nRedistributed" Nuc    -0.208  VCPki^R191Q  97315   250 small     
# 11 lfc   "Non\nRedistributed" Cyto   -0.368  VCPki^R191Q  97315   476 small     
# 12 lfc   "Nuc"                Cyto   -0.177  VCPki^R191Q    250   476 negligible

# FUS
RNAct.FUS.als.redistributed = bind_rows(mutate(RNAct.FUS.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.FUS.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.FUS.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.FUS.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.FUS_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.FUS.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 20), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(17,19,18),4), ylabel = expression(FUS:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.FUS_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin

# SFPQ
RNAct.SFPQ.als.redistributed = bind_rows(mutate(RNAct.SFPQ.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.SFPQ.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.SFPQ.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.SFPQ.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.SFPQ_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.SFPQ.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(SFPQ:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.SFPQ_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin

# HNRNPA1
RNAct.HNRNPA1.als.redistributed = bind_rows(mutate(RNAct.HNRNPA1.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.HNRNPA1.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.HNRNPA1.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.HNRNPA1.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.HNRNPA1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.HNRNPA1.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(HNRNPA1:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.HNRNPA1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# ATXN2
RNAct.ATXN2.als.redistributed = bind_rows(mutate(RNAct.ATXN2.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.ATXN2.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.ATXN2.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.ATXN2.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.ATXN2_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.ATXN2.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(ATXN2:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.ATXN2_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# MATR3
RNAct.MATR3.als.redistributed = bind_rows(mutate(RNAct.MATR3.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.MATR3.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.MATR3.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.MATR3.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.MATR3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.MATR3.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(MATR3:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.MATR3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# HNRNPA2B1
RNAct.HNRNPA2B1.als.redistributed = bind_rows(mutate(RNAct.HNRNPA2B1.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.HNRNPA2B1.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.HNRNPA2B1.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.HNRNPA2B1.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.HNRNPA2B1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.HNRNPA2B1.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(HNRNPA2B1:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.HNRNPA2B1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# TAF15
RNAct.TAF15.als.redistributed = bind_rows(mutate(RNAct.TAF15.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.TAF15.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.TAF15.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.TAF15.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.TAF15_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.TAF15.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(TAF15:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.TAF15_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# EWSR1
RNAct.EWSR1.als.redistributed = bind_rows(mutate(RNAct.EWSR1.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.EWSR1.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.EWSR1.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.EWSR1.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.EWSR1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.EWSR1.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(EWSR1:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.EWSR1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# ELAVL3
RNAct.ELAVL3.als.redistributed = bind_rows(mutate(RNAct.ELAVL3.tardbp.redistributed, mutation = "TARDBP^G298S"), mutate(RNAct.ELAVL3.vcp_r155c.redistributed, mutation = "VCP^R155C"), mutate(RNAct.ELAVL3.vcp_r191q.redistributed, mutation = "VCP^R191Q"), mutate(RNAct.ELAVL3.vcp_iso.redistributed, mutation = "VCPki^R191Q")) %>% drop_na(mislocalisation) %>% mutate(
  mislocalisation = gsub("Nuclear", "Nuc", mislocalisation), mislocalisation = gsub("Cytoplasm","Cyto",mislocalisation), mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")))
rnact.ELAVL3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct.ELAVL3.als.redistributed, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(0, 30), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(27,29,28),4), ylabel = expression(ELAVL3:~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# rnact.ELAVL3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin


# Merge
rnact.protein_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin = plot_grid(
  rnact.TDP43_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.FUS_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin,
  rnact.SFPQ_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.HNRNPA1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin,  
  rnact.ATXN2_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.MATR3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, 
  rnact.HNRNPA2B1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.TAF15_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin,  
  rnact.EWSR1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.ELAVL3_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin,  
  # rnact.NONO_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, rnact.PSPC1_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin,
  ncol = 2, nrow = 5, labels = "auto")
# rnact.protein_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin
ggsave(rnact.protein_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin, filename = here(proj_path, "expression/deseq2/rnact.protein_view.untreated.nuc_vs_cyt.als_vs_ctrl.violin.png"), height = 12, width = 10)

```


# Fig S9 Immunofluorescence TDP-43 FUS

```{r}
# import with magick 
tdp43_immuno_ipsn.png = magick::image_read(here(proj_path,"immuno/tdp43_immuno_ipsn.png"))
tdp43_immuno_ipsn.gg = ggdraw() + draw_image(tdp43_immuno_ipsn.png)
tdp43_immuno_ipsn.gg

fus_immuno_ipsn.png = magick::image_read(here(proj_path,"immuno/fus_immuno_ipsn.png"))
fus_immuno_ipsn.gg = ggdraw() + draw_image(fus_immuno_ipsn.png)
fus_immuno_ipsn.gg

# nucleocytoplasmic ratio
NC_FC_R155C.csv = read_csv(here(proj_path, "immuno/NC_R155C.csv")) %>% clean_names() %>% mutate(mutation = "VCP^R155C")
NC_FC_R191Q.csv = read_csv(here(proj_path, "immuno/NC_R191Q.csv")) %>% clean_names() %>% mutate(mutation = "VCP^R191Q")
NC_FC_TARDBP.csv = read_csv(here(proj_path, "immuno/NC_TARDBP.csv")) %>% clean_names() %>% mutate(mutation = "TARDBP^G298S")
# NC_FC_VCPki_VS_ISO.csv = read_csv(here(proj_path, "immuno/NC_FC_VCPki_VS_ISO.csv")) %>% clean_names() %>% mutate(mutation = "VCP^R191Q ki")
NC_FC_VCPki.csv = read_csv(here(proj_path, "immuno/NC_VCPki.csv")) %>% clean_names() %>% mutate(mutation = "VCPki^R191Q")
immuno_fus_tdp43.bind = bind_rows(NC_FC_R155C.csv, NC_FC_R191Q.csv, NC_FC_TARDBP.csv, NC_FC_VCPki.csv) %>% mutate(mutation = factor(mutation, levels = c("TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q")), als = case_when(mutant == "CTRL" ~ "CTRL", TRUE ~ "Mutant"))

immuno_fus.bind.violin = violin_plot(filter(immuno_fus_tdp43.bind, rbp == "FUS"), color_by = "als", continuous = "nc_ratio_fc", ylims = c(0.5,1.3), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.05, stat_label = "p.signif", stat_ypos = 1.3, ylabel = expression(FUS~nuclear/cytoplasm~FC~immunofluorescence), xlabel = "", dpi = 300, arrow_labels = FALSE, null_value = 1, plot_violin = FALSE) #, cols = c("firebrick2", "dodgerblue2")
immuno_fus.bind.violin
# facet        .y.   group1 group2    n1    n2 statistic        p p.signif y.position groups        xmin  xmax
#   <fct>        <chr> <chr>  <chr>  <int> <int>     <dbl>    <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP^G298S lfc   CTRL   Mutant    24    12       249 0.000199 ***             1.4 <chr [2]>        1     2
# 2 VCP^R155C    lfc   CTRL   Mutant    23    10       179 0.0111   *               1.4 <chr [2]>        1     2
# 3 VCP^R191Q    lfc   CTRL   Mutant    16     8       100 0.0274   *               1.4 <chr [2]>        1     2
# 4 VCPki^R191Q  lfc   CTRL   Mutant    12     6        32 0.75     ns              1.4 <chr [2]>        1     2

immuno_tdp43.bind.violin = violin_plot(filter(immuno_fus_tdp43.bind, rbp == "TDP-43"), color_by = "als", continuous = "nc_ratio_fc", ylims = c(0.4,1.3), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.05, stat_label = "p.signif", stat_ypos = 1.3, ylabel = expression(TDP-43~nuclear/cytoplasm~FC~immunofluorescence), xlabel = "", dpi = 300, arrow_labels = FALSE, null_value = 1, plot_violin = FALSE) #, cols = c("firebrick2", "dodgerblue2")
immuno_tdp43.bind.violin
# facet        .y.   group1 group2    n1    n2 statistic          p p.signif y.position groups        xmin  xmax
#   <fct>        <chr> <chr>  <chr>  <int> <int>     <dbl>      <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP^G298S lfc   CTRL   Mutant    18    12       208 0.00000155 ****            1.2 <chr [2]>        1     2
# 2 VCP^R155C    lfc   CTRL   Mutant    20    12       195 0.00272    **              1.2 <chr [2]>        1     2
# 3 VCP^R191Q    lfc   CTRL   Mutant    16     8        99 0.0324     *               1.2 <chr [2]>        1     2
# 4 VCPki^R191Q  lfc   CTRL   Mutant    12     6        64 0.0069     **              1.2 <chr [2]>        1     2

tdp43_fus_immuno_ipsn.merged = plot_grid(NULL, NULL, tdp43_immuno_ipsn.gg, fus_immuno_ipsn.gg, immuno_tdp43.bind.violin, immuno_fus.bind.violin, nrow = 3, ncol = 2, rel_heights = c(0.05,1.6,1), labels = c("a","b","","","c","d"))
# tdp43_fus_immuno_ipsn.merged
ggsave(tdp43_fus_immuno_ipsn.merged, filename = here(proj_path, "immuno/tdp43_fus_immuno_ipsn.merged.png"), height = 9, width = 9)
ggsave(tdp43_fus_immuno_ipsn.merged, filename = here(proj_path, "immuno/tdp43_fus_immuno_ipsn.merged.pdf"), height = 9, width = 9)

```



# Fig S9 CLIP countOverlaps regions

```{r}
clip_5utr.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_5utr.HepG2_K562_Hela.rds"))
clip_3utr.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_3utr.HepG2_K562_Hela.rds"))
clip_cds.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_cds.HepG2_K562_Hela.rds"))
# clip_exon.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_exon.HepG2_K562_Hela.rds"))
clip_intron.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_intron.HepG2_K562_Hela.rds"))
clip_transcript.HepG2_K562_Hela = readRDS(here(nemo_path,"genomes/rbp-databases/clip-bedgraphs/clip_transcript.HepG2_K562_Hela.rds"))


# 5UTR
clip_5utr.HepG2_K562_Hela # 153,186 5UTRs
clip_5utr.HepG2_K562_Hela %>% count(transcript_id) # 84,046 transcripts. Some transcripts have > 1 5UTR, as composed of multiple exons. Sum them.
clip_5utr.HepG2_K562_Hela.tx = clip_5utr.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_5utr.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl

clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)

clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,2.3), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(1.7, 2.1, 1.6),4), ylabel = expression(log[10]~CLIP~xlinks~5~UTR), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# facet        .y.   group1               group2    n1    n2 statistic            p        p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>  <int> <int>     <dbl>        <dbl>        <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    75078    88  2569415  0.000308     0.000924     ***                 1.7 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   75078   148  5026126. 0.045        0.09         ns                  2.1 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto      88   148     7384. 0.086        0.09         ns                  1.6 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    75094   111  3906231  0.253        0.506        ns                  1.7 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   75094   109  3738001  0.117        0.351        ns                  2.1 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto     111   109     5896. 0.747        0.747        ns                  1.6 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    74937   109  3651746. 0.056        0.112        ns                  1.7 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   74937   268  8704930. 0.000165     0.000495     ***                 2.1 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto     109   268    14230. 0.695        0.695        ns                  1.6 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    74915   120  3564576. 0.0000871    0.000174     ***                 1.7 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   74915   279  8450300. 0.0000000325 0.0000000975 ****                2.1 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto     120   279    17257  0.625        0.625        ns                  1.6 <chr [2]>        2     3
# .y.   group1               group2 effsize facet           n1    n2 magnitude 
#  * <chr> <chr>                <chr>    <dbl> <fct>        <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nuc    -0.398  TARDBP^G298S 83797    93 small     
#  2 lfc   "Non\nRedistributed" Cyto   -0.159  TARDBP^G298S 83797   156 negligible
#  3 lfc   "Nuc"                Cyto    0.269  TARDBP^G298S    93   156 small     
#  4 lfc   "Non\nRedistributed" Nuc    -0.133  VCP^R155C    83819   113 negligible
#  5 lfc   "Non\nRedistributed" Cyto   -0.149  VCP^R155C    83819   114 negligible
#  6 lfc   "Nuc"                Cyto   -0.0176 VCP^R155C      113   114 negligible
#  7 lfc   "Non\nRedistributed" Nuc    -0.179  VCP^R191Q    83644   115 negligible
#  8 lfc   "Non\nRedistributed" Cyto   -0.229  VCP^R191Q    83644   287 small     
#  9 lfc   "Nuc"                Cyto   -0.0545 VCP^R191Q      115   287 negligible
# 10 lfc   "Non\nRedistributed" Nuc    -0.363  VCPki^R191Q  83635   122 small     
# 11 lfc   "Non\nRedistributed" Cyto   -0.327  VCPki^R191Q  83635   289 small     
# 12 lfc   "Nuc"                Cyto    0.0427 VCPki^R191Q    122   289 negligible

# 3utr
clip_3utr.HepG2_K562_Hela # 155,153 3utrs
clip_3utr.HepG2_K562_Hela %>% count(transcript_id) # 77,060 transcripts. Some transcripts have > 1 3utr, as composed of multiple exons. Sum them.
clip_3utr.HepG2_K562_Hela.tx = clip_3utr.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_3utr.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl

clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)

clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,2.6), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(2.0, 2.4, 1.9),4), ylabel = expression(log[10]~CLIP~xlinks~3~UTR), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# facet        .y.   group1               group2    n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>  <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    71454   109  2576283  9.63e-10 2.89e- 9 ****                2   <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   71454   156  4493683  2.83e- 5 5.66e- 5 ****                2.4 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto     109   156     9950. 1.8 e- 2 1.8 e- 2 *                   1.9 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    71488   120  3380240. 5.87e- 5 1.76e- 4 ***                 2   <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   71488   111  3322616  3   e- 3 6   e- 3 **                  2.4 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto     120   111     6974  5.37e- 1 5.37e- 1 ns                  1.9 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    71292   129  3670756. 7.35e- 5 1.47e- 4 ***                 2   <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   71292   298  8508170. 2.87e- 9 8.61e- 9 ****                2.4 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto     129   298    19302. 9.46e- 1 9.46e- 1 ns                  1.9 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    71301   130  3807770. 4.32e- 4 8.64e- 4 ***                 2   <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   71301   288  7394617  2.25e-16 6.75e-16 ****                2.4 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto     130   288    16800. 9.3 e- 2 9.3 e- 2 ns                  1.9 <chr [2]>        2     3
#  .y.   group1               group2 effsize facet           n1    n2 magnitude 
#  * <chr> <chr>                <chr>    <dbl> <fct>        <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nuc    -0.553  TARDBP^G298S 76787   110 moderate  
#  2 lfc   "Non\nRedistributed" Cyto   -0.341  TARDBP^G298S 76787   163 small     
#  3 lfc   "Nuc"                Cyto    0.270  TARDBP^G298S   110   163 small     
#  4 lfc   "Non\nRedistributed" Nuc    -0.356  VCP^R155C    76825   120 small     
#  5 lfc   "Non\nRedistributed" Cyto   -0.289  VCP^R155C    76825   115 small     
#  6 lfc   "Nuc"                Cyto    0.0764 VCP^R155C      120   115 negligible
#  7 lfc   "Non\nRedistributed" Nuc    -0.347  VCP^R191Q    76622   132 small     
#  8 lfc   "Non\nRedistributed" Cyto   -0.330  VCP^R191Q    76622   306 small     
#  9 lfc   "Nuc"                Cyto    0.0195 VCP^R191Q      132   306 negligible
# 10 lfc   "Non\nRedistributed" Nuc    -0.320  VCPki^R191Q  76636   130 small     
# 11 lfc   "Non\nRedistributed" Cyto   -0.479  VCPki^R191Q  76636   294 small     
# 12 lfc   "Nuc"                Cyto   -0.206  VCPki^R191Q    130   294 small     

# cds
clip_cds.HepG2_K562_Hela # 763,579 cdss
clip_cds.HepG2_K562_Hela %>% count(transcript_id) # 100,592 transcripts. Transcripts have > 1 cds, as composed of multiple cdss. Sum them.
clip_cds.HepG2_K562_Hela.tx = clip_cds.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_cds.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl

clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)

clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,3), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(2.1, 2.7, 2.0),4), ylabel = expression(log[10]~CLIP~xlinks~CDS), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# facet        .y.   group1               group2    n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>  <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    95671   115  3330973  2.43e-13 7.29e-13 ****                2.1 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   95671   168  6107002  7.24e- 8 1.45e- 7 ****                2.7 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto     115   168    11516. 6   e- 3 6   e- 3 **                  2   <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    95706   129  4704956. 2.93e- 6 8.79e- 6 ****                2.1 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   95706   119  4378255  1.27e- 5 2.54e- 5 ****                2.7 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto     129   119     7725  9.31e- 1 9.31e- 1 ns                  2   <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    95505   136  4868498. 4.34e- 7 8.68e- 7 ****                2.1 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   95505   313 11737822. 5.11e-11 1.53e-10 ****                2.7 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto     136   313    22492  3.39e- 1 3.39e- 1 ns                  2   <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    95508   141  4636826. 1.56e-10 3.12e-10 ****                2.1 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   95508   305 10246548. 3.42e-19 1.03e-18 ****                2.7 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto     141   305    22158. 6.05e- 1 6.05e- 1 ns                  2   <chr [2]>        2     3
# .y.   group1               group2  effsize facet            n1    n2 magnitude 
#  * <chr> <chr>                <chr>     <dbl> <fct>         <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nuc    -0.646   TARDBP^G298S 100302   117 moderate  
#  2 lfc   "Non\nRedistributed" Cyto   -0.392   TARDBP^G298S 100302   173 small     
#  3 lfc   "Nuc"                Cyto    0.325   TARDBP^G298S    117   173 small     
#  4 lfc   "Non\nRedistributed" Nuc    -0.398   VCP^R155C    100341   129 small     
#  5 lfc   "Non\nRedistributed" Cyto   -0.405   VCP^R155C    100341   122 small     
#  6 lfc   "Nuc"                Cyto   -0.00845 VCP^R155C       129   122 negligible
#  7 lfc   "Non\nRedistributed" Nuc    -0.380   VCP^R191Q    100138   136 small     
#  8 lfc   "Non\nRedistributed" Cyto   -0.350   VCP^R191Q    100138   318 small     
#  9 lfc   "Nuc"                Cyto    0.0346  VCP^R191Q       136   318 negligible
# 10 lfc   "Non\nRedistributed" Nuc    -0.493   VCPki^R191Q  100144   141 small     
# 11 lfc   "Non\nRedistributed" Cyto   -0.459   VCPki^R191Q  100144   307 small     
# 12 lfc   "Nuc"                Cyto    0.0409  VCPki^R191Q     141   307 negligible


# intron
clip_intron.HepG2_K562_Hela # 1,239,213 introns
clip_intron.HepG2_K562_Hela %>% count(transcript_id) # 219,870 transcripts. Transcripts have > 1 intron, as composed of multiple introns. Sum them.
clip_intron.HepG2_K562_Hela.tx = clip_intron.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
  left_join(clip_intron.HepG2_K562_Hela.tx) %>%
  mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
         mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl

clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)

clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,2.7), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(1.8, 2.4, 1.7),4), ylabel = expression(log[10]~CLIP~xlinks~intron), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl.violin
# facet        .y.   group1               group2     n1    n2 statistic        p    p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>        <chr> <chr>                <chr>   <int> <int>     <dbl>    <dbl>    <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP^G298S lfc   "Non\nRedistributed" Nuc    189836   149  9565244. 7.89e-12 1.58e-11 ****                1.8 <chr [2]>        1     2
#  2 TARDBP^G298S lfc   "Non\nRedistributed" Cyto   189836   225 12305336. 3.63e-28 1.09e-27 ****                2.4 <chr [2]>        1     3
#  3 TARDBP^G298S lfc   "Nuc"                Cyto      149   225    14492. 2.6 e- 2 2.6 e- 2 *                   1.7 <chr [2]>        2     3
#  4 VCP^R155C    lfc   "Non\nRedistributed" Nuc    189865   176 11001888. 4.36e-15 8.72e-15 ****                1.8 <chr [2]>        1     2
#  5 VCP^R155C    lfc   "Non\nRedistributed" Cyto   189865   169 10309776  8.71e-16 2.61e-15 ****                2.4 <chr [2]>        1     3
#  6 VCP^R155C    lfc   "Nuc"                Cyto      176   169    14792  9.32e- 1 9.32e- 1 ns                  1.7 <chr [2]>        2     3
#  7 VCP^R191Q    lfc   "Non\nRedistributed" Nuc    189622   178 12286989  3.36e-10 6.72e-10 ****                1.8 <chr [2]>        1     2
#  8 VCP^R191Q    lfc   "Non\nRedistributed" Cyto   189622   410 25072378. 1.64e-35 4.92e-35 ****                2.4 <chr [2]>        1     3
#  9 VCP^R191Q    lfc   "Nuc"                Cyto      178   410    32862. 5.5 e- 2 5.5 e- 2 ns                  1.7 <chr [2]>        2     3
# 10 VCPki^R191Q  lfc   "Non\nRedistributed" Nuc    189655   185 11522122. 6.39e-16 1.28e-15 ****                1.8 <chr [2]>        1     2
# 11 VCPki^R191Q  lfc   "Non\nRedistributed" Cyto   189655   370 19715796  3.7 e-48 1.11e-47 ****                2.4 <chr [2]>        1     3
# 12 VCPki^R191Q  lfc   "Nuc"                Cyto      185   370    30813  5.5 e- 2 5.5 e- 2 ns                  1.7 <chr [2]>        2     3
#  .y.   group1               group2 effsize facet            n1    n2 magnitude 
#  * <chr> <chr>                <chr>    <dbl> <fct>         <int> <int> <ord>     
#  1 lfc   "Non\nRedistributed" Nuc    -0.567  TARDBP^G298S 201649   150 moderate  
#  2 lfc   "Non\nRedistributed" Cyto   -0.717  TARDBP^G298S 201649   228 moderate  
#  3 lfc   "Nuc"                Cyto   -0.205  TARDBP^G298S    150   228 small     
#  4 lfc   "Non\nRedistributed" Nuc    -0.591  VCP^R155C    201676   178 moderate  
#  5 lfc   "Non\nRedistributed" Cyto   -0.621  VCP^R155C    201676   173 moderate  
#  6 lfc   "Nuc"                Cyto   -0.0410 VCP^R155C       178   173 negligible
#  7 lfc   "Non\nRedistributed" Nuc    -0.474  VCP^R191Q    201429   180 small     
#  8 lfc   "Non\nRedistributed" Cyto   -0.601  VCP^R191Q    201429   418 moderate  
#  9 lfc   "Nuc"                Cyto   -0.161  VCP^R191Q       180   418 negligible
# 10 lfc   "Non\nRedistributed" Nuc    -0.581  VCPki^R191Q  201468   186 moderate  
# 11 lfc   "Non\nRedistributed" Cyto   -0.729  VCPki^R191Q  201468   373 moderate  
# 12 lfc   "Nuc"                Cyto   -0.200  VCPki^R191Q     186   373 negligible

# # Entire transcript
# clip_transcript.HepG2_K562_Hela # 227,818 transcripts
# clip_transcript.HepG2_K562_Hela %>% count(transcript_id) # 227,818 transcripts. Transcripts have > 1 transcript, as composed of multiple transcripts. Sum them.
# clip_transcript.HepG2_K562_Hela.tx = clip_transcript.HepG2_K562_Hela %>% group_by(transcript_id) %>% summarise(sum_iclip_score = sum(sum_iclip_score), mean_iclip_score = sum(mean_iclip_score)) 
# clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl =  bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP^G298S"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP^R155C"),
#   mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP^R191Q"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki^R191Q")) %>%
#   left_join(clip_transcript.HepG2_K562_Hela.tx) %>%
#   mutate(log10.sum_iclip_score = log10(sum_iclip_score), log10.mean_iclip_score = log10(mean_iclip_score), 
#          mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuc", padj < 0.05 & log2FoldChange < 0 ~ "Cyto", TRUE ~ "Non\nRedistributed"),
#          mislocalisation = factor(mislocalisation, levels = c("Non\nRedistributed", "Nuc", "Cyto")), mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))) %>% drop_na(log10.mean_iclip_score)
# clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl
# 
# clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(log10.mean_iclip_score, mislocalisation, mutation)
# 
# clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl, color_by = "mislocalisation", continuous = "log10.mean_iclip_score", ylims = c(-1,3.6), facet_by = "mutation", facet_ncol = 4, facet_labeller = "label_parsed", plot_stats = FALSE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(2.7, 3.2, 2.6),4), 
#                                                                ylabel = expression(log[10]~CLIP~xlinks~transcript), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE)
# clip_transcript.untreated.nuc_vs_cyt.als_vs_ctrl.violin


# Merge
clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.violin.merged = plot_grid(
  clip_5utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin, clip_3utr.untreated.nuc_vs_cyt.als_vs_ctrl.violin,
  clip_cds.untreated.nuc_vs_cyt.als_vs_ctrl.violin, clip_intron.untreated.nuc_vs_cyt.als_vs_ctrl.violin,
  nrow = 2, ncol = 2, labels = "auto")
# clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.violin.merged
ggsave(clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.violin.merged, filename = here(proj_path, "expression/deseq2/clip_regions.untreated.nuc_vs_cyt.als_vs_ctrl.violin.merged.png"), height = 6, width = 9)


```


# Table S4: RNAct RNA view

```{r}
RNAct_database.transcript.summarised = readRDS(here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.transcript.summarised.rds"))
rnact_rna_view.untreated.nuc_vs_cyt.tardbp_vs_ctrl = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% mutate(tardbp_mut_redist = case_when(padj < 0.05 & log2FoldChange > 0 ~ "nuclear", padj < 0.05 & log2FoldChange < 0 ~ "cytoplasm", TRUE ~ "none")) %>% select(transcript_id, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean) 
rnact_rna_view.untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% mutate(vcp_mut_redist = case_when(padj < 0.05 & log2FoldChange > 0 ~ "nuclear", padj < 0.05 & log2FoldChange < 0 ~ "cytoplasm", TRUE ~ "none")) %>% select(transcript_id, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean) 
rnact_rna_view.untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% mutate(vcp_mut_redist = case_when(padj < 0.05 & log2FoldChange > 0 ~ "nuclear", padj < 0.05 & log2FoldChange < 0 ~ "cytoplasm", TRUE ~ "none")) %>% select(transcript_id, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean) 
rnact_rna_view.untreated.nuc_vs_cyt.vcp_ki_vs_ctrl = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% mutate(tardbp_ki_redist = case_when(padj < 0.05 & log2FoldChange > 0 ~ "nuclear", padj < 0.05 & log2FoldChange < 0 ~ "cytoplasm", TRUE ~ "none")) %>% select(transcript_id, gene_name, vcp_ki.stat = stat, vcp_ki.padj = padj, vcp_ki.lfc = log2FoldChange, vcp_ki.baseMean = baseMean) 

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join = rnact_rna_view.untreated.nuc_vs_cyt.tardbp_vs_ctrl %>% left_join(rnact_rna_view.untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl) %>% left_join(rnact_rna_view.untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl) %>% left_join(rnact_rna_view.untreated.nuc_vs_cyt.vcp_ki_vs_ctrl) %>%
    filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_r191q.stat) == FALSE & is.na(vcp_ki.stat) == FALSE) %>%
  filter(tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_r191q.padj < 0.05 | vcp_ki.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_r191q.stat) + abs(vcp_ki.stat))) %>%
  mutate(overlapping = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat > 0 & vcp_r155c.stat > 0 & vcp_r191q.stat > 0 & vcp_ki.stat > 0) ~ "Nuclear", 
                             (tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat < 0 & vcp_r155c.stat < 0 & vcp_r191q.stat < 0 & vcp_ki.stat < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ "")) %>%
    # rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_noki", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_ki")), ~ str_replace_all(., "vcp_ki", "vcp_ki")) %>%
  inner_join(RNAct_database.transcript.summarised) %>% 
  select(transcript_id, gene_name, catrapid_score = sum_score, everything(), -mean_score, -number) %>% arrange(-catrapid_score)
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% print(n=50)
write_csv(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, here(proj_path, "expression/deseq2/untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join.csv"), na = "")
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S4 RNAct")

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join 
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(category != "")

rename(RNAct_database.transcript.summarised, catrapid_score = sum_score) %>% left_join(select(homo_sapiens.gtf.spliced.transcript, transcript_id, width)) %>% ggscatter(x = "width", y = "catrapid_score", cor.coef = TRUE, size = 0.5)
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% left_join(select(homo_sapiens.gtf.spliced.transcript, transcript_id, width)) %>% ggscatter(x = "width", y = "catrapid_score", cor.coef = TRUE, size = 0.5)


untreated.nuc_vs_cyt.vcp_vs_ctrl$res.tx %>% filter(padj < 0.05) #%>% inner_join(RNAct_database.transcript.summarised) # 
538 / 1017 #52.9
untreated.nuc_vs_cyt.vcp_vs_ctrl$res.tx %>% filter(padj < 0.05 & stat > 0) %>% inner_join(RNAct_database.transcript.summarised) # 
307 / 556
untreated.nuc_vs_cyt.vcp_vs_ctrl$res.tx %>% filter(padj < 0.05 & stat < 0) %>% inner_join(RNAct_database.transcript.summarised) # 
231 / 461
untreated.nuc_vs_cyt.vcp_vs_ctrl$res.tx %>% filter(padj > 0.05 | is.na(padj)) %>% inner_join(RNAct_database.transcript.summarised) # 
97503 / 226801 #43.0

untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) #%>% inner_join(RNAct_database.transcript.summarised) # 
449/789 # 56.9
untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05 & stat > 0) %>% inner_join(RNAct_database.transcript.summarised) # 
201 / 373
untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05 & stat < 0) %>% inner_join(RNAct_database.transcript.summarised) # 
248 / 416
untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj > 0.05 | is.na(padj)) #%>% inner_join(RNAct_database.transcript.summarised) # 
97592 / 227029 # 43.0%



```



# Table S5: RNAct Protein view

```{r}
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct.protein_view.tardbp_vcp_ki.redistributed.Rdata")) # script rnact_protein_tdp43_fus_sfpq.R

RNAct.TDP43.als.redistributed.join = rename(RNAct.TDP43.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.TDP43.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.TDP43.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.TDP43.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(TDP43_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.TDP43.als.redistributed.join
RNAct.FUS.als.redistributed.join = rename(RNAct.FUS.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.FUS.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.FUS.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.FUS.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(FUS_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.FUS.als.redistributed.join
RNAct.SFPQ.als.redistributed.join = rename(RNAct.SFPQ.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.SFPQ.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.SFPQ.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.SFPQ.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(SFPQ_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.SFPQ.als.redistributed.join
RNAct.HNRNPA1.als.redistributed.join = rename(RNAct.HNRNPA1.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.HNRNPA1.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPA1.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPA1.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(HNRNPA1_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.HNRNPA1.als.redistributed.join
RNAct.ATXN2.als.redistributed.join = rename(RNAct.ATXN2.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.ATXN2.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.ATXN2.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.ATXN2.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(ATXN2_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.ATXN2.als.redistributed.join
RNAct.MATR3.als.redistributed.join = rename(RNAct.MATR3.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.MATR3.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.MATR3.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.MATR3.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(MATR3_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.MATR3.als.redistributed.join
RNAct.HNRNPA2B1.als.redistributed.join = rename(RNAct.HNRNPA2B1.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.HNRNPA2B1.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPA2B1.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPA2B1.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(HNRNPA2B1_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.HNRNPA2B1.als.redistributed.join
RNAct.TAF15.als.redistributed.join = rename(RNAct.TAF15.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.TAF15.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.TAF15.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.TAF15.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(TAF15_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.TAF15.als.redistributed.join
RNAct.EWSR1.als.redistributed.join = rename(RNAct.EWSR1.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.EWSR1.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.EWSR1.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.EWSR1.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(EWSR1_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.EWSR1.als.redistributed.join
RNAct.ELAVL3.als.redistributed.join = rename(RNAct.ELAVL3.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.ELAVL3.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.ELAVL3.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.ELAVL3.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(ELAVL3_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.ELAVL3.als.redistributed.join
RNAct.HNRNPK.als.redistributed.join = rename(RNAct.HNRNPK.tardbp.redistributed, tardbp.redist = mislocalisation) %>% left_join(rename(RNAct.HNRNPK.vcp_r155c.redistributed, vcp_r155c.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPK.vcp_r191q.redistributed, vcp_r191q.redist = mislocalisation)) %>% left_join(rename(RNAct.HNRNPK.vcp_iso.redistributed, vcp_ki.redist = mislocalisation)) %>% filter(!(tardbp.redist == "Non\nRedistributed" & vcp_r155c.redist == "Non\nRedistributed" & vcp_r191q.redist == "Non\nRedistributed" & vcp_ki.redist == "Non\nRedistributed")) %>%
  select(HNRNPK_score = catrapid_score_max_normalised, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist) %>% distinct(transcript_id, .keep_all = TRUE)
RNAct.HNRNPK.als.redistributed.join

RNAct.RBPs.als.redistributed.join = RNAct.TDP43.als.redistributed.join %>% left_join(RNAct.FUS.als.redistributed.join) %>% left_join(RNAct.SFPQ.als.redistributed.join) %>% left_join(RNAct.HNRNPA1.als.redistributed.join) %>% left_join(RNAct.ATXN2.als.redistributed.join) %>% left_join(RNAct.MATR3.als.redistributed.join) %>% left_join(RNAct.HNRNPA2B1.als.redistributed.join) %>% left_join(RNAct.ELAVL3.als.redistributed.join) %>% left_join(RNAct.EWSR1.als.redistributed.join) %>% left_join(RNAct.TAF15.als.redistributed.join) %>%
  left_join(tx2gene) %>% left_join(gene2ens) %>% select(gene_name, transcript_id, tardbp.redist, vcp_r155c.redist, vcp_r191q.redist, vcp_ki.redist, everything(), -gene_id) %>% 
  mutate(sum_score = TDP43_score + FUS_score + SFPQ_score + TAF15_score + ATXN2_score + HNRNPA1_score + HNRNPA2B1_score + EWSR1_score + ELAVL3_score + MATR3_score) %>% arrange(-sum_score) %>% #select(-sum_score) %>%
  mutate(tardbp.redist = gsub("Non\nRedistributed", "", tardbp.redist), vcp_r155c.redist = gsub("Non\nRedistributed", "", vcp_r155c.redist), vcp_r191q.redist = gsub("Non\nRedistributed", "", vcp_r191q.redist), vcp_ki.redist = gsub("Non\nRedistributed", "", vcp_ki.redist))
RNAct.RBPs.als.redistributed.join
write_csv(RNAct.RBPs.als.redistributed.join, here(proj_path, "expression/deseq2/RNAct.RBPs.als.redistributed.join.csv"), na = "")
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(RNAct.RBPs.als.redistributed.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S5 RNAct RBP view")
  

RNAct.RBPs.als.redistributed.join %>% filter(gene_name %in% ALS_genes)
RNAct.RBPs.als.redistributed.join %>% filter(gene_name %in% RNA_BINDING_PROTEINS)
RNAct.RBPs.als.redistributed.join %>% filter(tardbp.redist == "Nuclear" | vcp_mut.redist == "Nuclear" | vcp_ki.redist == "Nuclear" ) # ILK, PDCD2, HOXB3
RNAct.RBPs.als.redistributed.join %>% filter(tardbp.redist == "Cytoplasm" | vcp_mut.redist == "Cytoplasm" | vcp_ki.redist == "Cytoplasm" ) # CD151, CHST12, CERS1


```


# Table S6 Transite motifs

```{r}
transite.tmsa.mrna.join = rename(select(tardbp_gene_exon.transite.nuclear,-p_value_n), tardbp.nuc.enrichment = enrichment, tardbp.nuc.pval = p_value, tardbp.nuc.padj = adj_p_value) %>% 
  left_join(rename(select(tardbp_gene_exon.transite.cytoplasm,-p_value_n), tardbp.cyt.enrichment = enrichment, tardbp.cyt.pval = p_value, tardbp.cyt.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_r155c_gene_exon.transite.nuclear,-p_value_n), vcp_r155c.nuc.enrichment = enrichment, vcp_r155c.nuc.pval = p_value, vcp_r155c.nuc.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_r155c_gene_exon.transite.cytoplasm,-p_value_n), vcp_r155c.cyt.enrichment = enrichment, vcp_r155c.cyt.pval = p_value, vcp_r155c.cyt.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_r191q_gene_exon.transite.nuclear,-p_value_n), vcp_r191q.nuc.enrichment = enrichment, vcp_r191q.nuc.pval = p_value, vcp_r191q.nuc.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_r191q_gene_exon.transite.cytoplasm,-p_value_n), vcp_r191q.cyt.enrichment = enrichment, vcp_r191q.cyt.pval = p_value, vcp_r191q.cyt.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_iso_gene_exon.transite.nuclear,-p_value_n), vcp_ki.nuc.enrichment = enrichment, vcp_ki.nuc.pval = p_value, vcp_ki.nuc.padj = adj_p_value)) %>% 
  left_join(rename(select(vcp_iso_gene_exon.transite.cytoplasm,-p_value_n), vcp_ki.cyt.enrichment = enrichment, vcp_ki.cyt.pval = p_value, vcp_ki.cyt.padj = adj_p_value)) %>% 
  filter(tardbp.nuc.padj < 0.05 | tardbp.cyt.padj < 0.05 | vcp_r155c.nuc.padj < 0.05 | vcp_r155c.cyt.padj < 0.05 | vcp_r191q.nuc.padj < 0.05 | vcp_r191q.cyt.padj < 0.05 | vcp_ki.nuc.padj < 0.05 | vcp_ki.cyt.padj < 0.05 ) %>%
  mutate(sum_enrichment = (tardbp.nuc.enrichment + tardbp.cyt.enrichment + vcp_r155c.nuc.enrichment + vcp_r155c.cyt.enrichment + vcp_r191q.nuc.enrichment + vcp_r191q.cyt.enrichment + vcp_ki.nuc.enrichment + vcp_ki.cyt.enrichment),
         sum_nuc_enrichment = (tardbp.nuc.enrichment + vcp_r155c.nuc.enrichment + vcp_r191q.nuc.enrichment + vcp_ki.nuc.enrichment),
         sum_cyt_enrichment = (tardbp.cyt.enrichment + vcp_r155c.cyt.enrichment + vcp_r191q.cyt.enrichment + vcp_ki.cyt.enrichment),
         motif_id = case_when(motif_id == "244_7543047" ~ "M274_0.6", motif_id == "790_10094314" ~ "M331_0.6", motif_id == "950_7908267" ~ "M349_0.6", motif_id == "682_10811881" ~ "M325_0.6", 
   motif_id == "784_7972035" ~ "M330_0.6", motif_id == "951_12324455" ~ "M350_0.6", motif_id == "953_7543047" ~ "M352_0.6", TRUE ~ motif_id)) %>% as_tibble() %>% arrange(-sum_enrichment)
transite.tmsa.mrna.join
write_csv(transite.tmsa.mrna.join, here(proj_path, "expression/rna-rbp-interaction/transite/TableS5.transite.tmsa.mrna.rbp_motifs.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(transite.tmsa.mrna.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S6 motif")

# For motifs without the CISBP IDs then find CISBP motif ID name from searching http://cisbp-rna.ccbr.utoronto.ca/TFreport.php?searchTF=T38010_0.6&orderby=M

```


# Fig 4 Splicing

MAJIQ 

## Volcano & GO terms

```{r}
# whole_untreated_tardbp_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_untreated_tardbp_vs_ctrl.tsv"), grp1 = "tardbp_untreated_whole", grp2 = "ctrl_untreated_whole")
# whole_untreated_vcp_r155c_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_untreated_vcp_r155c_vs_ctrl.tsv"), grp1 = "vcp_r155c_untreated_whole", grp2 = "ctrl_untreated_whole")
# whole_untreated_vcp_r191q_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_untreated_vcp_r191q_vs_ctrl.tsv"), grp1 = "vcp_r191q_untreated_whole", grp2 = "ctrl_untreated_whole")
# whole_untreated_vcp_iso_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_untreated_vcp_iso_vs_ctrl.tsv"), grp1 = "vcp_iso_untreated_whole", grp2 = "ctrl_untreated_whole")
# whole_untreated_als_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_untreated_als_vs_ctrl.tsv"), grp1 = "als_untreated_whole", grp2 = "ctrl_untreated_whole")
# whole_untreated_tardbp_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_untreated_tardbp_vs_ctrl"), skip_rows = 7)
# whole_untreated_vcp_r155c_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_untreated_vcp_r155c_vs_ctrl"), skip_rows = 7)
# whole_untreated_vcp_r191q_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_untreated_vcp_r191q_vs_ctrl"), skip_rows = 7)
# whole_untreated_vcp_iso_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_untreated_vcp_iso_vs_ctrl"), skip_rows = 7)
# whole_untreated_als_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_untreated_als_vs_ctrl"), skip_rows = 7)
# save(whole_untreated_tardbp_vs_ctrl.voila, whole_untreated_vcp_r155c_vs_ctrl.voila, whole_untreated_vcp_r191q_vs_ctrl.voila, whole_untreated_vcp_iso_vs_ctrl.voila, whole_untreated_als_vs_ctrl.voila,
#      whole_untreated_tardbp_vs_ctrl.modulizer, whole_untreated_vcp_r155c_vs_ctrl.modulizer, whole_untreated_vcp_r191q_vs_ctrl.modulizer, whole_untreated_vcp_iso_vs_ctrl.modulizer, whole_untreated_als_vs_ctrl.modulizer,
#      file = here(proj_path, "scripts/whole_untreated_majiq_objects.RData"))
# 
# whole_ml240_tardbp_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_ml240_tardbp_vs_ctrl.tsv"), grp1 = "tardbp_ml240_whole", grp2 = "ctrl_ml240_whole")
# whole_ml240_vcp_r155c_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_ml240_vcp_r155c_vs_ctrl.tsv"), grp1 = "vcp_r155c_ml240_whole", grp2 = "ctrl_ml240_whole")
# whole_ml240_vcp_r191q_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_ml240_vcp_r191q_vs_ctrl.tsv"), grp1 = "vcp_r191q_ml240_whole", grp2 = "ctrl_ml240_whole")
# whole_ml240_vcp_iso_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/whole_ml240_vcp_iso_vs_ctrl.tsv"), grp1 = "vcp_iso_ml240_whole", grp2 = "ctrl_ml240_whole")
# whole_ml240_tardbp_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_ml240_tardbp_vs_ctrl"), skip_rows = 7)
# whole_ml240_vcp_r155c_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_ml240_vcp_r155c_vs_ctrl"), skip_rows = 7)
# whole_ml240_vcp_r191q_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_ml240_vcp_r191q_vs_ctrl"), skip_rows = 7)
# whole_ml240_vcp_iso_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/whole_ml240_vcp_iso_vs_ctrl"), skip_rows = 7)
# save(whole_ml240_tardbp_vs_ctrl.voila, whole_ml240_vcp_r155c_vs_ctrl.voila, whole_ml240_vcp_r191q_vs_ctrl.voila, whole_ml240_vcp_iso_vs_ctrl.voila,
#      whole_ml240_tardbp_vs_ctrl.modulizer, whole_ml240_vcp_r155c_vs_ctrl.modulizer, whole_ml240_vcp_r191q_vs_ctrl.modulizer, whole_ml240_vcp_iso_vs_ctrl.modulizer,
#      file = here(proj_path, "scripts/whole_ml240_majiq_objects.RData"))
load(here(proj_path, "scripts/whole_untreated_majiq_objects.RData"))
# load(here(proj_path, "scripts/whole_ml240_majiq_objects.RData"))

# TARDBP
whole_untreated_tardbp_vs_ctrl.voila %>% glimpse
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 400
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 153
whole_untreated_tardbp_vs_ctrl.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) # 400
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 400
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) # IGHMBP2, MRPL43, RRP15
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # HLA-A
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, "CHMP7"))
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nuclear_export.markers) # 0


## Volcano
majiq.whole_untreated_tardbp_vs_ctrl.RBP.junction_labels <- whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_tardbp_vs_ctrl.ALS.junction_labels <- whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_tardbp_vs_ctrl.volcano <- majiq_deltapsi_volcano_plot(whole_untreated_tardbp_vs_ctrl.voila, title = "TARDBP mutant", numerator = "TARDBP", denomenator = "CTRL", ymax = 4.1, xmax = 0.8, dpi = 300, junction_labels = c(majiq.whole_untreated_tardbp_vs_ctrl.RBP.junction_labels, majiq.whole_untreated_tardbp_vs_ctrl.ALS.junction_labels), arrow_labels = FALSE)
# majiq.whole_untreated_tardbp_vs_ctrl.volcano

# GO of splicing events
whole_untreated_tardbp_vs_ctrl.voila.gost = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
whole_untreated_tardbp_vs_ctrl.voila.gost_filt <- whole_untreated_tardbp_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(whole_untreated_tardbp_vs_ctrl.voila.gost_filt$query) # 0
paste('"',unique(whole_untreated_tardbp_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
)
whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms <- whole_untreated_tardbp_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms$term_name) 
whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
whole_untreated_tardbp_vs_ctrl.voila.gost_filt.terms.curated

# # Cryptic Exons == denovo & cassette_exon
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 271 / 400
271 / 400 #67.8%
whole_untreated_tardbp_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(ctrl_untreated_whole_tardbp_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_tardbp_untreated_whole_probability_changing > 0.9) %>% count(modulizer) # 39 cryptic exons

whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 223 /400
223/400 # 55.8%
whole_untreated_tardbp_vs_ctrl.modulizer.cryptic = whole_untreated_tardbp_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
50/264

# Overlap splicing with transcript redistribution
whole_untreated_tardbp_vs_ctrl_splicing_events_genes = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% filter(gene_name %in% whole_untreated_tardbp_vs_ctrl_splicing_events_genes) # distinct(gene_name) %>% pull(gene_name)
# transcript_id   baseMean log2FoldChange lfcSE  stat   pvalue     padj gene_id         gene_name
#   <chr>              <dbl>          <dbl> <dbl> <dbl>    <dbl>    <dbl> <chr>           <chr>    
# 1 ENST00000293648    639.           12.7   1.55  8.16 3.28e-16 7.56e-12 ENSG00000171466 ZNF562   
# 2 ENST00000468924     30.8          -8.83  1.71 -5.15 2.60e- 7 2.04e- 4 ENSG00000182362 YBEY     

# Fisher exact overlap test
untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_tardbp_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=365
# Intersection size=2
# Overlapping p-value=0.24
# Jaccard Index=0.0


# VCP R155C
whole_untreated_vcp_r155c_vs_ctrl.voila %>% glimpse
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 531
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 218
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) # 531
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 531
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) # MRPL23, RBFOX1, IGHMBP2
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # HLA-B
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nuclear_export.markers) # NPIPA1, RGPD3, 

## Volcano
majiq.whole_untreated_vcp_r155c_vs_ctrl.RBP.junction_labels <- whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_r155c_vs_ctrl.ALS.junction_labels <- whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_r155c_vs_ctrl.volcano <- majiq_deltapsi_volcano_plot(whole_untreated_vcp_r155c_vs_ctrl.voila, title = "VCP R155C mutant", numerator = "VCP R155C", denomenator = "CTRL", ymax = 4.1, xmax = 0.8, dpi = 300, junction_labels = c(majiq.whole_untreated_vcp_r155c_vs_ctrl.RBP.junction_labels, majiq.whole_untreated_vcp_r155c_vs_ctrl.ALS.junction_labels), arrow_labels = FALSE)
# majiq.whole_untreated_vcp_r155c_vs_ctrl.volcano

# GO of splicing events
whole_untreated_vcp_r155c_vs_ctrl.voila.gost = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt <- whole_untreated_vcp_r155c_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt$query) # 0
paste('"',unique(whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"nervous system development",
"Focal adhesion",
"axon",
"cytoskeletal protein binding",)
whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms <- whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms$term_name) 
whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
whole_untreated_vcp_r155c_vs_ctrl.voila.gost_filt.terms.curated

# # Cryptic Exons == denovo & cassette_exon
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE)
296/531 # 55.7%
whole_untreated_vcp_r155c_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(ctrl_untreated_whole_vcp_r155c_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_vcp_r155c_untreated_whole_probability_changing > 0.9) %>% count(modulizer) # 55 cryptic exons
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type))
327/531 # 61.6%

# Overlap splicing with transcript redistribution
whole_untreated_vcp_r155c_vs_ctrl_splicing_events_genes = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% whole_untreated_vcp_r155c_vs_ctrl_splicing_events_genes) %>% arrange(stat)
# transcript_id   baseMean log2FoldChange lfcSE  stat   pvalue          padj gene_id         gene_name
#    <chr>              <dbl>          <dbl> <dbl> <dbl>    <dbl>         <dbl> <chr>           <chr>    
#  1 ENST00000614895    319.          -10.0  1.66  -6.03 1.65e- 9 0.00000310    ENSG00000150275 PCDH15   
#  2 ENST00000620199    200.           -6.68 1.41  -4.73 2.20e- 6 0.00127       ENSG00000278558 TMEM191B 
#  3 ENST00000370371     63.5         -24.8  5.38  -4.61 3.98e- 6 0.00196       ENSG00000101191 DIDO1    
#  4 ENST00000592904   1207.           -1.42 0.328 -4.32 1.53e- 5 0.00567       ENSG00000174652 ZNF266   
#  5 ENST00000667661     32.6         -17.0  4.41  -3.86 1.14e- 4 0.0288        ENSG00000236404 VLDLR-AS1
#  6 ENST00000416826     61.0          11.9  3.09   3.86 1.16e- 4 0.0291        ENSG00000236404 VLDLR-AS1
#  7 ENST00000396832   8077.            3.50 0.870  4.03 5.70e- 5 0.0165        ENSG00000213923 CSNK1E   
#  8 ENST00000430335    237.            1.24 0.306  4.04 5.27e- 5 0.0156        ENSG00000213923 CSNK1E   
#  9 ENST00000586293     58.0           8.24 2.03   4.07 4.71e- 5 0.0142        ENSG00000196605 ZNF846   
# 10 ENST00000405675   1515.            2.35 0.528  4.45 8.45e- 6 0.00358       ENSG00000213923 CSNK1E   
# 11 ENST00000613942    212.            2.21 0.434  5.08 3.70e- 7 0.000302      ENSG00000245694 CRNDE    
# 12 ENST00000442216    540.            2.79 0.501  5.58 2.43e- 8 0.0000322     ENSG00000213923 CSNK1E   
# 13 ENST00000583271    727.           11.0  1.52   7.26 3.90e-13 0.00000000339 ENSG00000270231 NBPF8    

# Fisher exact overlap test
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx,padj<0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_vcp_r155c_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=218
# listB size=345
# Intersection size=9
# Overlapping p-value=5.1e-06
# Jaccard Index=0.0

# VCP R191Q
whole_untreated_vcp_r191q_vs_ctrl.voila %>% glimpse
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 512
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) # 199
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) # 512
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% distinct(gene_name) # 8 RBFOX1, WDR4, RPS95, MRPL43
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-B
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nuclear_export.markers) # RGPD2, RGPD3, CCHCR1, SNUPN

## Volcano
majiq.whole_untreated_vcp_r191q_vs_ctrl.RBP.junction_labels <- whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_r191q_vs_ctrl.ALS.junction_labels <- whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_r191q_vs_ctrl.volcano <- majiq_deltapsi_volcano_plot(whole_untreated_vcp_r191q_vs_ctrl.voila, title = "VCP R191Q mutant", numerator = "VCP R191Q", denomenator = "CTRL", ymax = 4.1, xmax = 0.8, dpi = 300, junction_labels = c(majiq.whole_untreated_vcp_r191q_vs_ctrl.RBP.junction_labels, majiq.whole_untreated_vcp_r191q_vs_ctrl.ALS.junction_labels), arrow_labels = FALSE)
# majiq.whole_untreated_vcp_r191q_vs_ctrl.volcano

# GO of splicing events
whole_untreated_vcp_r191q_vs_ctrl.voila.gost = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt <- whole_untreated_vcp_r191q_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt$query) # 0
paste('"',unique(whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"calcium ion binding",
"homophilic cell adhesion via plasma membrane adhesion molecules",
"cell-cell adhesion via plasma-membrane adhesion molecules",
"GTPase regulator activity",
"nucleoside-triphosphatase regulator activity",
"GTPase activator activity")
whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms <- whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms$term_name) 
whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
whole_untreated_vcp_r191q_vs_ctrl.voila.gost_filt.terms.curated

# # Cryptic Exons == denovo & cassette_exon
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) 
308/512 #60.2%
whole_untreated_vcp_r191q_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(ctrl_untreated_whole_vcp_r191q_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_vcp_r191q_untreated_whole_probability_changing > 0.9) %>% count(modulizer) # 69 cryptic exons
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type))
310/512 #60.5%

# Overlap splicing with transcript redistribution
whole_untreated_vcp_r191q_vs_ctrl_splicing_events_genes = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% whole_untreated_vcp_r191q_vs_ctrl_splicing_events_genes) %>% arrange(stat) %>% print(n=Inf)
#  transcript_id   baseMean log2FoldChange lfcSE  stat   pvalue     padj gene_id         gene_name 
#    <chr>              <dbl>          <dbl> <dbl> <dbl>    <dbl>    <dbl> <chr>           <chr>     
#  1 ENST00000554573    254.         -10.2   1.45  -7.02 2.23e-12 7.95e- 9 ENSG00000174373 RALGAPA1  
#  2 ENST00000359624    285.         -11.5   1.66  -6.95 3.71e-12 9.97e- 9 ENSG00000160200 CBS       
#  3 ENST00000497329    154.          -8.86  1.34  -6.63 3.37e-11 6.57e- 8 ENSG00000196912 ANKRD36B  
#  4 ENST00000398158    754.         -10.4   1.70  -6.14 8.21e-10 9.56e- 7 ENSG00000160200 CBS       
#  5 ENST00000484836    113.          -8.63  1.73  -4.98 6.23e- 7 2.78e- 4 ENSG00000175061 SNHG29    
#  6 ENST00000468924     45.4         -7.87  1.59  -4.93 8.02e- 7 3.38e- 4 ENSG00000182362 YBEY      
#  7 ENST00000620199    635.          -6.29  1.29  -4.89 1.01e- 6 4.12e- 4 ENSG00000278558 TMEM191B  
#  8 ENST00000620507   3272.          -6.84  1.48  -4.63 3.68e- 6 1.17e- 3 ENSG00000078328 RBFOX1    
#  9 ENST00000583400     70.2         -7.09  1.56  -4.53 5.82e- 6 1.64e- 3 ENSG00000175061 SNHG29    
# 10 ENST00000598180   1445.          -1.13  0.275 -4.10 4.11e- 5 8.25e- 3 ENSG00000104936 DMPK      
# 11 ENST00000358537   2407.          -1.05  0.277 -3.78 1.59e- 4 2.49e- 2 ENSG00000136068 FLNB      
# 12 ENST00000414879     50.2         -6.99  1.92  -3.65 2.64e- 4 3.69e- 2 ENSG00000142949 PTPRF     
# 13 ENST00000665646     72.6         -6.22  1.75  -3.56 3.68e- 4 4.79e- 2 ENSG00000175061 SNHG29    
# 14 ENST00000354250    233.           1.34  0.374  3.59 3.30e- 4 4.41e- 2 ENSG00000160194 NDUFV3    
# 15 ENST00000618859     64.2          2.90  0.805  3.60 3.22e- 4 4.32e- 2 ENSG00000278558 TMEM191B  
# 16 ENST00000487214    401.           1.89  0.485  3.90 9.75e- 5 1.72e- 2 ENSG00000188976 NOC2L     
# 17 ENST00000267113    459.           1.36  0.339  4.02 5.81e- 5 1.11e- 2 ENSG00000139641 ESYT1     
# 18 ENST00000613386    499.           2.79  0.667  4.18 2.85e- 5 6.12e- 3 ENSG00000276550 HERC2P2   
# 19 ENST00000354665    744.           0.801 0.190  4.21 2.60e- 5 5.72e- 3 ENSG00000101191 DIDO1     
# 20 ENST00000340209   4761.           6.66  1.56   4.28 1.88e- 5 4.37e- 3 ENSG00000078328 RBFOX1    
# 21 ENST00000327044   4557.           1.08  0.249  4.33 1.49e- 5 3.62e- 3 ENSG00000188976 NOC2L     
# 22 ENST00000522356   2457.           0.611 0.129  4.75 2.05e- 6 7.37e- 4 ENSG00000206077 ZDHHC11B  
# 23 ENST00000361846    189.           4.15  0.861  4.83 1.40e- 6 5.39e- 4 ENSG00000007341 ST7L      
# 24 ENST00000373401   2963.           0.887 0.174  5.10 3.34e- 7 1.67e- 4 ENSG00000112139 MDGA1     
# 25 ENST00000416826     65.9          7.96  1.49   5.36 8.42e- 8 5.17e- 5 ENSG00000236404 VLDLR-AS1 
# 26 ENST00000433020    349.           9.09  1.57   5.80 6.82e- 9 6.12e- 6 ENSG00000205746 AC126755.1
# 27 ENST00000434585   6235.           1.21  0.128  9.41 5.07e-21 3.95e-16 ENSG00000090565 RAB11FIP3 

# Fisher exact overlap test
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx,padj<0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_vcp_r191q_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=199
# listB size=568
# Intersection size=21
# Overlapping p-value=4.6e-16
# Jaccard Index=0.0


# VCP knock-in
whole_untreated_vcp_iso_vs_ctrl.voila %>% glimpse
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 329
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) # 139
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) # 329
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% distinct(gene_name) # 5 IGHMBP2, MRPL43, RNH1, KHDRBS2, ZNF385A
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-B
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nuclear_export.markers) # 0

## Volcano
majiq.whole_untreated_vcp_iso_vs_ctrl.RBP.junction_labels <- whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_iso_vs_ctrl.ALS.junction_labels <- whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_vcp_iso_vs_ctrl.volcano <- majiq_deltapsi_volcano_plot(whole_untreated_vcp_iso_vs_ctrl.voila, title = "VCP R191 knock-in", numerator = "VCPki R191Q", denomenator = "CTRL", ymax = 4.1, xmax = 0.8, dpi = 300, junction_labels = c(majiq.whole_untreated_vcp_iso_vs_ctrl.RBP.junction_labels, majiq.whole_untreated_vcp_iso_vs_ctrl.ALS.junction_labels), arrow_labels = FALSE)
# majiq.whole_untreated_vcp_iso_vs_ctrl.volcano

# GO of splicing events
whole_untreated_vcp_iso_vs_ctrl.voila.gost = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt <- whole_untreated_vcp_iso_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt$query) # 0
paste('"',unique(whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"axon",
"neuron projection",
"calcium ion binding")
whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms <- whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms$term_name) 
whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
whole_untreated_vcp_iso_vs_ctrl.voila.gost_filt.terms.curated

# # Cryptic Exons == denovo & cassette_exon
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE)
233/329 #70.8%
whole_untreated_vcp_iso_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(ctrl_untreated_whole_vcp_iso_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_vcp_iso_untreated_whole_probability_changing > 0.9) %>% count(modulizer) # 30 cryptic exons

whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type))
183/329 #55.6%

# Overlap splicing with transcript redistribution
whole_untreated_vcp_iso_vs_ctrl_splicing_events_genes = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% whole_untreated_vcp_iso_vs_ctrl_splicing_events_genes) %>% arrange(stat) %>% print(n=Inf)
# transcript_id   baseMean log2FoldChange lfcSE  stat   pvalue          padj gene_id         gene_name    
#   <chr>              <dbl>          <dbl> <dbl> <dbl>    <dbl>         <dbl> <chr>           <chr>        
# 1 ENST00000352482    606.          -11.2  1.56  -7.15 8.92e-13 0.00000000332 ENSG00000101150 TPD52L2      
# 2 ENST00000348257     39.4         -30.0  5.56  -5.40 6.77e- 8 0.0000462     ENSG00000101150 TPD52L2      
# 3 ENST00000577841    198.           -8.87 1.95  -4.54 5.53e- 6 0.00175       ENSG00000233098 CCDC144NL-AS1
# 4 ENST00000504841    214.           -8.84 2.16  -4.09 4.34e- 5 0.00922       ENSG00000169246 NPIPB3       
# 5 ENST00000510157    277.            5.01 0.718  6.98 2.87e-12 0.00000000763 ENSG00000198589 LRBA         

# Fisher exact overlap test
whole_untreated_vcp_iso_vs_ctrl.voila.sig = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx,padj<0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_vcp_iso_vs_ctrl.voila.sig, untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=139
# listB size=541
# Intersection size=4
# Overlapping p-value=0.036
# Jaccard Index=0.0



# Combined
whole_untreated_als_vs_ctrl.voila %>% glimpse
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 174
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) #81
whole_untreated_als_vs_ctrl.voila %>% filter(abs(deltapsi) > 0.2, probability_changing > 0.9) #174
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #174
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #MRPL43
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # HLA-B
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, "CHMP7"))
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nuclear_export.markers) # 0


## Volcano
majiq.whole_untreated_als_vs_ctrl.RBP.junction_labels <- whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_als_vs_ctrl.ALS.junction_labels <- whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.whole_untreated_als_vs_ctrl.volcano <- majiq_deltapsi_volcano_plot(whole_untreated_als_vs_ctrl.voila, title = "Mutants combined", numerator = "ALS", denomenator = "CTRL", ymax = 4.1, xmax = 0.8, dpi = 300, junction_labels = c(majiq.whole_untreated_als_vs_ctrl.RBP.junction_labels, majiq.whole_untreated_als_vs_ctrl.ALS.junction_labels), arrow_labels = FALSE)
majiq.whole_untreated_als_vs_ctrl.volcano

# GO of splicing events
whole_untreated_als_vs_ctrl.voila.gost = whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
whole_untreated_als_vs_ctrl.voila.gost_filt <- whole_untreated_als_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(whole_untreated_als_vs_ctrl.voila.gost_filt$query) # 2
paste('"',unique(whole_untreated_als_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()

# # Cryptic Exons == denovo & cassette_exon
whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, de_novo_junctions == TRUE) # 115 / 174
115 / 174 #66.1
whole_untreated_als_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(ctrl_untreated_whole_als_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_als_untreated_whole_probability_changing > 0.9) %>% count(modulizer) # 18 cryptic exons

whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, grepl("i$",lsv_type)) # 104 /174
104/174 # 59.8
whole_untreated_als_vs_ctrl.modulizer.cryptic = whole_untreated_als_vs_ctrl.modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
50/264

# Overlap splicing with transcript redistribution
whole_untreated_als_vs_ctrl_splicing_events_genes = whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated.nuc_vs_cyt.als_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% filter(gene_name %in% whole_untreated_als_vs_ctrl_splicing_events_genes) # distinct(gene_name) %>% pull(gene_name)
# transcript_id   baseMean log2FoldChange lfcSE  stat       pvalue      padj gene_id         gene_name
#   <chr>              <dbl>          <dbl> <dbl> <dbl>        <dbl>     <dbl> <chr>           <chr>    
# 1 ENST00000425407    203.          -24.2   4.34 -5.57 0.0000000255 0.0000896 ENSG00000164828 SUN1     
# 2 ENST00000468924     46.1          -7.71  1.67 -4.63 0.00000365   0.00346   ENSG00000182362 YBEY     
# 3 ENST00000590161    658.           11.1   2.60  4.26 0.0000203    0.0106    ENSG00000099864 PALM     
# 4 ENST00000429178    108.            5.18  1.27  4.07 0.0000462    0.0195    ENSG00000164828 SUN1     

# Fisher exact overlap test
untreated.nuc_vs_cyt.als_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_als_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.als_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=365
# Intersection size=2
# Overlapping p-value=0.24
# Jaccard Index=0.0


```

## Modulizer

```{r}
whole_untreated_mutants_vs_ctrl.modulizer.bind = bind_rows(
  mutate(rename(whole_untreated_tardbp_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_tardbp_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_tardbp_untreated_whole_probability_changing), mutation = "TARDBP^G298S"), 
  mutate(rename(whole_untreated_vcp_r155c_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_r155c_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_r155c_untreated_whole_probability_changing), mutation = "VCP^R155C"),
  mutate(rename(whole_untreated_vcp_r191q_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_r191q_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_r191q_untreated_whole_probability_changing), mutation = "VCP^R191Q"), 
  mutate(rename(whole_untreated_vcp_iso_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_iso_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_iso_untreated_whole_probability_changing), mutation = "VCPki^R191Q"),
    mutate(rename(whole_untreated_als_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_als_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_als_untreated_whole_probability_changing), mutation = "Combined")) %>% 
  mutate(mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q","VCPki^R191Q", "Combined")), modulizer_simplified = gsub("_"," ",modulizer_simplified)) %>%
  filter(probability_changing > 0.9, abs(dpsi) > 0.2)

whole_untreated_mutants_vs_ctrl.modulizer.bind.count = whole_untreated_mutants_vs_ctrl.modulizer.bind %>% count(mutation, modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  group_by(mutation) %>% add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(pct > 11 ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
whole_untreated_mutants_vs_ctrl.modulizer.bind.count %>% print(n=Inf)

whole_untreated_mutants_vs_ctrl.modulizer.bind.barplot = ggbarplot(whole_untreated_mutants_vs_ctrl.modulizer.bind.count, x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "",  label = whole_untreated_mutants_vs_ctrl.modulizer.bind.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  # facet_grid(mutation ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
whole_untreated_mutants_vs_ctrl.modulizer.bind.barplot


# whole_untreated_tardbp_vs_ctrl.modulizer <- read_voila_modulizer(here(proj_path,"splicing/majiq/modulize/pan_als"), skip_rows = 9)

# # Modulizer barchart of differential events
# whole_untreated_tardbp_vs_ctrl.modulizer_simplfied.sig.summary = whole_untreated_tardbp_vs_ctrl.modulizer %>% filter(abs(ctrl_untreated_whole_tardbp_untreated_whole_median_dpsi) > 0.2, ctrl_untreated_whole_tardbp_untreated_whole_probability_changing > 0.9) %>% count(modulizer_simplified) %>% mutate(modulizer_simplified = gsub("_"," ",modulizer_simplified), modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>% arrange(desc(modulizer_simplified)) %>% 
#   mutate(pct = n/sum(n) *100, pct.label = case_when(pct > 8 ~ paste(as.character(n),"\n", as.character(round(pct,1)),"%"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) #%>% print
# whole_untreated_tardbp_vs_ctrl.modulizer_simplfied.sig.summary %>% print
# whole_untreated_tardbp_vs_ctrl.modulizer.sig.pie = ggplot(whole_untreated_tardbp_vs_ctrl.modulizer_simplfied.sig.summary, aes(x = "", y = pct, fill = fct_reorder(modulizer_simplified, n, .desc = TRUE))) + geom_bar(width = 1, stat = "identity", color="white") + coord_polar("y", start=0) + 
#   geom_text(aes(x = 1.2, y = ypos, label = pct.label), color = "black", size=2.5) + scale_fill_brewer(palette="Set1") + 
#   theme_oz() + theme_void() + theme(legend.title=element_blank(), legend.position = "right", legend.text=element_text(size=7), legend.spacing.y = unit(1, 'mm'), legend.spacing.x = unit(1, 'mm'), legend.key.height = unit(0.4, 'cm'), legend.key.width = unit(0.3, 'cm')) + guides(fill=guide_legend(ncol=1))
# whole_untreated_tardbp_vs_ctrl.modulizer.sig.pie


# Intron retention
whole_untreated_tardbp_vs_ctrl.modulizer.ir = whole_untreated_tardbp_vs_ctrl.modulizer %>% filter(modulizer_simplified == "intron_retention")
whole_untreated_tardbp_vs_ctrl.modulizer.ir %>% filter(abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # 60
whole_untreated_tardbp_vs_ctrl.modulizer.ir %>% filter(gene_name %in% ALS_genes, abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) # SIRT2
whole_untreated_tardbp_vs_ctrl.modulizer.ir %>% filter(gene_name %in% "SFPQ")


```


## Correlation heatmap

```{r}
# heatmap of correlation coefficients
select(whole_untreated_tardbp_vs_ctrl.voila, gene_name, gene_id, lsv_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, tardbp.dpsi = deltapsi, tardbp.probability_changing = probability_changing) %>%
  left_join(select(whole_untreated_vcp_r155c_vs_ctrl.voila, gene_name, gene_id, lsv_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_r155c.dpsi = deltapsi, vcp_r155c.probability_changing = probability_changing)) %>%
  left_join(select(whole_untreated_vcp_r191q_vs_ctrl.voila, gene_name, gene_id, lsv_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_r191q.dpsi = deltapsi, vcp_r191q.probability_changing = probability_changing)) %>%
  left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, gene_name, gene_id, lsv_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_ki.dpsi = deltapsi, vcp_ki.probability_changing = probability_changing))

whole_untreated_als_vs_ctrl.voila.deltapsi = select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, `TARDBP^G298S` = deltapsi) %>% 
  left_join(select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, `VCP^R155C` = deltapsi)) %>%
  left_join(select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, `VCP^R191Q` = deltapsi)) %>%
  left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, `VCPki^R191Q` = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")
cormat <- round(cor(whole_untreated_als_vs_ctrl.voila.deltapsi),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# reorder_cormat <- function(cormat){ # Use correlation between variables as distance
#   dd <- as.dist((1-cormat)/2)
#   hc <- hclust(dd)
#   cormat <-cormat[hc$order, hc$order]
# }
# cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
  # mutate(Var1 = fct_relevel(Var1, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"), Var2 = fct_relevel(Var2, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))
whole_untreated_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
whole_untreated_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap

```


## Upset plot

```{r}
whole_untreated_als_vs_ctrl.voila.bind = bind_rows(mutate(whole_untreated_tardbp_vs_ctrl.voila, mutation = "TARDBP G298S"), mutate(whole_untreated_vcp_r155c_vs_ctrl.voila, mutation = "VCP R155C"), mutate(whole_untreated_vcp_r191q_vs_ctrl.voila, mutation = "VCP R191Q"), mutate(whole_untreated_vcp_iso_vs_ctrl.voila, mutation = "VCP R191Q knock-in"))

whole_untreated_als_vs_ctrl.voila.bind.upset = majiq_upset_plot(whole_untreated_als_vs_ctrl.voila.bind, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 340, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
whole_untreated_als_vs_ctrl.voila.bind.upset

whole_untreated_als_vs_ctrl.voila.bind.up.upset = majiq_upset_plot(filter(whole_untreated_als_vs_ctrl.voila.bind, deltapsi > 0), overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 200, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "ALS splicing increased", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
whole_untreated_als_vs_ctrl.voila.bind.up.upset

whole_untreated_als_vs_ctrl.voila.bind.down.upset = majiq_upset_plot(filter(whole_untreated_als_vs_ctrl.voila.bind, deltapsi < 0), overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 200, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "ALS splicing decreased", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
whole_untreated_als_vs_ctrl.voila.bind.down.upset

# Fisher exact overlap test
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.up <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.down <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
# # # ggvenn
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes <- list("TARDBP" = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, "VCP" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes <- list("TARDBP" = untreated.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, "VCP" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn <- ggvenn(untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5 ,text_size = 4) + labs(caption = "Nuclear")  + theme(plot.caption = element_text(vjust = 4, hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn
# 
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn <- ggvenn(untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5,text_size = 4) + labs(caption = "Cytoplasm")  + theme(plot.caption = element_text(vjust = 4,hjust = 0.5, colour = "dodgerblue2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn

# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn = plot_grid(NULL, untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn, NULL, untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn, rel_heights =c(0.02,1,0.05,1), nrow = 4)
# untreated.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn

```



## Merge

```{r}
majiq.whole_untreated_als_vs_ctrl.merged = plot_grid(
  plot_grid(majiq.whole_untreated_tardbp_vs_ctrl.volcano, majiq.whole_untreated_vcp_r155c_vs_ctrl.volcano, majiq.whole_untreated_vcp_r191q_vs_ctrl.volcano, majiq.whole_untreated_vcp_iso_vs_ctrl.volcano, majiq.whole_untreated_als_vs_ctrl.volcano, ncol = 5, labels = "auto"),
  whole_untreated_mutants_vs_ctrl.modulizer.bind.barplot,
  plot_grid(whole_untreated_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap, whole_untreated_als_vs_ctrl.voila.bind.up.upset, whole_untreated_als_vs_ctrl.voila.bind.down.upset, ncol = 3, rel_widths = c(0.8,1,1), labels = letters[7:9]),
  nrow = 3, rel_heights = c(1,1,0.8), labels = c("","f",""))
# majiq.whole_untreated_als_vs_ctrl.merged
ggsave(majiq.whole_untreated_als_vs_ctrl.merged, filename = here(proj_path, "splicing/majiq/majiq.whole_untreated_als_vs_ctrl.merged.png"), height = 9, width = 12.5)

```

Coverage plot

```{r}
library(here)
proj_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation")
script_path = here("/nemo/lab/patanir/home/users/ziffo/projects/motor-neuron-mislocalisation/scripts/ipsc_mn_als_meta")
shared_path = here("/nemo/project/proj-luscombn-patani/working")
collab_path = here("/nemo/lab/patanir/home/users/ziffo/patani-collab")
source(here(nemo_path,"scripts/functions/R_workspace.R"))
# load(here(proj_path,"scripts/untreated_deseq_objects.RData"))

ctrl_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_nuclear.bam") # CTRL3 removed
ctrl_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/ctrl_cytoplasm.bam") # CTRL3 removed to balance reads
tardbp_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_nuclear.bam")
tardbp_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/tardbp_cytoplasm.bam")
vcp_r155c_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_nuclear.bam")
vcp_r155c_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r155c_cytoplasm.bam")
vcp_r191q_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_nuclear.bam")
vcp_r191q_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_r191q_cytoplasm.bam")
vcp_iso_nuclear.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_nuclear.bam")
vcp_iso_cytoplasm.bam <- here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/star_salmon/merged/vcp_iso_cytoplasm.bam")

library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(org.Hs.eg.db)
txdb <- makeTxDbFromGFF(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.chr_patch_hapl_scaff.gtf"), format="gtf") # make txdb from GTF - turn on only when needed
seqlevels(txdb, pruning.mode="coarse") <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y") # removes all the other chromsome annotations & scaffolds which are also in the BAM

track_viewer_nuc_cyt(level = "gene", gene_name = "SUN1", cond1.nuc.bam = ctrl_nuclear.bam, cond1.cyt.bam = ctrl_cytoplasm.bam, cond1.label = "CTRL", cond2.cyt.bam = tardbp_cytoplasm.bam, cond2.nuc.bam = tardbp_nuclear.bam, cond2.label = "TARDBP\nG298S", 
                     cond3.cyt.bam = vcp_r155c_cytoplasm.bam, cond3.nuc.bam = vcp_r155c_nuclear.bam, cond3.label = "VCP\nR155C", cond4.cyt.bam = vcp_r191q_cytoplasm.bam, cond4.nuc.bam = vcp_r191q_nuclear.bam, cond4.label = "VCP\nR191Q", 
                     cond5.cyt.bam = vcp_iso_cytoplasm.bam, cond5.nuc.bam = vcp_iso_nuclear.bam, cond5.label = "VCP\nR191Q ki", project_path = proj_path, ylab.size = 0.6)

```




# Table S7 MAJIQ

```{r}
load(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData")) #postmortem

whole_untreated_als_vs_ctrl.voila = mutate(select(whole_untreated_tardbp_vs_ctrl.voila, gene_name, gene_id, lsv_id, junction_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, tardbp.dpsi = deltapsi, tardbp.probability_changing = probability_changing),
                   tardbp.ml240 = case_when(lsv_junction_id %in% pull(filter(whole_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2==TRUE, (deltapsi > 0 & !lsv_junction_id %in% pull(filter(whole_ml240_tardbp_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi > 0),lsv_junction_id)) | (deltapsi < 0 & !lsv_junction_id %in% pull(filter(whole_ml240_tardbp_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi < 0),lsv_junction_id))),lsv_junction_id) ~ "rescued", TRUE ~ "")) %>%
  
  left_join(mutate(select(whole_untreated_vcp_r155c_vs_ctrl.voila, gene_name, gene_id, lsv_id, junction_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_r155c.dpsi = deltapsi, vcp_r155c.probability_changing = probability_changing),
                   vcp_r155c.ml240 = case_when(lsv_junction_id %in% pull(filter(whole_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, (deltapsi > 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi > 0),lsv_junction_id)) | (deltapsi < 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi < 0),lsv_junction_id))),lsv_junction_id) ~ "rescued", TRUE ~ ""))) %>%
  
  left_join(mutate(select(whole_untreated_vcp_r191q_vs_ctrl.voila, gene_name, gene_id, lsv_id, junction_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_r191q.dpsi = deltapsi, vcp_r191q.probability_changing = probability_changing),
                   vcp_r155c.ml240 = case_when(lsv_junction_id %in% pull(filter(whole_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, (deltapsi > 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi > 0),lsv_junction_id)) | (deltapsi < 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi < 0),lsv_junction_id))),lsv_junction_id) ~ "rescued", TRUE ~ ""))) %>%
              
  left_join(mutate(select(whole_untreated_vcp_iso_vs_ctrl.voila, gene_name, gene_id, lsv_id, junction_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, vcp_ki.dpsi = deltapsi, vcp_ki.probability_changing = probability_changing),
                   vcp_r191q.ml240 = case_when(lsv_junction_id %in% pull(filter(whole_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2==TRUE, (deltapsi > 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi > 0),lsv_junction_id)) | (deltapsi < 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi < 0),lsv_junction_id))),lsv_junction_id) ~ "rescued", TRUE ~ ""))) %>%
              
  left_join(mutate(select(whole_untreated_als_vs_ctrl.voila, gene_name, gene_id, lsv_id, junction_id, lsv_junction_id, seqid, strand, lsv_type, de_novo_junctions, combined.dpsi = deltapsi, combined.probability_changing = probability_changing),
                   vcp_ki.ml240 = case_when(lsv_junction_id %in% pull(filter(whole_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2==TRUE, (deltapsi > 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_iso_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi > 0),lsv_junction_id)) | (deltapsi < 0 & !lsv_junction_id %in% pull(filter(whole_ml240_vcp_iso_vs_ctrl.voila, changing_dpsi0.2==TRUE, deltapsi < 0),lsv_junction_id))),lsv_junction_id) ~ "rescued", TRUE ~ ""))) %>%
  
  filter(abs(tardbp.dpsi) > 0.2 | abs(vcp_r155c.dpsi) > 0.2 | abs(vcp_r191q.dpsi) > 0.2 | abs(vcp_ki.dpsi) > 0.2) %>%
  filter(tardbp.probability_changing > 0.9 | vcp_r155c.probability_changing > 0.9 | vcp_r191q.probability_changing > 0.9 | vcp_ki.probability_changing > 0.9) %>%
  arrange(-( abs(tardbp.dpsi) + abs(vcp_r155c.dpsi) + abs(vcp_r191q.dpsi) + abs(vcp_ki.dpsi))) %>%
  mutate(overlapping = case_when((tardbp.probability_changing > 0.9 & vcp_r155c.probability_changing > 0.9 & vcp_r191q.probability_changing > 0.9 & vcp_ki.probability_changing > 0.9 & tardbp.dpsi > 0.2 & vcp_r155c.dpsi > 0.2 & vcp_r191q.dpsi > 0.2 & vcp_ki.dpsi > 0.2) ~ "up", (tardbp.probability_changing > 0.9 & vcp_r155c.probability_changing > 0.9 & vcp_r191q.probability_changing > 0.9 & vcp_ki.probability_changing > 0.9 & tardbp.dpsi < 0.2 & vcp_r155c.dpsi < 0.2 & vcp_r191q.dpsi < 0.2 & vcp_ki.dpsi < 0.2) ~ "down", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore",  gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ ""),
         contains_ir = case_when(grepl("i$",lsv_type) ~ "ir", TRUE~""), de_novo = case_when(de_novo_junctions == 1 ~ "denovo", TRUE ~ "")) %>% select(-de_novo_junctions, -lsv_junction_id) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx,padj<0.05), tardbp.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, tardbp.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx,padj<0.05), vcp_r155c.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_r155c.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx,padj<0.05), vcp_r191q.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_r191q.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx,padj<0.05), vcp_ki.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_ki.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx,padj<0.05), combined.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, combined.redistributed)) %>% 
      mutate_if(is.character, ~replace_na(.,"")) %>%
  mutate( category = case_when(gene_name %in% ALS_genes ~ "ALS gene",  gene_name %in% nuclear_export.markers ~ "Nuclear export",  gene_name %in% RNA_BINDING_PROTEINS ~ "RBP", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ ""),
          postmortem = case_when(gene_name %in% pull(distinct(filter(nygc.als_ctrl.voila,changing_dpsi0.2 == TRUE),gene_name,.keep_all=TRUE),gene_name) ~ "altered", TRUE ~ ""))
whole_untreated_als_vs_ctrl.voila %>% glimpse
write_csv(whole_untreated_als_vs_ctrl.voila, here(proj_path, "splicing/majiq/whole_untreated_als_vs_ctrl.voila.csv"), na = "")
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(whole_untreated_als_vs_ctrl.voila, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S7 Splicing")

whole_untreated_als_vs_ctrl.voila %>% glimpse

whole_untreated_als_vs_ctrl.voila %>% filter(overlapping != "", gene_name %in% RNA_BINDING_PROTEINS)
whole_untreated_als_vs_ctrl.voila %>% filter(overlapping != "")
whole_untreated_als_vs_ctrl.voila %>% filter(tardbp.redistributed != "", vcp_r155c.redistributed != "", vcp_r191q.redistributed != "", vcp_ki.redistributed != "")
whole_untreated_als_vs_ctrl.voila %>% filter(tardbp.redistributed != "", vcp_r155c.redistributed != "", vcp_r191q.redistributed != "") %>% print(n=Inf)
whole_untreated_als_vs_ctrl.voila %>% filter(tardbp.redistributed != "", vcp_r191q.redistributed != "") %>% print(n=Inf)
whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r155c.redistributed != "", vcp_r191q.redistributed != "") %>% print(n=Inf)
whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r191q.redistributed != "", vcp_ki.redistributed != "") %>% print(n=Inf)

# # Voila view import with magick 
# SUN1_ctrl_untreated_whole_Combined_sg.svg = magick::image_read(here(proj_path,"splicing/majiq/voila-view/SUN1_ctrl_untreated_whole_Combined_sg.svg"))
# SUN1_ctrl_untreated_whole_Combined_sg.gg = ggdraw() + draw_image(SUN1_ctrl_untreated_whole_Combined_sg.svg)
# SUN1_ctrl_untreated_whole_Combined_sg.gg

```

# Fig S10 MAJIQ Scatter correlation


```{r}
whole_untreated_als_vs_ctrl.voila_list = list("TARDBP G298S" = whole_untreated_tardbp_vs_ctrl.voila, "VCP R155C" = whole_untreated_vcp_r155c_vs_ctrl.voila, "VCP R191Q" = whole_untreated_vcp_r191q_vs_ctrl.voila, "VCP R191Q knock-in" = whole_untreated_vcp_iso_vs_ctrl.voila)

tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))
tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))
tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))
vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))
vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))
vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi) %>% left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, gene_name, vcp.deltapsi = deltapsi)) %>% filter((vcp.deltapsi > 0.2 & tardbp.deltapsi  > 0.2) | (vcp.deltapsi < -0.2 & tardbp.deltapsi  < -0.2))

whole_untreated_als_vs_ctrl.voila_list_deltapsi_plot <- 
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R155C", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"), lsv_junction_labels = pull(tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"), lsv_junction_labels = pull(tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q knock-in", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"),  lsv_junction_labels = pull(tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"), lsv_junction_labels = pull(vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q knock-in", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"), lsv_junction_labels = pull(vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
plot_de_correlation(model_list1 = whole_untreated_als_vs_ctrl.voila_list, model_list2 = whole_untreated_als_vs_ctrl.voila_list, mutation1 = "VCP R191Q", mutation2 = "VCP R191Q knock-in", col = "deltapsi", join_by = c("lsv_junction_id", "gene_name"), lsv_junction_labels = pull(vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,lsv_junction_id))  +
  plot_layout(ncol = 2, nrow = 3) &  
  xlim(-1,1) & ylim(-1,1)
whole_untreated_als_vs_ctrl.voila_list_deltapsi_plot
ggsave(whole_untreated_als_vs_ctrl.voila_list_deltapsi_plot, filename = here(proj_path, "splicing/majiq/whole_untreated_als_vs_ctrl.voila_list_deltapsi_plot.png"), height = 10, width = 8.25)

```



# NYGC postmortem comparison

```{r}
nygc_path = here("/Volumes/lab-patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
# nygc_path = here("/nemo/lab/patanir/home/users/ziffo/proj-luscombn-patani/working/nygc-als-consortium")
# nygc.als_ctrl.voila = read_voila_het(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.tsv"), grp1 = "als", grp2 = "ctrl")
# nygc.pan_als.voila_modulizer <- read_voila_modulizer(here(nygc_path,"splicing/majiq/modulize/pan_als"), skip_rows = 9)
# save(nygc.als_ctrl.voila, nygc.pan_als.voila_modulizer, file = here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData"))
load(here(nygc_path,"splicing/majiq/voila-tsv/als_vs_ctrl.RData"))

nygc.als_ctrl.voila
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) # 842
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_name) # 445
nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) # 250
nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) # 130

nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% distinct(gene_name)  %>% print(n=Inf) #8
nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # TTC7A, DNM1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_genes) # CAMTA1, GRIN1, NEK1, ATXN1, DNM1, VEGF
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% sporadic_ALS_genes) #0
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_GWAS) # CAMTA1, NEK1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_EWAS) # TTC7A, DNM1
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% ALS_modifiers) # VEGFA
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% TDP43_spliced_genes) %>% distinct(gene_name)  %>% print(n=Inf)  #11

nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(nuclear_export.markers)) # AkAP8L, RTN4, ATXN1, SRSF5, RAPGEF3, XPO6, PKD1

nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c(p53.signalling.pathway, gprofiler_database$`DNA Damage Response`)) # CNOT3, EI24
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% gprofiler_database$`DNA repair`) #  APTX, CENPX, RIF1, RPAIN, KAT7
nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, -log10(tnom) > 4) # TVP23C-CDRT4, SREK1

## Volcano
majiq.als_vs_ctrl.RBP.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.ALS.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% c(ALS_genes)) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
majiq.als_vs_ctrl.NCT.junction_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% nuclear_export.markers) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)
# majiq.als_vs_ctrl.extra_labels <- nygc.als_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% c("METTL22","LINC00665")) %>% group_by(gene_name) %>% top_n(2, wt = abs(deltapsi)) %>% pull(lsv_junction_id)

nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, -log10(tnom)>3, -log10(tnom)<4) 

majiq.nygc.als_vs_ctrl.volcano <- majiq_het_volcano_plot(nygc.als_ctrl.voila, "Postmortem ALS", numerator = "ALS", denomenator = "CTRL", subtitle = "57 CTRL vs 214 ALS", ymax = 15, xmax = 0.5, dpi = 300, junction_labels = c(majiq.als_vs_ctrl.ALS.junction_labels, majiq.als_vs_ctrl.NCT.junction_labels, majiq.als_vs_ctrl.RBP.junction_labels))
majiq.nygc.als_vs_ctrl.volcano

# GO of splicing events
nygc.als_ctrl.voilagost = nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
nygc.als_ctrl.voilagost_filt <- nygc.als_ctrl.voilagost$result %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
table(nygc.als_ctrl.voilagost_filt$query) # 167
paste('"',unique(nygc.als_ctrl.voilagost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
majiq_als_go_list = c(
"vesicle localization",
"regulation of cellular component organization",
"cytoskeleton",
"axon",
"protein binding",
"synapse",
"RHO GTPase cycle",
"cytoplasm",
"actin cytoskeleton",
"cytoskeletal protein binding"
  )
nygc.als_ctrl.voilagost_filt.terms <- nygc.als_ctrl.voilagost_filt %>% filter(term_name %in% majiq_als_go_list)
# nygc.als_ctrl.voilagost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", nygc.als_ctrl.voilagost_filt.terms$term_name) 
nygc.als_ctrl.voilagost_filt.terms.curated <- curated_profiles(nygc.als_ctrl.voilagost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
nygc.als_ctrl.voilagost_filt.terms.curated


# Modulize
nygc.als_ctrl.voila.modulizer.count = nygc.pan_als.voila_modulizer %>% count(modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(pct > 14 ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
nygc.als_ctrl.voila.modulizer.count %>% print(n=Inf)

nygc.als_ctrl.voila.modulizer.barplot = ggbarplot(mutate(nygc.als_ctrl.voila.modulizer.count,mutation=""), x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "",  label = nygc.als_ctrl.voila.modulizer.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  # facet_grid(mutation ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
nygc.als_ctrl.voila.modulizer.barplot


# # Cryptic Exons == denovo & cassette_exon
 nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, de_novo_junctions == TRUE) # 44 / 250
44 / 250 #17.6
nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.2, als_ctrl_het_tnom < 0.05) %>% count(modulizer) # 7 cryptic exons

nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, grepl("i$",lsv_type)) # 120 /250
120/250 # 48%
nygc.pan_als.voila_modulizer.cryptic = nygc.pan_als.voila_modulizer %>% filter(denovo == TRUE, modulizer == "cassette_exon", abs(als_ctrl_het_median_dpsi) > 0.1, als_ctrl_het_tnom < 0.05) %>% arrange(-abs(als_ctrl_het_median_dpsi)) %>% pull(gene_name)  # "RELCH"         "HOXC4"         "RBM26"         "SLC35B3"       "TENM3"         "TPTEP2-CSNK1E" "ZSCAN29"   
50/264

# Overlap splicing with transcript redistribution
nygc.als_ctrl.voila_splicing_events_genes =  nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
nygc_postmortem_spinal_cord.tx$res.tx %>% filter(padj < 0.05) %>% filter(gene_name %in% nygc.als_ctrl.voila_splicing_events_genes) #568 

# Fisher exact overlap test
untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
newGeneOverlap(whole_untreated_tardbp_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=365
# Intersection size=2
# Overlapping p-value=0.24
# Jaccard Index=0.0



# Overlap postmortem and iPSNs 

### gene level
nygc.als_ctrl.splicing_events_genes = nygc.als_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated_als_ctrl.splicing_events_genes = whole_untreated_als_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated_tardbp_ctrl.splicing_events_genes = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated_vcp_r155c_ctrl.splicing_events_genes = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated_vcp_r191q_ctrl.splicing_events_genes = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
untreated_vcp_iso_ctrl.splicing_events_genes = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)

newGeneOverlap(nygc.als_ctrl.splicing_events_genes, untreated_als_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# listA size=130
# listB size=81
# Intersection size=0
# Overlapping p-value=1
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% untreated_tardbp_ctrl.splicing_events_genes] #"ARHGEF7" "EPB41L2" "SDHAP3"  "SDHAP3"  "SYNGR1"  "SYNGR1"  "ERICH1"  "ERICH1" 

newGeneOverlap(nygc.als_ctrl.splicing_events_genes, untreated_tardbp_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=130
# Intersection size=5
# Overlapping p-value=2.1e-05
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% untreated_tardbp_ctrl.splicing_events_genes] #"ARHGEF7" "EPB41L2" "SDHAP3"  "SDHAP3"  "SYNGR1"  "SYNGR1"  "ERICH1"  "ERICH1" 

newGeneOverlap(nygc.als_ctrl.splicing_events_genes, untreated_vcp_r155c_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=130
# listB size=218
# Intersection size=5
# Overlapping p-value=1.1e-04
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% untreated_vcp_r155c_ctrl.splicing_events_genes] #"GARS-DT" "FMNL3"   "ACAP3"   "EPB41L2" "ERICH1" 

newGeneOverlap(nygc.als_ctrl.splicing_events_genes, untreated_vcp_r191q_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=130
# listB size=199
# Intersection size=2
# Overlapping p-value=0.069
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% untreated_vcp_r191q_ctrl.splicing_events_genes] # "RALGAPA1" "PRUNE2" 

newGeneOverlap(nygc.als_ctrl.splicing_events_genes, untreated_vcp_iso_ctrl.splicing_events_genes, genome.size=nrow(distinct(gtf,gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=130
# listB size=139
# Intersection size=1
# Overlapping p-value=0.26
# Jaccard Index=0.0
nygc.als_ctrl.splicing_events_genes[nygc.als_ctrl.splicing_events_genes %in% untreated_vcp_iso_ctrl.splicing_events_genes] # "ERICH1"



# Merge
nygc.majiq.als_vs_ctrl.merged = plot_grid(
  plot_grid(majiq.nygc.als_vs_ctrl.volcano, nygc.als_ctrl.voilagost_filt.terms.curated,ncol=2,labels="auto"),
          nygc.als_ctrl.voila.modulizer.barplot,
  nrow = 2, labels = c("","c"))
# nygc.majiq.als_vs_ctrl.merged
ggsave(nygc.majiq.als_vs_ctrl.merged, filename = here(proj_path, "splicing/majiq/nygc.majiq.als_vs_ctrl.merged.png"), height = 5, width = 9)


```



# Fig S11 Fraction splicing

```{r}
# nuclear_untreated_tardbp_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/nuclear_untreated_tardbp_vs_ctrl.tsv"), grp1 = "tardbp_untreated_nuclear", grp2 = "ctrl_untreated_nuclear")
# nuclear_untreated_vcp_r155c_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/nuclear_untreated_vcp_r155c_vs_ctrl.tsv"), grp1 = "vcp_r155c_untreated_nuclear", grp2 = "ctrl_untreated_nuclear")
# nuclear_untreated_vcp_r191q_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/nuclear_untreated_vcp_r191q_vs_ctrl.tsv"), grp1 = "vcp_r191q_untreated_nuclear", grp2 = "ctrl_untreated_nuclear")
# nuclear_untreated_vcp_iso_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/nuclear_untreated_vcp_iso_vs_ctrl.tsv"), grp1 = "vcp_iso_untreated_nuclear", grp2 = "ctrl_untreated_nuclear")
# # nuclear_untreated_als_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/nuclear_untreated_als_vs_ctrl.tsv"), grp1 = "als_untreated_nuclear", grp2 = "ctrl_untreated_nuclear")
# nuclear_untreated_tardbp_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/nuclear_untreated_tardbp_vs_ctrl"), skip_rows = 7)
# nuclear_untreated_vcp_r155c_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/nuclear_untreated_vcp_r155c_vs_ctrl"), skip_rows = 7)
# nuclear_untreated_vcp_r191q_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/nuclear_untreated_vcp_r191q_vs_ctrl"), skip_rows = 7)
# nuclear_untreated_vcp_iso_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/nuclear_untreated_vcp_iso_vs_ctrl"), skip_rows = 7)
# # nuclear_untreated_als_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/nuclear_untreated_als_vs_ctrl"), skip_rows = 7)
# save(nuclear_untreated_tardbp_vs_ctrl.voila, nuclear_untreated_vcp_r155c_vs_ctrl.voila, nuclear_untreated_vcp_r191q_vs_ctrl.voila, nuclear_untreated_vcp_iso_vs_ctrl.voila, #nuclear_untreated_als_vs_ctrl.voila,
#      nuclear_untreated_tardbp_vs_ctrl.modulizer, nuclear_untreated_vcp_r155c_vs_ctrl.modulizer, nuclear_untreated_vcp_r191q_vs_ctrl.modulizer, nuclear_untreated_vcp_iso_vs_ctrl.modulizer, #nuclear_untreated_als_vs_ctrl.modulizer,
#      file = here(proj_path, "scripts/nuclear_untreated_majiq_objects.RData"))
# cytoplasm_untreated_tardbp_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/cytoplasm_untreated_tardbp_vs_ctrl.tsv"), grp1 = "tardbp_untreated_cytoplasm", grp2 = "ctrl_untreated_cytoplasm")
# cytoplasm_untreated_vcp_r155c_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/cytoplasm_untreated_vcp_r155c_vs_ctrl.tsv"), grp1 = "vcp_r155c_untreated_cytoplasm", grp2 = "ctrl_untreated_cytoplasm")
# cytoplasm_untreated_vcp_r191q_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/cytoplasm_untreated_vcp_r191q_vs_ctrl.tsv"), grp1 = "vcp_r191q_untreated_cytoplasm", grp2 = "ctrl_untreated_cytoplasm")
# cytoplasm_untreated_vcp_iso_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/cytoplasm_untreated_vcp_iso_vs_ctrl.tsv"), grp1 = "vcp_iso_untreated_cytoplasm", grp2 = "ctrl_untreated_cytoplasm")
# # cytoplasm_untreated_als_vs_ctrl.voila = read_voila_deltapsi(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/voila-tsv/cytoplasm_untreated_als_vs_ctrl.tsv"), grp1 = "als_untreated_cytoplasm", grp2 = "ctrl_untreated_cytoplasm")
# cytoplasm_untreated_tardbp_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/cytoplasm_untreated_tardbp_vs_ctrl"), skip_rows = 7)
# cytoplasm_untreated_vcp_r155c_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/cytoplasm_untreated_vcp_r155c_vs_ctrl"), skip_rows = 7)
# cytoplasm_untreated_vcp_r191q_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/cytoplasm_untreated_vcp_r191q_vs_ctrl"), skip_rows = 7)
# cytoplasm_untreated_vcp_iso_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/cytoplasm_untreated_vcp_iso_vs_ctrl"), skip_rows = 7)
# # cytoplasm_untreated_als_vs_ctrl.modulizer <- read_voila_modulizer(here(collab_path,"motor-neuron-vcp-inhibitor-harley-2022/splicing/majiq/modulize/cytoplasm_untreated_als_vs_ctrl"), skip_rows = 7)
# save(cytoplasm_untreated_tardbp_vs_ctrl.voila, cytoplasm_untreated_vcp_r155c_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_iso_vs_ctrl.voila, #cytoplasm_untreated_als_vs_ctrl.voila,
#      cytoplasm_untreated_tardbp_vs_ctrl.modulizer, cytoplasm_untreated_vcp_r155c_vs_ctrl.modulizer, cytoplasm_untreated_vcp_r191q_vs_ctrl.modulizer, cytoplasm_untreated_vcp_iso_vs_ctrl.modulizer, #cytoplasm_untreated_als_vs_ctrl.modulizer,
#      file = here(proj_path, "scripts/cytoplasm_untreated_majiq_objects.RData"))
load(here(proj_path, "scripts/nuclear_untreated_majiq_objects.RData"))
load(here(proj_path, "scripts/cytoplasm_untreated_majiq_objects.RData"))
load(here(proj_path, "scripts/whole_untreated_majiq_objects.RData"))

# TARDBP
nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #579
nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) # 243
nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #15
nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-A, SETX
nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nucleocytoplasmic_genes) #16
cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #303
cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #136
cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #5
cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-A
cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nucleocytoplasmic_genes) #6
# unique splicing changes
distinct(filter(bind_rows(nuclear_untreated_tardbp_vs_ctrl.voila, cytoplasm_untreated_tardbp_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE) #817
# overlap fraction & whole
pull(filter(whole_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(whole_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id) %in% pull(filter(bind_rows(nuclear_untreated_tardbp_vs_ctrl.voila, cytoplasm_untreated_tardbp_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id)] #170
170/400 #42.5
# overlap nuclear & cytoplasm
pull(filter(cytoplasm_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(cytoplasm_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)
 %in% pull(filter(nuclear_untreated_tardbp_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)] #65
65/303 #21.5


# vcp_r155c
nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #558
nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) # 224
nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #15
nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # MAPK12, SETX
nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nucleocytoplasmic_genes) #17
cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #440
cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #185
cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #10
cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-A, HLA-B, OPTN
cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% nucleocytoplasmic_genes) #14
# unique splicing changes
distinct(filter(bind_rows(select(nuclear_untreated_vcp_r155c_vs_ctrl.voila,-num_exons), cytoplasm_untreated_vcp_r155c_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE) #867
# overlap fraction & whole
pull(filter(whole_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(whole_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id) %in% pull(filter(bind_rows(select(nuclear_untreated_vcp_r155c_vs_ctrl.voila,-num_exons), cytoplasm_untreated_vcp_r155c_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id)] #170
282/531 #53.1
# overlap nuclear & cytoplasm
pull(filter(cytoplasm_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(cytoplasm_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)
 %in% pull(filter(nuclear_untreated_vcp_r155c_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)] #131
131/440 #29.8


# vcp_r191q
nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #1585
nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #658
nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #108
nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-A, HLA-C, GRIA1, SQSTM1, DAXX, UNC13B
nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nucleocytoplasmic_genes) #113
cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #774
cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #323
cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #25
cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) # HLA-B, APOE
cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% nucleocytoplasmic_genes) #27
# unique splicing changes
distinct(filter(bind_rows(nuclear_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE) #2170
# overlap fraction & whole
pull(filter(whole_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(whole_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id) %in% pull(filter(bind_rows(nuclear_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id)] #170
323/512 #63.1
# how many fraction events were identifiable in whole cell
pull(distinct(filter(bind_rows(nuclear_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE),lsv_junction_id)[pull(distinct(filter(bind_rows(nuclear_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE),lsv_junction_id) %in% pull(filter(whole_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)] #323
# overlap nuclear & cytoplasm
pull(filter(cytoplasm_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(cytoplasm_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)
 %in% pull(filter(nuclear_untreated_vcp_r191q_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)] #189
189/774 #24.4


# vcp_iso
nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #2157
nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #950
nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #237
nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) %>% print(n=Inf)#35
nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% nucleocytoplasmic_genes) #239
cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #343
cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% distinct(gene_name) #159
cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% RNA_BINDING_PROTEINS) #15
cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, gene_name %in% ALS_genes) #0
cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE, gene_name %in% nucleocytoplasmic_genes) #18
# unique splicing changes
distinct(filter(bind_rows(nuclear_untreated_vcp_iso_vs_ctrl.voila, cytoplasm_untreated_vcp_iso_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id, .keep_all=TRUE) #2429
# overlap fraction & whole
pull(filter(whole_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(whole_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id) %in% pull(filter(bind_rows(nuclear_untreated_vcp_iso_vs_ctrl.voila, cytoplasm_untreated_vcp_iso_vs_ctrl.voila), changing_dpsi0.2 == TRUE), lsv_junction_id)] #170
134/329 #40.7
# overlap nuclear & cytoplasm
pull(filter(cytoplasm_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)[pull(filter(cytoplasm_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)
 %in% pull(filter(nuclear_untreated_vcp_iso_vs_ctrl.voila, changing_dpsi0.2 == TRUE), lsv_junction_id)] #71
71/343 #20.7


# UpSet nuc, cyt, whole events
whole_nuc_cyt_untreated_tardbp_vs_ctrl.voila.bind = bind_rows(mutate(whole_untreated_tardbp_vs_ctrl.voila, fraction = "Whole"), mutate(nuclear_untreated_tardbp_vs_ctrl.voila, fraction = "Nuclear"), mutate(cytoplasm_untreated_tardbp_vs_ctrl.voila, fraction = "Cytoplasm"))
whole_nuc_cyt_untreated_tardbp_vs_ctrl.voila.bind.upset = majiq_upset_plot(whole_nuc_cyt_untreated_tardbp_vs_ctrl.voila.bind, overlap_by = "fraction", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 500, order = "freq", order_group_list = c("Whole", "Nuclear", "Cytoplasm"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen"), title = "TARDBP")
whole_nuc_cyt_untreated_tardbp_vs_ctrl.voila.bind.upset

whole_nuc_cyt_untreated_vcp_r155c_vs_ctrl.voila.bind = bind_rows(mutate(whole_untreated_vcp_r155c_vs_ctrl.voila, fraction = "Whole"), mutate(nuclear_untreated_vcp_r155c_vs_ctrl.voila, fraction = "Nuclear"), mutate(select(cytoplasm_untreated_vcp_r155c_vs_ctrl.voila,-num_exons), fraction = "Cytoplasm"))
whole_nuc_cyt_untreated_vcp_r155c_vs_ctrl.voila.bind.upset = majiq_upset_plot(whole_nuc_cyt_untreated_vcp_r155c_vs_ctrl.voila.bind, overlap_by = "fraction", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 370, order = "freq", order_group_list = c("Whole", "Nuclear", "Cytoplasm"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen"), title = "VCP R155C")
whole_nuc_cyt_untreated_vcp_r155c_vs_ctrl.voila.bind.upset

whole_nuc_cyt_untreated_vcp_r191q_vs_ctrl.voila.bind = bind_rows(mutate(whole_untreated_vcp_r191q_vs_ctrl.voila, fraction = "Whole"), mutate(nuclear_untreated_vcp_r191q_vs_ctrl.voila, fraction = "Nuclear"), mutate(cytoplasm_untreated_vcp_r191q_vs_ctrl.voila, fraction = "Cytoplasm"))
whole_nuc_cyt_untreated_vcp_r191q_vs_ctrl.voila.bind.upset = majiq_upset_plot(whole_nuc_cyt_untreated_vcp_r191q_vs_ctrl.voila.bind, overlap_by = "fraction", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 1400, order = "freq", order_group_list = c("Whole", "Nuclear", "Cytoplasm"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen"), title = "VCP R191Q")
whole_nuc_cyt_untreated_vcp_r191q_vs_ctrl.voila.bind.upset

whole_nuc_cyt_untreated_vcp_iso_vs_ctrl.voila.bind = bind_rows(mutate(whole_untreated_vcp_iso_vs_ctrl.voila, fraction = "Whole"), mutate(nuclear_untreated_vcp_iso_vs_ctrl.voila, fraction = "Nuclear"), mutate(cytoplasm_untreated_vcp_iso_vs_ctrl.voila, fraction = "Cytoplasm"))
whole_nuc_cyt_untreated_vcp_iso_vs_ctrl.voila.bind.upset = majiq_upset_plot(whole_nuc_cyt_untreated_vcp_iso_vs_ctrl.voila.bind, overlap_by = "fraction", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 2100, order = "freq", order_group_list = c("Whole", "Nuclear", "Cytoplasm"), label_colours = c("dodgerblue2", "firebrick2", "forestgreen"), title = "VCP knock-in")
whole_nuc_cyt_untreated_vcp_iso_vs_ctrl.voila.bind.upset


# heatmap of correlation coefficients
whole_nuclear_cytoplasm_untreated_als_vs_ctrl.voila.deltapsi =
  select(nuclear_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, `TARDBP^G298S\nnuclear` = deltapsi) %>%
  left_join(select(nuclear_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, `VCP^R155C\nnuclear` = deltapsi)) %>%
  left_join(select(nuclear_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, `VCP^R191Q\nnuclear` = deltapsi)) %>%
  left_join(select(nuclear_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, `VCPki^R191Q\nnuclear` = deltapsi)) %>%
  left_join(select(cytoplasm_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, `TARDBP^G298S\ncytoplasm` = deltapsi)) %>% 
  left_join(select(cytoplasm_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, `VCP^R155C\ncytoplasm` = deltapsi)) %>%
  left_join(select(cytoplasm_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, `VCP^R191Q\ncytoplasm` = deltapsi)) %>%
  left_join(select(cytoplasm_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, `VCPki^R191Q\ncytoplasm` = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")
cormat <- round(cor(whole_nuclear_cytoplasm_untreated_als_vs_ctrl.voila.deltapsi),2)
head(cormat)
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) %>% # Melt the correlation matrix
  filter(grepl("nuclear",Var1)&grepl("cytoplasm",Var2), (grepl("TARDBP",Var1) & grepl("TARDBP",Var2)) | (grepl("VCP\\^R155C",Var1) & grepl("VCP\\^R155C",Var2)) | (grepl("VCP\\^R191Q",Var1) & grepl("VCP\\^R191Q",Var2)) | (grepl("VCPki",Var1) & grepl("VCPki",Var2)) ) 

whole_nuclear_cytoplasm_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  + labs(y="Nuclear",x="Cytoplasm") +
  theme_minimal() + theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) + scale_x_discrete(labels = scales::parse_format()) +  scale_y_discrete(labels = scales::parse_format()) 
whole_nuclear_cytoplasm_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap


# Modulize
nuclear_cytoplasm_mutants_vs_ctrl.modulizer.bind = bind_rows(
  mutate(rename(nuclear_untreated_tardbp_vs_ctrl.modulizer, dpsi = ctrl_untreated_nuclear_tardbp_untreated_nuclear_median_dpsi, probability_changing = ctrl_untreated_nuclear_tardbp_untreated_nuclear_probability_changing), mutation = "TARDBP^G298S", fraction = "Nuclear"), 
  mutate(rename(cytoplasm_untreated_tardbp_vs_ctrl.modulizer, dpsi = ctrl_untreated_cytoplasm_tardbp_untreated_cytoplasm_median_dpsi, probability_changing = ctrl_untreated_cytoplasm_tardbp_untreated_cytoplasm_probability_changing), mutation = "TARDBP^G298S", fraction = "Cytoplasm"), 
  mutate(rename(nuclear_untreated_vcp_r155c_vs_ctrl.modulizer, dpsi = ctrl_untreated_nuclear_vcp_r155c_untreated_nuclear_median_dpsi, probability_changing = ctrl_untreated_nuclear_vcp_r155c_untreated_nuclear_probability_changing), mutation = "VCP^R155C", fraction = "Nuclear"),
  mutate(rename(cytoplasm_untreated_vcp_r155c_vs_ctrl.modulizer, dpsi = ctrl_untreated_cytoplasm_vcp_r155c_untreated_cytoplasm_median_dpsi, probability_changing = ctrl_untreated_cytoplasm_vcp_r155c_untreated_cytoplasm_probability_changing), mutation = "VCP^R155C", fraction = "Cytoplasm"),
  mutate(rename(nuclear_untreated_vcp_r191q_vs_ctrl.modulizer, dpsi = ctrl_untreated_nuclear_vcp_r191q_untreated_nuclear_median_dpsi, probability_changing = ctrl_untreated_nuclear_vcp_r191q_untreated_nuclear_probability_changing), mutation = "VCP^R191Q", fraction = "Nuclear"), 
  mutate(rename(cytoplasm_untreated_vcp_r191q_vs_ctrl.modulizer, dpsi = ctrl_untreated_cytoplasm_vcp_r191q_untreated_cytoplasm_median_dpsi, probability_changing = ctrl_untreated_cytoplasm_vcp_r191q_untreated_cytoplasm_probability_changing), mutation = "VCP^R191Q", fraction = "Cytoplasm"), 
  mutate(rename(nuclear_untreated_vcp_iso_vs_ctrl.modulizer, dpsi = ctrl_untreated_nuclear_vcp_iso_untreated_nuclear_median_dpsi, probability_changing = ctrl_untreated_nuclear_vcp_iso_untreated_nuclear_probability_changing), mutation = "VCPki^R191Q", fraction = "Nuclear"),
  mutate(rename(cytoplasm_untreated_vcp_iso_vs_ctrl.modulizer, dpsi = ctrl_untreated_cytoplasm_vcp_iso_untreated_cytoplasm_median_dpsi, probability_changing = ctrl_untreated_cytoplasm_vcp_iso_untreated_cytoplasm_probability_changing), mutation = "VCPki^R191Q", fraction = "Cytoplasm")) %>% 
  mutate(mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q","VCPki^R191Q")), modulizer_simplified = gsub("_"," ",modulizer_simplified)) %>%
  filter(probability_changing > 0.9, abs(dpsi) > 0.2)

nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.count = nuclear_cytoplasm_mutants_vs_ctrl.modulizer.bind %>% count(mutation, fraction, modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  group_by(mutation, fraction) %>% add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(modulizer_simplified == "intron retention" ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.count %>% print(n=Inf)

nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.barplot = ggbarplot(nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.count, x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "",  label = nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  facet_grid(fraction ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.barplot



fraction_splicing.merged = plot_grid(
  plot_grid(whole_nuc_cyt_untreated_tardbp_vs_ctrl.voila.bind.upset, whole_nuc_cyt_untreated_vcp_r155c_vs_ctrl.voila.bind.upset, whole_nuc_cyt_untreated_vcp_r191q_vs_ctrl.voila.bind.upset, whole_nuc_cyt_untreated_vcp_iso_vs_ctrl.voila.bind.upset, ncol = 4),
  plot_grid(whole_nuclear_cytoplasm_als_vs_ctrl.voila.deltapsi_correlation_ggheatmap, nuclear_cytoplasm_untreated_mutants_vs_ctrl.modulizer.bind.barplot, ncol = 2, labels = letters[2:3], rel_widths = c(1,3)),
  nrow = 2, rel_heights = c(0.8,1.2), labels = c("a",""))
# fraction_splicing.merged
ggsave(fraction_splicing.merged, filename = here(proj_path, "splicing/majiq/fraction_splicing.merged.png"), height = 7, width = 12)

```


# Table S8 MAJIQ fractions

```{r}
whole_tardbp.voila.sig_lsv = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
nuclear_tardbp.voila.sig_lsv = nuclear_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
cytoplasm_tardbp.voila.sig_lsv = cytoplasm_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
whole_vcp_r155c.voila.sig_lsv = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
nuclear_vcp_r155c.voila.sig_lsv = nuclear_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
cytoplasm_vcp_r155c.voila.sig_lsv = cytoplasm_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
whole_vcp_r191q.voila.sig_lsv = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
nuclear_vcp_r191q.voila.sig_lsv = nuclear_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
cytoplasm_vcp_r191q.voila.sig_lsv = cytoplasm_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
whole_vcp_iso.voila.sig_lsv = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
nuclear_vcp_iso.voila.sig_lsv = nuclear_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)
cytoplasm_vcp_iso.voila.sig_lsv = cytoplasm_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) %>% pull(lsv_junction_id)


whole_nuclear_cytoplasm.overlap.voila = select(whole_untreated_tardbp_vs_ctrl.voila, gene_name, gene_id, lsv_junction_id) %>%
  mutate(tardbp.spliced = case_when(lsv_junction_id %in% whole_tardbp.voila.sig_lsv & lsv_junction_id %in% nuclear_tardbp.voila.sig_lsv & lsv_junction_id %in% cytoplasm_tardbp.voila.sig_lsv ~ "Whole, Nuc, & Cyto",
                            lsv_junction_id %in% whole_tardbp.voila.sig_lsv & lsv_junction_id %in% nuclear_tardbp.voila.sig_lsv ~ "Whole & Nuc",
                            lsv_junction_id %in% whole_tardbp.voila.sig_lsv & lsv_junction_id %in% cytoplasm_tardbp.voila.sig_lsv ~ "Whole & Cyto",
                            lsv_junction_id %in% nuclear_tardbp.voila.sig_lsv & lsv_junction_id %in% cytoplasm_tardbp.voila.sig_lsv ~ "Nuc & Cyto",
                            lsv_junction_id %in% whole_tardbp.voila.sig_lsv ~ "Whole only", lsv_junction_id %in% nuclear_tardbp.voila.sig_lsv ~ "Nuc only", lsv_junction_id %in% cytoplasm_tardbp.voila.sig_lsv ~ "Cyto only", TRUE ~ ""),
         vcp_r155c.spliced = case_when(lsv_junction_id %in% whole_vcp_r155c.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_r155c.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r155c.voila.sig_lsv ~ "Whole, Nuc, & Cyto",
                            lsv_junction_id %in% whole_vcp_r155c.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_r155c.voila.sig_lsv ~ "Whole & Nuc",
                            lsv_junction_id %in% whole_vcp_r155c.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r155c.voila.sig_lsv ~ "Whole & Cyto",
                            lsv_junction_id %in% nuclear_vcp_r155c.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r155c.voila.sig_lsv ~ "Nuc & Cyto",
                            lsv_junction_id %in% whole_vcp_r155c.voila.sig_lsv ~ "Whole only", lsv_junction_id %in% nuclear_vcp_r155c.voila.sig_lsv ~ "Nuc only", lsv_junction_id %in% cytoplasm_vcp_r155c.voila.sig_lsv ~ "Cyto only", TRUE ~ ""),
         vcp_r191q.spliced = case_when(lsv_junction_id %in% whole_vcp_r191q.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_r191q.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r191q.voila.sig_lsv ~ "Whole, Nuc, & Cyto",
                            lsv_junction_id %in% whole_vcp_r191q.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_r191q.voila.sig_lsv ~ "Whole & Nuc",
                            lsv_junction_id %in% whole_vcp_r191q.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r191q.voila.sig_lsv ~ "Whole & Cyto",
                            lsv_junction_id %in% nuclear_vcp_r191q.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_r191q.voila.sig_lsv ~ "Nuc & Cyto",
                            lsv_junction_id %in% whole_vcp_r191q.voila.sig_lsv ~ "Whole only", lsv_junction_id %in% nuclear_vcp_r191q.voila.sig_lsv ~ "Nuc only", lsv_junction_id %in% cytoplasm_vcp_r191q.voila.sig_lsv ~ "Cyto only", TRUE ~ ""),
         vcp_ki.spliced = case_when(lsv_junction_id %in% whole_vcp_iso.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_iso.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_iso.voila.sig_lsv ~ "Whole, Nuc, & Cyto",
                            lsv_junction_id %in% whole_vcp_iso.voila.sig_lsv & lsv_junction_id %in% nuclear_vcp_iso.voila.sig_lsv ~ "Whole & Nuc",
                            lsv_junction_id %in% whole_vcp_iso.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_iso.voila.sig_lsv ~ "Whole & Cyto",
                            lsv_junction_id %in% nuclear_vcp_iso.voila.sig_lsv & lsv_junction_id %in% cytoplasm_vcp_iso.voila.sig_lsv ~ "Nuc & Cyto",
                            lsv_junction_id %in% whole_vcp_iso.voila.sig_lsv ~ "Whole only", lsv_junction_id %in% nuclear_vcp_iso.voila.sig_lsv ~ "Nuc only", lsv_junction_id %in% cytoplasm_vcp_iso.voila.sig_lsv ~ "Cyto only", TRUE ~ "")) %>% filter(!(tardbp.spliced == "" & vcp_r155c.spliced == "" & vcp_r191q.spliced == "" & vcp_ki.spliced == "")) %>%
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx,padj<0.05), tardbp.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, tardbp.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx,padj<0.05), vcp_r155c.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_r155c.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx,padj<0.05), vcp_r191q.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_r191q.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx,padj<0.05), vcp_ki.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, vcp_ki.redistributed)) %>% 
  left_join(select(mutate(filter(untreated.nuc_vs_cyt.als_vs_ctrl$res.tx,padj<0.05), combined.redistributed = case_when(stat>0~"nuclear",stat<0~"cytoplasmic")), gene_name, gene_id, combined.redistributed)) %>% 
      mutate_if(is.character, ~replace_na(.,"")) %>%
  mutate(category = case_when(gene_name %in% ALS_genes ~ "ALS gene",  gene_name %in% nuclear_export.markers ~ "Nuclear export",  gene_name %in% RNA_BINDING_PROTEINS ~ "RBP", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ ""))
whole_nuclear_cytoplasm.overlap.voila
write_csv(whole_nuclear_cytoplasm.overlap.voila, here(proj_path, "splicing/majiq/whole_nuclear_cytoplasm.overlap.voila.csv"), na = "")
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(whole_nuclear_cytoplasm.overlap.voila, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S8 Splicing fractions")


whole_nuclear_cytoplasm.overlap.voila %>% filter(tardbp.spliced != "" & tardbp.redistributed != "") #18
whole_nuclear_cytoplasm.overlap.voila %>% filter(tardbp.spliced != "" & tardbp.redistributed != "") %>% distinct(gene_name, .keep_all = TRUE) # 6 genes
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_r155c.spliced != "" & vcp_r155c.redistributed != "") #92
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_r155c.spliced != "" & vcp_r155c.redistributed != "") %>% distinct(gene_name, .keep_all = TRUE) # 19 genes
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_r191q.spliced != "" & vcp_r191q.redistributed != "") #256
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_r191q.spliced != "" & vcp_r191q.redistributed != "") %>% distinct(gene_name, .keep_all = TRUE) # 61 genes
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_ki.spliced != "" & vcp_ki.redistributed != "") #173
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_ki.spliced != "" & vcp_ki.redistributed != "") %>% distinct(gene_name, .keep_all = TRUE) # 54 genes

whole_nuclear_cytoplasm.overlap.voila %>% filter(tardbp.spliced != "" & tardbp.redistributed != "" & vcp_r155c.spliced != "" & vcp_r155c.redistributed != "" & vcp_r191q.spliced != "" & vcp_r191q.redistributed != "") # YBEY
whole_nuclear_cytoplasm.overlap.voila %>% filter(tardbp.spliced != "" & tardbp.redistributed != "" & vcp_r155c.spliced != "" & vcp_r155c.redistributed != "" & vcp_ki.spliced != "" & vcp_ki.redistributed != "") # 
whole_nuclear_cytoplasm.overlap.voila %>% filter(tardbp.spliced != "" & tardbp.redistributed != "" & vcp_r191q.spliced != "" & vcp_r191q.redistributed != "" & vcp_ki.spliced != "" & vcp_ki.redistributed != "") # 
whole_nuclear_cytoplasm.overlap.voila %>% filter(vcp_r191q.spliced != "" & vcp_r191q.redistributed != "" & vcp_r155c.spliced != "" & vcp_r155c.redistributed != "" & vcp_ki.spliced != "" & vcp_ki.redistributed != "") # 


# Fisher exact overlap test
untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
whole_nuclear_cytoplasm_untreated_tardbp_vs_ctrl_splicing_events_genes = bind_rows(whole_untreated_tardbp_vs_ctrl.voila, nuclear_untreated_tardbp_vs_ctrl.voila, cytoplasm_untreated_tardbp_vs_ctrl.voila) %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
newGeneOverlap(whole_nuclear_cytoplasm_untreated_tardbp_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.tardbp_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=385
# listB size=365
# Intersection size=8
# Overlapping p-value=2.5e-03
# Jaccard Index=0.0

untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
whole_nuclear_cytoplasm_untreated_vcp_r155c_vs_ctrl_splicing_events_genes = bind_rows(whole_untreated_vcp_r155c_vs_ctrl.voila, nuclear_untreated_vcp_r155c_vs_ctrl.voila, select(cytoplasm_untreated_vcp_r155c_vs_ctrl.voila,-num_exons)) %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
newGeneOverlap(whole_nuclear_cytoplasm_untreated_vcp_r155c_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=416
# listB size=345
# Intersection size=20
# Overlapping p-value=5.9e-13
# Jaccard Index=0.0

untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
whole_nuclear_cytoplasm_untreated_vcp_r191q_vs_ctrl_splicing_events_genes = bind_rows(whole_untreated_vcp_r191q_vs_ctrl.voila, nuclear_untreated_vcp_r191q_vs_ctrl.voila, cytoplasm_untreated_vcp_r191q_vs_ctrl.voila) %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
newGeneOverlap(whole_nuclear_cytoplasm_untreated_vcp_r191q_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=914
# listB size=568
# Intersection size=64
# Overlapping p-value=9e-36
# Jaccard Index=0.0

untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.sig = filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05) %>% pull(gene_name)
whole_nuclear_cytoplasm_untreated_vcp_iso_vs_ctrl_splicing_events_genes = bind_rows(whole_untreated_vcp_iso_vs_ctrl.voila, nuclear_untreated_vcp_iso_vs_ctrl.voila, cytoplasm_untreated_vcp_iso_vs_ctrl.voila) %>% filter(changing_dpsi0.2 == TRUE) %>% pull(gene_name)
newGeneOverlap(whole_nuclear_cytoplasm_untreated_vcp_iso_vs_ctrl_splicing_events_genes, untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.sig, genome.size=nrow(distinct(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, gene_name))) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=1100
# listB size=541
# Intersection size=55
# Overlapping p-value=1e-24
# Jaccard Index=0.0

```


# Fig 5 Proteomics ALS vs CTRL & Nuc vs Cyt

## TARDBP Volcano  & GO

multi-factor design

```{r}
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) # 577
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0) # 169
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange < 0) # 408
169/577#29.3
408/577 #70.7
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 140
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 11
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) %>% print(n=Inf) # 22
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(grepl("TARDBP",gene_name))

# transcript level volcano
untreated.tardbp_vs_ctrl.nuc_vs_cyt.rbp.labels <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.als.labels <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(untreated.tardbp_vs_ctrl.nuc_vs_cyt.als.labels, untreated.tardbp_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 7, ymax = 9, dpi = 300, distinct_labels = TRUE, title = "TARDBP mutant") + 
  xlab(expression(TARDBP~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 7*0.25, xend = 7*0.95, y= 9*0.88, yend= 9 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -7*0.25, xend = -7*0.95, y= 9*0.88, yend= 9*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 7*0.6, y = 9*0.93, label = "Nuclear", size = 3) + annotate("text", x = -7*0.6, y = 9*0.93, label = "Cytoplasm", size = 3)
# untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano

# # # # density plot
# untreated.tardbp_vs_ctrl.nuc_vs_cyt.density = density_plot(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, ymax = 0.45, xlims = c(-7,7), xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# untreated.tardbp_vs_ctrl.nuc_vs_cyt.density

### GO mislocalised
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       210        64 
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"Nucleocytoplasmic transport",
# "regulation of mRNA processing",
"chromosome organization",
"protein binding",
"regulation of RNA splicing",
"translation",
"nucleic acid binding",
"Ribosome, cytoplasmic",
# "intracellular anatomical structure",
"Nonsense-Mediated Decay (NMD)",
"Metabolism of RNA",
"mRNA binding"
)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"RNA binding",
"Cellular responses to stress",
"protein localization",
"protein transport",
"translation",
# "protein metabolic process",
# "nucleotide binding",
"actin cytoskeleton",
"mitochondrion",
"intracellular transport",
"cellular protein metabolic process",
# "protein-containing complex binding",
"protein binding")
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" \\(NMD\\)", "", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated

```


## VCP R155C mutant volcano & GO

multi-factor design

```{r}
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) # 169
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0) # 55
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange < 0) # 114
55/169 #32.5
114/169 #67.5
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 29
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat <0, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 21
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # TBP, ELAVL3, FUS, NOL4L
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) %>% print(n=Inf) # 5
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(grepl("TARDBP",gene_name))

# transcript level volcano
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rbp.labels <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.als.labels <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.als.labels, untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 7, dpi = 300, distinct_labels = TRUE, title = "VCP R155C mutant") + 
  xlab(expression(VCP~R155C~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 7*0.88, yend= 7*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 7*0.88, yend= 7*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 7*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 7*0.93, label = "Cytoplasm", size = 3)
# untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano

### GO mislocalised
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#        13         3 
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cytoplasm",
"Dsl1/NZR complex",
"Biosynthesis of amino acids"
)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"intracellular anatomical structure",
"protein binding",
"mitochondrial matrix",
"histone binding",
"nucleoplasm",
"cytoplasm")
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub("COPII-coated ER to Golgi transport vesicle", "ER to Golgi transport", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, rel_heights = c(1,2), align = "v") 
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated


untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_CYTOSOLIC_PART)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_AXON)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 89
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% ALS_genes) #14

```


## VCP R191Q mutant volcano & GO

multi-factor design

```{r}
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) # 867
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0) # 113
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange < 0) # 754
113/867#13
754/867#87
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 172
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes) %>% print(n=Inf)# 21
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf)# 5
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) %>% print(n=Inf) # 31
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS) %>% print(n=Inf) # 31
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(grepl("TARDBP",gene_name))

# transcript level volcano
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rbp.labels <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.als.labels <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.als.labels, untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 8, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q mutant") + 
  xlab(expression(VCP~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 8*0.88, yend= 8*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 8*0.88, yend= 8*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 8*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 8*0.93, label = "Cytoplasm", size = 3)
# untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano

### GO mislocalised
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       723       237 
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
# "chromosome organization",
"protein binding",
# "mRNA processing",
"RNA metabolic process",
# "regulation of gene expression",
# "protein-containing complex",
"ribonucleoprotein complex",
"regulation of RNA splicing",
"Metabolism of RNA",
"RNA binding",
"mRNA Splicing",
"spliceosomal complex"
)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
# "Nonsense-Mediated Decay (NMD)",
"Metabolism of RNA",
"protein transport",
"translation",
"Cellular responses to stress",
"cytoskeleton organization",
# "Axon guidance",
# "intracellular anatomical structure",
"protein localization",
"axon",
# "cytosolic ribosome",
"synapse",
# "Eukaryotic Translation Initiation",
"protein binding")
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_filt_combined %<>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated

untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_CYTOSOLIC_PART)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_AXON)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 89
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% ALS_genes) #14

```



## VCP knock-in volcano & GO

multi-factor design

```{r}
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) # 639
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0) # 203
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange < 0) # 436
436/639#68.2
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 117
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 17 FUS, TAF15, TARDBP
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # FUS, TAF15
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) %>% print(n=Inf) # 17
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(grepl("TARDBP",gene_name))
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, grepl("HNRN",gene_name))

# transcript level volcano
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rbp.labels <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.als.labels <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.als.labels, untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 8, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q knock-in") + 
  xlab(expression(VCPki~R191~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 8*0.88, yend= 8 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 8*0.88, yend= 8*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 8*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 8*0.93, label = "Cytoplasm", size = 3)
# untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano


### GO mislocalised
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       213        64 
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"Metabolism of RNA",
# "nuclear import signal receptor activity",
# "endoplasmic reticulum",
"DNA repair",
"DNA metabolic process",
"cellular response to stress",
"endoplasmic reticulum to Golgi vesicle-mediated transport",
"Cholesterol biosynthesis",
"mitochondrion",
# "nucleoside phosphate binding",
# "nucleotide binding",
"protein binding",
"endoplasmic reticulum exit site"
# "catalytic activity"
)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"RNA binding",
"protein localization",
"cytoskeleton organization",
"translation",
"axon",
"regulation of mRNA metabolic process",
"neuron projection",
# "mRNA catabolic process",
# "protein-containing complex binding",
"synapse",
"Nervous system development",
"protein binding")
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" exit site", "", term_name), term_name = gsub("endoplasmic reticulum to Golgi vesicle-mediated transport","ER to Golgi transport",term_name))
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated

```


## Mutants combined

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.dep = readRDS(file = here(proj_path, "scripts/untreated.als_vs_ctrl.nuc_vs_cyt.dep.rds"))

untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) # 578
untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0) # 70
untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange < 0) # 508
70/578#12.1
508/578 #87.9
untreated.als_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) #92
untreated.als_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% ALS_genes) #12
untreated.als_vs_ctrl.nuc_vs_cyt.dep$res  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) %>% print(n=Inf) #15

# transcript level volcano
untreated.als_vs_ctrl.nuc_vs_cyt.rbp.labels <- untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.als_vs_ctrl.nuc_vs_cyt.als.labels <- untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
untreated.als_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(untreated.als_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(untreated.als_vs_ctrl.nuc_vs_cyt.als.labels, untreated.als_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 5, ymax = 7, dpi = 300, distinct_labels = TRUE, title = "Mutants combined") + 
  xlab(expression(ALS~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 5*0.25, xend = 5*0.95, y= 7*0.88, yend= 7 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -5*0.25, xend = -5*0.95, y= 7*0.88, yend= 7*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 5*0.6, y = 7*0.93, label = "Nuclear", size = 3) + annotate("text", x = -5*0.6, y = 7*0.93, label = "Cytoplasm", size = 3)
untreated.als_vs_ctrl.nuc_vs_cyt.dep.volcano


### GO mislocalised
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost <- untreated.als_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       608        42 
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"Processing of Capped Intron-Containing Pre-mRNA",
"DNA/RNA helicase activity",
"regulation of mRNA splicing, via spliceosome",
"single-stranded DNA binding",
"ATP binding",
"catalytic activity",
"ribonucleotide binding",
# "telomeric DNA binding",
"cytoplasm",
"nucleotide binding"
)
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub("Capped Intron-Containing ", "", term_name), term_name = gsub("regulation of ","",term_name), term_name = gsub(", via spliceosome","",term_name))

paste('"',unique(untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"intracellular transport",
"protein localization",
"Amyotrophic lateral sclerosis",
"Pathways of neurodegeneration - multiple diseases",
# "protein-containing complex",
"proteasome core complex",
"Axon guidance",
"cellular localization",
"synapse",
# "mitochondrion",
"protein binding")
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm %<>% mutate(term_name = gsub("Pathways of neurodegeneration - multiple diseases", "neurodegeneration pathways", term_name))
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.go_curated

```

## Correlation heatmap

```{r}
# heatmap of correlation coefficients
untreated.als_vs_ctrl.nuc_vs_cyt.dep_stat = select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `TARDBP^G298S` = stat) %>% 
  left_join(select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R155C` = stat)) %>%
  left_join(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R191Q` = stat)) %>%
  left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCPki^R191Q` = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_name")
cormat <- round(cor(untreated.als_vs_ctrl.nuc_vs_cyt.dep_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
untreated.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
# untreated.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap

```


## RNAct

```{r}
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.tardbp.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r155c.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r191q.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_iso.nuclear_cytoplasm_protein_transcript.rds"))

RNAct_database.tardbp.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.tardbp.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"),  mutate(RNAct_database.tardbp.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_r155c.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r155c.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_r191q.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r191q.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_iso.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_iso.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))

RNAct_database.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.protein_transcript.bind, mutation = "TARDBP mutant"), mutate(RNAct_database.vcp_r155c.protein_transcript.bind, mutation = "VCP R155C mutant"), mutate(RNAct_database.vcp_r191q.protein_transcript.bind, mutation = "VCP R191Q mutant"), mutate(RNAct_database.vcp_iso.protein_transcript.bind, mutation = "VCPki R191Q")) %>% mutate(mutation = factor(mutation, levels = c("TARDBP mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCPki R191Q")), mislocalisation = factor(mislocalisation, levels = c("Non Tx\nNon Pr", "Nuc Tx\nNuc Pr", "Cyt Tx\nCyt Pr")))

RNAct_database.tardbp.protein_transcript.bind %>% my_summarise(catrapid_score_max_normalised)

# Mislocalisation Nuclear vs Cytoplasm vs None
rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct_database.protein_transcript.bind, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(5,25), facet_by = "mutation", facet_ncol = 4, plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(22, 24, 22.5),4), 
                                                               ylabel = expression(RNA~protein~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2","forestgreen","gold2"), dpi = 300, arrow_labels = FALSE)
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin
# facet            .y.   group1           group2               n1     n2 statistic      df         p     p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>            <chr> <chr>            <chr>             <int>  <int>     <dbl>   <dbl>     <dbl>     <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP mutant    lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  10836   -40.9    15747. 0         0         ****               22   <chr [2]>        1     2
#  2 TARDBP mutant    lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000  50355   -69.2   117396. 0         0         ****               24   <chr [2]>        1     3
#  3 TARDBP mutant    lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  10836  50355    -3.55   17804. 3.92e-  4 3.92e-  4 ***                22.5 <chr [2]>        2     3
#  4 VCP R155C mutant lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000   3648   -22.2     4026. 5.68e-103 1.14e-102 ****               22   <chr [2]>        1     2
#  5 VCP R155C mutant lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000   8455   -29.7    10227. 8.76e-186 2.63e-185 ****               24   <chr [2]>        1     3
#  6 VCP R155C mutant lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"   3648   8455     0.502   7441. 6.16e-  1 6.16e-  1 ns                 22.5 <chr [2]>        2     3
#  7 VCP R191Q mutant lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  10560   -30.5    14468. 6.19e-198 1.24e-197 ****               22   <chr [2]>        1     2
#  8 VCP R191Q mutant lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000 157528   -80.8   194130. 0         0         ****               24   <chr [2]>        1     3
#  9 VCP R191Q mutant lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  10560 157528    -7.76   12450. 9   e- 15 9   e- 15 ****               22.5 <chr [2]>        2     3
# 10 VCPki R191Q      lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  20651   -64.6    37833. 0         0         ****               22   <chr [2]>        1     2
# 11 VCPki R191Q      lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000  90792   -70.3   190684. 0         0         ****               24   <chr [2]>        1     3
# 12 VCPki R191Q      lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  20651  90792    14.8    35299. 1.99e- 49 1.99e- 49 ****               22.5 <chr [2]>        2     3



# # other groups of non-redistributed & non-mislocalised for comparison:
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.tardbp.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r155c.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r191q.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_iso.nuclear_cytoplasm_protein_transcript.rds"))
# 
# RNAct_database.tardbp.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.tardbp.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.tardbp.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.tardbp.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.tardbp.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"),  mutate(RNAct_database.tardbp.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"),  mutate(RNAct_database.tardbp.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_r155c.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r155c.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_r155c.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_r191q.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r191q.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_r191q.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_iso.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_iso.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_iso.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# 
# RNAct_database.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.protein_transcript.bind, mutation = "TARDBP mutant"), mutate(RNAct_database.vcp_r155c.protein_transcript.bind, mutation = "VCP R155C mutant"), mutate(RNAct_database.vcp_r191q.protein_transcript.bind, mutation = "VCP R191Q mutant"), mutate(RNAct_database.vcp_iso.protein_transcript.bind, mutation = "VCPki R191Q")) %>% 
#   mutate(mutation = factor(mutation, levels = c("TARDBP mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCPki R191Q")), 
#          mislocalisation = factor(mislocalisation, levels = c("Non Tx\nNon Pr", "Non Tx\nCyt Pr", "Non Tx\nNuc Pr", "Cyt Tx\nNon Pr", "Nuc Tx\nNon Pr", "Nuc Tx\nNuc Pr", "Cyt Tx\nCyt Pr")))
# 
# RNAct_database.protein_transcript.bind %>% my_summarise(catrapid_score_max_normalised, mislocalisation, mutation)
# 
# # Mislocalisation Nuclear vs Cytoplasm vs Non
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct_database.protein_transcript.bind, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(5,25), facet_by = "mutation", facet_ncol = 4, 
#                                                               plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = 24,#rep(c(22, 24, 22.5),4), 
#                                                               ylabel = expression(RNA~protein~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2","forestgreen","gold2"), dpi = 30, arrow_labels = FALSE)
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin

```

## Merge

```{r}
vcp_tardbp_protein_redistribution.merged <- plot_grid(
  plot_grid(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano, untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano, untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano, untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano, untreated.als_vs_ctrl.nuc_vs_cyt.dep.volcano, ncol = 5, labels = c("a","c","e","g","i")),
  plot_grid(untreated.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, untreated.als_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, ncol = 5, labels = c("b","d","f","h","j")), 
  plot_grid(untreated.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap, rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin, ncol = 2, rel_widths = c(0.8,2), labels = c("k","l")),
  nrow = 3, rel_heights = c(1,1.2,1))
# vcp_tardbp_protein_redistribution.merged
ggsave(vcp_tardbp_protein_redistribution.merged, filename = here(proj_path, "mass-spec/figures/vcp_tardbp_protein_mislocalisation.merged.png"), height = 11, width = 14)
ggsave(vcp_tardbp_protein_redistribution.merged, filename = here(proj_path, "mass-spec/figures/vcp_tardbp_protein_mislocalisation.merged.pdf"), height = 11, width = 14)

```



# Table S8 Protein mislocalisation

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.dep = readRDS(file = here(proj_path, "scripts/untreated.als_vs_ctrl.nuc_vs_cyt.dep.rds"))
uniprot2gene = readRDS(here(nemo_path, "genomes/annotation/uniprot2gene.rds"))
LLPSDB <- read_excel(here(nemo_path, "genomes/protein-aggregation/LLPSDB/LLPS_Proteins/protein.xls")) %>% clean_names() %>%  rename(gene_symbols = gene_name) %>% left_join(select(uniprot2gene, uniprot_id = uniprot_accession, gene_name)) %>% mutate(protein_symbol = toupper(protein_name_abbreviation)) %>% distinct() 

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join = mutate(select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean),
                                                             tardbp.ml240 = case_when(gene_name %in% pull(filter(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, (stat > 0 & !gene_name %in% pull(filter(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat > 0),gene_name)) | (stat < 0 & !gene_name %in% pull(filter(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat < 0),gene_name))),gene_name) ~ "rescued", TRUE ~ "")) %>%
  
  left_join(mutate(select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean),
            vcp_r155c.ml240 = case_when(gene_name %in% pull(filter(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, (stat > 0 & !gene_name %in% pull(filter(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat > 0),gene_name)) | (stat < 0 & !gene_name %in% pull(filter(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat < 0),gene_name))),gene_name) ~ "rescued", TRUE ~ ""))) %>%
  
  left_join(mutate(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean),
            vcp_r191q.ml240 = case_when(gene_name %in% pull(filter(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, (stat > 0 & !gene_name %in% pull(filter(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat > 0),gene_name)) | (stat < 0 & !gene_name %in% pull(filter(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat < 0),gene_name))),gene_name) ~ "rescued", TRUE ~ ""))) %>%
  
  left_join(mutate(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_ki.stat = stat, vcp_ki.padj = padj, vcp_ki.lfc = log2FoldChange, vcp_ki.baseMean = baseMean),
            vcp_ki.ml240 = case_when(gene_name %in% pull(filter(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, (stat > 0 & !gene_name %in% pull(filter(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat > 0),gene_name)) | (stat < 0 & !gene_name %in% pull(filter(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05, stat < 0),gene_name))),gene_name) ~ "rescued", TRUE ~ ""))) %>%
  
  left_join(select(untreated.als_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, combined.stat = stat, combined.padj = padj, combined.lfc = log2FoldChange, combined.baseMean = baseMean)) %>% 
  
  filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_r191q.stat) == FALSE & is.na(vcp_ki.stat) == FALSE) %>%
  # filter(tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_r191q.padj < 0.05 | vcp_ki.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_r191q.stat) + abs(vcp_ki.stat))) %>%
  mutate(overlap = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05) ~ "overlap", TRUE ~ ""),
         overlap_direction = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat > 0 & vcp_r155c.stat > 0 & vcp_r191q.stat > 0 & vcp_ki.stat > 0) ~ "Nuclear", 
                             (tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat < 0 & vcp_r155c.stat < 0 & vcp_r191q.stat < 0 & vcp_ki.stat < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes & gene_name %in% RNA_BINDING_PROTEINS & gene_name %in% LLPSDB$gene_name ~ "ALS gene, RBP, LLPS",
                              gene_name %in% ALS_genes & gene_name %in% RNA_BINDING_PROTEINS ~ "ALS gene, RBP", 
                              gene_name %in% RNA_BINDING_PROTEINS & gene_name %in% LLPSDB$gene_name ~ "RBP, LLPS",
                              gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP", gene_name %in% LLPSDB$gene_name ~ "LLPS",
                              gene_name %in% nuclear_export.markers ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ ""))
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join
write_csv(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join, here(proj_path,"expression/untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S8 protein misloc")
sheet_properties("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% select(gene_name, protein.overlapping = overlapping, category) %>% left_join(select(untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, gene_name, transcript_id, transcript.overlapping = overlapping)) %>% drop_na() %>% filter(gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf)

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(gene_name %in% c("XIST",'RPS4Y1', 'EIFAY', 'DDX3Y', 'KDM5D'))

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter((tardbp.padj < 0.05 | vcp_mut.padj < 0.05 | vcp_iso.padj < 0.05), gene_name %in% c(ALS_genes))
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter((tardbp.padj < 0.05 | vcp_mut.padj < 0.05 | vcp_iso.padj < 0.05) gene_name %in% c(RNA_BINDING_PROTEINS))

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(overlap_direction == "Nuclear")

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(overlap != "", gene_name %in% ALS_genes)
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(overlap_direction != "")

untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(grepl("TARDBP", gene_name))
untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(grepl("FUS", gene_name))

```


# Fig S12 Compare mutations

## RNAseq & mass spec correlation

```{r}
# TARDBP
untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.labels = select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, rna.stat = stat, rna.padj = padj) %>% left_join(select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, protein.stat = stat, protein.padj = padj)) %>% filter(rna.padj < 0.05, protein.padj < 0.05)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.list = list("Protein mislocalisation" = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, "mRNA redistribution" = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr = plot_de_correlation(model_list1 = untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.list, model_list2 = untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.list, mutation2 = "Protein mislocalisation", mutation1 = "mRNA redistribution", gene_labels = pull(untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", title = "TARDBP mutant") #
untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr

# VCP R155C
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.labels = select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, rna.stat = stat, rna.padj = padj) %>% left_join(select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, protein.stat = stat, protein.padj = padj)) %>% filter(rna.padj < 0.05, protein.padj < 0.05)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.list = list("Protein mislocalisation" = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, "mRNA redistribution" = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr = plot_de_correlation(model_list1 = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.list, model_list2 = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.list, mutation2 = "Protein mislocalisation", mutation1 = "mRNA redistribution", gene_labels = pull(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", title = "VCP R155C mutant") #
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr

# VCP r191q
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.labels = select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, rna.stat = stat, rna.padj = padj) %>% left_join(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, protein.stat = stat, protein.padj = padj)) %>% filter(rna.padj < 0.05, protein.padj < 0.05)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.list = list("Protein mislocalisation" = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, "mRNA redistribution" = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx)
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr = plot_de_correlation(model_list1 = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.list, model_list2 = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.list, mutation2 = "Protein mislocalisation", mutation1 = "mRNA redistribution", gene_labels = pull(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", title = "VCP R191Q mutant") #
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr

# VCP knock-in
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.labels = select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, rna.stat = stat, rna.padj = padj) %>% left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, protein.stat = stat, protein.padj = padj)) %>% filter(rna.padj < 0.05, protein.padj < 0.05)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.list = list("Protein mislocalisation" = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, "mRNA redistribution" = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr = plot_de_correlation(model_list1 = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.list, model_list2 = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.list, mutation2 = "Protein mislocalisation", mutation1 = "mRNA redistribution", gene_labels = pull(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.labels,gene_name), plot_corr_line = TRUE, join_by = "gene_name", title = "VCP R191Q knock-in") #
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr

```


## Scatter correlations

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.dep_list = list("TARDBP G298S" = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, "VCP R155C" = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, "VCP R191Q" = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, "VCP R191Q knock-in" = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res)

tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)
tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)
tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)
vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)
vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)
vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels = select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj) %>% left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp.stat = stat, vcp.padj = padj)) %>% filter(vcp.padj < 0.05 & tardbp.padj < 0.05)

untreated.tardbp_vcp_r155c.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R155C", join_by = "gene_name", gene_labels = pull(tardbp_vcp_r155c.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
untreated.tardbp_vcp_r191q.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q", join_by = "gene_name", gene_labels = pull(tardbp_vcp_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
untreated.tardbp_vcp_iso.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "TARDBP G298S", mutation2 = "VCP R191Q knock-in", join_by = "gene_name",  gene_labels = pull(tardbp_vcp_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
untreated.vcp_r191q_vcp_r155c.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q", join_by = "gene_name", gene_labels = pull(vcp_r155c_r191q.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
untreated.vcp_iso_vcp_r155c.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "VCP R155C", mutation2 = "VCP R191Q knock-in", join_by = "gene_name", gene_labels = pull(vcp_r155c_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
untreated.vcp_r191q_vcp_iso.dep_list.scatter = plot_de_correlation(model_list1 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, model_list2 = untreated.als_vs_ctrl.nuc_vs_cyt.dep_list, mutation1 = "VCP R191Q", mutation2 = "VCP R191Q knock-in", join_by = "gene_name", gene_labels = pull(vcp_r191q_iso.untreated.nuc_vs_cyt.als_vs_ctrl.labels,gene_name)) + xlim(-8,8) & ylim(-8,8)
  
```

## Upset plot

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind = bind_rows(mutate(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "TARDBP G298S"), mutate(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "VCP R155C"), mutate(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "VCP R191Q"), mutate(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "VCP R191Q knock-in"))

untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.nuclear.upset = upset_plot(filter(untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind, stat > 0), overlap_by = "mutation", overlap_level = "gene_name", split_direction = FALSE, ymax = 190, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Nuclear mislocalised", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.nuclear.upset

untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.cytoplasm.upset = upset_plot(filter(untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind, stat < 0), overlap_by = "mutation", overlap_level = "gene_name", split_direction = FALSE, ymax = 510, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Cytoplasm mislocalised", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.cytoplasm.upset

```

## Merge

```{r}
untreated.als_vs_ctrl.nuc_vs_cyt.dep_list_stat_scatter_upset = plot_grid(
  plot_grid(untreated.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr,  untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, ncol = 4, labels = "auto"),
  plot_grid(untreated.tardbp_vcp_r155c.dep_list.scatter,untreated.tardbp_vcp_r191q.dep_list.scatter,untreated.tardbp_vcp_iso.dep_list.scatter, untreated.vcp_r191q_vcp_r155c.dep_list.scatter,  untreated.vcp_iso_vcp_r155c.dep_list.scatter, untreated.vcp_r191q_vcp_iso.dep_list.scatter, ncol = 3, nrow = 2, labels = letters[5:10]),
  plot_grid(untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.nuclear.upset, untreated.als_vs_ctrl.nuc_vs_cyt.dep.bind.cytoplasm.upset, ncol = 2, labels = letters[11:12]),
nrow = 3, rel_heights = c(1,2,1))
# untreated.als_vs_ctrl.nuc_vs_cyt.dep_list_stat_scatter_upset
ggsave(untreated.als_vs_ctrl.nuc_vs_cyt.dep_list_stat_scatter_upset, filename = here(proj_path, "mass-spec/figures/untreated.als_vs_ctrl.nuc_vs_cyt.dep_list_stat_scatter_upset.png"), height = 11, width = 8.25)

```


# Table S9 RNAct protein summarised

```{r}
RNAct.tardbp.nuclear.protein_sum = RNAct_database.tardbp.nuclear_protein_transcript %>% group_by(uniprot_accession) %>% summarise(tardbp.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-tardbp.nuclear)
RNAct.tardbp.cytoplasm.protein_sum = RNAct_database.tardbp.cytoplasm_protein_transcript %>% group_by(uniprot_accession) %>% summarise(tardbp.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-tardbp.cytoplasm)
RNAct.vcp_r155c.nuclear.protein_sum = RNAct_database.vcp_r155c.nuclear_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_r155c.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r155c.nuclear)
RNAct.vcp_r155c.cytoplasm.protein_sum = RNAct_database.vcp_r155c.cytoplasm_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_r155c.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r155c.cytoplasm)
RNAct.vcp_r191q.nuclear.protein_sum = RNAct_database.vcp_r191q.nuclear_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_r191q.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r191q.nuclear)
RNAct.vcp_r191q.cytoplasm.protein_sum = RNAct_database.vcp_r191q.cytoplasm_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_r191q.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r191q.cytoplasm)
RNAct.vcp_ki.nuclear.protein_sum = RNAct_database.vcp_iso.nuclear_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_ki.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_ki.nuclear)
RNAct.vcp_ki.cytoplasm.protein_sum = RNAct_database.vcp_iso.cytoplasm_protein_transcript %>% group_by(uniprot_accession) %>% summarise(vcp_ki.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_ki.cytoplasm)

uniprot2gene = readRDS(here(nemo_path, "genomes/annotation/uniprot2gene.rds"))
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum = RNAct.tardbp.nuclear.protein_sum %>% full_join(RNAct.tardbp.cytoplasm.protein_sum) %>% full_join(RNAct.vcp_r155c.nuclear.protein_sum) %>% full_join(RNAct.vcp_r155c.cytoplasm.protein_sum) %>% full_join(RNAct.vcp_r191q.nuclear.protein_sum) %>% full_join(RNAct.vcp_r191q.cytoplasm.protein_sum) %>% full_join(RNAct.vcp_ki.nuclear.protein_sum) %>% full_join(RNAct.vcp_ki.cytoplasm.protein_sum) %>% 
  rowwise(uniprot_accession) %>% mutate(total = sum(c_across(where(is.numeric)), na.rm = TRUE)) %>% arrange(-total) %>% left_join(uniprot2gene) %>% 
  select(gene_name, uniprot_accession, everything(), -gene_id) %>% distinct() 
  # rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_ki")), ~ str_replace_all(., "vcp_ki", "vcp_ki"))
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum
write_csv(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum, here(proj_path,"expression/RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S9 RNAct protein scores")

RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum %>% filter(gene_name %in% RNA_BINDING_PROTEINS)
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum %>% filter(!is.na(tardbp.nuclear) | !is.na(vcp_r155c.nuclear) | !is.na(vcp_r191q.nuclear) | !is.na(vcp_ki.nuclear)) # ANP32E, CLTA, PPP1R14B

```


# Table S10 RNAct transcript summarised

```{r}
RNAct.tardbp.nuclear.transcript_sum = RNAct_database.tardbp.nuclear_protein_transcript %>% group_by(transcript_id) %>% summarise(tardbp.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-tardbp.nuclear)
RNAct.tardbp.cytoplasm.transcript_sum = RNAct_database.tardbp.cytoplasm_protein_transcript %>% group_by(transcript_id) %>% summarise(tardbp.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-tardbp.cytoplasm)
RNAct.vcp_r155c.nuclear.transcript_sum = RNAct_database.vcp_r155c.nuclear_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_r155c.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r155c.nuclear)
RNAct.vcp_r155c.cytoplasm.transcript_sum = RNAct_database.vcp_r155c.cytoplasm_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_r155c.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r155c.cytoplasm)
RNAct.vcp_r191q.nuclear.transcript_sum = RNAct_database.vcp_r191q.nuclear_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_r191q.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r191q.nuclear)
RNAct.vcp_r191q.cytoplasm.transcript_sum = RNAct_database.vcp_r191q.cytoplasm_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_r191q.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_r191q.cytoplasm)
RNAct.vcp_ki.nuclear.transcript_sum = RNAct_database.vcp_iso.nuclear_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_ki.nuclear = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_ki.nuclear)
RNAct.vcp_ki.cytoplasm.transcript_sum = RNAct_database.vcp_iso.cytoplasm_protein_transcript %>% group_by(transcript_id) %>% summarise(vcp_ki.cytoplasm = sum(catrapid_score_max_normalised)) %>% arrange(-vcp_ki.cytoplasm)

uniprot2gene = readRDS(here(nemo_path, "genomes/annotation/uniprot2gene.rds"))
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum = RNAct.tardbp.nuclear.transcript_sum %>% full_join(RNAct.tardbp.cytoplasm.transcript_sum) %>% full_join(RNAct.vcp_r155c.nuclear.transcript_sum) %>% full_join(RNAct.vcp_r155c.cytoplasm.transcript_sum) %>% full_join(RNAct.vcp_r191q.nuclear.transcript_sum) %>% full_join(RNAct.vcp_r191q.cytoplasm.transcript_sum) %>% full_join(RNAct.vcp_ki.nuclear.transcript_sum) %>% full_join(RNAct.vcp_ki.cytoplasm.transcript_sum) %>% 
  rowwise(transcript_id) %>% mutate(total = sum(c_across(where(is.numeric)), na.rm = TRUE)) %>% arrange(-total) %>% left_join(tx2gene) %>% left_join(gene2ens) %>%
  select(gene_name, transcript_id, everything(), -gene_id) %>% distinct()
RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum
write_csv(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum, here(proj_path,"expression/RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S10 RNAct Tx scores")

RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum %>% filter(gene_name %in% RNA_BINDING_PROTEINS)

RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum %>% select(gene_name, total.tx = total) %>% ungroup %>% left_join(select(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum, gene_name, total.pr = total)) %>% drop_na()
select(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.protein_sum, gene_name, total.pr = total) %>% ungroup() %>% left_join(select(RNAct.tardbp_vcp_r155c_ki.nuclear_cytoplasm.transcript_sum, gene_name, total.tx = total)) %>% drop_na() %>% arrange(-(total.pr + total.tx))

```


# Fig S13 Heatmap nuclear export NPC proteins

```{r}
load(file = here(proj_path, "scripts/dep_mass_spec.untreated.als_vs_ctrl.nuc_cyt.RData"))

untreated.tardbp_vs_ctrl.nuc_cyt.dep$res %>% filter(padj < 0.05)#311
untreated.vcp_r155c_vs_ctrl.nuc_cyt.dep$res %>% filter(padj < 0.05)#509
untreated.vcp_r191q_vs_ctrl.nuc_cyt.dep$res %>% filter(padj < 0.05)#37
untreated.vcp_iso_vs_ctrl.nuc_cyt.dep$res %>% filter(padj < 0.05)#58

### nuclear export markers
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))
nuclear_export.markers.df = tibble("gene_name" = nuclear_export.markers) %>% filter(gene_name %in% nuclear_export.markers) %>% 
  mutate(group = case_when(gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE_NUCLEAR_BASKET ~ "nuclear.basket", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION ~ "npc.organisation", 
                           gene_name %in% go.pathways.msigdb$GO_NUCLEAR_EXPORT_SIGNAL_RECEPTOR_ACTIVITY ~ "export.signal", gene_name %in% go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY ~ "carrier.activity", gene_name %in% go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX ~ "TREX", gene_name %in% go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS ~ "rna.export", 
    gene_name %in% go.pathways.msigdb$GO_NUCLEAR_EXPORT ~ "nuclear.export", gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "nuclear.pore"),
         group = factor(group, levels = c("nuclear.basket", "npc.organisation", "nuclear.pore", "export.signal","carrier.activity","TREX","rna.export", "nuclear.export"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)

# L2FC
nuclear_export.l2fc.marker <- select(untreated.tardbp_vs_ctrl.nuc_cyt.dep$res, gene_name, TARDBP = log2FoldChange) %>% 
  full_join(select(untreated.vcp_r155c_vs_ctrl.nuc_cyt.dep$res, gene_name, VCP.R155C = log2FoldChange)) %>% 
  full_join(select(untreated.vcp_r191q_vs_ctrl.nuc_cyt.dep$res, gene_name, VCP.R191Q = log2FoldChange)) %>% 
  full_join(select(untreated.vcp_iso_vs_ctrl.nuc_cyt.dep$res, gene_name, VCP.KI = log2FoldChange)) %>%
  filter(gene_name %in% nuclear_export.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  arrange(factor(gene_name, levels = nuclear_export.markers.df$gene_name))
nuclear_export.markers.df.filt = nuclear_export.markers.df %>% filter(gene_name %in% nuclear_export.l2fc.marker$gene_name)
nuclear_export.l2fc.marker.mat <- make_matrix(select(nuclear_export.l2fc.marker,-gene_name), pull(nuclear_export.l2fc.marker,gene_name))

### Cluster samples ###
nuclear_export.l2fc.marker.centered_mat = t(scale(t(nuclear_export.l2fc.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
nuclear_export.l2fc.marker.scaled_mat = t(scale(t(nuclear_export.l2fc.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered

nuclear_export.markers.mass_spec.cluster.heatmap <- Heatmap(nuclear_export.l2fc.marker.mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          column_names_rot = 90, column_names_gp = gpar(fontsize = 10), 
          row_order = nuclear_export.markers.df.filt$gene_name, 
          row_split = nuclear_export.markers.df.filt$group,
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",8)), labels = c("NPC\nBasket", "NPC\nOrganisation", "Nuclear\nPore","Export\nSignal","Carrier\nActivity","TREX", "RNA\nExport","Nuclear\nExport"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_title = NULL,
          row_names_gp = gpar(fontsize = 5), #show_row_names = FALSE, 
          column_title = NULL, cluster_columns = FALSE, cluster_row_slices = FALSE, cluster_rows = FALSE)
nuclear_export.markers.mass_spec.cluster.heatmap
ggsave(as.ggplot(nuclear_export.markers.mass_spec.cluster.heatmap), filename = here(proj_path,"expression/deseq2/nuclear_export.markers.mass_spec.cluster.heatmap.png"), width = 9, height = 13)

nuclear_export.l2fc.marker %>% left_join(nuclear_export.markers.df) %>% filter(TARDBP < 0, VCP.R155C < 0, VCP.R191Q < 0, VCP.KI < 0) %>% mutate(sum = TARDBP + VCP.R155C + VCP.R191Q + VCP.KI) %>% arrange(sum)

nuclear_export.marker.bind <- mutate(untreated.tardbp_vs_ctrl.nuc_cyt.dep$res, mutation = "tardbp") %>% 
  bind_rows(mutate(untreated.vcp_r155c_vs_ctrl.nuc_cyt.dep$res, mutation = "vcp.r155c")) %>% 
  bind_rows(mutate(untreated.vcp_r191q_vs_ctrl.nuc_cyt.dep$res, mutation = "vcp.r191q")) %>% 
  bind_rows(mutate(untreated.vcp_iso_vs_ctrl.nuc_cyt.dep$res, mutation = "cxp.ki")) %>%
  filter(gene_name %in% nuclear_export.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE)
nuclear_export.marker.bind %>% arrange(padj)


```


# Fig S14 QC ML240 samples

## Viability

```{r}
# import with magick 
ml240_viability.png = magick::image_read(here(proj_path,"ml240-assays/ml240_viability.png"))
ml240_viability.gg = ggdraw() + draw_image(ml240_viability.png)
# ml240_viability.gg

PlateResults.20220905 = read_csv(here(proj_path, "ml240-assays/20220905_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 1)
PlateResults.20220913 = read_csv(here(proj_path, "ml240-assays/20220913_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 2)
PlateResults.20220920 = read_csv(here(proj_path, "ml240-assays/20220920_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 3)
PlateResults.bind = bind_rows(PlateResults.20220905, PlateResults.20220913, PlateResults.20220920) %>% mutate(mutation = case_when(grepl("CB1",cell_line)~"VCP\nR155C", grepl("GLI",cell_line)~"VCP\nR191Q", TRUE ~"CTRL"), mutant_treatment = paste(mutant, treatment, sep = " +\n"), mutation_treatment = paste(mutation, treatment, sep = "\n"))
PlateResults.bind %>% glimpse

ml240_assay_viability.violin = violin_plot(data = drop_na(filter(PlateResults.bind, treatment %in% c("DMSO","ML240")), percent_live_cells_sytox), color_by = "mutation_treatment", continuous = "percent_live_cells_sytox", ylims = c(0,100), cols = brewer.pal(6, "Paired"), plot_stats = TRUE, stats_test = "wilcox_test", stat_label = "p.adj.signif", one_sample = FALSE, stat_ypos = 100, ylabel = "% survival", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE) #c("dodgerblue1", "dodgerblue3", "firebrick1", "firebrick3", "forestgreen", "forestgreen"), 
ml240_assay_viability.violin
# [1] "Using two sample wilcox_test"
# # A tibble: 6 × 10
#   .y.   group1          group2             n1    n2 statistic     p p.adj p.adj.signif y.position
#   <chr> <chr>           <chr>           <int> <int>     <dbl> <dbl> <dbl> <chr>             <dbl>
# 1 lfc   "CTRL +\nDMSO"  "CTRL +\nML240"    25    26      383  0.279     1 ns                  100
# 2 lfc   "CTRL +\nDMSO"  "VCP +\nDMSO"      25    28      418  0.231     1 ns                  100
# 3 lfc   "CTRL +\nDMSO"  "VCP +\nML240"     25    27      384  0.399     1 ns                  100
# 4 lfc   "CTRL +\nML240" "VCP +\nDMSO"      26    28      361  0.965     1 ns                  100
# 5 lfc   "CTRL +\nML240" "VCP +\nML240"     26    27      332. 0.735     1 ns                  100
# 6 lfc   "VCP +\nDMSO"   "VCP +\nML240"     28    27      366  0.846     1 ns                  100
# # A tibble: 6 × 7
#   .y.   group1          group2          effsize    n1    n2 magnitude 
# * <chr> <chr>           <chr>             <dbl> <int> <int> <ord>     
# 1 lfc   "CTRL +\nDMSO"  "CTRL +\nML240"  0.253     25    26 small     
# 2 lfc   "CTRL +\nDMSO"  "VCP +\nDMSO"    0.419     25    28 small     
# 3 lfc   "CTRL +\nDMSO"  "VCP +\nML240"   0.212     25    27 small     
# 4 lfc   "CTRL +\nML240" "VCP +\nDMSO"    0.187     26    28 negligible
# 5 lfc   "CTRL +\nML240" "VCP +\nML240"  -0.0457    26    27 negligible
# 6 lfc   "VCP +\nDMSO"   "VCP +\nML240"  -0.233     28    27 small   
Sytox_LysoTracker_PlateResults %>% count(mutant, treatment)
Sytox_LysoTracker_PlateResults %>% filter(treatment %in% c("DMSO","ML240")) %>% cohens_d(percent_live_cells ~ group, var.equal = TRUE) 
Sytox_LysoTracker_PlateResults %>% drop_na(percent_live_cells) %>% my_summarise(percent_live_cells, mutant, treatment)
Sytox_LysoTracker_PlateResults %>% drop_na(percent_live_cells) %>% violin_plot(data, color_by = "mutant_treatment", continuous = "percent_live_cells", ylims = c(75,100), cols = brewer.pal(n = 10, name = "Paired"), plot_stats = FALSE, stats_test = "wilcox", one_sample = FALSE, stat_ypos = 100, ylabel = "% survival", dpi = 300, arrow_labels = FALSE)

```

## Western blot

```{r}
ml240.western_blot.png =  magick::image_read(here(proj_path,"expression/jacob_western_fractionation_ml240.png"))
ml240.western_blot.gg = ggdraw() + draw_image(ml240.western_blot.png)
ml240.western_blot.gg


```

##  PCA RNAseq & Mass spec

```{r}
untreated_ml240.nuc_vs_cyt.pcaData <- plotPCA(untreated_ml240.nuc_vs_cyt$vsd, intgroup=c("condition", "sample", "cellline", "treatment", "condition", "fraction", "mutation"), returnData=TRUE) %>% 
  mutate(cellline = gsub("cb1d", "R155C.1", cellline), cellline = gsub("cb1e", "R155C.2", cellline), cellline = gsub("glia", "R191Q.1", cellline), cellline = gsub("glib", "R191Q.2", cellline))

untreated_ml240.nuc_vs_cyt.pca <- ggplot(untreated_ml240.nuc_vs_cyt.pcaData, aes(PC1, PC2, color=fraction, shape = treatment)) + 
  geom_point(size=1) + 
  geom_text_repel(aes(label=toupper(cellline)), size = 2.4) +
 scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
  theme_oz() + theme(legend.text=element_text(size=8), legend.title=element_text(size=7), legend.position = "top", legend.spacing = unit(0.01, 'cm')) + 
  guides(color=guide_legend(title="Fraction", keywidth=0.1, keyheight=0.15, default.unit="inch")) +
    xlab(paste0("PC1: ",round(100 * attr(untreated_ml240.nuc_vs_cyt.pcaData, "percentVar"))[1],"% variance")) +  ylab(paste0("PC2: ",round(100 * attr(untreated_ml240.nuc_vs_cyt.pcaData, "percentVar"))[2],"% variance")) + ggtitle("RNA sequencing") +
  ggforce::geom_mark_ellipse(show.legend = FALSE) +
  guides(color = guide_legend(reverse = TRUE, nrow = 1))
untreated_ml240.nuc_vs_cyt.pca


## Mass spec PCA
colData(untreated_ml240.nuc_vs_cyt.dep$dep) = colData(untreated_ml240.nuc_vs_cyt.dep$dep) %>% 
  mutate(cellline = gsub("cb1d", "R155C.1", cellline), cellline = gsub("cb1e", "R155C.2", cellline), cellline = gsub("glia", "R191Q.1", cellline), cellline = gsub("glib", "R191Q.2", cellline))

untreated_ml240.nuc_vs_cyt.dep.pca = plot_pca_oz(untreated_ml240.nuc_vs_cyt.dep$dep, point_size = 1, indicate = c("condition", "treatment")) + 
   scale_shape_manual(values=c(1, 16)) + scale_color_manual( values = c("dodgerblue2", "firebrick2", "forestgreen") ) +  
    theme_oz() + theme(panel.grid = element_blank(), axis.text=element_text(size=8), axis.title=element_text(size=10), legend.position = "none", legend.spacing = unit(0.01, 'cm')) + 
    # guides(color=guide_legend(title="", keywidth=0.1, keyheight=0.15, default.unit="inch", reverse = TRUE, nrow = 1)) + 
  ggtitle("Mass Spectrometry") +
    ggforce::geom_mark_ellipse(aes(color = condition, shape = treatment), show.legend = FALSE) + 
  geom_text_repel(aes(label=toupper(colData(untreated_ml240.nuc_vs_cyt.dep$dep)$cellline)), size = 2.4, max.overlaps = 5) #+ coord_equal(ratio = 0.4)
untreated_ml240.nuc_vs_cyt.dep.pca

```


## QualiMap read origin

Export data from multiQC QualiMap origin of reads & RSeQC Read Distribution

```{r}
rseqc_read_distribution = read_tsv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/nfcore/multiqc/star_salmon/multiqc_data/multiqc_rseqc_read_distribution.txt")) %>%
  clean_names() %>% select(sample, CDS = cds_exons_tag_pct, `5UTR` = x5_utr_exons_tag_pct, `3UTR` = x3_utr_exons_tag_pct, Intron = introns_tag_pct) %>% mutate(Intergenic = 100 - (CDS + `5UTR` + `3UTR` + Intron)) %>%
  pivot_longer(!sample, names_to = "region", values_to = "percent") %>%
  mutate(fraction = str_split_fixed(sample, "_", 3)[,3], cellline = toupper(str_split_fixed(sample, "_", 3)[,1]), treatment = str_split_fixed(sample, "_", 3)[,2],
         fraction = gsub("nuclear","nucleus",fraction),
         cellline = gsub("CB1D","R155C.1",cellline), cellline = gsub("CB1E","R155C.2",cellline), cellline = gsub("GLIA","R191Q.1",cellline), cellline = gsub("GLIB","R191Q.2",cellline),
         cellline = factor(cellline, levels = rev(c("CTRL1", "CTRL3", "CTRL4", "CTRL6", "R191Q.1", "R191Q.2", "R155C.1", "R155C.2", "NCRMC2", "NCRME6", "TDPN1", "TDPN2"))),
         region = factor(region, levels = c("5UTR", "CDS", "3UTR", "Intron", "Intergenic"))) %>% drop_na(cellline) %>%
  filter(treatment != "untreated", fraction != "whole")

rseqc_read_distribution.summarised.barplot <- ggbarplot(rseqc_read_distribution, x="cellline" , y="percent", fill="region", label = FALSE, position = position_fill(), xlab = "")  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 1)) +
  facet_grid(fraction ~ .,scales="free_y") +
  coord_flip() +
  ylab(expression(Read~type~proportions))
rseqc_read_distribution.summarised.barplot

rseqc_read_distribution %>% filter(fraction == "cytoplasm", region %in% c("5UTR","3UTR","CDS")) %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum)) #88.0
rseqc_read_distribution %>% filter(fraction == "nucleus", region %in% c("5UTR","3UTR","CDS")) %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum)) #67.5
rseqc_read_distribution %>% filter(fraction == "cytoplasm", region == "Intron") %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum))#9.4
rseqc_read_distribution %>% filter(fraction == "nucleus", region == "Intron") %>% group_by(sample) %>% summarise(sum = sum(percent))  %>% ungroup() %>% summarise(mean(sum))#28.1
  
```


## Cyto & Nuc Heatmap & Density

Keep nuc R155C.1 in heatmap but exclude in all downstream analyses

```{r}
# Import gene list of Nuc & Cyto markers
price_2020_nuclear_cyto_genes <- read_excel(here(nemo_path, "genomes/nuc-cyt-localisation/price_2020_nuclear_cytoplasmic_transcriptomes_supp_tables.xlsx"), sheet = "Table S2 DEGs by Fraction") %>% filter(Symbol %in% gene2ens$gene_name) # 4284
price_2020_nuclear_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Nuclear: Both","Nuclear: Prenatal Only", "Nuclear: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_cyto_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Cytoplasmic: Both","Cytoplasmic: Prenatal Only", "Cytoplasmic: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_nuc_cyto_markers <- c(price_2020_cyto_genes, price_2020_nuclear_genes)
nuc_cyt_markers.df = data.frame("gene_name" = price_2020_nuc_cyto_markers) %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>% mutate(group = case_when(gene_name %in% price_2020_nuclear_genes ~ "nuclear", gene_name %in% price_2020_cyto_genes ~ "cytoplasm")) %>% distinct()
nuc_cyt_markers.annot <- tibble(gene_name = nuc_cyt_markers.df$gene_name, group = factor(nuc_cyt_markers.df$group)) %>% arrange(group) # Annotation 

ml240.nuc_vs_cyt$vsd.counts = as_tibble(assay(ml240.nuc_vs_cyt$vsd), rownames = "gene_id") %>% left_join(gene2ens)
mn_price.nuc.cyt.marker <- ml240.nuc_vs_cyt$vsd.counts %>% distinct(gene_name, .keep_all = TRUE) %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>% #, rowSums(across(where(is.numeric))) > 162) %>% 
  select(-gene_id, -contains("whole")) 
mn_price.nuc.cyt.marker.topVar <- mn_price.nuc.cyt.marker %>%
  dplyr::mutate_if(is.numeric, funs(ifelse(. <0, 0, .))) %>%   # dplyr::mutate_if(is.numeric, funs(log2(.))) %>%
  pivot_longer(!gene_name, names_to = "sample", values_to = "count") %>% 
  mutate(fraction = case_when(grepl("cyt", sample) ~ "cytoplasm", grepl("nuc", sample) ~ "nuclear", TRUE ~ "whole")) %>%
  group_by(gene_name, fraction) %>% summarise(mean = mean(count, na.rm = TRUE)) %>% ungroup %>% pivot_wider(names_from = fraction, values_from = mean) %>% mutate(delta.mean = nuclear - cytoplasm) %>% 
  slice_max(abs(delta.mean), n = 3000) %>% # retain the most variable nuc vs cyt genes between samples
  mutate(fraction_database = case_when(gene_name %in% price_2020_cyto_genes ~ "cytoplasm", gene_name %in% price_2020_nuclear_genes ~ "nuclear")) %>% arrange(fraction_database)

colnames(mn_price.nuc.cyt.marker) = gsub("ml240_", "", colnames(mn_price.nuc.cyt.marker))
mn_price.nuc.cyt.marker.filt <- mn_price.nuc.cyt.marker %>% filter(gene_name %in% mn_price.nuc.cyt.marker.topVar$gene_name) %>% filter_all(all_vars(!is.infinite(.))) %>% drop_na()
mn_price.nuc.cyt.marker.mat <- make_matrix(select(mn_price.nuc.cyt.marker.filt,-gene_name), pull(mn_price.nuc.cyt.marker.filt,gene_name))
# colnames(mn_price.nuc.cyt.marker.mat) <- mn.metadata$name.long[ mn.metadata$sample %in% colnames(mn_price.nuc.cyt.marker.mat) ]
# colnames(mn_price.nuc.cyt.marker.mat) <- gsub("mn_|d35_|_R", "", colnames(mn_price.nuc.cyt.marker.mat) )

# price.nuc.cyt.marker.topVar[price.nuc.cyt.marker.topVar > 14] <- 14 # crop outliers to improve colour distribution
mn_price.nuc.cyt.marker.scaled_mat = t(scale(t(mn_price.nuc.cyt.marker.mat))) # row scale
mn_price.nuc.cyt.marker.scaled_mat <- mn_price.nuc.cyt.marker.scaled_mat[!is.infinite(rowSums(mn_price.nuc.cyt.marker.scaled_mat)),]

cytoplasm_markers = mn_price.nuc.cyt.marker.filt$gene_name[mn_price.nuc.cyt.marker.filt$gene_name %in% price_2020_cyto_genes]
nuclear_markers = mn_price.nuc.cyt.marker.filt$gene_name[mn_price.nuc.cyt.marker.filt$gene_name %in% price_2020_nuclear_genes]
mn_price.nuc.cyt.marker.column_order = ml240.metadata %>% filter(!grepl("whole",sample)) %>% arrange(fraction)

colnames(mn_price.nuc.cyt.marker.scaled_mat) = colnames(mn_price.nuc.cyt.marker.scaled_mat) %>% gsub("cb1d", "R155C.1", .) %>% gsub("cb1e", "R155C.2", .) %>% gsub("glia", "R191Q.1", .) %>% gsub("glib", "R191Q.2", .)

mn_price.nuc.cyt.marker.heatmap <- as.ggplot(
  Heatmap(mn_price.nuc.cyt.marker.scaled_mat,
          row_km = 2,
          # row_order = mn_price.nuc.cyt.marker.topVar$gene_name,
          show_row_names = FALSE,
          # row_split =  factor(mn_price.nuc.cyt.marker.topVar$fraction_database, levels = c("nuclear", "cytoplasm")), #c(rep("cyt", length(cytoplasm_markers)), rep("nuc", length(nuclear_markers))),
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Nuclear\nmarkers", "Cytoplasm\nmarkers"), labels_gp = gpar(fontsize = 8))), row_title = NULL,
          # column_order = gsub("ml240_", "", mn_price.nuc.cyt.marker.column_order$sample), 
          column_km = 2,
          # column_split = factor(mn_price.nuc.cyt.marker.column_order$fraction, levels = c("cytoplasm", "nuclear", "whole")),
          bottom_annotation = HeatmapAnnotation(markers = anno_block(gp = gpar(fill = c("white", "white")), labels = c("Nuclear samples", "Cytoplasm samples"), labels_gp = gpar(fontsize = 8))), column_title = NULL,
          column_names_gp = gpar(fontsize = 6),
          cluster_row_slices = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE,cluster_rows = FALSE,
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical")))
mn_price.nuc.cyt.marker.heatmap


## Cyto & Nuclear Density
price_2020_nuclear_cyto_genes <- read_excel(here(nemo_path, "genomes/nuc-cyt-localisation/price_2020_nuclear_cytoplasmic_transcriptomes_supp_tables.xlsx"), sheet = "Table S2 DEGs by Fraction") %>% filter(Symbol %in% gene2ens$gene_name) # 4284
price_2020_nuclear_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Nuclear: Both","Nuclear: Prenatal Only", "Nuclear: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_cyto_genes <- price_2020_nuclear_cyto_genes %>% filter(Group %in% c("Cytoplasmic: Both","Cytoplasmic: Prenatal Only", "Cytoplasmic: Adult Only"), Symbol != "NA") %>% dplyr::pull(Symbol)
price_2020_nuc_cyto_markers <- c(price_2020_cyto_genes, price_2020_nuclear_genes, "LMNA", "NEAT1", "GCGR", "HC-4")
nuc_vs_cyt.volcano.labels <-  c("MALAT1", "MLXIPL", "XIST", "SOX9", "RBFOX3", "LMNA", "HC-4",  # nuclear
                                      "GLUL",  "STAU1") # cytoplasmic

# Density
# redraw figure 1e (volcano plot of nuclear and cytoplasmic markers) to instead be a superimposed density plot of the two distributions of marker intensities - the volcano plot is quite misleading as it squashes the key points near log-foldchange=0 to a very small vertical region preventing us from seeing whether the two distributions are properly balanced around the origin (which represents the point where the _normalised_ cytoplasmic = _normalised_ nuclear).
ml240.nuc_vs_cyt.res.price_markers <- ml240.nuc_vs_cyt$res %>% filter(gene_name %in% price_2020_nuc_cyto_markers) %>%
  mutate(Markers = case_when(gene_name %in% price_2020_cyto_genes ~ "Cytoplasmic", TRUE ~ "Nuclear"))
ml240.nuc_vs_cyt.price_markers.density = density_plot(ml240.nuc_vs_cyt.res.price_markers, color_by = Markers, cols = c("dodgerblue2", "firebrick2"), ymax = 0.8, xlims = c(-2.5,4), numerator = "Nucleus", denomenator = "Cytoplasm", dpi = 300)
# ml240.nuc_vs_cyt.price_markers.density

```



## Cell type marker heatmap

```{r}
astrocyte.markers = c("ALDH1L1", "AQP4", "GFAP","EDNRB",  "SLC4A4", "DIO2")#"SOX9", ,"MLC1""FGFR3",  
endo.markers = c("CLDN5",  "ESAM", "ICAM2", "APOLD1", "NOSTRIN", "ADGRL4", "PECAM1")
myeloid.markers = c("AIF1", "C1QA", "CSF1R", "ITGAM", "PTPRC", "CCL3","IL1A", "TREM2") # "RUNX1", "CD83",  "CCL2", "TLR2",  "MERTK", "CX3CR1","P2RY12", "TNF",  "TMEM119"
oligo.markers = c("MBP", "MOG",  "GPR37", "SOX10", "OPALIN", "MOBP", "PLP1")
nonspecific.neuronal = c("SYT1","MAP2","STMN2","NEFM","SNAP25","NCAM1","GAP43","L1CAM","TUBB3","NEFL","ELAVL3")#"NEFH",
motor.neuron.progenitor <- c("OLIG2","NKX6-1","NKX6-2","PAX6","GLI2","FOXA2","NEUROG2","OLIG1","NKX2-2") #"CREBBP","EP300",
postmitotic.motor.neuron <- c("CHAT","MNX1","ISL1","LHX3","LMO4","ALCAM","SMAD1","ZEB2","SLC18A3") #"ETV1","ETV4","FOXP1","ALDH1A2", "NOS1",,"RUNX1","NOTCH1","ARHGAP36","SLC10A4","POU3F1","FOXP2","ISL2"
# oculomotor.markers = c("PHOX2A", "PHOX2B", "TBX20", "RGS4","SEMA6D","PLXNA4","EYA2","EYA1","FGF10","EN1")
# interneuron.markers <- c(V0.domain,V1,V2a.domain,V2b.domain,V3.domain,"SHOX2")#dl.domains LDB1 "LHX1","POU4F1",,"LBX1""LHX2","LHX9","DRGX","LMX1B","VSX2","EVX2","LMO4"
mn.markers = c(postmitotic.motor.neuron,nonspecific.neuronal)

### Cell type markers
celltype.markers = c(astrocyte.markers, endo.markers, myeloid.markers, oligo.markers, mn.markers)
celltype.markers[!celltype.markers %in% gtf$gene_name]
celltype.markers.df = tibble("gene_name" = celltype.markers) %>% filter(gene_name %in% celltype.markers) %>% 
  mutate(group = case_when(gene_name %in% astrocyte.markers ~ "astrocyte.markers", gene_name %in% endo.markers ~ "endo.markers", 
                           gene_name %in% myeloid.markers ~ "myeloid.markers", gene_name %in% oligo.markers ~ "oligo.markers", 
                           gene_name %in% mn.markers ~ "mn.markers"), 
         group = factor(group, levels = c("astrocyte.markers", "endo.markers", "myeloid.markers", "oligo.markers","mn.markers"))) %>% 
  arrange(group) %>% distinct(gene_name, .keep_all = TRUE)
dataset.samples.df = untreated.metadata %>% filter(fraction != "whole") %>% select(sample, fraction)

ml240.nuc_vs_cyt$vsd.counts = as_tibble(assay(ml240.nuc_vs_cyt$vsd), rownames = "gene_id") %>% left_join(gene2ens)
min_vst = min(ml240.nuc_vs_cyt$vsd.counts[,2])
celltype.markers.marker <- ml240.nuc_vs_cyt$vsd.counts %>% 
  filter(gene_name %in% celltype.markers.df$gene_name) %>% distinct(gene_name, .keep_all = TRUE) %>% 
  dplyr::mutate_if(is.numeric, funs((. - min_vst)))  %>% 
  arrange(factor(gene_name, levels = celltype.markers.df$gene_name)) %>% select(-gene_id) %>% 
  rename_with(~ gsub("mn_d25_", "", .x, fixed = TRUE)) 
celltype.markers.df.filt = celltype.markers.df %>% filter(gene_name %in% celltype.markers.marker$gene_name)
celltype.markers.marker.mat <- make_matrix(select(celltype.markers.marker,-gene_name), pull(celltype.markers.marker,gene_name))

### Cluster samples ###
# celltype.markers.marker.centered_mat = t(scale(t(celltype.markers.marker.mat), center=TRUE,scale=FALSE)) # centered, not scaled
# celltype.markers.marker.scaled_mat = t(scale(t(celltype.markers.marker.mat), center=FALSE,scale=TRUE)) # scaled, not centered
# rownames(celltype.markers.marker.centered_mat)[!rownames(celltype.markers.marker.centered_mat) %in% dataset.samples.df$sample]
# dataset.samples.df$sample[!dataset.samples.df$sample %in% rownames(celltype.markers.marker.centered_mat)]
celltype.markers.rnaseq.cluster.heatmap <- as.ggplot(
  Heatmap(celltype.markers.marker.mat, 
          border = TRUE, heatmap_legend_param = list(title = "", direction = "vertical"), #, annotation_legend_side = "bottom"), show_heatmap_legend = FALSE,
          row_order = celltype.markers.df.filt$gene_name,
          row_names_gp = gpar(fontsize = 5), 
          row_split = celltype.markers.df.filt$group,
          right_annotation = rowAnnotation(markers = anno_block(gp = gpar(fill = rep("white",5)), labels = c("Astrocyte", "Endothelia", "Myeloid", "Oligodend","Motor\nneuron"), labels_gp = gpar(fontsize = 7), labels_rot = 0)), row_title = NULL, 
          show_column_names = FALSE, #row_names_gp = gpar(fontsize = 6),
          # row_order = dataset.samples.df$dataset_sample,
          column_split = factor(dataset.samples.df$fraction, levels = c("nucleus", "cytoplasm")),
          bottom_annotation = columnAnnotation(datasets = anno_block(gp = gpar(fill = rep("white",2)),  labels = c("Nuclear samples", "Cytoplasm samples"), labels_gp = gpar(fontsize = 7), labels_rot = 0)),
          column_title = NULL, cluster_rows = FALSE, cluster_column_slices = FALSE, cluster_columns = FALSE))
celltype.markers.rnaseq.cluster.heatmap

```


## Merge

```{r}
ml240.pca_characterisation.merged <- plot_grid(
  plot_grid(NULL, ml240_viability.gg, ml240_assay_viability.violin, ncol = 3, rel_widths = c(0.07,1.5,1), labels = c("a","","b")),
  ml240.western_blot.gg,
  plot_grid(NULL, celltype.markers.rnaseq.cluster.heatmap, rseqc_read_distribution.summarised.barplot, ncol = 3, rel_widths = c(0.05,1,1), labels = c("d", "", "e")),
  plot_grid(NULL, mn_price.nuc.cyt.marker.heatmap, ml240.nuc_vs_cyt.price_markers.density, ncol = 3,  rel_widths = c(0.05,1.1,1), labels = c("f", "", "g")),
  untreated_ml240.nuc_vs_cyt.pca,
  untreated_ml240.nuc_vs_cyt.dep.pca,
  nrow = 6, labels = c("", "c","", "", "h", "","i"), rel_heights = c(1.2,0.7,1.6,1.3,1,0.9))
# ml240.pca_characterisation.merged
ggsave(ml240.pca_characterisation.merged, filename = here(proj_path, "expression/deseq2/ml240.pca_characterisation.merged.png"), height = 15, width = 9)

```


# Fig 6 ML240

## RNAseq Upset & heatmap

```{r}
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) #423
untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) #381
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) #412
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) #358
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) #689
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) #602
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) #906
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) #569

# Upset plot
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "Untreated"), mutate(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.tx.upset = upset_plot(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.tx, overlap_by = "mutation", overlap_level = "transcript_id", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 280, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "TARDBP mutant", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.tx.upset
 
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "Untreated"), mutate(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.tx.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.tx, overlap_by = "mutation", overlap_level = "transcript_id", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 280, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCP R155C", label_colours = c("#1F78B4", "#A6CEE3"))+  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(color = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.tx.upset

untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "Untreated"), mutate(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.tx.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.tx, overlap_by = "mutation", overlap_level = "transcript_id", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 460, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCP R191Q", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(colour = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.tx.upset

untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "Untreated"), mutate(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.tx.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.tx, overlap_by = "mutation", overlap_level = "transcript_id", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 570, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCPki R191Q", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(colour = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.tx.upset

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.res.tx.upset = plot_grid(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.tx.upset, untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.tx.upset, untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.tx.upset, untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.tx.upset, ncol = 4, rel_widths = c(1,0.55,0.55,0.55))
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.res.tx.upset

# heatmap of correlation coefficients
untreated_ml240.nuc_vs_cyt.als_vs_ctrl_stat = select(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, `TARDBP^G298S\nML240` = stat) %>% 
    left_join(select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, `TARDBP^G298S\nUntreated` = stat)) %>%
  left_join(select(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, `VCP^R155C\nML240` = stat)) %>%
  left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, `VCP^R155C\nUntreated` = stat)) %>%
  left_join(select(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, `VCP^R191Q\nML240` = stat)) %>%
  left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, `VCP^R191Q\nUntreated` = stat)) %>%
  left_join(select(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, `VCPki^R191Q\nML240` = stat)) %>%
    left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, `VCPki^R191Q\nUntreated` = stat)) %>%
  drop_na() %>%
  column_to_rownames("transcript_id")
cormat <- round(cor(untreated_ml240.nuc_vs_cyt.als_vs_ctrl_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
melted_cormat %<>% filter(Var1=="TARDBP^G298S\nML240" & Var2 =="TARDBP^G298S\nUntreated" | Var1=="VCP^R155C\nML240" & Var2 =="VCP^R155C\nUntreated" | Var1=="VCP^R191Q\nML240" & Var2 =="VCP^R191Q\nUntreated" | Var1=="VCPki^R191Q\nML240" & Var2 =="VCPki^R191Q\nUntreated")
rnaseq.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) + labs(x = "Untreated", y = "ML240") +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
rnaseq.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap

# How many of the untreated redistributed transcripts are not redistributed with ML240?
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts %>% str #153
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts) # 3
150/153 # 98.0
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj <= 0.05, stat <= 0) %>% pull(transcript_id) 
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts %>% str #228
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj <= 0.05, stat <= 0, transcript_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts) # 5
223/228 # #97.8%

untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0) %>% pull(transcript_id) 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts %>% str # 181
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts) # 7
174/181 # 96.1
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0) %>% pull(transcript_id) 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts %>% str # 177
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts) # 5
172/177 # 97.2

untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0) %>% pull(transcript_id) 
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts %>% str # 181
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts) # 11
170/181 #93.9
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0) %>% pull(transcript_id) 
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts %>% str # 421
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts) # 25
396/421 #94.1

untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts %>% str # 190
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, stat > 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts) # 8
182/190 # 95.8
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0) %>% pull(transcript_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts %>% str # 379
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, stat < 0, transcript_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts) # 23
356/379 # 93.9%

```


## Proteomics Upset & heatmap

```{r}
ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #111
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #577
(577-111)/577 # 80.8
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #116
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #169
(169-116)/169 #31.4
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #400
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #867
(867-400)/867 #53.9
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #433
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) #639
(639-433)/639 #32.2

# Upset plot
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.dep = bind_rows(mutate(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "Untreated"), mutate(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.dep.upset = upset_plot(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.dep, overlap_by = "mutation", overlap_level = "gene_name", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 480, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "TARDBP mutant", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.dep.upset
 
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.dep = bind_rows(mutate(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "Untreated"), mutate(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.dep.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.dep, overlap_by = "mutation", overlap_level = "gene_name", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 160, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCP R155C", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(color = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.dep.upset

untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.dep = bind_rows(mutate(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "Untreated"), mutate(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.dep.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.dep, overlap_by = "mutation", overlap_level = "gene_name", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 780, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCP R191Q", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(color = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.dep.upset

untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.dep = bind_rows(mutate(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "Untreated"), mutate(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.dep.upset = upset_plot(untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.dep, overlap_by = "mutation", overlap_level = "gene_name", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 420, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "VCPki R191Q", label_colours = c("#1F78B4", "#A6CEE3")) + 
  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3") + theme(axis.text.y = element_text(color = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,-30, "mm")) 
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.dep.upset

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.dep.upset = plot_grid(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.dep.upset, untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.dep.upset, untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.dep.upset, untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.dep.upset, ncol = 4, rel_widths = c(1,0.55,0.55,0.55))
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.dep.upset


# heatmap of correlation coefficients
untreated_ml240.nuc_vs_cyt.als_vs_ctrl_dep = select(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `TARDBP^G298S\nML240` = stat) %>% 
    left_join(select(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `TARDBP^G298S\nUntreated` = stat)) %>%
  left_join(select(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R155C\nML240` = stat)) %>%
  left_join(select(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R155C\nUntreated` = stat)) %>%
  left_join(select(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R191Q\nML240` = stat)) %>%
  left_join(select(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R191Q\nUntreated` = stat)) %>%
  left_join(select(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCPki^R191Q\nML240` = stat)) %>%
    left_join(select(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCPki^R191Q\nUntreated` = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_name")
cormat <- round(cor(untreated_ml240.nuc_vs_cyt.als_vs_ctrl_dep),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
melted_cormat %<>% filter(Var1=="TARDBP^G298S\nML240" & Var2 =="TARDBP^G298S\nUntreated" | Var1=="VCP^R155C\nML240" & Var2 =="VCP^R155C\nUntreated" | Var1=="VCP^R191Q\nML240" & Var2 =="VCP^R191Q\nUntreated" | Var1=="VCPki^R191Q\nML240" & Var2 =="VCPki^R191Q\nUntreated")
mass.spec.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) + labs(x = "Untreated", y = "ML240") +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
mass.spec.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap



# How many of the untreated redistributed transcripts are not redistributed with ML240?
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str #169
ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0, gene_name %in% untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 2
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)
untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 408
ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0, gene_name %in% untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 3

untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name) 
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 55
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0, gene_name %in% untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 1
54/55 #98.2
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name) 
untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 114
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0, gene_name %in% untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 6
108/114 # 94.7

untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name) 
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 113
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0, gene_name %in% untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 8
105/113
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name) 
untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 754
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0, gene_name %in% untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 48
(754-48)/754 # 93.6

untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0) %>% pull(gene_name)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 203
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat > 0, gene_name %in% untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 23
180/203 #88.7
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0) %>% pull(gene_name)
untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins %>% str # 436
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, stat < 0, gene_name %in% untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.mislocalised_proteins) # 82
(436-82)/436 #81.2%

```


## Functional assays ML240

import fluorescence images

```{r}
# import with magick 
ml240_lysotracker.png = magick::image_read(here(proj_path,"ml240-assays/ml240_lysotracker.png"))
ml240_lysotracker.gg = ggdraw() + draw_image(ml240_lysotracker.png)
# ml240_lysotracker.gg

ml240_cellrox.png = magick::image_read(here(proj_path,"ml240-assays/ml240_cellrox.png"))
ml240_cellrox.gg = ggdraw() + draw_image(ml240_cellrox.png)
# ml240_cellrox.gg

ml240_yH2AX.png = magick::image_read(here(proj_path,"ml240-assays/yH2AX/yH2AX_immuno.png"))
ml240_yH2AX.gg = ggdraw() + draw_image(ml240_yH2AX.png)
# ml240_yH2AX.gg

```


Lysosome soma:neurite ratio

```{r}
lysosome_sytox.PlateResults.20220905 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20220905_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 1) %>% select(-temperature)
lysosome_sytox.PlateResults.20220913 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20220913_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 2) %>% select(-temperature)
lysosome_sytox.PlateResults.20220920 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20220920_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 3) %>% select(-temperature)
lysosome_sytox.PlateResults.20221013 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20221013_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 4) %>% select(-temperature) # vcp ki & TARDBP
lysosome_sytox.PlateResults.20221017 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20221017_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 5) %>% select(-temperature) # vcp ki & TARDBP
lysosome_sytox.PlateResults.20221020 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20221020_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 6) %>% select(-temperature) # vcp ki & TARDBP
lysosome_sytox.PlateResults.20221025 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20221025_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 7) %>% select(-temperature) # vcp ki & TARDBP
lysosome_sytox.PlateResults.20221027 = read_csv(here(proj_path, "ml240-assays/lysosome_sytox/20221027_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 8) %>% select(-temperature) # vcp ki & TARDBP
lysosome_sytox.PlateResults.20221027 %>% count(cell_line)
lysosome_sytox.PlateResults.bind = bind_rows(lysosome_sytox.PlateResults.20220905, lysosome_sytox.PlateResults.20220913, lysosome_sytox.PlateResults.20220920, lysosome_sytox.PlateResults.20221013, lysosome_sytox.PlateResults.20221017, lysosome_sytox.PlateResults.20221020, lysosome_sytox.PlateResults.20221025, lysosome_sytox.PlateResults.20221027) %>% mutate(mutation = case_when(grepl("CB1",cell_line)~"VCP R155C", grepl("GLI",cell_line)~"VCP R191Q", grepl("NCRM",cell_line)~"VCP ki", grepl("TDP",cell_line)~"TARDBP", TRUE ~"CTRL"), mutant_treatment = paste(mutant, treatment, sep = "\n"), mutation_treatment = paste(mutation, treatment, sep = "\n"), neurite_soma_ratio = 1/lyso_soma_neurites_ratio)
lysosome_sytox.PlateResults.bind %>% count(mutant)

lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors = lysosome_sytox.PlateResults.bind %>% filter(live_nuclei_number_of_objects>=800) %>% mutate(treatment = factor(treatment, levels = c("DMSO","ML240","CB-5083","DBeQ","NMS-873")), mutation = factor(mutation, levels = c("CTRL","TARDBP","VCP R155C","VCP R191Q","VCP ki")))
lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors %>% glimpse
lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors %>% drop_na(lyso_soma_neurites_ratio) %>% my_summarise(neurite_soma_ratio, mutation_treatment) %>% print(n=Inf)

vcp_inhibitors_lysotracker.violin.stat = lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors %>% group_by(mutation) %>% wilcox_test(neurite_soma_ratio ~ treatment) %>% add_significance() %>% add_xy_position(x = "treatment") %>% 
  mutate(p.signif = case_when(group1=="DMSO"&p<0.001~"****", group1=="DMSO"&p<0.001~"***", group1=="DMSO"&p<0.01~"**", group1=="DMSO"&p<0.05~"*", TRUE ~ "ns"), facet=mutation, .y. = "lfc", 
         y.position = case_when(mutation=="CTRL"&group2=="ML240"~1.7, mutation=="CTRL"&group2=="NMS-873"~2.0,  
                                mutation=="VCP R155C"&group2=="ML240"~1.7, mutation=="VCP R155C"&group2=="NMS-873"~2.0, 
                                mutation=="VCP R191Q"&group2=="CB-5083"~2.1)) %>% select(-contains("adj"))
vcp_inhibitors_lysotracker.violin.stat %>% filter(group1 == "DMSO")
#  mutation  .y.   group1 group2     n1    n2 statistic     p y.position groups        xmin  xmax p.signif facet    
#    <fct>     <chr> <chr>  <chr>   <int> <int>     <dbl> <dbl>      <dbl> <named list> <dbl> <dbl> <chr>    <fct>    
#  1 CTRL      lfc   DMSO   ML240      56    54      1180 0.048        1.7 <chr [2]>        1     2 *        CTRL     
#  2 CTRL      lfc   DMSO   CB-5083    56    56      1452 0.502       NA   <chr [2]>        1     3 ns       CTRL     
#  3 CTRL      lfc   DMSO   DBeQ       56    61      1652 0.762       NA   <chr [2]>        1     4 ns       CTRL     
#  4 CTRL      lfc   DMSO   NMS-873    56    57      1117 0.006        2   <chr [2]>        1     5 **       CTRL     
#  5 TARDBP    lfc   DMSO   ML240      16    20       139 0.519       NA   <chr [2]>        1     2 ns       TARDBP   
#  6 TARDBP    lfc   DMSO   CB-5083    16    17       109 0.345       NA   <chr [2]>        1     3 ns       TARDBP   
#  7 TARDBP    lfc   DMSO   DBeQ       16    15       123 0.922       NA   <chr [2]>        1     4 ns       TARDBP   
#  8 TARDBP    lfc   DMSO   NMS-873    16    18       109 0.237       NA   <chr [2]>        1     5 ns       TARDBP   
#  9 VCP R155C lfc   DMSO   ML240      14    11        32 0.013        1.7 <chr [2]>        1     2 *        VCP R155C
# 10 VCP R155C lfc   DMSO   CB-5083    14    12        68 0.432       NA   <chr [2]>        1     3 ns       VCP R155C
# 11 VCP R155C lfc   DMSO   DBeQ       14    12        51 0.095       NA   <chr [2]>        1     4 ns       VCP R155C
# 12 VCP R155C lfc   DMSO   NMS-873    14    12        40 0.023        2   <chr [2]>        1     5 *        VCP R155C
# 13 VCP R191Q lfc   DMSO   ML240      12    10        32 0.069       NA   <chr [2]>        1     2 ns       VCP R191Q
# 14 VCP R191Q lfc   DMSO   CB-5083    12     8        13 0.005        2.1 <chr [2]>        1     3 **       VCP R191Q
# 15 VCP R191Q lfc   DMSO   DBeQ       12    12        44 0.114       NA   <chr [2]>        1     4 ns       VCP R191Q
# 16 VCP R191Q lfc   DMSO   NMS-873    12    11        53 0.449       NA   <chr [2]>        1     5 ns       VCP R191Q
# 17 VCP ki    lfc   DMSO   ML240      11    12        42 0.151       NA   <chr [2]>        1     2 ns       VCP ki   
# 18 VCP ki    lfc   DMSO   CB-5083    11    13        53 0.303       NA   <chr [2]>        1     3 ns       VCP ki   
# 19 VCP ki    lfc   DMSO   DBeQ       11    13        49 0.207       NA   <chr [2]>        1     4 ns       VCP ki   
# 20 VCP ki    lfc   DMSO   NMS-873    11    11        39 0.171       NA   <chr [2]>        1     5 ns       VCP ki   

lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors.filt = drop_na(lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors,neurite_soma_ratio) %>% filter(neurite_soma_ratio<2.1)

vcp_inhibitors_lysotracker.violin = violin_plot(data = lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors.filt, color_by = "treatment", facet_by = "mutation", facet_nrow = 1, continuous = "neurite_soma_ratio",cols = rep(c("dodgerblue2","firebrick2","forestgreen","gold2","purple"),5), plot_stats = FALSE,  ylabel = "Lysosome\nNeurite:Soma ratio", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE)+
  stat_pvalue_manual(vcp_inhibitors_lysotracker.violin.stat, label = "p.signif", tip.length = .02, size = 4, hide.ns = TRUE)
vcp_inhibitors_lysotracker.violin

# stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = rep(c(1.8,1.9,1.9,2.05,rep(10,6)),5), tip_length = 0.003,
 # ylims = c(0.1,2.0), 

# # whole image intensity
# vcp_inhibitors_lysotracker.intensity.violin = violin_plot(data = drop_na(lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors,lysosome_in_whole_image_relative_spot_intensity_mean_per_well), color_by = "mutation_treatment", continuous = "lysosome_in_whole_image_relative_spot_intensity_mean_per_well", ylims = c(0.12,0.2), cols = rep(c("dodgerblue2","firebrick2","forestgreen","gold2","purple"),5), plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 0.2, ylabel = "Lysosome intensity", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE)
# # vcp_inhibitors_lysotracker.intensity.violin
# 
# # number of lysosomes per cell
# vcp_inhibitors_lysotracker.number.violin = violin_plot(data = filter(drop_na(lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors,number_of_lyso_per_cell)), color_by = "mutation_treatment", continuous = "number_of_lyso_per_cell", ylims = c(0,26), cols = rep(c("dodgerblue2","firebrick2","forestgreen","gold2","purple"),5), plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = 24, tip_length = 0.005, ylabel = "Lysosome number per cell", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE)
# # vcp_inhibitors_lysotracker.number.violin
# 

# 
# # ML240 & DMSO only:
# lysosome_sytox.PlateResults.bind.live_800.ml240 = lysosome_sytox.PlateResults.bind %>% filter(live_nuclei_number_of_objects>=800, treatment %in% c ("DMSO","ML240"))
# lysosome_sytox.PlateResults.bind.live_800.ml240 %>% glimpse
# lysosome_sytox.PlateResults.bind.live_800.ml240 %>% drop_na(lyso_soma_neurites_ratio) %>% my_summarise(lyso_soma_neurites_ratio, mutation_treatment)
# lysosome_sytox.PlateResults.bind %>% count(treatment)
# ml240_assay_lysotracker.violin = violin_plot(data = drop_na(lysosome_sytox.PlateResults.bind.live_800.ml240,neurite_soma_ratio), color_by = "mutation_treatment", continuous = "neurite_soma_ratio", ylims = c(0.1,2.0), cols = brewer.pal(10, "Paired"), plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos =c(1.9,rep(10,16),2.1,rep(10,13),2.1,rep(10,8),2.1,rep(10,3),1.9), tip_length = 0.003, ylabel = "Lysosome Neurite:Soma ratio", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE)
# # ml240_assay_lysotracker.violin

```


Oxidative stress

```{r}
CellRox_Fix_PlateResults.20220728 = read_csv(here(proj_path, "ml240-assays/cellrox/20220728_CellRox_Fix_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 1)
CellRox_Fix_PlateResults.20220913 = read_csv(here(proj_path, "ml240-assays/cellrox/20220913_CellRox_Fix_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 2)
# CellRox_Fix_PlateResults.20220920 = read_csv(here(proj_path, "ml240-assays/cellrox/20220920_CellRox_Fix_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 3)
CellRox_Fix_PlateResults.20221013 = read_csv(here(proj_path, "ml240-assays/cellrox/20221013_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 4) # vcp ki & TARDBP
CellRox_Fix_PlateResults.20221017 = read_csv(here(proj_path, "ml240-assays/cellrox/20221017_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 5) # vcp ki & TARDBP
CellRox_Fix_PlateResults.20221020 = read_csv(here(proj_path, "ml240-assays/cellrox/20221020_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 6) # vcp ki & TARDBP
CellRox_Fix_PlateResults.20221025 = read_csv(here(proj_path, "ml240-assays/cellrox/20221025_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 7) # vcp ki & TARDBP
CellRox_Fix_PlateResults.20221027 = read_csv(here(proj_path, "ml240-assays/cellrox/20221027_PlateResults.csv")) %>% clean_names() %>% mutate(batch = 8) # vcp ki & TARDBP
CellRox_Fix_PlateResults = bind_rows(CellRox_Fix_PlateResults.20220728, CellRox_Fix_PlateResults.20220913, CellRox_Fix_PlateResults.20221013, CellRox_Fix_PlateResults.20221017, CellRox_Fix_PlateResults.20221020, CellRox_Fix_PlateResults.20221025, CellRox_Fix_PlateResults.20221027) %>% mutate(mutation = case_when(grepl("CB1",cell_line)~"VCP R155C", grepl("GLI",cell_line)~"VCP R191Q", grepl("NCRM",cell_line)~"VCP ki", grepl("TDP",cell_line)~"TARDBP", TRUE ~"CTRL"), mutant_treatment = paste(mutant, treatment, sep = "\n"), mutation_treatment = paste(mutation, treatment, sep = "\n"))
CellRox_Fix_PlateResults.vcp_inhibitors.800 = CellRox_Fix_PlateResults %>% filter(live_nuclei_number_of_objects >= 800)%>% mutate(treatment = factor(treatment, levels = c("DMSO","ML240","CB-5083","DBeQ","NMS-873")), mutation = factor(mutation, levels = c("CTRL","TARDBP","VCP R155C","VCP R191Q","VCP ki")))
CellRox_Fix_PlateResults.vcp_inhibitors.800 %>% glimpse
CellRox_Fix_PlateResults.vcp_inhibitors.800 %>% drop_na(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well) %>% my_summarise(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well, mutation_treatment)

vcp_inhibitors_cellrox.violin.stat = CellRox_Fix_PlateResults.vcp_inhibitors.800 %>% group_by(mutation) %>% wilcox_test(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well ~ treatment) %>% add_significance() %>% add_xy_position(x = "treatment") %>% 
  mutate(p.signif = case_when(group1=="DMSO"&p<0.001~"****", group1=="DMSO"&p<0.001~"***", group1=="DMSO"&p<0.01~"**", group1=="DMSO"&p<0.05~"*", TRUE ~ "ns"), facet=mutation, .y. = "lfc", 
         y.position = case_when(mutation=="CTRL"&group2=="ML240"~950000, mutation=="CTRL"&group2=="CB-5083"~1000000, mutation=="CTRL"&group2=="NMS-873"~1050000,  
                                mutation=="TARDBP"&group2=="ML240"~710000, mutation=="TARDBP"&group2=="CB-5083"~750000,mutation=="TARDBP"&group2=="DBeQ"~790000, 
                                mutation=="VCP R191Q"&group2=="DBeQ"~1000000,mutation=="VCP R191Q"&group2=="NMS-873"~1040000, 
                                mutation=="VCP ki"&group2=="ML240"~710000,mutation=="VCP ki"&group2=="DBeQ"~750000,mutation=="VCP ki"&group2=="NMS-873"~790000)) %>% select(-contains("adj"))
vcp_inhibitors_cellrox.violin.stat
      
CellRox_Fix_PlateResults.vcp_inhibitors.800.filt = CellRox_Fix_PlateResults.vcp_inhibitors.800 %>% filter( (mutation=="CTRL"&live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well < 1000000) | (mutation=="TARDBP"&live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well < 800000) | (mutation=="VCP R155C") | (mutation=="VCP R191Q"&live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well < 1000000) | (mutation=="VCP ki"&live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well < 700000) )

vcp_inhibitors_cellrox.violin = violin_plot(data = CellRox_Fix_PlateResults.vcp_inhibitors.800.filt, color_by = "treatment", facet_by = "mutation", facet_nrow = 1,  continuous = "live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well",cols = rep(c("dodgerblue2","firebrick2","forestgreen","gold2","purple"),5), plot_stats = FALSE, ylabel = "Reactive Oxygen Species\nNuclei intensity", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE) + scale_y_continuous(labels = scales::comma) + theme(axis.text.y = element_text(angle = 70, vjust = 0.5)) +
  stat_pvalue_manual(vcp_inhibitors_cellrox.violin.stat, label = "p.signif", tip.length = .03, size = 4, hide.ns = TRUE)
vcp_inhibitors_cellrox.violin
 
# ylims = c(302146,2250000), 

# TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = rep(c(1800000,1850000,1900000,2000000,rep(0,6)),5)


# # ML240 & DMSO only
# CellRox_Fix_PlateResults.ML240.800 = CellRox_Fix_PlateResults %>% filter(treatment %in% c("DMSO","ML240"), live_nuclei_number_of_objects >= 800)
# CellRox_Fix_PlateResults.ML240.800 %>% glimpse
# CellRox_Fix_PlateResults.ML240.800 %>% drop_na(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well) %>% my_summarise(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well, mutation_treatment)
# ml240_assay_cellrox.violin = violin_plot(data = CellRox_Fix_PlateResults.ML240.800, color_by = "mutation_treatment", continuous = "live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well", ylims = c(302146,2250000), cols = brewer.pal(10, "Paired"), plot_stats = TRUE, stats_test = "wilcox_test", adjust_p = FALSE, stat_label = "p.signif", stat_ypos = c(1600000,rep(0,4),2240000,rep(0,11),1000000,rep(0,12),1000000,rep(0,8),2210000,rep(0,4),1900000), ylabel = "Reactive Oxygen Species Nuclei intensity", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE) + scale_y_continuous(labels = scales::comma) + theme(axis.text.y = element_text(angle = 45, vjust = 0.5))
# ml240_assay_cellrox.violin

```


yH2AX

```{r}
pH2AX_norm_6reps.csv = read_csv(here(proj_path, "ml240-assays/yH2AX/pH2AX_6reps.csv")) %>% clean_names()
pH2AX_norm_6reps.csv %>% glimpse

pH2AX_norm_6reps = pH2AX_norm_6reps.csv %>% mutate(mutation = case_when(grepl("CB1",cell_line)~"VCP R155C", grepl("GLI",cell_line)~"VCP R191Q", grepl("NCRM",cell_line)~"VCP ki", grepl("TDP",cell_line)~"TARDBP", TRUE ~"CTRL"), mutant_treatment = paste(mutant, treatment, sep = "\n"), mutation_treatment = paste(mutation, treatment, sep = "\n"), treatment = factor(treatment, levels = c("DMSO","ML240","CB-5083","DBeQ","NMS-873")), mutation = factor(mutation, levels = c("CTRL","TARDBP","VCP R155C","VCP R191Q","VCP ki")))

# filter out rows with less than 100 nuclei
pH2AX_norm_6reps.100 = pH2AX_norm_6reps %>% filter(live_nuclei_number_of_objects >= 100) 
pH2AX_norm_6reps.100 %>% glimpse
pH2AX_norm_6reps.100 %>% drop_na(neuronal_population_2_number_of_spots_mean_per_well) %>% my_summarise(neuronal_population_2_number_of_spots_mean_per_well, mutation_treatment)

vcp_inhibitors_yh2ax.stat = pH2AX_norm_6reps.100 %>% group_by(mutation) %>% wilcox_test(neuronal_population_2_number_of_spots_mean_per_well ~ treatment) %>% add_significance() %>% add_xy_position(x = "treatment") %>% 
  mutate(p.signif = case_when(group1=="DMSO"&p<0.001~"****", group1=="DMSO"&p<0.001~"***", group1=="DMSO"&p<0.01~"**", group1=="DMSO"&p<0.05~"*", TRUE ~ "ns"), facet=mutation, .y. = "lfc", 
         y.position = case_when(mutation=="CTRL"&group2=="ML240"~11, mutation=="CTRL"&group2=="CB-5083"~11, mutation=="CTRL"&group2=="NMS-873"~11,  
                                mutation=="TARDBP"&group2=="ML240"~12, mutation=="TARDBP"&group2=="CB-5083"~12,mutation=="TARDBP"&group2=="DBeQ"~12, 
                                mutation=="VCP R155C"&group2=="ML240"~12, mutation=="VCP R155C"&group2=="DBeQ"~12,mutation=="VCP R155C"&group2=="NMS-873"~12, 
                                mutation=="VCP R191Q"&group2=="ML240"~12, mutation=="VCP R191Q"&group2=="DBeQ"~12,mutation=="VCP R191Q"&group2=="NMS-873"~12, 
                                mutation=="VCP ki"&group2=="ML240"~12,mutation=="VCP ki"&group2=="DBeQ"~12,mutation=="VCP ki"&group2=="NMS-873"~12)) %>% select(-contains("adj"))
vcp_inhibitors_yh2ax.stat
#  mutation  .y.   group1 group2    n1    n2 statistic      p p.signif y.position groups        xmin  xmax facet    
#   <fct>     <chr> <chr>  <chr>  <int> <int>     <dbl>  <dbl> <chr>         <dbl> <named list> <dbl> <dbl> <fct>    
# 1 CTRL      lfc   DMSO   ML240     44    45      1150 0.192  ns               11 <chr [2]>        1     2 CTRL     
# 2 TARDBP    lfc   DMSO   ML240     14    13       100 0.685  ns               12 <chr [2]>        1     2 TARDBP   
# 3 VCP R155C lfc   DMSO   ML240     19    20       240 0.166  ns               12 <chr [2]>        1     2 VCP R155C
# 4 VCP R191Q lfc   DMSO   ML240     18    23       292 0.0253 *                12 <chr [2]>        1     2 VCP R191Q
# 5 VCP ki    lfc   DMSO   ML240     16    20       236 0.0149 *                12 <chr [2]>        1     2 VCP ki   

# pH2AX_norm_6reps.filt = pH2AX_norm_6reps %>% filter( (mutation=="CTRL"&neuronal_population_2_number_of_spots_mean_per_well < 9) | (mutation=="TARDBP"&neuronal_population_2_number_of_spots_mean_per_well < 9) | (mutation=="VCP R155C") | (mutation=="VCP R191Q"&neuronal_population_2_number_of_spots_mean_per_well < 9) | (mutation=="VCP ki"&neuronal_population_2_number_of_spots_mean_per_well < 11) )

vcp_inhibitors_yh2ax.violin = violin_plot(data = pH2AX_norm_6reps, color_by = "treatment", facet_by = "mutation", facet_nrow = 1,  continuous = "neuronal_population_2_number_of_spots_mean_per_well",cols = rep(c("dodgerblue2","firebrick2","forestgreen","gold2","purple"),5), plot_stats = FALSE, ylabel = "yH2AX foci per cell", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE) + scale_y_continuous(labels = scales::comma) + theme(axis.text.y = element_text(angle = 70, vjust = 0.5)) +
  stat_pvalue_manual(vcp_inhibitors_yh2ax.stat, label = "p.signif", tip.length = .03, size = 4, hide.ns = TRUE)
vcp_inhibitors_yh2ax.violin

```

## Merge

```{r}
ml240.transcript_redistribution.protein_mislocalisation.merged <- plot_grid(
  NULL,
  plot_grid(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.res.tx.upset, rnaseq.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap, ncol = 2, labels = c("a","b"), rel_widths = c(3,1.1)) + theme(panel.background = element_rect(fill = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,1, "mm"), plot.background = element_rect(fill = adjustcolor( "white", alpha.f = 0), colour = "black", size = 1)),
  NULL,
    plot_grid(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.dep.upset, mass.spec.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap, ncol = 2, labels = c("c","d"), rel_widths = c(3,1.1)) + theme(panel.background = element_rect(fill = adjustcolor( "white", alpha.f = 0)), plot.margin = margin(1,1,1,1, "mm"), plot.background = element_rect(fill = adjustcolor( "white", alpha.f = 0), colour = "black", size = 1)),
  rel_heights = c(0.13,1.6,0.13,1.6), nrow = 4, labels = c("Transcript Redistribution","","Protein Mislocalisation",""), label_size = 12, label_x = -0.05, label_y = 1.1)
ml240_functional_assays.merged = plot_grid(
  plot_grid(ml240_yH2AX.gg, vcp_inhibitors_yh2ax.violin, nrow = 2, labels = c("e", "f"), rel_heights = c(1.3,1)),
  plot_grid(NULL,ml240_lysotracker.gg,NULL,ml240_cellrox.gg, ncol = 4, rel_widths = c(0.03,1,0.03,1), labels = c("g","","h","")),
  plot_grid(vcp_inhibitors_lysotracker.violin, vcp_inhibitors_cellrox.violin, nrow = 2, labels = c("i","j")),
  nrow = 3, rel_heights = c(1.2,0.8,1))
ml240.transcript_protein.lysosome.ros.merged = plot_grid(ml240.transcript_redistribution.protein_mislocalisation.merged, NULL, ml240_functional_assays.merged,
  nrow = 3, rel_heights = c(0.9,0.02,2.0))
ggsave(ml240.transcript_protein.lysosome.ros.merged, filename = here(proj_path, "expression/deseq2/ml240.transcript_protein.lysosome.ros.merged.png"), height = 17, width = 13)
ggsave(ml240.transcript_protein.lysosome.ros.merged, filename = here(proj_path, "expression/deseq2/ml240.transcript_protein.lysosome.ros.merged.pdf"), height = 17, width = 13)


# ml240_functional_assays.merged = plot_grid(
#   NULL, ml240_lysotracker.gg, ml240_assay_lysotracker.violin, 
#   NULL, ml240_cellrox.gg, ml240_assay_cellrox.violin,
#   nrow = 2, ncol = 3, rel_widths = c(0.07,2,1), labels = c("a","","b","c","","d"))
# ml240_functional_assays.merged
# ggsave(ml240_functional_assays.merged, filename = here(proj_path, "ml240-assays/ml240_functional_assays.merged.png"), height = 6, width = 8.25)

```


# Table S12 ML240 Transcript redistribution

```{r}
untreated.tardbp.nuc = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj<0.05,stat>0) %>% pull(transcript_id)
untreated.tardbp.cyt = untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj<0.05,stat<0) %>% pull(transcript_id)
untreated.vcp_r155c.nuc = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj<0.05,stat>0) %>% pull(transcript_id)
untreated.vcp_r155c.cyt = untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj<0.05,stat<0) %>% pull(transcript_id)
untreated.vcp_r191q.nuc = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj<0.05,stat>0) %>% pull(transcript_id)
untreated.vcp_r191q.cyt = untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj<0.05,stat<0) %>% pull(transcript_id)
untreated.vcp_iso.nuc = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj<0.05,stat>0) %>% pull(transcript_id)
untreated.vcp_iso.cyt = untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj<0.05,stat<0) %>% pull(transcript_id)

ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join = mutate(select(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean), 
                                                         tardbp.remain = case_when(tardbp.padj < 0.05 & tardbp.stat > 0 & transcript_id %in% untreated.tardbp.nuc ~ "remains nuc", tardbp.padj<0.05 & tardbp.stat<0 & transcript_id %in% untreated.tardbp.cyt ~ "remains cyto", TRUE ~ "")) %>% 
  left_join(mutate(select(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean),
            vcp_r155c.remain = case_when(vcp_r155c.padj < 0.05 & vcp_r155c.stat > 0 & transcript_id %in% untreated.vcp_r155c.nuc ~ "remains nuc", vcp_r155c.padj<0.05 & vcp_r155c.stat<0 & transcript_id %in% untreated.vcp_r155c.cyt ~ "remains cyto", TRUE ~ ""))) %>%
  left_join(mutate(select(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean),
            vcp_r191q.remain = case_when(vcp_r191q.padj < 0.05 & vcp_r191q.stat > 0 & transcript_id %in% untreated.vcp_r191q.nuc ~ "remains nuc", vcp_r191q.padj<0.05 & vcp_r191q.stat<0 & transcript_id %in% untreated.vcp_r191q.cyt ~ "remains cyto", TRUE ~ ""))) %>%
  left_join(mutate(select(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, vcp_iso.stat = stat, vcp_iso.padj = padj, vcp_iso.lfc = log2FoldChange, vcp_iso.baseMean = baseMean),
            vcp_iso.remain = case_when(vcp_iso.padj < 0.05 & vcp_iso.stat > 0 & transcript_id %in% untreated.vcp_iso.nuc ~ "remains nuc", vcp_iso.padj<0.05 & vcp_iso.stat<0 & transcript_id %in% untreated.vcp_iso.cyt ~ "remains cyto", TRUE ~ ""))) %>% 
  filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_r191q.stat) == FALSE & is.na(vcp_iso.stat) == FALSE) %>%
  filter(tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_r191q.padj < 0.05 | vcp_iso.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_r191q.stat) + abs(vcp_iso.stat))) %>%
  mutate(overlapping = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_iso.padj < 0.05 & tardbp.stat > 0 & vcp_r155c.stat > 0 & vcp_r191q.stat > 0 & vcp_iso.stat > 0) ~ "Nuclear", 
                             (tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_iso.padj < 0.05 & tardbp.stat < 0 & vcp_r155c.stat < 0 & vcp_r191q.stat < 0 & vcp_iso.stat < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ "")) 
    # rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_iso")), ~ str_replace_all(., "vcp_iso", "vcp_ki"))
ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(tardbp.remain!="")
write_csv(ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, here(proj_path,"expression/ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S11 ML240 Tx redist")

```


# Fig S16 ML240 Transcript redistribution


## Compare nuc/cyt ML240 with untreated controls

Correlation

```{r}
rna.untreated_ml240.nuc_vs_cyt.list = list("Untreated Nuclear/Cytoplasm" = ctrl.untreated.nuc_vs_cyt$res.tx, "ML240 Nuclear/Cytoplasm" = ctrl.ml240.nuc_vs_cyt$res.tx)
rna.untreated_ml240.nuc_vs_cyt.list.scatter = plot_de_correlation(model_list1 = rna.untreated_ml240.nuc_vs_cyt.list, model_list2 = rna.untreated_ml240.nuc_vs_cyt.list, mutation1 = "Untreated Nuclear/Cytoplasm", mutation2 = "ML240 Nuclear/Cytoplasm", join_by = c("gene_name", "gene_id","transcript_id"), title = "RNAseq Controls") + xlim(-20,20) & ylim(-20,30)
# rna.untreated_ml240.nuc_vs_cyt.list.scatter

protein.untreated_ml240.nuc_vs_cyt.list = list("Untreated Nuclear/Cytoplasm" = untreated.ctrl_nuc_vs_cyt.dep$res, "ML240 Nuclear/Cytoplasm" = ml240.ctrl_nuc_vs_cyt.dep$res)
protein.untreated_ml240.nuc_vs_cyt.list.scatter = plot_de_correlation(model_list1 = protein.untreated_ml240.nuc_vs_cyt.list, model_list2 = protein.untreated_ml240.nuc_vs_cyt.list, mutation1 = "Untreated Nuclear/Cytoplasm", mutation2 = "ML240 Nuclear/Cytoplasm", join_by = "gene_name", title = "Mass Spec Controls") + xlim(-10,10) & ylim(-10,10)
# protein.untreated_ml240.nuc_vs_cyt.list.scatter

# Upset
# untreated.nuc_vs_cyt$res %>% filter(padj < 0.05) #20055
# ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) #751
# ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) #975
# 
# # Upset plot
# untreated_ml240.nuc_vs_cyt.res.tx = bind_rows(mutate(untreated.nuc_vs_cyt$res, mutation = "Untreated"), mutate(ml240.nuc_vs_cyt$res, mutation = "ML240"))
# untreated_ml240.nuc_vs_cyt.res.tx.upset = upset_plot(untreated_ml240.nuc_vs_cyt.res.tx, overlap_by = "mutation", overlap_level = "gene_id", split_direction = TRUE,split_up = "Nuclear", split_down = "Cytoplasm", ymax = 10700, order = "freq", order_group_list = c("Untreated Nuclear", "ML240 Nuclear", "Untreated Cytoplasm", "ML240 Cytoplasm"), title = "CTRLs", label_colours = c("#1F78B4", "#A6CEE3")) + 
#   theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
# untreated_ml240.nuc_vs_cyt.res.tx.upset

```

## Redistributed transcript numbers

```{r}
nrow(filter(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05)) #381
nrow(filter(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, padj < 0.05)) #423
nrow(filter(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05)) #358
nrow(filter(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, padj < 0.05)) #412
nrow(filter(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05)) #602
nrow(filter(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, padj < 0.05)) #689
nrow(filter(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05)) #569
nrow(filter(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, padj < 0.05)) #906
mutant_tx_ml240_rescue = tibble(
  mutation = c("TARDBP mutant", "TARDBP mutant", "VCP R155C mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCP R191Q mutant", "VCPki R191Q", "VCPki R191Q"),
  treatment = factor(c("Untreated", "ML240", "Untreated", "ML240", "Untreated", "ML240", "Untreated", "ML240"), levels = c("Untreated", "ML240")),
  n = c(381,423,358,412,602,689,569,906))
mutant_tx_ml240_rescue

mutant_tx_ml240_rescue.plot <- ggplot(mutant_tx_ml240_rescue, aes(x=treatment, y = n, fill = treatment)) +
    geom_bar(stat = "identity") + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
  # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust = 1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(x = "", y = "Redistributed transcripts") + #ylim(c(-12,12)) +
  facet_wrap(~mutation, scales = "free", nrow = 1)
mutant_tx_ml240_rescue.plot

```


## TARDBP Volcano  & GO

multi-factor design

```{r}
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) # 423
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 213
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 210
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 7
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) # KIF5A
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 34
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) # MATR3, ELP3, HNRNPA2B1
ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 3
nuclear_export.markers = unique(c(go.pathways.msigdb$GO_NUCLEAR_EXPORT, go.pathways.msigdb$GO_NUCLEAR_PORE, go.pathways.msigdb$GO_NUCLEAR_PORE_ORGANIZATION, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS))

# transcript level volcano
ml240.nuc_vs_cyt.tardbp_vs_ctrl.rbp.labels <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.tardbp_vs_ctrl.als.labels <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.tardbp_vs_ctrl.volcano <- volcano_plot(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, gene_labels = c(ml240.nuc_vs_cyt.tardbp_vs_ctrl.als.labels, ml240.nuc_vs_cyt.tardbp_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 20, dpi = 300, distinct_labels = TRUE, title = "TARDBP mutant") + 
  xlab(expression(TARDBP~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 20*0.88, yend= 20 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 20*0.88, yend= 20*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 20*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 20*0.93, label = "Cytoplasm", size = 3)
# ml240.nuc_vs_cyt.tardbp_vs_ctrl.volcano

# # # # density plot
# ml240.nuc_vs_cyt.tardbp_vs_ctrl.density = density_plot(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, color_by = "condition", cols = "forestgreen", title = NULL, subtitle = NULL, ymax = 1, xpos = 0.5, numerator = NULL, denomenator = NULL, arrow_labels = TRUE, dpi = 300)
# ml240.nuc_vs_cyt.tardbp_vs_ctrl.density

### GO mislocalised
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt <- ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        26        19 
ml240.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_nuclear <- ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
ml240.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_cytoplasm <- ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"Autophagy - animal",
"positive regulation of RNA metabolic process",
"regulation of signaling",
"histone modification",
"intracellular anatomical structure",
"synapse",
"protein binding"
)
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_nuclear <- ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_nuclear %<>% mutate(term_name = gsub("positive regulation of ", "", term_name))
paste('"',unique(ml240.nuc_vs_cyt.tardbp_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"tubulin binding",
"somatodendritic compartment",
"ATP-dependent activity",
"trans-Golgi network",
"cell junction",
"intracellular anatomical structure",
"ribonucleotide binding",
"ATP binding",
"cytoplasm")
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm <- ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
ml240.nuc_vs_cyt.tardbp_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.nuc_vs_cyt.tardbp_vs_ctrl.go_curated


# Rescued splicing events GO terms
# untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join == Table S3
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(tardbp.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(tardbp.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt <- ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#        91        24 
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"endomembrane system",
"cellular localization",
"cellular response to stress",
"intracellular anatomical structure",
"regulation of cellular component organization",
"protein modification process",
"microtubule organizing center",
"protein binding",
"protein metabolic process"
)
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"protein metabolic process",
"Chromatin organization",
"cellular response to stress",
"nuclear speck",
"RNA biosynthetic process",
"regulation of RNA metabolic process",
"protein modification process",
"macromolecule modification",
"intracellular anatomical structure",
"protein binding")
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated

```


## VCP R155c volcano & GO

multi-factor design

```{r}
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res %>% filter(padj < 0.05) # 75
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) # 412
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 231
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 181
# ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj > 0.05) # 1017
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 39
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) #3
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) #0
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) #0
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 5 

# transcript level volcano
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.rbp.labels <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.als.labels <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano <- volcano_plot(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, gene_labels = c(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.als.labels, ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 20, dpi = 300, distinct_labels = TRUE, title = "VCP R155C mutant") + 
  xlab(expression(VCP~R155C~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 20*0.88, yend= 20*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 20*0.88, yend= 20*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 20*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 20*0.93, label = "Cytoplasm", size = 3)
# ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano

### GO mislocalised
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        17        16 
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_nuclear <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_cytoplasm <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"macromolecule localization",
"protein modification process",
"intracellular anatomical structure",
"macromolecule modification",
"regulation of signaling",
"protein binding"
)
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_nuclear <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"nuclear lumen",
"ribonucleotide binding",
"ion binding",
"ATP binding",
"protein binding")
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("cellular ","",term_name), term_name = gsub("acting on a ","",term_name))
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated

# Rescued transcripts
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>%filter(vcp_r155c.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_r155c.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt <- ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#         8         9 
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cell junction organization",
"synapse",
"site of polarized growth",
"growth cone",
"cell junction",
"postsynapse",
"cytosol",
"cytoplasm",
"protein binding"
)
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"regulation of cellular metabolic process",
"cytoplasm",
"RNA polymerase",
"cytosol",
"ion binding",
"peptidyl-amino acid modification",
"cell cortex",
"positive regulation of double-strand break repair")
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("regulation of ","",term_name))
ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated

```


## VCP R191Q volcano & GO

multi-factor design

```{r}
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res %>% filter(padj < 0.05) #138
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) # 689
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 311
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 378
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 64
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) #11
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) #TNIP1
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) #MATR3
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) #7

# transcript level volcano
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.rbp.labels <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.als.labels <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano <- volcano_plot(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, gene_labels = c(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.als.labels, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 20, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q mutant") + 
  xlab(expression(VCP~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 20*0.88, yend= 20*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 20*0.88, yend= 20*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 20*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 20*0.93, label = "Cytoplasm", size = 3)
# ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano

### GO mislocalised
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#        75        30 
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_nuclear <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_cytoplasm <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cytoskeleton-dependent intracellular transport",
"catalytic activity",
"ion binding",
"ribonucleotide binding",
"anatomical structure development",
"ATP binding",
"protein binding"
)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear %<>% mutate(term_name = gsub("cytoskeleton-dependent ", "", term_name))
paste('"',unique(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"ATP binding",
"cytoskeleton",
"axon",
"ribonucleotide binding",
"ATP-dependent activity",
"focal adhesion",
"enzyme binding",
"peptide metabolic process",
"ion binding",
"intracellular anatomical structure",
"protein binding")
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name), term_name = gsub("acting on a ","",term_name), term_name = gsub(" \\(FBXW7, CUL1, SKP1A, RBX1\\)","",term_name))
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated


# Rescued transcripts
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>%filter(vcp_r191q.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_r191q.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt <- ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#         5         6 
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"distal axon",
"3-hydroxyacyl-CoA dehydrogenase activity",
"site of polarized growth",
"growth cone",
"spanning component of membrane",
"spanning component of plasma membrane"
)
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"matrix side of mitochondrial inner membrane",
"cytosol",
"cell death",
"regulation of cell death",
"cytoplasm")
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("matrix side of ", "", term_name))#, term_name = gsub("and","&",term_name))
ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated

```


## VCP R191Q knock-in volcano & GO

multi-factor design

```{r}
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) # 906
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) # 379
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) # 527
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 86
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_RBPs) # MATR3
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_GWAS) # NEK1, TNIP1
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% ALS_genes) # 12
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx  %>% filter(padj < 0.05, gene_name %in% nuclear_export.markers) # 20

# transcript level volcano
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.rbp.labels <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.als.labels <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano <- volcano_plot(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, gene_labels = c(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.als.labels, ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.rbp.labels), arrow_labels = FALSE, xmax = 30, ymax = 20, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q knock-in") + 
  xlab(expression(VCPki~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 30*0.25, xend = 30*0.95, y= 20*0.88, yend= 20 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -30*0.25, xend = -30*0.95, y= 20*0.88, yend= 20*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 30*0.6, y = 20*0.93, label = "Nuclear", size = 3) + annotate("text", x = -30*0.6, y = 20*0.93, label = "Cytoplasm", size = 3)
# ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano

### GO mislocalised
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt$query)
# Cytoplasm   Nuclear 
#       110        80 
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_nuclear <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_cytoplasm <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cellular localization",
"intracellular anatomical structure",
"ion binding",
"catalytic activity",
"neuron projection",
"nucleotide binding",
"synapse",
"regulation of signaling",
"regulation of response to stimulus",
"protein modification process",
"ATP binding",
"protein binding")
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_nuclear <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"ATPase complex",
"focal adhesion",
"regulation of RNA metabolic process",
"ATP binding",
"synapse",
"protein modification process",
"axon",
"intracellular anatomical structure",
"protein metabolic process",
"nucleotide binding",
"catalytic activity",
"protein binding",
"ion binding")
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated


# Rescued transcripts
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.tx.join %>% filter(vcp_ki.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_ki_r191q.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt <- ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#        66        28 
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"enzyme binding",
"regulation of apoptotic process",
"Hsp90 protein binding",
"protein C-terminus binding",
"regulation of programmed cell death",
"spanning component of membrane",
"intracellular protein-containing complex",
"postsynapse",
"synapse"
# "spanning component of plasma membrane"
)
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"endomembrane system",
"chromatin organization",
"translation",
"protein metabolic process",
"regulation of cellular response to stress",
"synapse",
"axon",
"cytoplasmic stress granule",
"protein binding",
"intracellular anatomical structure")
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))#, term_name = gsub("and","&",term_name))
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated


```


Correlation heatmap

```{r}
# heatmap of correlation coefficients
ml240.nuc_vs_cyt.als_vs_ctrl_stat = select(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, `TARDBP^G298S` = stat) %>% 
  left_join(select(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, `VCP^R155C` = stat)) %>%
  left_join(select(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, `VCP^R191Q` = stat)) %>%
  left_join(select(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, `VCPki^R191Q` = stat)) %>%
  drop_na() %>%
  column_to_rownames("transcript_id")
cormat <- round(cor(ml240.nuc_vs_cyt.als_vs_ctrl_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
# reorder_cormat <- function(cormat){ # Use correlation between variables as distance
#   dd <- as.dist((1-cormat)/2)
#   hc <- hclust(dd)
#   cormat <-cormat[hc$order, hc$order]
# }
# cormat <- reorder_cormat(cormat)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
  # mutate(Var1 = fct_relevel(Var1, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"), Var2 = fct_relevel(Var2, "TARDBP^G298S", "VCP^R155C", "VCP^R191Q", "VCPki^R191Q"))
ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap

```


Upset plot

```{r}
ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx = bind_rows(mutate(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP G298S"), mutate(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C"), mutate(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q"), mutate(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCP R191Q knock-in"))

ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset = upset_plot(filter(ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx, stat > 0), overlap_by = "mutation", overlap_level = "transcript_id", split_direction = FALSE, ymax = 340, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Nuclear redistributed", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset

ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset = upset_plot(filter(ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx, stat < 0), overlap_by = "mutation", overlap_level = "transcript_id", split_direction = FALSE, ymax = 480, order = "freq", order_group_list = c("TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Cytoplasm redistributed", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset

# Fisher exact overlap test
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.up <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange > 0) %>% pull(transcript_id)
ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.down <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05, log2FoldChange < 0) %>% pull(transcript_id)

newGeneOverlap(ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up, genome.size=nrow(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=181
# Intersection size=20
# Overlapping p-value=1.7e-38
# Jaccard Index=0.1
newGeneOverlap(ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down, genome.size=nrow(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=228
# listB size=177
# Intersection size=34
# Overlapping p-value=2.1e-67
# Jaccard Index=0.1

newGeneOverlap(ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=153
# listB size=181
# Intersection size=16
# Overlapping p-value=2.2e-29
# Jaccard Index=0.1
newGeneOverlap(ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=228
# listB size=421
# Intersection size=53
# Overlapping p-value=1e-94
# Jaccard Index=0.1

newGeneOverlap(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=181
# listB size=181
# Intersection size=25
# Overlapping p-value=1.6e-48
# Jaccard Index=0.1
newGeneOverlap(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=177
# listB size=421
# Intersection size=50
# Overlapping p-value=3.3e-94
# Jaccard Index=0.1

newGeneOverlap(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.up, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.up, genome.size=nrow(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=190
# listB size=181
# Intersection size=22
# Overlapping p-value=5.1e-41
# Jaccard Index=0.1
newGeneOverlap(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.res.dge.down, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.res.dge.down, genome.size=nrow(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=379
# listB size=421
# Intersection size=65
# Overlapping p-value=1.1e-106
# Jaccard Index=0.1


# # # ggvenn
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes <- list("TARDBP" = ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.up, "VCP" = ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.up)
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes <- list("TARDBP" = ml240.nuc_vs_cyt.tardbp_vs_ctrl.res.dge.down, "VCP" = ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.res.dge.down)
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn <- ggvenn(ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5 ,text_size = 4) + labs(caption = "Nuclear")  + theme(plot.caption = element_text(vjust = 4, hjust = 0.5, colour = "firebrick2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn
# 
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn <- ggvenn(ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_genes, fill_color = c("forestgreen", "gold2"), show_percentage = FALSE, stroke_size = 0.4, set_name_size = 2.5,text_size = 4) + labs(caption = "Cytoplasm")  + theme(plot.caption = element_text(vjust = 4,hjust = 0.5, colour = "dodgerblue2"), plot.margin = unit(c(0, 0, 0, 0), "lines"))
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn

# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn = plot_grid(NULL, ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.up_venn, NULL, ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.down_venn, rel_heights =c(0.02,1,0.05,1), nrow = 4)
# ml240.nuc_vs_cyt.tardbp_vcp_r155c_vs_ctrl.venn

```





## Merge

```{r}
ml240.transcript_redistribution.merged <- plot_grid(
  plot_grid(rna.untreated_ml240.nuc_vs_cyt.list.scatter, protein.untreated_ml240.nuc_vs_cyt.list.scatter, ncol = 2, labels = c("a","b")),
  mutant_tx_ml240_rescue.plot,
  plot_grid(ml240.nuc_vs_cyt.tardbp_vs_ctrl.volcano, ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.volcano, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.volcano, ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.volcano, ncol = 4, labels = letters[4:7]),
  plot_grid(ml240.nuc_vs_cyt.tardbp_vs_ctrl.go_curated, ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.go_curated, ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.go_curated, ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.go_curated, ncol = 4, labels = letters[8:11]), 
    plot_grid(ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated, ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated, ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated, ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated, ncol = 4, labels = letters[12:15]), 
  nrow = 5, rel_heights = c(1,0.7,0.9,1,1), labels = c("","c","","",""))
# ml240.transcript_redistribution.merged
ggsave(ml240.transcript_redistribution.merged, filename = here(proj_path, "expression/deseq2/ml240.transcript_redistribution.merged.png"), height = 16.5, width = 12)

# plot_grid(ml240.tardbp_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.rna_protein.stat.corr, ncol = 3),
#   plot_grid(ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap, ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.nuclear.upset, ml240.als_vs_ctrl.nuc_vs_cyt.bind_datasets.res.tx.cytoplasm.upset, ncol = 3, rel_widths = c(0.8,1,1), labels = letters[11:13]), 
#   rnact.rna_view.ml240.nuc_vs_cyt.als_vs_ctrl.violin, 
#   
# ml240.vcp_tardbp_tx_redistribution.merged <- plot_grid(
#   # analysis_schematic.gg
#   
#   nrow = 3, labels = c("","","","l"), rel_heights = c(1,1,0.6))
# # ml240.vcp_tardbp_tx_redistribution.merged
# ggsave(ml240.vcp_tardbp_tx_redistribution.merged, filename = here(proj_path, "expression/deseq2/ml240.vcp_tardbp_tx_redistribution.merged.png"), height = 11, width = 12)
# 
# plot_grid(tardbp_vcp_r155c.ml240.nuc_vs_cyt.als_vs_ctrl_stat.corr, vcp_iso_vcp_r155c.ml240.nuc_vs_cyt.als_vs_ctrl_stat.corr, ncol = 2, rel_widths = c(1,1), labels = letters[8:9]), 

```




# Fig S17 Redistributed transcript features


```{r}
homo_sapiens.gtf = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.rds"))
Homo_sapiens.GRCh38.99.gc.phast = readRDS(here(nemo_path, "genomes/ensembl/Homo_sapiens.GRCh38.99.gc.phase.rds"))
homo_sapiens.gtf.gene <- homo_sapiens.gtf %>% filter(type == "gene")
homo_sapiens.gtf.transcript <- homo_sapiens.gtf %>% filter(type == "transcript") # NB this is unspliced width
homo_sapiens.gtf.spliced.transcript <- homo_sapiens.gtf %>% group_by(transcript_id) %>% select(transcript_id, transcript_biotype, width, type) %>% pivot_wider(c(transcript_id, transcript_biotype), names_from = type, values_from = width, values_fn = sum) %>% mutate(width = sum(exon, CDS, five_prime_utr, three_prime_utr, na.rm = TRUE)) %>% select(transcript_id, transcript_biotype, width)
homo_sapiens.gtf.5UTR.transcript <- homo_sapiens.gtf %>% filter(type == "five_prime_utr") %>% group_by(transcript_id) %>% summarise(five_prime_utr = mean(width, na.rm = TRUE)) # mean 5'UTR length isoform
homo_sapiens.gtf.3UTR.transcript <- homo_sapiens.gtf %>% filter(type == "three_prime_utr") %>% group_by(transcript_id) %>% summarise(three_prime_utr = mean(width, na.rm = TRUE)) # mean 3UTR length isoform
homo_sapiens.gtf.CDS.transcript <- homo_sapiens.gtf %>% filter(type == "CDS") %>% group_by(transcript_id) %>% summarise(CDS = mean(width, na.rm = TRUE))
homo_sapiens.gtf.exon.transcript <- homo_sapiens.gtf %>% filter(type == "exon") %>% group_by(transcript_id) %>% summarise(exon_number = n()) # total exon number - capped at 9???

# mean lenths of biotypes
homo_sapiens.gtf.transcript %>% group_by(transcript_biotype) %>% summarise(lengths = mean(width, na.rm = TRUE)) %>% print(n=Inf)
homo_sapiens.gtf.transcript %>% summarise(lengths = mean(width, na.rm = TRUE))
homo_sapiens.gtf.CDS.transcript %>% summarise(lengths = mean(CDS, na.rm = TRUE))

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features = bind_rows(mutate(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP", treatment = "ML240"), mutate(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C", treatment = "ML240"),  mutate(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q", treatment = "ML240"), mutate(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki R191Q", treatment = "ML240"),
mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP", treatment = "Untreated"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C", treatment = "Untreated"), mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q", treatment = "Untreated"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki R191Q", treatment = "Untreated")) %>% 
  filter(padj < 0.05) %>%
  mutate(mutation = factor(mutation, levels = c("TARDBP", "VCP R155C", "VCP R191Q", "VCPki R191Q")), treatment = factor(treatment, levels = c("Untreated", "ML240"))) %>%
  left_join(select(homo_sapiens.gtf.spliced.transcript, transcript_id, transcript_biotype, width)) %>%
  left_join(select(homo_sapiens.gtf.5UTR.transcript, transcript_id, five_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.3UTR.transcript, transcript_id, three_prime_utr)) %>%
  left_join(select(homo_sapiens.gtf.CDS.transcript, transcript_id, CDS)) %>%
  left_join(select(homo_sapiens.gtf.exon.transcript, transcript_id, exon_number)) %>%
  left_join(select(Homo_sapiens.GRCh38.99.gc.phast, transcript_id, gc_content, phast)) %>%
  mutate(mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear\nRedistributed", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm\nRedistributed", TRUE ~ "Non"), mislocalisation = factor(mislocalisation, levels = c("Non", "Nuclear\nRedistributed", "Cytoplasm\nRedistributed")), 
        log10.width = log10(width), log10.five_prime_utr = log10(five_prime_utr), log10.three_prime_utr = log10(three_prime_utr), log10.CDS = log10(CDS), log10.exon_number = log10(exon_number),
        intronless = case_when(exon_number == 1 ~ "Intronless", exon_number >1 ~ "Intron Containing", TRUE ~ "NA"))
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features %>% filter(padj < 0.05 & log2FoldChange > 0 & treatment == "ML240")

# define resistant transcripts
tardbp_nuclear_resistant = select(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc > 0, ml240.padj < 0.05, ml240.lfc > 0) %>% pull(transcript_id)
tardbp_cytoplasm_resistant = select(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc < 0, ml240.padj < 0.05, ml240.lfc < 0) %>% pull(transcript_id)
vcp_r155c_nuclear_resistant = select(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc > 0, ml240.padj < 0.05, ml240.lfc > 0) %>% pull(transcript_id)
vcp_r155c_cytoplasm_resistant = select(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc < 0, ml240.padj < 0.05, ml240.lfc < 0) %>% pull(transcript_id)
vcp_r191q_nuclear_resistant = select(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc > 0, ml240.padj < 0.05, ml240.lfc > 0) %>% pull(transcript_id)
vcp_r191q_cytoplasm_resistant = select(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc < 0, ml240.padj < 0.05, ml240.lfc < 0) %>% pull(transcript_id)
vcp_iso_nuclear_resistant = select(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc > 0, ml240.padj < 0.05, ml240.lfc > 0) %>% pull(transcript_id)
vcp_iso_cytoplasm_resistant = select(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, ml240.padj = padj, ml240.lfc = log2FoldChange) %>% 
  left_join(select(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, transcript_id, gene_name, untx.padj = padj, untx.lfc = log2FoldChange)) %>% 
  filter(untx.padj < 0.05, untx.lfc < 0, ml240.padj < 0.05, ml240.lfc < 0) %>% pull(transcript_id)

### Transcript lengths
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features %>% group_by(treatment, mutation, mislocalisation) %>% summarise(lengths = mean(width, na.rm = TRUE))
# treatment mutation      mislocalisation            lengths
#    <fct>     <fct>         <fct>                        <dbl>
#  1 Untreated TARDBP "Nuclear\nRedistributed"     5848.
#  2 Untreated TARDBP "Cytoplasm\nRedistributed"   6007.
#  3 Untreated VCP R155C     "Nuclear\nRedistributed"     5141.
#  4 Untreated VCP R155C     "Cytoplasm\nRedistributed"   5719.
#  5 Untreated VCP R191Q     "Nuclear\nRedistributed"     5755.
#  6 Untreated VCP R191Q     "Cytoplasm\nRedistributed"   5323.
#  7 Untreated VCPki R191Q   "Nuclear\nRedistributed"     5363.
#  8 Untreated VCPki R191Q   "Cytoplasm\nRedistributed"   4926.
#  9 ML240     TARDBP "Nuclear\nRedistributed"     5476.
# 10 ML240     TARDBP "Cytoplasm\nRedistributed"   6223.
# 11 ML240     VCP R155C     "Nuclear\nRedistributed"     5139.
# 12 ML240     VCP R155C     "Cytoplasm\nRedistributed"   5769.
# 13 ML240     VCP R191Q     "Nuclear\nRedistributed"     4705.
# 14 ML240     VCP R191Q     "Cytoplasm\nRedistributed"   5608.
# 15 ML240     VCPki R191Q   "Nuclear\nRedistributed"     5829.
# 16 ML240     VCPki R191Q   "Cytoplasm\nRedistributed"   6412.

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.width.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "log10.width", ylims = c(3,4.5), 
                                                           facet_by = "mutation", facet_by2 = "mislocalisation",
                                                           plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 4.4, 
                                                           ylabel = expression(log[10]~Transcript~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust=1)) #+
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.width.tx.violin
# [1] "Faceting by mutation, mislocalisation"
# [1] "Using two sample wilcox_test"
# # A tibble: 6 × 14
#   facet         facet2                     .y.   group1    group2    n1    n2 statistic        p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>    <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    373   355    70721  1.12e- 1 ns              4.4 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    416   348    68898. 2.51e- 1 ns              4.4 <chr [2]>        1     2
# 3 VCP mutant    "Nuclear\nRedistributed"   lfc   Untreated ML240    560   844   304966. 2.76e-20 ****            4.4 <chr [2]>        1     2
# 4 VCP mutant    "Cytoplasm\nRedistributed" lfc   Untreated ML240    440   575   115582. 1.83e- 2 *               4.4 <chr [2]>        1     2
# 5 VCP knock-in  "Nuclear\nRedistributed"   lfc   Untreated ML240    466   253    56618  3.81e- 1 ns              4.4 <chr [2]>        1     2
# 6 VCP knock-in  "Cytoplasm\nRedistributed" lfc   Untreated ML240    770   744   234060. 7.32e-10 ****            4.4 <chr [2]>        1     2
# # A tibble: 6 × 9
#   .y.   group1    group2 effsize facet         facet2                        n1    n2 magnitude 
# * <chr> <chr>     <chr>    <dbl> <fct>         <fct>                      <int> <int> <ord>     
# 1 lfc   Untreated ML240   0.122  TARDBP "Nuclear\nRedistributed"     373   355 negligible
# 2 lfc   Untreated ML240  -0.0689 TARDBP "Cytoplasm\nRedistributed"   416   348 negligible
# 3 lfc   Untreated ML240   0.526  VCP mutant    "Nuclear\nRedistributed"     560   844 moderate  
# 4 lfc   Untreated ML240  -0.157  VCP mutant    "Cytoplasm\nRedistributed"   440   575 negligible
# 5 lfc   Untreated ML240  -0.0782 VCP knock-in  "Nuclear\nRedistributed"     466   253 negligible
# 6 lfc   Untreated ML240  -0.341  VCP knock-in  "Cytoplasm\nRedistributed"   770   744 small     

### Exon number
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "log10.exon_number", ylims = c(0,1.7), 
                                                               facet_by = "mutation", facet_by2 = "mislocalisation",
                                                               plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 1.6, 
                                                               ylabel = expression(log[10]~Exon~Number), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE, violin_adjust = 2.5) + theme(axis.text.x = element_text(angle = 30, hjust=1))
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin
# facet         facet2                     .y.   group1    group2    n1    n2 statistic             p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>         <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    360   358    66998. 0.357         ns              4.4 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    419   353    73181  0.803         ns              4.4 <chr [2]>        1     2
# 3 VCP R155C     "Nuclear\nRedistributed"   lfc   Untreated ML240    407   379    80286. 0.321         ns              4.4 <chr [2]>        1     2
# 4 VCP R155C     "Cytoplasm\nRedistributed" lfc   Untreated ML240    343   372    63566. 0.933         ns              4.4 <chr [2]>        1     2
# 5 VCP R191Q     "Nuclear\nRedistributed"   lfc   Untreated ML240    373   603   129664. 0.0000581     ****            4.4 <chr [2]>        1     2
# 6 VCP R191Q     "Cytoplasm\nRedistributed" lfc   Untreated ML240    578   624   175740. 0.445         ns              4.4 <chr [2]>        1     2
# 7 VCPki R191Q   "Nuclear\nRedistributed"   lfc   Untreated ML240    476   238    53136. 0.177         ns              4.4 <chr [2]>        1     2
# 8 VCPki R191Q   "Cytoplasm\nRedistributed" lfc   Untreated ML240    775   737   231040  0.00000000013 ****            4.4 <chr [2]>        1     2

### 5'UTR length
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "log10.five_prime_utr", ylims = c(1,3), 
                                                          facet_by = "mutation", facet_by2 = "mislocalisation",
                                                          plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 2.9, 
                                                          ylabel = expression(log[10]~5~UTR~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust=1))
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin
#  facet         facet2                     .y.   group1    group2    n1    n2 statistic      p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>  <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    227   225    25934  0.775  ns              2.9 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    296   255    38338. 0.748  ns              2.9 <chr [2]>        1     2
# 3 VCP R155C     "Nuclear\nRedistributed"   lfc   Untreated ML240    268   203    26562. 0.662  ns              2.9 <chr [2]>        1     2
# 4 VCP R155C     "Cytoplasm\nRedistributed" lfc   Untreated ML240    244   251    30211  0.796  ns              2.9 <chr [2]>        1     2
# 5 VCP R191Q     "Nuclear\nRedistributed"   lfc   Untreated ML240    234   338    40426  0.651  ns              2.9 <chr [2]>        1     2
# 6 VCP R191Q     "Cytoplasm\nRedistributed" lfc   Untreated ML240    401   438    88438  0.86   ns              2.9 <chr [2]>        1     2
# 7 VCPki R191Q   "Nuclear\nRedistributed"   lfc   Untreated ML240    297   145    23750. 0.0788 ns              2.9 <chr [2]>        1     2
# 8 VCPki R191Q   "Cytoplasm\nRedistributed" lfc   Untreated ML240    541   577   146009  0.062  ns              2.9 <chr [2]>        1     2

### 3'UTR length
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "log10.three_prime_utr", ylims = c(1.5,4), 
                                                                    facet_by = "mutation", facet_by2 = "mislocalisation",
                                                               plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 3.9, 
                                                               ylabel = expression(log[10]~3~UTR~length), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust=1))
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin
#  facet         facet2                     .y.   group1    group2    n1    n2 statistic        p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>    <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    249   241    31540. 0.327    ns              3.9 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    306   258    39456  0.993    ns              3.9 <chr [2]>        1     2
# 3 VCP R155C     "Nuclear\nRedistributed"   lfc   Untreated ML240    278   217    29652. 0.747    ns              3.9 <chr [2]>        1     2
# 4 VCP R155C     "Cytoplasm\nRedistributed" lfc   Untreated ML240    250   265    32164. 0.569    ns              3.9 <chr [2]>        1     2
# 5 VCP R191Q     "Nuclear\nRedistributed"   lfc   Untreated ML240    246   332    47806. 0.000446 ***             3.9 <chr [2]>        1     2
# 6 VCP R191Q     "Cytoplasm\nRedistributed" lfc   Untreated ML240    435   470   102862. 0.871    ns              3.9 <chr [2]>        1     2
# 7 VCPki R191Q   "Nuclear\nRedistributed"   lfc   Untreated ML240    316   153    22972. 0.383    ns              3.9 <chr [2]>        1     2
# 8 VCPki R191Q   "Cytoplasm\nRedistributed" lfc   Untreated ML240    557   593   150594. 0.00971  **              3.9 <chr [2]>        1     2

# GC content
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "gc_content", ylims = c(0.3,0.7), 
                                                                  facet_by = "mutation", facet_by2 = "mislocalisation",
                                                           stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 0.68, 
                                                               ylabel = "GC content", xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust=1))
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin
# facet         facet2                     .y.   group1    group2    n1    n2 statistic        p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>    <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    342   343    60502. 0.475    ns             0.68 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    400   332    63827  0.366    ns             0.68 <chr [2]>        1     2
# 3 VCP R155C     "Nuclear\nRedistributed"   lfc   Untreated ML240    377   359    60873  0.0184   *              0.68 <chr [2]>        1     2
# 4 VCP R155C     "Cytoplasm\nRedistributed" lfc   Untreated ML240    330   353    57947  0.908    ns             0.68 <chr [2]>        1     2
# 5 VCP R191Q     "Nuclear\nRedistributed"   lfc   Untreated ML240    351   592    90532  0.000949 ***            0.68 <chr [2]>        1     2
# 6 VCP R191Q     "Cytoplasm\nRedistributed" lfc   Untreated ML240    547   581   147848  0.0432   *              0.68 <chr [2]>        1     2
# 7 VCPki R191Q   "Nuclear\nRedistributed"   lfc   Untreated ML240    443   230    50738  0.931    ns             0.68 <chr [2]>        1     2
# 8 VCPki R191Q   "Cytoplasm\nRedistributed" lfc   Untreated ML240    741   697   269757  0.143    ns             0.68 <chr [2]>        1     2

# PhastCons
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.phast.tx.violin = violin_plot(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.tx.features, color_by = "treatment", continuous = "phast", ylims = c(0,0.35), 
                                                                     facet_by = "mutation", facet_by2 = "mislocalisation",
                                                           stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 0.33, 
                                                               ylabel = "Conservation score", xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust=1))
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.phast.tx.violin
# facet         facet2                     .y.   group1    group2    n1    n2 statistic      p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>                      <chr> <chr>     <chr>  <int> <int>     <dbl>  <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP "Nuclear\nRedistributed"   lfc   Untreated ML240    269   268    37276. 0.494  ns             0.33 <chr [2]>        1     2
# 2 TARDBP "Cytoplasm\nRedistributed" lfc   Untreated ML240    289   260    33481  0.0276 *              0.33 <chr [2]>        1     2
# 3 VCP R155C     "Nuclear\nRedistributed"   lfc   Untreated ML240    307   292    43018  0.394  ns             0.33 <chr [2]>        1     2
# 4 VCP R155C     "Cytoplasm\nRedistributed" lfc   Untreated ML240    248   279    35136. 0.757  ns             0.33 <chr [2]>        1     2
# 5 VCP R191Q     "Nuclear\nRedistributed"   lfc   Untreated ML240    272   475    63824. 0.785  ns             0.33 <chr [2]>        1     2
# 6 VCP R191Q     "Cytoplasm\nRedistributed" lfc   Untreated ML240    430   469    93148. 0.0481 *              0.33 <chr [2]>        1     2
# 7 VCPki R191Q   "Nuclear\nRedistributed"   lfc   Untreated ML240    365   179    35120. 0.155  ns             0.33 <chr [2]>        1     2
# 8 VCPki R191Q   "Cytoplasm\nRedistributed" lfc   Untreated ML240    577   536   145868. 0.102  ns             0.33 <chr [2]>        1     2

## RNAct interactions RNA view
RNAct_database.transcript.summarised = readRDS(here(nemo_path, "genomes/rbp-databases/rnact/RNAct_database.transcript.summarised.rds"))
rnact_rna_view.ml240.nuc_vs_cyt.als_vs_ctrl = bind_rows(mutate(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP", treatment = "Untreated"), mutate(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C", treatment = "Untreated"),  mutate(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q", treatment = "Untreated"), mutate(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki R191Q", treatment = "Untreated"), mutate(ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx, mutation = "TARDBP", treatment = "ML240"), mutate(ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx, mutation = "VCP R155C", treatment = "ML240"), mutate(ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx, mutation = "VCP R191Q", treatment = "ML240"), mutate(ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx, mutation = "VCPki R191Q", treatment = "ML240")) %>%
  filter(padj < 0.05) %>%
  left_join(RNAct_database.transcript.summarised) %>%
  mutate(log10.mean_score = log10(mean_score), log10.sum_score = log10(sum_score), log10.number = log10(number), 
         mislocalisation = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear\nRedistributed", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm\nRedistributed", TRUE ~ "Non"),
         mislocalisation = factor(mislocalisation, levels = c("Non", "Nuclear\nRedistributed", "Cytoplasm\nRedistributed")), mutation = factor(mutation, levels = c("TARDBP", "VCP R155C", "VCP R191Q", "VCPki R191Q")), treatment = factor(treatment, levels = c("Untreated", "ML240"))) %>% drop_na(mean_score)#, 
         # across(contains(c("score","number")), ~replace(., is.na(.)|is.infinite(.), 0))) # replace NA with 0 - this is crucial before taking the mean otherwise doesnt account for number of RBP-RNA interactions
rnact_rna_view.ml240.nuc_vs_cyt.als_vs_ctrl

rnact_rna_view.ml240.nuc_vs_cyt.als_vs_ctrl %>% my_summarise(sum_score, mislocalisation, mutation)
# mislocalisation mutation      mean_sum_score median_sum_score sd_sum_score Q1_sum_score Q3_sum_score min_sum_score max_sum_score sum_sum_score n_sum_score
#   <fct>           <fct>                  <dbl>            <dbl>        <dbl>        <dbl>        <dbl>         <dbl>         <dbl>         <dbl>       <int>
# 1 Nuc             TARDBP        250209.          245530.      103838.      175616.      313703.        17221.       600449.     92577254.         370
# 2 Nuc             VCP R155C            257636.          255558.      103465.      181043.      321444.        24895.       664032.    106403860.         413
# 3 Nuc             VCP R191Q            255012.          253762.      101783.      179040.      321822.        13839.       631727.    136176542.         534
# 4 Nuc             VCPki R191Q          250588.          250182.       99483.      170578.      318251.        36906.       535929.     93720055.         374
# 5 Cyto            TARDBP        253315.          248288.      106031.      173841.      323879.        28145.       626094.    113484995.         448
# 6 Cyto            VCP R155C            251269.          247827.      106393.      170474.      333387.        24895.       543095.    103522655.         412
# 7 Cyto            VCP R191Q            266418.          264305.      107321.      179800.      342937.        42360.       650328.    185959754.         698
# 8 Cyto            VCPki R191Q          267730.          267983.      107852.      187020.      341372.        22098.       614839.    258626736.         966

# Mislocalisation Nuclear vs Cytoplasm vs None
rnact.rna_view.ml240.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(rnact_rna_view.ml240.nuc_vs_cyt.als_vs_ctrl, color_by = "treatment", continuous = "sum_score", ylims = c(0,550000), 
                                                                     facet_by = "mutation", facet_by2 = "mislocalisation",
                                                               plot_stats = TRUE, stats_test = "wilcox_test", tip_length = 0.001, stat_label = "p.signif", stat_ypos = 520000, 
                                                               ylabel = expression(RNA~view:RBP~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2"), dpi = 300, arrow_labels = FALSE) +
	scale_y_continuous(labels = label_number(suffix = " M", scale = 1e-6))  + theme(axis.text.x = element_text(angle = 30, hjust=1))
# rnact.rna_view.ml240.nuc_vs_cyt.als_vs_ctrl.violin
# facet         facet2 .y.   group1    group2    n1    n2 statistic      p p.signif y.position groups        xmin  xmax
#   <fct>         <fct>  <chr> <chr>     <chr>  <int> <int>     <dbl>  <dbl> <chr>         <dbl> <named list> <dbl> <dbl>
# 1 TARDBP Nuc    lfc   Untreated ML240    185   185    17688. 0.577  ns           520000 <chr [2]>        1     2
# 2 TARDBP Cyto   lfc   Untreated ML240    250   198    26580  0.179  ns           520000 <chr [2]>        1     2
# 3 VCP R155C     Nuc    lfc   Untreated ML240    213   200    21920  0.609  ns           520000 <chr [2]>        1     2
# 4 VCP R155C     Cyto   lfc   Untreated ML240    205   207    21886. 0.581  ns           520000 <chr [2]>        1     2
# 5 VCP R191Q     Nuc    lfc   Untreated ML240    204   330    29646. 0.0205 *            520000 <chr [2]>        1     2
# 6 VCP R191Q     Cyto   lfc   Untreated ML240    325   373    59268. 0.613  ns           520000 <chr [2]>        1     2
# 7 VCPki R191Q   Nuc    lfc   Untreated ML240    250   124    15989  0.62   ns           520000 <chr [2]>        1     2
# 8 VCPki R191Q   Cyto   lfc   Untreated ML240    476   490   120818. 0.333  ns           520000 <chr [2]>        1     2

# # correlate stat with sum_score
# ggscatter(rnact_rna_view.ml240.nuc_vs_cyt.als_vs_ctrl, x = "stat", y = "sum_score", #color = "Classification", palette = rev(c("firebrick2", "grey", "dodgerblue2")),
#           cor.coef = TRUE, cor.coeff.args = list(method = "pearson", label.y = 7.5, label.x = -4), cor.coef.size = 3, xlab = expression(Stat~ALS~vs~CTRL), ylab = expression(RNAct~score), size = 0.5) + 
#   theme_classic() + theme(legend.position = "none", axis.text=element_text(size=8), axis.title=element_text(size=10)) +
#   geom_hline(yintercept = 0, linetype = 3, colour = "darkgrey") +
#   facet_wrap(~mutation)



## Merge
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.sequence_features.violin.merged =plot_grid(
  untreated_ml240.nuc_vs_cyt.als_vs_ctrl.width.tx.violin, untreated_ml240.nuc_vs_cyt.als_vs_ctrl.exon_number.tx.violin,
  untreated_ml240.nuc_vs_cyt.als_vs_ctrl.5UTR.tx.violin, untreated_ml240.nuc_vs_cyt.als_vs_ctrl.3UTR.tx.violin,
  untreated_ml240.nuc_vs_cyt.als_vs_ctrl.gc.tx.violin, rnact.rna_view.ml240.nuc_vs_cyt.als_vs_ctrl.violin,
  nrow = 3, ncol = 2, labels = "auto")
# untreated_ml240.nuc_vs_cyt.als_vs_ctrl.sequence_features.violin.merged
ggsave(untreated_ml240.nuc_vs_cyt.als_vs_ctrl.sequence_features.violin.merged, filename = here(proj_path, "expression/deseq2/untreated_ml240.nuc_vs_cyt.als_vs_ctrl.sequence_features.violin.merged.png"), height = 11.75, width = 8.25)

```


# Fig S18 ML240 splicing

```{r}
load(here(proj_path, "scripts/whole_ml240_majiq_objects.RData"))

whole_ml240_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #364
whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #400
whole_ml240_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #553
whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #531
whole_ml240_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #578
whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #512
whole_ml240_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #336
whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE) #329

# UpSet plots
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.voila = bind_rows(mutate(select(whole_untreated_tardbp_vs_ctrl.voila,-num_exons), mutation = "Untreated"), mutate(whole_ml240_tardbp_vs_ctrl.voila, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.voila.upset = majiq_upset_plot(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.voila, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 290, title = "TARDBP", label_colours = c("#1F78B4", "#A6CEE3"), order_group_list = c("Untreated", "ML240")) +  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
# untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.voila.upset
400-135
1-135/400 #66.3

untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.voila = bind_rows(mutate(select(whole_untreated_vcp_r155c_vs_ctrl.voila,-num_exons), mutation = "Untreated"), mutate(whole_ml240_vcp_r155c_vs_ctrl.voila, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.voila.upset = majiq_upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.voila, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 340, title = "VCP R155C", label_colours = c("#1F78B4", "#A6CEE3"), order_group_list = c("Untreated", "ML240")) +  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.voila.upset
531-245
1-245/531 #53.9

untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.voila = bind_rows(mutate(select(whole_untreated_vcp_r191q_vs_ctrl.voila,-num_exons), mutation = "Untreated"), mutate(whole_ml240_vcp_r191q_vs_ctrl.voila, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.voila.upset = majiq_upset_plot(untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.voila, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 360, title = "VCP R191Q", label_colours = c("#1F78B4", "#A6CEE3"), order_group_list = c("Untreated", "ML240")) +  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.voila.upset
512-240
1-240/512 #53.1%

untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.voila = bind_rows(mutate(select(whole_untreated_vcp_iso_vs_ctrl.voila,-num_exons), mutation = "Untreated"), mutate(whole_ml240_vcp_iso_vs_ctrl.voila, mutation = "ML240"))
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.voila.upset = majiq_upset_plot(untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.voila, overlap_by = "mutation", overlap_level = "lsv_junction_id", split_direction = FALSE, ymax = 250, title = "VCP knock-in", label_colours = c("#1F78B4", "#A6CEE3"), order_group_list = c("Untreated", "ML240")) +  theme_combmatrix(combmatrix.panel.striped_background.color.two = "#1F78B4", combmatrix.panel.striped_background.color.one = "#A6CEE3")
untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.voila.upset
329-114
1-114/329 #65.3%

untreated_ml240.nuc_vs_cyt.als_vs_ctrl.voila.upset = plot_grid(untreated_ml240.nuc_vs_cyt.tardbp_vs_ctrl.voila.upset, untreated_ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl.voila.upset, untreated_ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl.voila.upset, untreated_ml240.nuc_vs_cyt.vcp_iso_vs_ctrl.voila.upset, ncol = 4)
untreated_ml240.nuc_vs_cyt.als_vs_ctrl.voila.upset


# Heatmap of correlation coefficients
untreated_ml240.nuc_vs_cyt.als_vs_ctrl_deltapsi = select(whole_ml240_tardbp_vs_ctrl.voila, lsv_junction_id, `TARDBP^G298S\nML240` = deltapsi) %>% 
  left_join(select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, `TARDBP^G298S\nUntreated` = deltapsi)) %>%
  left_join(select(whole_ml240_vcp_r155c_vs_ctrl.voila, lsv_junction_id, `VCP^R155C\nML240` = deltapsi)) %>%
  left_join(select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, `VCP^R155C\nUntreated` = deltapsi)) %>%
  left_join(select(whole_ml240_vcp_r191q_vs_ctrl.voila, lsv_junction_id, `VCP^R191Q\nML240` = deltapsi)) %>%
  left_join(select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, `VCP^R191Q\nUntreated` = deltapsi)) %>%
  left_join(select(whole_ml240_vcp_iso_vs_ctrl.voila, lsv_junction_id, `VCPki^R191Q\nML240` = deltapsi)) %>%
    left_join(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, `VCPki^R191Q\nUntreated` = deltapsi)) %>%
  drop_na() %>%
  column_to_rownames("lsv_junction_id")
cormat <- round(cor(untreated_ml240.nuc_vs_cyt.als_vs_ctrl_deltapsi),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
melted_cormat %<>% filter(Var1=="TARDBP^G298S\nML240" & Var2 =="TARDBP^G298S\nUntreated" | Var1=="VCP^R155C\nML240" & Var2 =="VCP^R155C\nUntreated" | Var1=="VCP^R191Q\nML240" & Var2 =="VCP^R191Q\nUntreated" | Var1=="VCPki^R191Q\nML240" & Var2 =="VCPki^R191Q\nUntreated")
voila.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) + labs(x = "Untreated", y = "ML240") +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
voila.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap

# How many of the untreated redistributed transcripts are not redistributed with ML240?
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0) %>% pull(lsv_junction_id)
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts %>% str #153
whole_ml240_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0, lsv_junction_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts) # 3
150/153 # 98.0
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts = whole_untreated_tardbp_vs_ctrl.voila %>% filter(padj <= 0.05, deltapsi <= 0) %>% pull(lsv_junction_id) 
untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts %>% str #228
whole_ml240_tardbp_vs_ctrl.voila %>% filter(padj <= 0.05, deltapsi <= 0, lsv_junction_id %in% untreated.nuc_vs_cyt.tardbp_vs_ctrl.redistrubted.transcripts) # 5
223/228 # #97.8%

untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0) %>% pull(lsv_junction_id) 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts %>% str # 181
whole_ml240_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts) # 7
174/181 # 96.1
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0) %>% pull(lsv_junction_id) 
untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts %>% str # 177
whole_ml240_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl.redistrubted.transcripts) # 5
172/177 # 97.2

untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0) %>% pull(lsv_junction_id) 
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts %>% str # 181
whole_ml240_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts) # 11
170/181 #93.9
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0) %>% pull(lsv_junction_id) 
untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts %>% str # 421
whole_ml240_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl.redistrubted.transcripts) # 25
396/421 #94.1

untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0) %>% pull(lsv_junction_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts %>% str # 190
whole_ml240_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi > 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts) # 8
182/190 # 95.8
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0) %>% pull(lsv_junction_id)
untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts %>% str # 379
whole_ml240_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2 == TRUE, deltapsi < 0, lsv_junction_id %in% untreated.nuc_vs_cyt.vcp_iso_vs_ctrl.redistrubted.transcripts) # 23
356/379 # 93.9%



# # GO of splicing events
# whole_ml240_tardbp_vs_ctrl.voila.gost = whole_ml240_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
# whole_ml240_tardbp_vs_ctrl.voila.gost_filt <- whole_ml240_tardbp_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
# table(whole_ml240_tardbp_vs_ctrl.voila.gost_filt$query) # 0
# paste('"',unique(whole_ml240_tardbp_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
# majiq_als_go_list = c(
# "cell-cell adhesion via plasma-membrane adhesion molecules",
# "homophilic cell adhesion via plasma membrane adhesion molecules",
# "membrane",
# "calcium ion binding",
# "ankyrin binding"
# )
# whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms <- whole_ml240_tardbp_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
# # whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms$term_name) 
# whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
# whole_ml240_tardbp_vs_ctrl.voila.gost_filt.terms.curated
# 
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost = whole_ml240_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt <- whole_ml240_vcp_r155c_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
# table(whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt$query) # 0
# paste('"',unique(whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
# majiq_als_go_list = c(
# "cortical microtubule",
# "cortical cytoskeleton",
# "membrane",
# "actin filament binding",
# "cortical microtubule plus-end",
# "cytoplasmic microtubule plus-end",
# "actin binding")
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms <- whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms$term_name) 
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
# whole_ml240_vcp_r155c_vs_ctrl.voila.gost_filt.terms.curated
# 
# whole_ml240_vcp_r191q_vs_ctrl.voila.gost = whole_ml240_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
# whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt <- whole_ml240_vcp_r191q_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
# table(whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt$query) # 0
# paste('"',unique(whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
# majiq_als_go_list = c(
# "cell-cell adhesion via plasma-membrane adhesion molecules",
# "homophilic cell adhesion via plasma membrane adhesion molecules",
# "intrinsic component of membrane",
# "calcium ion binding",
# "membrane")
# whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms <- whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
# # whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms$term_name) 
# whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
# whole_ml240_vcp_r191q_vs_ctrl.voila.gost_filt.terms.curated
# 
# whole_ml240_vcp_iso_vs_ctrl.voila.gost = whole_ml240_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.1 == TRUE) %>% distinct(gene_id) %>% pull(gene_id) %>% gost()
# whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt <- whole_ml240_vcp_iso_vs_ctrl.voila.gost$result %>% filter(!str_detect(source, "HPA|TF|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = "1")
# table(whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt$query) # 0
# paste('"',unique(whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt$term_name),'",', sep = "", collapse = '\n') %>% cat()
# majiq_als_go_list = c(
# "titin binding",
# "beta-Alanine metabolism",
# "Pantothenate and CoA biosynthesis")
# whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms <- whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt %>% filter(term_name %in% majiq_als_go_list)
# whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms$term_name = gsub("positive regulation of |regulation of ", "", whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms$term_name) 
# whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms.curated <- curated_profiles(whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms, cols = c("forestgreen")) + labs(title = "Splicing functional enrichment") + theme(plot.title=element_text(size=8), strip.text.y.right = element_blank())
# whole_ml240_vcp_iso_vs_ctrl.voila.gost_filt.terms.curated
# 
# 
# # Rescued splicing events GO terms
# # whole_untreated_als_vs_ctrl.voila == Table S7
# ml240_rescued.tardbp_vs_ctrl.voila.gost <- whole_untreated_als_vs_ctrl.voila %>% filter(tardbp.ml240 == "rescued") %>%  group_split(tardbp.dpsi > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
# ml240_rescued.tardbp_vs_ctrl.voila.gost_filt <- ml240_rescued.tardbp_vs_ctrl.voila.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
# table(ml240_rescued.tardbp_vs_ctrl.voila.gost_filt$query)
# # Cytoplasm   Nuclear 
# #         1         1 
# ml240_rescued.tardbp_vs_ctrl.voila.gprofiles_nuclear <- ml240_rescued.tardbp_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")
# ml240_rescued.tardbp_vs_ctrl.voila.gprofiles_cytoplasm <- ml240_rescued.tardbp_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")
# paste('"',unique(ml240_rescued.tardbp_vs_ctrl.voila.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
# nuclear_list <- c(
# "Focal adhesion"
# )
# ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_nuclear <- ml240_rescued.tardbp_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
# ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
# paste('"',unique(ml240_rescued.tardbp_vs_ctrl.voila.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
# cytoplasm_list <- c(
# "Regulation of actin cytoskeleton")
# ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_cytoplasm <- ml240_rescued.tardbp_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# # ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
# ml240_rescued.tardbp_vs_ctrl.voila.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.tardbp_vs_ctrl.voila.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# # ml240_rescued.tardbp_vs_ctrl.voila.go_curated
# 
# 
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gost <- whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r155c.ml240 == "rescued") %>%  group_split(vcp_r155c.dpsi > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt <- ml240_rescued.vcp_r155c_vs_ctrl.voila.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
# table(ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt$query)
# # Cytoplasm   Nuclear 
# #         4         4 
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gprofiles_nuclear <- ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gprofiles_cytoplasm <- ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")
# paste('"',unique(ml240_rescued.vcp_r155c_vs_ctrl.voila.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
# nuclear_list <- c(
# "asymmetric synapse",
# "postsynaptic density",
# "axon",
# "site of polarized growth"
# )
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_nuclear <- ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
# paste('"',unique(ml240_rescued.vcp_r155c_vs_ctrl.voila.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
# cytoplasm_list <- c(
# "neuron to neuron synapse",
# "postsynaptic specialization",
# "asymmetric synapse",
# "postsynaptic density")
# ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_cytoplasm <- ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# # ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
# ml240_rescued.vcp_r155c_vs_ctrl.voila.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_r155c_vs_ctrl.voila.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# # ml240_rescued.vcp_r155c_vs_ctrl.voila.go_curated
# 
# 
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gost <- whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r191q.ml240 == "rescued") %>%  group_split(vcp_r191q.dpsi > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt <- ml240_rescued.vcp_r191q_vs_ctrl.voila.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
# table(ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt$query)
# # Cytoplasm   Nuclear 
# #         1         3 
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gprofiles_nuclear <- ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gprofiles_cytoplasm <- ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")
# paste('"',unique(ml240_rescued.vcp_r191q_vs_ctrl.voila.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
# nuclear_list <- c(
# "GTPase regulator activity",
# "nucleoside-triphosphatase regulator activity",
# "GTPase activator activity"
# )
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_nuclear <- ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
# paste('"',unique(ml240_rescued.vcp_r191q_vs_ctrl.voila.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
# cytoplasm_list <- c(
# "recycling endosome membrane")
# ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_cytoplasm <- ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# # ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
# ml240_rescued.vcp_r191q_vs_ctrl.voila.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_r191q_vs_ctrl.voila.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# # ml240_rescued.vcp_r191q_vs_ctrl.voila.go_curated
# 
# 
# ml240_rescued.vcp_iso_vs_ctrl.voila.gost <- whole_untreated_als_vs_ctrl.voila %>% filter(vcp_ki.ml240 == "rescued") %>%  group_split(vcp_ki.dpsi > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
# ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt <- ml240_rescued.vcp_iso_vs_ctrl.voila.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
# table(ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt$query)
# # 0
# ml240_rescued.vcp_iso_vs_ctrl.voila.gprofiles_nuclear <- ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")
# ml240_rescued.vcp_iso_vs_ctrl.voila.gprofiles_cytoplasm <- ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")
# paste('"',unique(ml240_rescued.vcp_iso_vs_ctrl.voila.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
# nuclear_list <- c(
# "Focal adhesion"
# )
# ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_nuclear <- ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
# ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
# paste('"',unique(ml240_rescued.vcp_iso_vs_ctrl.voila.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
# cytoplasm_list <- c(
# "Regulation of actin cytoskeleton")
# ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_cytoplasm <- ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# # ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
# ml240_rescued.vcp_iso_vs_ctrl.voila.go_curated <- plot_grid(curated_profiles(gprofiles = ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240_rescued.vcp_iso_vs_ctrl.voila.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# # ml240_rescued.vcp_iso_vs_ctrl.voila.go_curated


whole_untreated_als_vs_ctrl.voila %>% filter(tardbp.ml240 == "rescued", category != "") %>% select(gene_name, category)
whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r155c.ml240 == "rescued", category != "") %>% select(gene_name, category)
whole_untreated_als_vs_ctrl.voila %>% filter(vcp_r191q.ml240 == "rescued", category != "") %>% select(gene_name, category)
whole_untreated_als_vs_ctrl.voila %>% filter(vcp_ki.ml240 == "rescued", category != "") %>% select(gene_name, category)



## Modulizer
untreated_ml240_mutants_vs_ctrl.modulizer.bind = bind_rows(
  mutate(rename(whole_untreated_tardbp_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_tardbp_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_tardbp_untreated_whole_probability_changing), mutation = "TARDBP^G298S", treatment = "Untreated"), 
  mutate(rename(whole_ml240_tardbp_vs_ctrl.modulizer, dpsi = ctrl_ml240_whole_tardbp_ml240_whole_median_dpsi, probability_changing = ctrl_ml240_whole_tardbp_ml240_whole_probability_changing), mutation = "TARDBP^G298S", treatment = "ML240"), 
  mutate(rename(whole_untreated_vcp_r155c_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_r155c_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_r155c_untreated_whole_probability_changing), mutation = "VCP^R155C", treatment = "Untreated"),
  mutate(rename(whole_ml240_vcp_r155c_vs_ctrl.modulizer, dpsi = ctrl_ml240_whole_vcp_r155c_ml240_whole_median_dpsi, probability_changing = ctrl_ml240_whole_vcp_r155c_ml240_whole_probability_changing), mutation = "VCP^R155C", treatment = "ML240"),
  mutate(rename(whole_untreated_vcp_r191q_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_r191q_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_r191q_untreated_whole_probability_changing), mutation = "VCP^R191Q", treatment = "Untreated"), 
  mutate(rename(whole_ml240_vcp_r191q_vs_ctrl.modulizer, dpsi = ctrl_ml240_whole_vcp_r191q_ml240_whole_median_dpsi, probability_changing = ctrl_ml240_whole_vcp_r191q_ml240_whole_probability_changing), mutation = "VCP^R191Q", treatment = "ML240"), 
  mutate(rename(whole_untreated_vcp_iso_vs_ctrl.modulizer, dpsi = ctrl_untreated_whole_vcp_iso_untreated_whole_median_dpsi, probability_changing = ctrl_untreated_whole_vcp_iso_untreated_whole_probability_changing), mutation = "VCPki^R191Q", treatment = "Untreated"),
  mutate(rename(whole_ml240_vcp_iso_vs_ctrl.modulizer, dpsi = ctrl_ml240_whole_vcp_iso_ml240_whole_median_dpsi, probability_changing = ctrl_ml240_whole_vcp_iso_ml240_whole_probability_changing), mutation = "VCPki^R191Q", treatment = "ML240")) %>% 
  mutate(mutation = factor(mutation, levels = c("TARDBP^G298S","VCP^R155C", "VCP^R191Q","VCPki^R191Q")), modulizer_simplified = gsub("_"," ",modulizer_simplified)) %>%
  filter(probability_changing > 0.9, abs(dpsi) > 0.2)

untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.count = untreated_ml240_mutants_vs_ctrl.modulizer.bind %>% count(mutation, treatment, modulizer_simplified) %>% 
  mutate(modulizer_simplified = fct_reorder(modulizer_simplified, n, .desc = TRUE)) %>%  
  group_by(mutation, treatment) %>% add_tally(n) %>% mutate(pct = n/nn *100, pct.label = case_when(modulizer_simplified == "intron retention" ~ paste0(as.character(n)," (", as.character(round(pct,1)),"%)"), TRUE ~ ""), ypos = cumsum(pct) - 0.5*pct) %>% ungroup()
untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.count %>% print(n=Inf)

untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.barplot = ggbarplot(untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.count, x="mutation", y="pct", fill="modulizer_simplified", position = position_fill(), xlab = "", label = untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.count$pct.label, lab.size = 3, lab.vjust = 0.5, lab.hjust = 1.5)  +
  scale_fill_npg() +
  theme_classic() +  theme(panel.grid = element_blank(), legend.title = element_blank(), legend.position = "top", legend.spacing.x = unit(0.2, "cm"), legend.text=element_text(size=8), legend.key.size = unit(0.4, "cm"), axis.text=element_text(size=8), axis.title=element_text(size=8), strip.text.y = element_text(angle = 0)) +
  guides(fill = guide_legend(reverse = TRUE, nrow = 2)) +
  facet_grid(treatment ~ .,scales="free_y", labeller = "label_parsed") +
  coord_flip() + ylab(expression(Splicing~type~proportions)) + scale_x_discrete(labels = scales::parse_format()) 
untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.barplot


ml240_splicing.merged = plot_grid(
  untreated_ml240.nuc_vs_cyt.als_vs_ctrl.voila.upset,
  plot_grid(voila.untreated_ml240.nuc_vs_cyt.als_vs_ctrl_correlation_ggheatmap, untreated_ml240_untreated_mutants_vs_ctrl.modulizer.bind.barplot, ncol = 2, labels = letters[2:3], rel_widths = c(1,3)),
  nrow = 2, rel_heights = c(0.8,1.2), labels = c("a",""))
# ml240_splicing.merged
ggsave(ml240_splicing.merged, filename = here(proj_path, "splicing/majiq/ml240_splicing.merged.png"), height = 7, width = 12)


```



# Table S13 ML240 Splicing

```{r}
untreated.tardbp.up = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi>0) %>% pull(lsv_junction_id)
untreated.tardbp.cyt = whole_untreated_tardbp_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi<0) %>% pull(lsv_junction_id)
untreated.vcp_r155c.up = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi>0) %>% pull(lsv_junction_id)
untreated.vcp_r155c.cyt = whole_untreated_vcp_r155c_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi<0) %>% pull(lsv_junction_id)
untreated.vcp_r191q.up = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi>0) %>% pull(lsv_junction_id)
untreated.vcp_r191q.cyt = whole_untreated_vcp_r191q_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi<0) %>% pull(lsv_junction_id)
untreated.vcp_ki.up = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi>0) %>% pull(lsv_junction_id)
untreated.vcp_ki.cyt = whole_untreated_vcp_iso_vs_ctrl.voila %>% filter(changing_dpsi0.2==TRUE,deltapsi<0) %>% pull(lsv_junction_id)

ml240.mutants_vs_ctrl.res.voila.join = mutate(select(whole_untreated_tardbp_vs_ctrl.voila, lsv_junction_id, gene_name, tardbp.deltapsi = deltapsi, tardbp.probability_changing = probability_changing, tardbp.deltapsi = deltapsi, tardbp.changing_dpsi0.2 = changing_dpsi0.2), 
                                              tardbp.remain = case_when(tardbp.changing_dpsi0.2==TRUE & tardbp.deltapsi > 0 & lsv_junction_id %in% untreated.tardbp.nuc ~ "remains up", tardbp.changing_dpsi0.2==TRUE & tardbp.deltapsi<0 & lsv_junction_id %in% untreated.tardbp.cyt ~ "remains down", TRUE ~ "")) %>% 
  left_join(mutate(select(whole_untreated_vcp_r155c_vs_ctrl.voila, lsv_junction_id, gene_name, vcp_r155c.deltapsi = deltapsi, vcp_r155c.probability_changing = probability_changing, vcp_r155c.deltapsi = deltapsi, vcp_r155c.changing_dpsi0.2 = changing_dpsi0.2),
            vcp_r155c.remain = case_when(vcp_r155c.changing_dpsi0.2==TRUE & vcp_r155c.deltapsi > 0 & lsv_junction_id %in% untreated.vcp_r155c.nuc ~ "remains up", vcp_r155c.changing_dpsi0.2==TRUE & vcp_r155c.deltapsi<0 & lsv_junction_id %in% untreated.vcp_r155c.cyt ~ "remains down", TRUE ~ ""))) %>%
  left_join(mutate(select(whole_untreated_vcp_r191q_vs_ctrl.voila, lsv_junction_id, gene_name, vcp_r191q.deltapsi = deltapsi, vcp_r191q.probability_changing = probability_changing, vcp_r191q.deltapsi = deltapsi, vcp_r191q.changing_dpsi0.2 = changing_dpsi0.2),
            vcp_r191q.remain = case_when(vcp_r191q.changing_dpsi0.2==TRUE & vcp_r191q.deltapsi > 0 & lsv_junction_id %in% untreated.vcp_r191q.nuc ~ "remains up", vcp_r191q.changing_dpsi0.2==TRUE & vcp_r191q.deltapsi<0 & lsv_junction_id %in% untreated.vcp_r191q.cyt ~ "remains down", TRUE ~ ""))) %>%
  left_join(mutate(select(whole_untreated_vcp_iso_vs_ctrl.voila, lsv_junction_id, gene_name, vcp_ki.deltapsi = deltapsi, vcp_ki.probability_changing = probability_changing, vcp_ki.deltapsi = deltapsi, vcp_ki.changing_dpsi0.2 = changing_dpsi0.2),
            vcp_iso.remain = case_when(vcp_ki.changing_dpsi0.2==TRUE & vcp_ki.deltapsi > 0 & lsv_junction_id %in% untreated.vcp_ki.nuc ~ "remains up", vcp_ki.changing_dpsi0.2==TRUE & vcp_ki.deltapsi<0 & lsv_junction_id %in% untreated.vcp_ki.cyt ~ "remains down", TRUE ~ ""))) %>% 
  filter(tardbp.changing_dpsi0.2==TRUE | vcp_r155c.changing_dpsi0.2==TRUE | vcp_r191q.changing_dpsi0.2==TRUE | vcp_ki.changing_dpsi0.2==TRUE) %>%
  arrange(-( abs(tardbp.deltapsi) + abs(vcp_r155c.deltapsi) + abs(vcp_r191q.deltapsi) + abs(vcp_ki.deltapsi))) %>%
  mutate(overlapping = case_when((tardbp.changing_dpsi0.2==TRUE & vcp_r155c.changing_dpsi0.2==TRUE & vcp_r191q.changing_dpsi0.2==TRUE & vcp_ki.changing_dpsi0.2==TRUE & tardbp.deltapsi > 0 & vcp_r155c.deltapsi > 0 &  vcp_r191q.deltapsi > 0 & vcp_ki.deltapsi > 0) ~ "Nuclear", 
                             (tardbp.changing_dpsi0.2==TRUE & vcp_r155c.changing_dpsi0.2==TRUE & vcp_r191q.changing_dpsi0.2==TRUE & vcp_ki.changing_dpsi0.2==TRUE & tardbp.deltapsi < 0 & vcp_r155c.deltapsi < 0 & vcp_r191q.deltapsi < 0 & vcp_ki.deltapsi < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ "")) #%>%  select(-contains("changing_dpsi"))
ml240.mutants_vs_ctrl.res.voila.join
write_csv(ml240.mutants_vs_ctrl.res.voila.join, here(proj_path,"splicing/majiq/majiq.ml240.mutants_vs_ctrl.res.voila.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(ml240.mutants_vs_ctrl.res.voila.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S12 ML240 Splicing")



ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join = select(ml240.ctrl_nuc_vs_cyt.dep$res, gene_name, ctrl.deltapsi = deltapsi, ctrl.padj = padj, ctrl.lfc = log2FoldChange) %>% 
  left_join(select(ml240.tardbp_nuc_vs_cyt.dep$res, gene_name, tardbp.deltapsi = deltapsi, tardbp.padj = padj, tardbp.lfc = log2FoldChange)) %>% 
  left_join(select(ml240.vcp_r155c_nuc_vs_cyt.dep$res, gene_name, vcp_r155c.deltapsi = deltapsi, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange)) %>%
  left_join(select(ml240.vcp_iso_nuc_vs_cyt.dep$res, gene_name, vcp_iso.deltapsi = deltapsi, vcp_iso.padj = padj, vcp_iso.lfc = log2FoldChange))


ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt = ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join %>% filter(is.na(tardbp.deltapsi) == FALSE & is.na(vcp_r155c.deltapsi) == FALSE & is.na(vcp_iso.deltapsi) == FALSE) %>%
  filter(ctrl.changing_dpsi0.2==TRUE | tardbp.changing_dpsi0.2==TRUE | vcp_r155c.changing_dpsi0.2==TRUE | vcp_iso.changing_dpsi0.2==TRUE) %>%
  arrange(-(abs(ctrl.deltapsi) + abs(tardbp.deltapsi) + abs(vcp_r155c.deltapsi) + abs(vcp_iso.deltapsi))) %>%
  rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_iso")), ~ str_replace_all(., "vcp_iso", "vcp_ki"))
ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt %>% filter(gene_name == "TARDBP")

```


# Fig S19 ML240 Protein mislocalisation

## TARDBP Volcano  & GO

multi-factor design

```{r}
ml240.tardbp_vs_ctrl.nuc_vs_cyt.rbp.labels <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.tardbp_vs_ctrl.nuc_vs_cyt.als.labels <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(ml240.tardbp_vs_ctrl.nuc_vs_cyt.als.labels, ml240.tardbp_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 7, ymax = 9, dpi = 300, distinct_labels = TRUE, title = "TARDBP mutant") + 
  xlab(expression(TARDBP~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 7*0.25, xend = 7*0.95, y= 9*0.88, yend= 9 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -7*0.25, xend = -7*0.95, y= 9*0.88, yend= 9*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 7*0.6, y = 9*0.93, label = "Nuclear", size = 3) + annotate("text", x = -7*0.6, y = 9*0.93, label = "Cytoplasm", size = 3)
# ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano

### GO mislocalised
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#        19        24 
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"icosanoid binding",
"icosatetraenoic acid binding",
"arachidonic acid binding",
"secretory vesicle",
"Toll-like receptor 4 binding",
"cytoskeletal protein binding"
)
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"spliceosomal complex",
"Metabolism of RNA",
"protein binding",
"mRNA Splicing",
"regulation of RNA splicing")
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" \\(NMD\\)", "", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated



# Rescued transcripts GO terms
# untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join == Table S8
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(tardbp.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(tardbp.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt <- dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#       316       125 
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"cytoplasmic translation",
"Nonsense-Mediated Decay (NMD)",
"intracellular anatomical structure",
"Peptide chain elongation",
"ncRNA metabolic process",
"cytosolic ribosome",
"ncRNA processing",
"mRNA binding",
"Metabolism of RNA",
"RNA binding"
)
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"protein transport",
"cellular localization",
"intracellular transport",
"translation",
"protein metabolic process",
"mitochondrion",
"actin cytoskeleton",
"cytoskeletal protein binding",
# "regulation of cellular component organization",
"protein-containing complex",
"intracellular anatomical structure",
"protein binding")
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated


```


## VCP R155C mutant volcano & GO

multi-factor design

```{r}
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.rbp.labels <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.als.labels <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.als.labels, ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 8, dpi = 300, distinct_labels = TRUE, title = "VCP R155C mutant") + 
  xlab(expression(VCP~R155C~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 8*0.88, yend= 8*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 8*0.88, yend= 8*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 8*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 8*0.93, label = "Cytoplasm", size = 3)
# ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano

### GO mislocalised
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#         9         6 
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"organelle envelope",
"primary lysosome",
"azurophil granule",
"GTPase activating protein binding"
)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"protein binding",
"membrane-bounded organelle",
"cytoplasm")
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub("COPII-coated ER to Golgi transport vesicle", "ER to Golgi transport", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, rel_heights = c(1,1), align = "v") 
# ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated


ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_CYTOSOLIC_PART)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_AXON)
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 89
ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% ALS_genes) #14


# Rescued transcripts GO terms
# untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join == Table S8
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(vcp_r155c.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_r155c.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt <- dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#        16         4 
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"intracellular anatomical structure",
"cytoplasm",
"Dsl1/NZR complex",
"Biosynthesis of amino acids"
)
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"catalytic activity",
"protein binding",
"intracellular anatomical structure",
"mitochondrial matrix",
"histone binding",
"nucleoplasm",
"cytoplasm")
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated

```


## VCP R191Q mutant volcano & GO

multi-factor design

```{r}
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.rbp.labels <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.als.labels <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.als.labels, ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 8, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q mutant") + 
  xlab(expression(VCP~R191Q~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 8*0.88, yend= 8*0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 8*0.88, yend= 8*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 8*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 8*0.93, label = "Cytoplasm", size = 3)
# ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano

### GO mislocalised
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       158        68 
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"lysosome",
"Glycolysis / Gluconeogenesis",
"protein binding",
"small molecule binding",
"Metabolic pathways",
"carbohydrate derivative metabolic process",
"nucleotide metabolic process",
"intracellular anatomical structure",
"mitochondrial matrix",
"carboxylic acid metabolic process",
"oxoacid metabolic process"
)
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"Nonsense-Mediated Decay (NMD)",
"RNA binding",
"cytoskeleton organization",
"Axon guidance",
"intracellular anatomical structure",
"protein binding",
"protein-containing complex",
"mRNA Splicing",
"focal adhesion",
"spliceosomal complex",
"Metabolism of RNA")
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_filt_combined %<>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated

ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_CYTOSOLIC_PART)
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, log2FoldChange > 0, gene_name %in% go.pathways.msigdb$GO_AXON)
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 89
ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% ALS_genes) #14



# Rescued transcripts GO terms
# untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join == Table S8
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(vcp_r191q.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_r191q.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt <- dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#       655       184 
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"mRNA transport",
"DNA repair",
"protein-containing complex",
"mRNA metabolic process",
"ribonucleoprotein complex",
"Metabolism of RNA",
"RNA binding",
"mRNA Splicing",
"spliceosomal complex"
)
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"RNA binding",
"protein transport",
"synapse",
"axon",
"cytoskeleton",
"cytoskeletal protein binding",
"protein localization",
"Axon guidance",
"cellular localization",
"cytoplasmic translation",
"protein binding")
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
# dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("positive regulation of ", "", term_name), term_name = gsub("and","&",term_name))
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated

```



## VCP knock-in volcano & GO

multi-factor design

```{r}
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.rbp.labels <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.als.labels <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano <- volcano_plot(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_labels = c(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.als.labels, ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.rbp.labels), arrow_labels = FALSE, xmax = 6, ymax = 8, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q knock-in") + 
  xlab(expression(VCPki~R191~vs~CTRL~log[2]~FC~Nucleus/Cytoplasm)) +
       geom_segment(aes(x = 6*0.25, xend = 6*0.95, y= 8*0.88, yend= 8 * 0.88), arrow=arrow(length=unit(0.3,"cm"))) + geom_segment(aes(x = -6*0.25, xend = -6*0.95, y= 8*0.88, yend= 8*0.88),arrow=arrow(length=unit(0.3,"cm"))) +
      annotate("text", x = 6*0.6, y = 8*0.93, label = "Nuclear", size = 3) + annotate("text", x = -6*0.6, y = 8*0.93, label = "Cytoplasm", size = 3)
# ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano


### GO mislocalised
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Nuclear", padj < 0.05 & log2FoldChange < 0 ~ "Cytoplasm", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"))
table(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt$query)
# Cytoplasm   Nuclear 
#       257        53 
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"protein binding",
"ion binding",
"ribonucleotide binding",
"NADH metabolic process",
"oxidoreductase activity",
"Metabolic pathways",
"Cholesterol biosynthesis",
"nucleotide binding",
"small molecule binding",
"catalytic activity"
)
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
paste('"',unique(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"mRNA transport",
"RNA transport",
"histone modification",
"Transport of Mature Transcript to Cytoplasm",
"Spliceosome",
"RNA localization",
"intracellular anatomical structure",
"protein binding",
"spliceosomal complex",
"mRNA binding",
"mRNA Splicing",
"nucleic acid binding",
"Metabolism of RNA",
"RNA binding")
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm <- ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" exit site", "", term_name), term_name = gsub("endoplasmic reticulum to Golgi vesicle-mediated transport","ER to Golgi transport",term_name))
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated


# Rescued transcripts GO terms
# untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join == Table S8
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost <- untreated.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join %>% filter(vcp_ki.ml240 == "rescued") %>% left_join(gene2ens) %>%  group_split(vcp_ki.stat > 0) %>% map("gene_id") %>% gost(sources = c("GO", "KEGG", "REAC", "CORUM"))
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt <- dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost$result %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Cytoplasm", query == "query_2" ~ "Nuclear"), term_name = as.factor(term_name))
table(dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt$query)
# Cytoplasm   Nuclear 
#       409        55 
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear <- dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm <- dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")
paste('"',unique(dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
nuclear_list <- c(
"Metabolism of RNA",
"nuclear import signal receptor activity",
"Mismatch repair",
"nucleotide binding",
"mitochondrion",
"protein binding",
"endoplasmic reticulum exit site",
"intracellular anatomical structure"
)
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear <- dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Nuclear")  %>% filter(term_name %in% nuclear_list)
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear %<>% mutate(term_name = gsub("regulation of ", "", term_name))
paste('"',unique(dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
cytoplasm_list <- c(
"Metabolism of RNA",
"vesicle-mediated transport",
"mitochondrion",
"presynapse",
"Cellular responses to stress",
"protein-containing complex",
"cytoplasmic translation",
"Axon guidance",
"synapse",
"protein binding")
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm <- dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt %>% filter(query == "Cytoplasm")  %>% filter(term_name %in% cytoplasm_list)
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm %<>% mutate(term_name = gsub("pre", "", term_name))#, term_name = gsub("and","&",term_name))
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated <- plot_grid(curated_profiles(gprofiles = dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_nuclear) + xlab(""), curated_profiles(gprofiles = dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated


```


Correlation heatmap

```{r}
# heatmap of correlation coefficients
ml240.als_vs_ctrl.nuc_vs_cyt.dep_stat = select(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `TARDBP^G298S` = stat) %>% 
  left_join(select(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R155C` = stat)) %>%
  left_join(select(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCP^R191Q` = stat)) %>%
  left_join(select(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, `VCPki^R191Q` = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_name")
cormat <- round(cor(ml240.als_vs_ctrl.nuc_vs_cyt.dep_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
ml240.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
ml240.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap

```

## Mislocalised protein numbers

```{r}
nrow(filter(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05)) # 577
nrow(filter(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05)) # 111
nrow(filter(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05)) #169
nrow(filter(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#116
nrow(filter(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#867
nrow(filter(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#400
nrow(filter(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#639
nrow(filter(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#433
nrow(filter(untreated.als_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#639
nrow(filter(ml240.als_vs_ctrl.nuc_vs_cyt.dep$res, padj < 0.05))#433

mutant_dep_ml240_rescue = tibble(
  mutation = c("TARDBP mutant", "TARDBP mutant", "VCP R155C mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCP R191Q mutant", "VCPki R191Q", "VCPki R191Q"),
  treatment = factor(c("Untreated", "ML240", "Untreated", "ML240", "Untreated", "ML240", "Untreated", "ML240"), levels = c("Untreated", "ML240")),
  n = c(577, 111,169, 116, 867, 400, 639, 433))

mutant_dep_ml240_rescue

mutant_dep_ml240_rescue.plot <- ggplot(mutant_dep_ml240_rescue, aes(x=treatment, y = n, fill = treatment)) +
    geom_bar(stat = "identity") + scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
  # scale_fill_gradient2(low = "darkblue", high = "indianred", mid = "whitesmoke", midpoint = 0) + #scale_fill_npg() +
  theme_oz(xaxis_arrow = FALSE) + theme(axis.text.x = element_text(angle = 30, hjust = 1), plot.title = element_text(hjust = 0.5), legend.position = "none") +
  geom_hline(yintercept = 0, linetype = 3) +
  labs(x = "", y = "Mislocalised proteins") + #ylim(c(-12,12)) +
  facet_wrap(~mutation, scales = "free", nrow = 1)
mutant_dep_ml240_rescue.plot

```

## RNAct

```{r}
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.tardbp.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r155c.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r191q.nuclear_cytoplasm_protein_transcript.rds"))
load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_iso.nuclear_cytoplasm_protein_transcript.rds"))

RNAct_database.tardbp.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.tardbp.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"),  mutate(RNAct_database.tardbp.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_r155c.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r155c.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_r191q.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r191q.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
RNAct_database.vcp_iso.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_iso.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))

RNAct_database.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.protein_transcript.bind, mutation = "TARDBP mutant"), mutate(RNAct_database.vcp_r155c.protein_transcript.bind, mutation = "VCP R155C mutant"), mutate(RNAct_database.vcp_r191q.protein_transcript.bind, mutation = "VCP R191Q mutant"), mutate(RNAct_database.vcp_iso.protein_transcript.bind, mutation = "VCPki R191Q")) %>% mutate(mutation = factor(mutation, levels = c("TARDBP mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCPki R191Q")), mislocalisation = factor(mislocalisation, levels = c("Non Tx\nNon Pr", "Nuc Tx\nNuc Pr", "Cyt Tx\nCyt Pr")))

RNAct_database.tardbp.protein_transcript.bind %>% my_summarise(catrapid_score_max_normalised)

# Mislocalisation Nuclear vs Cytoplasm vs None
rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct_database.protein_transcript.bind, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(5,25), facet_by = "mutation", facet_ncol = 4, plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = rep(c(22, 24, 22.5),4), 
                                                               ylabel = expression(RNA~protein~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2","forestgreen","gold2"), dpi = 300, arrow_labels = FALSE)
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin
# facet            .y.   group1           group2               n1     n2 statistic      df         p     p.adj p.adj.signif y.position groups        xmin  xmax
#    <fct>            <chr> <chr>            <chr>             <int>  <int>     <dbl>   <dbl>     <dbl>     <dbl> <chr>             <dbl> <named list> <dbl> <dbl>
#  1 TARDBP mutant    lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  10836   -40.9    15747. 0         0         ****               22   <chr [2]>        1     2
#  2 TARDBP mutant    lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000  50355   -69.2   117396. 0         0         ****               24   <chr [2]>        1     3
#  3 TARDBP mutant    lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  10836  50355    -3.55   17804. 3.92e-  4 3.92e-  4 ***                22.5 <chr [2]>        2     3
#  4 VCP R155C mutant lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000   3648   -22.2     4026. 5.68e-103 1.14e-102 ****               22   <chr [2]>        1     2
#  5 VCP R155C mutant lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000   8455   -29.7    10227. 8.76e-186 2.63e-185 ****               24   <chr [2]>        1     3
#  6 VCP R155C mutant lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"   3648   8455     0.502   7441. 6.16e-  1 6.16e-  1 ns                 22.5 <chr [2]>        2     3
#  7 VCP R191Q mutant lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  10560   -30.5    14468. 6.19e-198 1.24e-197 ****               22   <chr [2]>        1     2
#  8 VCP R191Q mutant lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000 157528   -80.8   194130. 0         0         ****               24   <chr [2]>        1     3
#  9 VCP R191Q mutant lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  10560 157528    -7.76   12450. 9   e- 15 9   e- 15 ****               22.5 <chr [2]>        2     3
# 10 VCPki R191Q      lfc   "Non Tx\nNon Pr" "Nuc Tx\nNuc Pr" 100000  20651   -64.6    37833. 0         0         ****               22   <chr [2]>        1     2
# 11 VCPki R191Q      lfc   "Non Tx\nNon Pr" "Cyt Tx\nCyt Pr" 100000  90792   -70.3   190684. 0         0         ****               24   <chr [2]>        1     3
# 12 VCPki R191Q      lfc   "Nuc Tx\nNuc Pr" "Cyt Tx\nCyt Pr"  20651  90792    14.8    35299. 1.99e- 49 1.99e- 49 ****               22.5 <chr [2]>        2     3



# # other groups of non-redistributed & non-mislocalised for comparison:
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.tardbp.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r155c.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_r191q.nuclear_cytoplasm_protein_transcript.rds"))
# load(here(proj_path, "expression/rna-rbp-interaction/rnact/RNAct_database.vcp_iso.nuclear_cytoplasm_protein_transcript.rds"))
# 
# RNAct_database.tardbp.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.tardbp.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.tardbp.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.tardbp.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.tardbp.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"),  mutate(RNAct_database.tardbp.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"),  mutate(RNAct_database.tardbp.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_r155c.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r155c.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_r155c.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r155c.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r155c.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_r191q.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_r191q.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_r191q.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_r191q.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_r191q.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# RNAct_database.vcp_iso.protein_transcript.bind = bind_rows(mutate(RNAct_database.vcp_iso.cytoplasm_protein.nonredistributed, mislocalisation = "Non Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nuclear_protein.nonredistributed, mislocalisation = "Non Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein.cytoplasm_transcript, mislocalisation = "Cyt Tx\nNon Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein.nuclear_transcript, mislocalisation = "Nuc Tx\nNon Pr"), mutate(RNAct_database.vcp_iso.nuclear_protein_transcript, mislocalisation = "Nuc Tx\nNuc Pr"), mutate(RNAct_database.vcp_iso.cytoplasm_protein_transcript, mislocalisation = "Cyt Tx\nCyt Pr"), mutate(RNAct_database.vcp_iso.nonmislocalised_protein_transcript, mislocalisation = "Non Tx\nNon Pr"))
# 
# RNAct_database.protein_transcript.bind = bind_rows(mutate(RNAct_database.tardbp.protein_transcript.bind, mutation = "TARDBP mutant"), mutate(RNAct_database.vcp_r155c.protein_transcript.bind, mutation = "VCP R155C mutant"), mutate(RNAct_database.vcp_r191q.protein_transcript.bind, mutation = "VCP R191Q mutant"), mutate(RNAct_database.vcp_iso.protein_transcript.bind, mutation = "VCPki R191Q")) %>% 
#   mutate(mutation = factor(mutation, levels = c("TARDBP mutant", "VCP R155C mutant", "VCP R191Q mutant", "VCPki R191Q")), 
#          mislocalisation = factor(mislocalisation, levels = c("Non Tx\nNon Pr", "Non Tx\nCyt Pr", "Non Tx\nNuc Pr", "Cyt Tx\nNon Pr", "Nuc Tx\nNon Pr", "Nuc Tx\nNuc Pr", "Cyt Tx\nCyt Pr")))
# 
# RNAct_database.protein_transcript.bind %>% my_summarise(catrapid_score_max_normalised, mislocalisation, mutation)
# 
# # Mislocalisation Nuclear vs Cytoplasm vs Non
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin = violin_plot(RNAct_database.protein_transcript.bind, color_by = "mislocalisation", continuous = "catrapid_score_max_normalised", ylims = c(5,25), facet_by = "mutation", facet_ncol = 4, 
#                                                               plot_stats = TRUE, stats_test = "t_test", tip_length = 0.001, stat_label = "p.adj.signif", stat_ypos = 24,#rep(c(22, 24, 22.5),4), 
#                                                               ylabel = expression(RNA~protein~interaction~score), xlabel = "", cols = c("grey", "firebrick2", "dodgerblue2","forestgreen","gold2"), dpi = 30, arrow_labels = FALSE)
# rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin

```

## Merge

```{r}
ml240.vcp_tardbp_protein_redistribution.merged <- plot_grid(
  mutant_dep_ml240_rescue.plot,
  plot_grid(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep.volcano, ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep.volcano, ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep.volcano, ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep.volcano, ncol = 4, labels = letters[2:5]),
  plot_grid(ml240.tardbp_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.res.dep.go_curated, ncol = 4, labels = letters[6:9]), 
    plot_grid(dep.ml240_rescued.tardbp_vs_ctrl.nuc_vs_cyt.go_curated, dep.ml240_rescued.vcp_r155c_vs_ctrl.nuc_vs_cyt.go_curated, dep.ml240_rescued.vcp_r191q_vs_ctrl.nuc_vs_cyt.go_curated, dep.ml240_rescued.vcp_iso_vs_ctrl.nuc_vs_cyt.go_curated, ncol = 4, labels = letters[10:13]), 
  rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin,# plot_grid(ml240.als_vs_ctrl.nuc_vs_cyt.dep_correlation_ggheatmap, rnact.rna_protein.nuc_vs_cyt.als_vs_ctrl.violin, ncol = 2, rel_widths = c(0.8,2), labels = letters[13:14]),
  nrow = 5, rel_heights = c(0.8,1,1.2,1.2,1), labels = c("a","","","","n"))
# vcp_tardbp_protein_redistribution.merged
ggsave(ml240.vcp_tardbp_protein_redistribution.merged, filename = here(proj_path, "mass-spec/figures/ml240.vcp_tardbp_protein_redistribution.merged.png"), height = 14, width = 12)

```


# Table S14 ML240 Protein mislocalisation

```{r}
untreated.tardbp.nuc = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat>0) %>% pull(gene_name)
untreated.tardbp.cyt = untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat<0) %>% pull(gene_name)
untreated.vcp_r155c.nuc = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat>0) %>% pull(gene_name)
untreated.vcp_r155c.cyt = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat<0) %>% pull(gene_name)
untreated.vcp_r191q.nuc = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat>0) %>% pull(gene_name)
untreated.vcp_r191q.cyt = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat<0) %>% pull(gene_name)
untreated.vcp_ki.nuc = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat>0) %>% pull(gene_name)
untreated.vcp_ki.cyt = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj<0.05,stat<0) %>% pull(gene_name)


ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join = mutate(select(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean), 
                                                         tardbp.remain = case_when(tardbp.padj < 0.05 & tardbp.stat > 0 & gene_name %in% untreated.tardbp.nuc ~ "remains nuc", tardbp.padj<0.05 & tardbp.stat<0 & gene_name %in% untreated.tardbp.cyt ~ "remains cyto", TRUE ~ "")) %>% 
  left_join(mutate(select(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean),
            vcp_r155c.remain = case_when(vcp_r155c.padj < 0.05 & vcp_r155c.stat > 0 & gene_name %in% untreated.vcp_r155c.nuc ~ "remains nuc", vcp_r155c.padj<0.05 & vcp_r155c.stat<0 & gene_name %in% untreated.vcp_r155c.cyt ~ "remains cyto", TRUE ~ ""))) %>%
  left_join(mutate(select(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean),
            vcp_r191q.remain = case_when(vcp_r191q.padj < 0.05 & vcp_r191q.stat > 0 & gene_name %in% untreated.vcp_r191q.nuc ~ "remains nuc", vcp_r191q.padj<0.05 & vcp_r191q.stat<0 & gene_name %in% untreated.vcp_r191q.cyt ~ "remains cyto", TRUE ~ ""))) %>%
  left_join(mutate(select(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res, gene_name, vcp_ki.stat = stat, vcp_ki.padj = padj, vcp_ki.lfc = log2FoldChange, vcp_ki.baseMean = baseMean),
            vcp_iso.remain = case_when(vcp_ki.padj < 0.05 & vcp_ki.stat > 0 & gene_name %in% untreated.vcp_ki.nuc ~ "remains nuc", vcp_ki.padj<0.05 & vcp_ki.stat<0 & gene_name %in% untreated.vcp_ki.cyt ~ "remains cyto", TRUE ~ ""))) %>% 
  filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_r191q.stat) == FALSE & is.na(vcp_ki.stat) == FALSE) %>%
  # filter(tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_r191q.padj < 0.05 | vcp_ki.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_r191q.stat) + abs(vcp_ki.stat))) %>%
  mutate(overlapping = case_when((tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat > 0 & vcp_r155c.stat > 0 &  vcp_r191q.stat > 0 & vcp_ki.stat > 0) ~ "Nuclear", 
                             (tardbp.padj < 0.05 & vcp_r155c.padj < 0.05 & vcp_r191q.padj < 0.05 & vcp_ki.padj < 0.05 & tardbp.stat < 0 & vcp_r155c.stat < 0 & vcp_r191q.stat < 0 & vcp_ki.stat < 0) ~ "Cytoplasm", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ "")) 
ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join
write_csv(ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join, here(proj_path,"expression/ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(ml240.vcp_tardbp_vs_ctrl.nuc_vs_cyt.res.dep.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S12 ML240 protein misloc")



ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join = select(ml240.ctrl_nuc_vs_cyt.dep$res, gene_name, ctrl.stat = stat, ctrl.padj = padj, ctrl.lfc = log2FoldChange) %>% 
  left_join(select(ml240.tardbp_nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange)) %>% 
  left_join(select(ml240.vcp_r155c_nuc_vs_cyt.dep$res, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange)) %>%
  left_join(select(ml240.vcp_iso_nuc_vs_cyt.dep$res, gene_name, vcp_iso.stat = stat, vcp_iso.padj = padj, vcp_iso.lfc = log2FoldChange))


ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt = ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join %>% filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_iso.stat) == FALSE) %>%
  filter(ctrl.padj < 0.05 | tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_iso.padj < 0.05) %>%
  arrange(-(abs(ctrl.stat) + abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_iso.stat))) %>%
  rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_iso")), ~ str_replace_all(., "vcp_iso", "vcp_ki"))
ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt %>% filter(gene_name == "TARDBP")

```


# Fig S20 Co-immunoprecipitation

VCP protein pull down blot

```{r}
# import with magick 
vcp_pull_down_blot.png = magick::image_read(here(proj_path,"mass-spec/vcp_co_ip/vcp pull down validation.png"))
vcp_pull_down_blot.gg = ggdraw() + draw_image(vcp_pull_down_blot.png)
# vcp_pull_down_blot.gg

```

Comparison ML240 vs untreated

```{r}
load(file = here(proj_path, "scripts/dep_coip_mass_spec.nuc_cyt.ml240_vs_untreated.RData"))

coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat > 0) #165
coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat < 0) #337

coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat > 0) #0
coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat < 0) #0

coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat > 0) #78
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat < 0) #64

coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat > 0) #37
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat < 0) #13

coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat > 0) #23
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, stat < 0) #87


```

### CTRL Volcano & GO

```{r}
coip.ctrl.nuc_cyt.ml240_vs_untreated.rbp.labels <- coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.ctrl.nuc_cyt.ml240_vs_untreated.als.labels <- coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.ctrl.nuc_cyt.ml240_vs_untreated.dep.volcano <- volcano_plot(coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res, gene_labels = c(coip.ctrl.nuc_cyt.ml240_vs_untreated.als.labels, coip.ctrl.nuc_cyt.ml240_vs_untreated.rbp.labels), xmax = 5, ymax = 15, dpi = 300, distinct_labels = TRUE, title = "CTRL", numerator = "ML240", denomenator = "Untreated")
# coip.ctrl.nuc_cyt.ml240_vs_untreated.dep.volcano

### GO mislocalised
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost <- coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Up in ML240", padj < 0.05 & log2FoldChange < 0 ~ "Down in ML240", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt <- coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Down in ML240", query == "query_2" ~ "Up in ML240"))
table(coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt$query)
# Down in ML240   Up in ML240 
#           425            54 
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear <- coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm <- coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")
paste('"',unique(coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"ATPase complex",
"protein binding",
"Metabolism of RNA",
"protein-containing complex",
"mRNA Splicing",
"intracellular anatomical structure",
"spliceosomal complex",
"cytoplasm",
"nucleoplasm"
)
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear <- coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")  %>% filter(term_name %in% up_list)
paste('"',unique(coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
# "Nonsense-Mediated Decay (NMD)",
"ubiquitin protein ligase binding",
"nucleocytoplasmic transport",
"mRNA transport",
# "intracellular transport",
"peptide metabolic process",
"protein-containing complex",
# "cellular localization",
"Cellular responses to stress",
"VCP-NPL4-UFD1 AAA ATPase complex",
"ATP-dependent activity",
"ATP binding",
# "Axon guidance",
"Metabolism of RNA",
"mRNA binding",
"protein binding")
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm <- coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")  %>% filter(term_name %in% down_list)
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" \\(NMD\\)", "", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear, title = "CTRL") + xlab(""), curated_profiles(gprofiles = coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v", rel_heights = c(0.8,1)) 
# coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.go_curated


"VCP-NPL4-UFD1 AAA ATPase complex"
coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% gprofiler_database$`VCP-NPL4-UFD1 AAA ATPase complex`)
```


### TARDBP Volcano  & GO

multi-factor design

```{r}
coip.tardbp.nuc_cyt.ml240_vs_untreated.rbp.labels <- coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.tardbp.nuc_cyt.ml240_vs_untreated.als.labels <- coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.volcano <- volcano_plot(coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res, gene_labels = c(coip.tardbp.nuc_cyt.ml240_vs_untreated.als.labels, coip.tardbp.nuc_cyt.ml240_vs_untreated.rbp.labels), xmax = 5, ymax = 6, dpi = 300, distinct_labels = TRUE, title = "TARDBP mutant", numerator = "ML240", denomenator = "Untreated")
# coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.volcano

# ### GO mislocalised
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost <- coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "TARDBP: Up in ML240", padj < 0.05 & log2FoldChange < 0 ~ "TARDBP: Down in ML240", TRUE ~ "None")) %>% 
#   group_split(mislocalised) %>% map("gene_id") %>% gost()
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt <- coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "TARDBP: Down in ML240", query == "query_2" ~ "TARDBP: Up in ML240"))
# table(coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt$query)
# # Cytoplasm   Nuclear 
# #       323        90 
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear <- coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "TARDBP: Up in ML240")
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm <- coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "TARDBP: Down in ML240")
# paste('"',unique(coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
# up_list <- c(
# "mRNA binding",
# "cytoskeleton organization",
# "synapse",
# "dendrite",
# "nucleocytoplasmic transport",
# "cell junction",
# "neuron development",
# "somatodendritic compartment",
# "cytoskeletal protein binding",
# "axon"
# )
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear <- coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "TARDBP: Up in ML240")  %>% filter(term_name %in% up_list)
# paste('"',unique(coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
# down_list <- c(
# "ATP binding",
# "proteasome complex",
# "RNA binding",
# "DNA repair",
# "Metabolism of RNA",
# "small molecule binding",
# "chromosome organization",
# "Cell Cycle",
# "intracellular anatomical structure",
# "protein binding")
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm <- coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "TARDBP: Down in ML240")  %>% filter(term_name %in% down_list)
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" \\(NMD\\)", "", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear, title = "TARDBP") + xlab(""), curated_profiles(gprofiles = coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# coip.tardbp.nuc_cyt.ml240_vs_untreated.res.dep.go_curated

```


### VCP R155C mutant volcano & GO

multi-factor design

```{r}
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.rbp.labels <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.als.labels <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.volcano <- volcano_plot(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res, gene_labels = c(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.als.labels, coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.rbp.labels), xmax = 4, ymax = 6, dpi = 300, distinct_labels = TRUE, title = "VCP R155C mutant", numerator = "ML240", denomenator = "Untreated") 
# coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.volcano


### GO mislocalised
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Up in ML240", padj < 0.05 & log2FoldChange < 0 ~ "Down in ML240", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Down in ML240", query == "query_2" ~ "Up in ML240"))
table(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt$query)
# Down in ML240   Up in ML240 
#            78            11 
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")
paste('"',unique(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"intracellular anatomical structure",
"catalytic complex",
"endomembrane system",
"protein binding",
"nucleotide binding",
"small molecule binding",
"cytosol",
"cytoplasm"
)
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")  %>% filter(term_name %in% up_list)
paste('"',unique(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"nuclear export",
"RNA export from nucleus",
"protein-containing complex",
"Axon guidance",
"RNA localization",
"Nervous system development",
"spliceosomal complex",
"Metabolism of RNA",
"mRNA Splicing",
"RNA binding")
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")  %>% filter(term_name %in% down_list)
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub("COPII-coated ER to Golgi transport vesicle", "ER to Golgi transport", term_name))#, term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear, title = "VCP R155C mutant") + xlab(""), curated_profiles(gprofiles = coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, rel_heights = c(1,1), align = "v") 
# coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.go_curated

```


### VCP R191Q mutant volcano & GO

multi-factor design

```{r}
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.rbp.labels <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.als.labels <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.volcano <- volcano_plot(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res, gene_labels = c(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.als.labels, coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.rbp.labels), xmax = 4, ymax = 6, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q mutant", numerator = "ML240", denomenator = "Untreated") 
# coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.volcano

### GO mislocalised
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Up in ML240", padj < 0.05 & log2FoldChange < 0 ~ "Down in ML240", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Down in ML240", query == "query_2" ~ "Up in ML240"))
table(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt$query)
# Down in ML240   Up in ML240 
#             2             1 
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")
paste('"',unique(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"Neurotrophin signaling pathway"
)
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")  %>% filter(term_name %in% up_list)
paste('"',unique(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"myosin V complex",
"Glycolysis")
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")  %>% filter(term_name %in% down_list)
# coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_filt_combined %<>% mutate(term_name = gsub("miRNA regulation of DNA damage response", "miRNA regulation of DNA damage", term_name), term_name = gsub("and","&",term_name), term_name = gsub("DNA-binding transcription factor activity, RNA polymerase II-specific","transcription factor activity",term_name), term_name = gsub("cis-regulatory region sequence-specific DNA binding","DNA binding",term_name), term_name = gsub("cell ","",term_name))
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear, title = "VCP R191Q mutant") + xlab(""), curated_profiles(gprofiles = coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v") 
# coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.go_curated

coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% RNA_BINDING_PROTEINS) %>% print(n=Inf) # 5
coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% ALS_genes) #PRPH

```



### VCP knock-in volcano & GO

multi-factor design

```{r}
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.rbp.labels <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(RNA_BINDING_PROTEINS)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.als.labels <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05, gene_name %in% c(ALS_genes)) %>% top_n(n = 20, wt = abs(stat)) %>% pull(gene_name)
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.volcano <- volcano_plot(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res, gene_labels = c(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.als.labels, coip.vcp_iso.nuc_cyt.ml240_vs_untreated.rbp.labels), xmax = 4, ymax = 7, dpi = 300, distinct_labels = TRUE, title = "VCP R191Q knock-in", numerator = "ML240", denomenator = "Untreated") 
# coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.volcano


### GO mislocalised
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% left_join(gene2ens) %>% select(gene_name, gene_id, log2FoldChange, padj) %>% 
  mutate(mislocalised = case_when(padj < 0.05 & log2FoldChange > 0 ~ "Up in ML240", padj < 0.05 & log2FoldChange < 0 ~ "Down in ML240", TRUE ~ "None")) %>% 
  group_split(mislocalised) %>% map("gene_id") %>% gost()
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost$result %>% filter(str_detect(source, "KEGG|GO:BP|GO:MF|GO:CC|REAC|CORUM")) %>% arrange( -log10(p_value) ) %>% mutate(query = case_when(query == "query_1" ~ "Down in ML240", query == "query_2" ~ "Up in ML240"))
table(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt$query)
# Down in ML240   Up in ML240 
#            90             4 
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")
paste('"',unique(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_nuclear$term_name),'",', sep = "", collapse = '\n') %>% cat()
up_list <- c(
"cytoplasm",
"ProTalpha C2 complex",
"cytosol"
)
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Up in ML240")  %>% filter(term_name %in% up_list)
paste('"',unique(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gprofiles_cytoplasm$term_name),'",', sep = "", collapse = '\n') %>% cat()
down_list <- c(
"Transport of Mature Transcript to Cytoplasm",
"protein binding",
"spliceosomal complex",
"regulation of RNA splicing",
"mRNA binding",
"Metabolism of RNA",
"Nucleocytoplasmic transport",
"mRNA transport",
"nuclear pore",
"RNA localization",
"RNA binding")
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt %>% filter(query == "Down in ML240")  %>% filter(term_name %in% down_list)
# coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear %<>% mutate(term_name = gsub(" exit site", "", term_name), term_name = gsub("endoplasmic reticulum to Golgi vesicle-mediated transport","ER to Golgi transport",term_name))
# coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm %<>% mutate(term_name = gsub("regulation of ", "", term_name))
coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.go_curated <- plot_grid(curated_profiles(gprofiles = coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_nuclear, title = "VCP R155C knock-in") + xlab(""), curated_profiles(gprofiles = coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.gost_filt_cytoplasm, cols = "dodgerblue2"), nrow = 2, align = "v", rel_heights = c(0.8,1.2)) 
# coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.go_curated

```


## UpSet and correlation heatmap

```{r}
coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep = bind_rows(mutate(coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res, mutation = "CTRL"), mutate(coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res, mutation = "TARDBP G298S"), mutate(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res, mutation = "VCP R155C"), mutate(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res, mutation = "VCP R191Q"), mutate(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res, mutation = "VCP R191Q knock-in"))

coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.increased.upset = upset_plot(filter(coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep, stat > 0), overlap_by = "mutation", overlap_level = "gene_name", split_direction = FALSE, ymax = 170, order = "freq", order_group_list = c("CTRL", "TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Increased VCP binding", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))
coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.increased.upset

coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.decreased.upset = upset_plot(filter(coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep, stat < 0), overlap_by = "mutation", overlap_level = "gene_name", split_direction = FALSE, ymax = 300, order = "freq", order_group_list = c("CTRL", "TARDBP G298S", "VCP R155C", "VCP R191Q", "VCP R191Q knock-in"), title = "Decreased VCP binding", label_colours = c("dodgerblue2", "firebrick2", "forestgreen", "gold2"))  
coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.decreased.upset

```

Correlation heatmap

```{r}
# heatmap of correlation coefficients
coip.ctrl_als.nuc_cyt.ml240_vs_untreated.dep_stat = select(coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, CTRL = stat) %>%
  left_join(select(coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, `TARDBP^G298S` = stat)) %>% 
  left_join(select(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, `VCP^R155C` = stat)) %>%
  left_join(select(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, `VCP^R191Q` = stat)) %>%
  left_join(select(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, `VCPki^R191Q` = stat)) %>%
  drop_na() %>%
  column_to_rownames("gene_name")
cormat <- round(cor(coip.ctrl_als.nuc_cyt.ml240_vs_untreated.dep_stat),2)
head(cormat)
melted_cormat <- melt(cormat)
head(melted_cormat)
get_lower_tri<-function(cormat){ # Get lower triangle of the correlation matrix
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
get_upper_tri <- function(cormat){ # Get upper triangle of the correlation matrix
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
  }
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE) #%>% # Melt the correlation matrix
coip.ctrl_als.nuc_cyt.ml240_vs_untreated_correlation_ggheatmap <- ggplot(melted_cormat, aes(Var2, Var1, fill = value))+
  geom_tile(color = "white") +  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limit = c(-1,1), space = "Lab", name="Pearson\nCorrelation") + 
  geom_text(aes(Var2, Var1, label = value), color = "black", size = 3)  +
  theme_minimal() + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.ticks = element_blank(),
                          legend.position = "top", legend.direction = "horizontal", legend.title = element_text(size = 8), axis.text=element_text(size=7.5)) + # coord_fixed()+ 
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1, title.position = "top", title.hjust = 0.5)) +
  scale_x_discrete(labels = scales::parse_format())  +  scale_y_discrete(labels = scales::parse_format()) 
# coip.ctrl_als.nuc_cyt.ml240_vs_untreated_correlation_ggheatmap

```

Fisher test

```{r}
# overlap mislocalised proteins with ML240 changed VCP interactors
coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
tardbp.ml240_relocalised <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
newGeneOverlap(tardbp.ml240_relocalised, coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=567
# listB size=0
# Intersection size=0
# Overlapping p-value=1
# Jaccard Index=0.0

coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
vcp_r155c.ml240_relocalised <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
newGeneOverlap(vcp_r155c.ml240_relocalised, coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=161
# listB size=142
# Intersection size=11
# Overlapping p-value=6.8e-03
# Jaccard Index=0.0

coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
vcp_r191q.ml240_relocalised <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
newGeneOverlap(vcp_r191q.ml240_relocalised, coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=778
# listB size=50
# Intersection size=6
# Overlapping p-value=0.83
# Jaccard Index=0.0

coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
vcp_iso.ml240_relocalised <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
newGeneOverlap(vcp_iso.ml240_relocalised, coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res)) %>% testGeneOverlap()
# GeneOverlap object:
# listA size=529
# listB size=101
# Intersection size=18
# Overlapping p-value=0.021
# Jaccard Index=0.0


# # overlap redistributed transcripts with ML240 changed VCP interactors
# coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
# untreated.tardbp_vs_ctrl.nuc_vs_cyt.det <- untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# ml240.tardbp_vs_ctrl.nuc_vs_cyt.det <- ml240.nuc_vs_cyt.tardbp_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# tardbp.ml240_tx_relocalised = untreated.tardbp_vs_ctrl.nuc_vs_cyt.det[!untreated.tardbp_vs_ctrl.nuc_vs_cyt.det %in% ml240.tardbp_vs_ctrl.nuc_vs_cyt.det]
# newGeneOverlap(tardbp.ml240_tx_relocalised, coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.nuc_vs_cyt.tardbp_vs_ctrl$res.tx)) %>% testGeneOverlap()
# # GeneOverlap object:
# # listA size=328
# # listB size=0
# # Intersection size=0
# # Overlapping p-value=1
# # Jaccard Index=0.0
# 
# coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
# untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.det <- untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.det <- ml240.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# vcp_r155c.ml240_tx_relocalised = untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.det[!untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.det %in% ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.det]
# newGeneOverlap(vcp_r155c.ml240_tx_relocalised, coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r155c_vs_ctrl$res.tx)) %>% testGeneOverlap()
# # GeneOverlap object:
# # listA size=310
# # listB size=142
# # Intersection size=3
# # Overlapping p-value=1e-03
# # Jaccard Index=0.0
# 
# coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
# untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.det <- untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.det <- ml240.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# vcp_r191q.ml240_tx_relocalised = untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.det[!untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.det %in% ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.det]
# newGeneOverlap(vcp_r191q.ml240_tx_relocalised, coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.nuc_vs_cyt.vcp_r191q_vs_ctrl$res.tx)) %>% testGeneOverlap()
# # GeneOverlap object:
# # listA size=464
# # listB size=50
# # Intersection size=3
# # Overlapping p-value=1.5e-04
# # Jaccard Index=0.0
# 
# coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.dpe <- coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res %>% filter(padj < 0.05) %>% pull(gene_name)
# untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.det <- untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.det <- ml240.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx %>% filter(padj < 0.05) %>% pull(gene_name)
# vcp_iso.ml240_tx_relocalised = untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.det[!untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.det %in% ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.det]
# newGeneOverlap(vcp_iso.ml240_tx_relocalised, coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.dpe, genome.size=nrow(untreated.nuc_vs_cyt.vcp_iso_vs_ctrl$res.tx)) %>% testGeneOverlap()
# # GeneOverlap object:
# # listA size=442
# # listB size=101
# # Intersection size=5
# # Overlapping p-value=1.8e-06
# # Jaccard Index=0.0

```



## DMSO Lysotracker & CellROX only

```{r}
# DMSO only:
dmso_assay_lysotracker.violin = violin_plot(data = drop_na(filter(lysosome_sytox.PlateResults.bind.live_800.vcp_inhibitors, treatment == "DMSO"),neurite_soma_ratio), color_by = "mutation_treatment", continuous = "neurite_soma_ratio", ylims = c(0.1,1.4), cols = brewer.pal(10, "Paired"), plot_stats = FALSE, ylabel = "Lysosome Neurite:Soma ratio", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE)
dmso_assay_lysotracker.violin


dmso_cellrox.violin.stat = filter(CellRox_Fix_PlateResults.vcp_inhibitors.800, treatment == "DMSO") %>% wilcox_test(live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well ~ mutation) %>% add_significance() %>% add_xy_position(x = "mutation") %>% filter(group1 == "CTRL")
dmso_cellrox.violin.stat

dmso_assay_cellrox.violin = violin_plot(data = filter(CellRox_Fix_PlateResults.vcp_inhibitors.800, treatment == "DMSO"), color_by = "mutation", continuous = "live_nuclei_intensity_nucleus_alexa_488_sum_mean_per_well", ylims = c(302146,2250000), cols = brewer.pal(10, "Paired"), plot_stats = FALSE, ylabel = "Reactive Oxygen Species Nuclei intensity", dpi = 300, arrow_labels = FALSE, plot_violin = FALSE) + scale_y_continuous(labels = scales::comma) + theme(axis.text.y = element_text(angle = 45, vjust = 0.5)) +
    stat_pvalue_manual(dmso_cellrox.violin.stat, label = "p.adj.signif", tip.length = .03, size = 4, hide.ns = TRUE, y.position = c(2000000,2100000))
dmso_assay_cellrox.violin
```


## Merge

```{r}
coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged <- plot_grid(
  vcp_pull_down_blot.gg,
  plot_grid(coip.ctrl.nuc_cyt.ml240_vs_untreated.dep.volcano, coip.tardbp.nuc_cyt.ml240_vs_untreated.dep.volcano, coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep.volcano, coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep.volcano, coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep.volcano, ncol = 5, labels = letters[2:6]),
  plot_grid(coip.ctrl.nuc_cyt.ml240_vs_untreated.res.dep.go_curated, coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.res.dep.go_curated, coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.res.dep.go_curated, coip.vcp_iso.nuc_cyt.ml240_vs_untreated.res.dep.go_curated, ncol = 4, labels = letters[7:10]), 
  plot_grid(coip.ctrl_als.nuc_cyt.ml240_vs_untreated_correlation_ggheatmap, coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.increased.upset, coip.nuc_cyt.ml240_vs_untreated.bind_datasets.dep.decreased.upset, ncol = 3, labels = letters[11:13]),
  plot_grid(dmso_assay_lysotracker.violin, dmso_assay_cellrox.violin, ncol = 2, labels = letters[14:15]),
nrow = 5, rel_heights = c(0.7,1,1.3,0.8,0.9), labels = c("a","","",""))
# coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged
ggsave(coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged, filename = here(proj_path, "mass-spec/figures/coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged.png"), height = 15, width = 14)
ggsave(coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged, filename = here(proj_path, "mass-spec/figures/coip.ctrl_als.nuc_cyt.ml240_vs_untreated.volano_go.merged.pdf"), height = 15, width = 14)

```

# Table S13 Co-IP ML240 vs untreated

```{r}
tardbp.ml240_relocalised <- untreated.tardbp_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.tardbp_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
vcp_r155c.ml240_relocalised <- untreated.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_r155c_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
vcp_r191q.ml240_relocalised <- untreated.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_r191q_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)
vcp_iso.ml240_relocalised <- untreated.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res %>% filter(padj < 0.05, !gene_name %in% pull(filter(ml240.vcp_iso_vs_ctrl.nuc_vs_cyt.dep$res,padj<0.05),gene_name)) %>% pull(gene_name)

coip.nuc_cyt.ml240_vs_untreated.dep.join = select(coip.ctrl.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, ctrl.stat = stat, ctrl.padj = padj, ctrl.lfc = log2FoldChange, ctrl.baseMean = baseMean) %>%
  full_join(select(coip.tardbp.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange, tardbp.baseMean = baseMean)) %>% 
  full_join(select(coip.vcp_r155c.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange, vcp_r155c.baseMean = baseMean)) %>%
  full_join(select(coip.vcp_r191q.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, vcp_r191q.stat = stat, vcp_r191q.padj = padj, vcp_r191q.lfc = log2FoldChange, vcp_r191q.baseMean = baseMean)) %>%
  full_join(select(coip.vcp_iso.nuc_cyt.ml240_vs_untreated.dep$res, gene_name, vcp_ki.stat = stat, vcp_ki.padj = padj, vcp_ki.lfc = log2FoldChange, vcp_ki.baseMean = baseMean)) %>% 
  filter(is.na(ctrl.stat) == FALSE & is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_r191q.stat) == FALSE & is.na(vcp_ki.stat) == FALSE) %>%
  # filter(ctrl.padj < 0.05 | tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_r191q.padj < 0.05 | vcp_ki.padj < 0.05) %>%
  arrange(-( abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_r191q.stat) + abs(vcp_ki.stat))) %>%
  mutate(tardbp.rescued = case_when(gene_name %in% tardbp.ml240_relocalised ~ "rescued", TRUE ~ ""),
         vcp_r155c.rescued = case_when(gene_name %in% vcp_r155c.ml240_relocalised ~ "rescued", TRUE ~ ""),
         vcp_r191q.rescued = case_when(gene_name %in% vcp_r191q.ml240_relocalised ~ "rescued", TRUE ~ ""),
         vcp_ki.rescued = case_when(gene_name %in% vcp_iso.ml240_relocalised ~ "rescued", TRUE ~ ""),
         category = case_when(gene_name %in% ALS_genes ~ "ALS gene", gene_name %in% RNA_BINDING_PROTEINS ~ "RBP",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_PORE ~ "Nuclear pore", 
                              gene_name %in% c(go.pathways.msigdb$GO_NEGATIVE_REGULATION_OF_NUCLEOCYTOPLASMIC_TRANSPORT, go.pathways.msigdb$GO_NUCLEOCYTOPLASMIC_CARRIER_ACTIVITY, go.pathways.msigdb$GO_RNA_EXPORT_FROM_NUCLEUS, go.pathways.msigdb$GO_TRANSCRIPTION_EXPORT_COMPLEX) ~ "Nuclear export",
                              gene_name %in% go.pathways.msigdb$GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY ~ "NMD", TRUE ~ "")) 
coip.nuc_cyt.ml240_vs_untreated.dep.join
write_csv(coip.nuc_cyt.ml240_vs_untreated.dep.join, here(proj_path,"expression/coip.nuc_cyt.ml240_vs_untreated.dep.join.csv"))
library(googlesheets4)
gs4_browse("https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267")
write_sheet(coip.nuc_cyt.ml240_vs_untreated.dep.join, ss = "https://docs.google.com/spreadsheets/d/188f703jFBmSzS42syPbSQ7AFp7IL7VrHzuLLwdlkiMw/edit#gid=1363320267", sheet = "S13 CO-IP")

coip.nuc_cyt.ml240_vs_untreated.dep.join %>% filter(ctrl.padj < 0.05, vcp_r155c.padj < 0.05, vcp_r191q.padj < 0.05, vcp_ki.padj < 0.05)


ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join = select(ml240.ctrl_nuc_vs_cyt.dep$res, gene_name, ctrl.stat = stat, ctrl.padj = padj, ctrl.lfc = log2FoldChange) %>% 
  left_join(select(ml240.tardbp_nuc_vs_cyt.dep$res, gene_name, tardbp.stat = stat, tardbp.padj = padj, tardbp.lfc = log2FoldChange)) %>% 
  left_join(select(ml240.vcp_r155c_nuc_vs_cyt.dep$res, gene_name, vcp_r155c.stat = stat, vcp_r155c.padj = padj, vcp_r155c.lfc = log2FoldChange)) %>%
  left_join(select(ml240.vcp_iso_nuc_vs_cyt.dep$res, gene_name, vcp_iso.stat = stat, vcp_iso.padj = padj, vcp_iso.lfc = log2FoldChange))


ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt = ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join %>% filter(is.na(tardbp.stat) == FALSE & is.na(vcp_r155c.stat) == FALSE & is.na(vcp_iso.stat) == FALSE) %>%
  filter(ctrl.padj < 0.05 | tardbp.padj < 0.05 | vcp_r155c.padj < 0.05 | vcp_iso.padj < 0.05) %>%
  arrange(-(abs(ctrl.stat) + abs(tardbp.stat) + abs(vcp_r155c.stat) + abs(vcp_iso.stat))) %>%
  rename_at(vars(starts_with("vcp_r155c")), ~ str_replace_all(., "vcp_r155c", "vcp_mut")) %>% rename_at(vars(starts_with("vcp_iso")), ~ str_replace_all(., "vcp_iso", "vcp_ki"))
ctrl_vcp_tardbp.ml240.nuc_vs_cyt.dep.join.filt %>% filter(gene_name == "TARDBP")

```


# GEO submission

```{r}
geo_metadata = read_csv(here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/sample-details/samplesheet.csv")) %>% distinct(sample, .keep_all = TRUE) %>% 
  mutate(raw_file_1 = paste0(sample,"_1.merged.fastq.gz"), raw_file_2 = paste0(sample,"_2.merged.fastq.gz"), organism = "Homo sapiens",
         title = paste0(fraction,"_",treatment,"_",cellline)) %>%
  filter(fraction != "whole") %>%
  select(sample_name = sample, raw_file_1, raw_file_2, title, organism, strandedness, fraction, treatment, cellline, library_type, read_length = requested_read_length, paired_end = requested_run_type, sequencing_platform = requested_sequencing_platform)
geo_metadata
write_csv(geo_metadata, here(collab_path, "motor-neuron-vcp-inhibitor-harley-2022/sample-details/geo_metadata.csv"))

```
